# GxG and GxE Interactions, Power and Sample Size Calculation
In this section we select the top 5 SNPs for continuous Z-endoxifen ratio, and demonstrate how to use PLINK to perform tests of GxG [G(Gene) by G(Gene)] and GxE [G(Gene) by E(Environment)] interactions.

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_knit$set(root.dir = "./Data/case_interaction")
getwd
```

```{r eval=T, echo=F, message=FALSE, warnings=FALSE}
# R codes- run in R Console
#load R packages
library(data.table)
library(GEOquery)
library(qqman)
library(png)
```

```{bash eval=F}
# change the path below to where you saved "cshl_2024" folder
## run in R Terminal
mydir="/path/to/cshl_2024/Data/case_interaction"
cd $mydir
```

```{r eval=F}
# R codes- run in R Console
setwd("/path/to/cshl_2024/Data/case_interaction")
mydir<-getwd()
```

## Prepare Files
We can obtain the top 5 SNPs and p-values for continuous Z-endoxifen ratio (using R)

```{r}
# R codes- run in R Console
qassoc=fread("../case_data/significant_snps_with_rs_for_z-endoxifen-ratio.txt",header=T) #load association results for continuous r_met
ordered_qassoc=qassoc[order(qassoc$FDR_BH),] #order association results based on p-value
head(ordered_qassoc,5) #association results for the top 5 SNPs with the 5 smallest p-values
```

We want to test Epistasis between the top 5 SNPs for continuous r_met using PLINK. As shown in the table above, the top 5 SNPs for continuous Z-endoxifen ratio are:

    chr22 rs8138080
    chr22 rs4467371
    chr22 rs2011944
    chr22 rs1062753
    chr22 rs4453786
    
The following code generates **map** idx & **map** file of the top 5 SNPs (using R)

```{r message=FALSE, warnings=FALSE}
# R codes- run in R Console
map=read.table("genotype_1k.map") #read map file
map=as.matrix(map) #create a matrix from the MAP file
top5SNPs=c("rs8138080", "rs4467371","rs2011944","rs1062753","rs4453786") #SNP ID of the top 5 SNPs
top5_idx=matrix(0,5,1) #create an empty matrix
for (i in 1:5){
  top5_idx[i]=which(map[,2]==top5SNPs[i]) #index of the top 5 SNPs in the MAP file
}
as.vector(top5_idx)
top5SNPs.map=map[top5_idx,] #MAP file of the top 5 SNPs
write.table(top5SNPs.map,"top5SNPs.map", row.names=F, col.names=F, quote=F, sep=" ") #save MAP file of the top 5 SNPs
```

The following code generates **ped** file of the top 5 SNPs (in Linux) using the index of these 5 SNPs above. Since the first 6 columns are sample information and each SNP has two alleles in .ped file, the index of each pair of allele for each SNP= [6+2(top5_idx)-1, 6+2(top5_idx)].

```{bash}
# bash codes- run in terminal
awk '{print $1,$2,$3,$4,$5,$6, $1901,$1902, $1891,$1892, $1859, $1860, $1849,$1850, $1861, $1862}' \
genotype_1k.ped > top5SNPs.ped
```

Phenotype data:
```{bash}
# bash codes- run in terminal
paste famID.txt samID.txt z_endoxifen_ratio.txt > ZR.txt
```

## Test for Epistasis (GxG)
For disease-trait population-based samples, it is possible to test for epistasis. All pairwise combinations of SNPs can be tested using PLINK (computationally feasible). For more detailed information for this test, please refer: http://zzz.bwh.harvard.edu/plink/epi.shtml

PLINK produces a model based on allele dosage for each SNP, A and B, and fits the model in the form of

     Y ~ b0 + b1.A + b2.B + b3.AB + e 

```{bash eval=F}
## # both Mac and Windows can run in Terminal
# windows users
../../Tools/PLINK/plink_win64_20231211/plink.exe \
--noweb --file top5SNPs --epistasis --epi1 1 --pheno ZR.txt --out younameit
```
```{bash}
# macOS users
../../Tools/PLINK/plink_mac_20231211/plink \
--noweb --file top5SNPs --epistasis --epi1 1 --pheno ZR.txt --out younameit
```

```{r}
# R codes- run in R Console
epi_res <- fread('younameit.epi.qt')
epi_res[!is.na(epi_res$P),]
```

The output is in the form:

     CHR1    Chromosome of the first SNP

     SNP1    Identifier for the first SNP
         
     CHR2    Chromosome of the second SNP
         
     SNP2    Identifier for the second SNP
         
     OR_INT  Odds ratio for the interaction
         
     STAT    Chi-square statistic, df=1
         
     P       Asymptotic p-value

```{bash eval=F}
## # both Mac and Windows can run in Terminal
# windows users
../../Tools/PLINK/plink_win64_20231211/plink.exe \
--noweb --file top5SNPs --epistasis --epi1 0.05 --pheno ZR.txt --out younameit
```
```{bash}
# macOS users
../../Tools/PLINK/plink_mac_20231211/plink \
--noweb --file top5SNPs --epistasis --epi1 0.05 --pheno ZR.txt --out younameit
```

It only records results that are significant, i.e., p<=0.05, which prevents generating too many outputs.

The results are shown as below. There is nothing in the table so no Epistasis.
```{r}
# R codes- run in R Console
epi_res <- fread('younameit.epi.qt')
epi_res
```

## Gene-Environment Interaction (GxE)
PLINK provides the ability to test for a difference in association with a quantitative trait between two environments (or, more generally, two groups). This test is simply based on comparing the difference between two regression coefficients.

We need to specify a covariate file for this test:

**cover_age.txt**
   
    Col 1 is family ID

    Col 2 is sample ID

    Col 3 is age (need to be in 2 groups: 1: old, 2: young)
    
Our current covariate file `age.txt` is a continuous variable. So we need to manually separate it in 2 groups: old(recode as 1) and young(recode as 2). As median age is 55.5, we will assign the sample to "old" if its age is older than 55.5, otherwise to "young".

```{r message=FALSE, warnings=FALSE}
# R codes- run in R Console
age<- fread("age.txt", header=F) # read in age
names(age)<-"age"
age_recode <- matrix(1,192,1) # create a new data with all 1
idx <- which(age$age<55.5) # which sample has age < 55.5
age_recode[idx,]<-2 # assign those with age <55.5 as 2
fwrite(age_recode,"age_binary.txt",quote=F,col.name=F, row.name=F) # save data
```

```{bash}
# bash codes- run in terminal
paste famID.txt samID.txt age_binary.txt > covar_age.txt
```

```{bash}
## # both Mac and Windows can run in Terminal
# windows users
../../Tools/PLINK/plink_win64_20231211/plink.exe \
--noweb --file top5SNPs --gxe --covar covar_age.txt \
--pheno ZR.txt --out younameit

# macOS users
../../Tools/PLINK/plink_mac_20231211/plink \
--noweb --file top5SNPs --gxe --covar covar_age.txt \
--pheno ZR.txt --out younameit
```

The output is in the form:

     CHR       Chromosome number

     SNP       SNP identifier

     NMISS1    Number of non-missing genotypes in the first group

     BETA1     Regression coefficient in the first group

     SE1       Standard error of coefficient in the first group

     NMISS2    Number of non-missing genotypes in the second group

     BETA2     Regression coefficient in the second group

     SE2       Standard error of coefficient in the second group

     Z_GXE     Z score, test for the interaction
         
     P_GXE     Asymptotic p-value for this test

The results are shown as below:
```{r}
# R codes- run in R Console
gxe_res <- fread('younameit.qassoc.gxe')
gxe_res
```
By looking at the P-value, there is still no GxE interaction.

## Power and sample size calculation for GWAS

In GWAS studies, it's also crucial to perform power analysis and sample size calculation. This helps researchers to determine the appropriate number of participants to detect significant genetic associations with sufficient statistical power. A review paper of sample size calculation definitions and tools can be found at [Politi et. al, 2023](https://www.mdpi.com/2075-1729/13/1/235).

A number of factors must be considered for GWAS power analysis and sample size calculations.

- **Outcome**: The phenotype of a study. It can be continuous or binary (case/control). For continuous pahenotype, variability is assessed with SD (standard deviation). While for binary phenotype, the paramater used is the proportion of individuals with disease.
- **Effect size**: The expected size of the genetic effect on the phenotype. This can be estimated based on previous studies or biological knowledge. The parameter used is odds ratio (OR) for binary phenotype or percentage of phenotypic variance for continuous phenotype.
- **Significance level (alpha)**: The threshold used to determine statistical significance. The most common level is 0.05 (5%).
- **Power (1 - beta)**: The probability of correctly rejecting the null hypothesis if the alternative hypothesis is true. The total power target is 0.8 (80%).
- **Minor Allele frequency (MAF)**: The frequency of a genetic variant in a population. This information can be obtained from benchmarking databases or pilot studies. A genetic variant can be classified as common (MAF > 5%), uncommon (1â€“5% MAF) and rare (MAF < 1%). Typically, GWAS studies are sufficiently powered only to test common genetic variants.
- **Genetic Inheritance Model**: The assumed genetic model to assess the clinical effect of an SNP. Three genetic models are usually adopted: a dominant, recessive and additive model.

Below, we will use an R package `genpwr` to show you an example of power and sample size calculation. Package `genpwr` can calculate power, sample size or detectable effect size. It handle both continuous and binary outcomes, and is capable of considering GxE interactions when calculating.

Below, we will show you examples for calculating:

- power
- sample size
- detectable effect size (or odds ratio in the case of a binary outcome variable)
provided that two of the three above variables are entered into the appropriate genpwr function.

First, we need to install this package from GitHub using "devtools". The codes below show you how to install it on your laptop.
```{r message=FALSE, warnings=FALSE}
# R codes- run in R Console

# installation
#install.packages("devtools")
#library(devtools)
#install_github("camillemmoore/Power_Genetics", subdir="genpwr")

# use package
library(genpwr)
```

We will determine the above parameters using our case study data.

### Power calculation with binary outcome:

Based on Chapter 4, we have a case-control study with 192 individuals, including 95 cases and 97 controls (case rate of approximately 50%). The SNPs included in the GWAS analysis have MAF over a range of 0 to 0.5. (This can be calculated using plink `--freq`. Check Chapter 2.) But we will only focus on common alleles with MAF >0.1. We want to calculate the statistical power to detect an odds ratio of 3, for all possible combinations of true and test models, assuming an alpha of 0.05.

We need to specify the calculation is for power with `calc="power"` and use `model="logistic"` for binary outcome. The function `genpwr.calc` calculates power for all alleles with MAF between 0.1 and 0.5 with an increment of 0.01 based on the study design. Then a plot is generated to visualize the power change vs MAF under different genetic models.

```{r powerplot,message=FALSE, warnings=FALSE}
# R codes- run in R Console
pw <- genpwr.calc(calc = "power", model = "logistic", 
                  ge.interaction = NULL, N=192, Case.Rate=0.5, k=NULL,
                  MAF=seq(0.1,0.5,0.01), OR=3, Alpha=0.05,
                  True.Model=c("Additive","Dominant", "Recessive"), 
                  Test.Model=c("Additive","Dominant", "Recessive"))
power.plot(pw)
```

#### Case study with continuous outcome:

For continuous phenotype, we need to obtain the phenotype SD instead of case rate. The codes below read in the phenotype data from Chapter 3 and calculate the standard deviation for all metabolites.

```{r}
# R codes- run in R Console
# Read data
phenotype=read.table("../case_data/phenotype.txt", header=T)
head(phenotype)
```
```{r}
# R codes- run in R Console
# get sd
apply(phenotype[,-1],2,function(x){sd(x,na.rm=T)})
```

We will use tamoxifen as an example of continuous phenotype. The SD of tamoxifen is approximately 67.

We still focus on common alleles with MAF >0.1. and the goal is to calculate the statistical power to detect an effect size of 30 (~45% of the phenotype SD) under alpha of 0.05. Note that the effect size (ES) can't exceed the total SD of outcome.

Similar as in binary study, we use `calc="power"` but specify continuous study model with `model="linear"`.

```{r pp2,message=FALSE, warnings=FALSE, error=F}
# R codes- run in R Console
pw <- genpwr.calc(calc = "power", model = "linear",
   N=192, sd_y=67, k=NULL,
   MAF=seq(0.1, 0.5, 0.01), ES=30, Alpha=0.05,
   True.Model=c("Additive","Dominant", "Recessive"), 
   Test.Model=c("Additive","Dominant", "Recessive"))

power.plot(pw)
```

#### Continuous study with continuous GxE interaction:

In addition to power calculation for main effects model, `genpwr` is also capable to calculate power for studies with GxE interactions. We just show you a continuous study with continuous GxE interactions here. You can also practice for a binary study with continuous GxE interactions, or study with binary GxE interactions.

Besides the parameters above, we need to specify the effect size for both main effect environmental variables and GxE interactions. Also note that the sum of all effect size parameters shouldn't exceed the total SD of phenotype. We also specify the type of GxE interactions with `ge.interaction="continuous"`.

```{r pp3,message=FALSE, warnings=FALSE}
# R codes- run in R Console
pec <- genpwr.calc(calc = "power", model = "linear",
                 ge.interaction = "continuous",
                 N=192, sd_y=67, MAF=seq(0.1,0.5,0.01), ES_G=30, 
                 ES_E=5, ES_GE=20, sd_e =1,  Alpha=0.05, 
                   True.Model=c("Additive","Dominant", "Recessive"), 
                   Test.Model=c("Additive","Dominant", "Recessive"))
power.plot(pec)
```

### Sample size calculation
For sample size calculation, we want to determine an appropriate study size that is sufficient to detect the effect size/OR under a desired power.

#### Case study with binary outcome:

We still use the above binary case study with 192 individuals. Assuming the case rate is 50% and MAF of variants is 0.2-0.5, we want to calculate the sample size needed to detect an odds ratio of 3 with a power of 0.7 or 0.8.

We need to spcific calculation type with `calc="ss"` and `Power=c(0.7,0.8)`.

```{r ss1,message=FALSE, warnings=FALSE}
# R codes- run in R Console
ss <- genpwr.calc(calc = "ss", model = "logistic", ge.interaction = NULL,
                  OR=3, Case.Rate=0.5, k=NULL,
                  MAF=seq(0.2, 0.5, 0.01), Power=c(0.7,0.8), Alpha=0.05,
                  True.Model=c("Additive","Dominant", "Recessive"), 
                  Test.Model=c("Additive","Dominant", "Recessive"))
ss.plot(ss)
```

As a general rule, under the same assumptions, the dominant model needs the smallest sample size to achieve adequate statistical power, whereas the recessive model is the one that requires the largest number of participants.

We could check the dominant and additive model only and ignore the recessive model.

```{r ss2,message=FALSE, warnings=FALSE}
# R codes- run in R Console
ss <- genpwr.calc(calc = "ss", model = "logistic", ge.interaction = NULL,
                  OR=3, Case.Rate=0.5, k=NULL,
                  MAF=seq(0.2, 0.5, 0.01), Power=c(0.7,0.8), Alpha=0.05,
                  True.Model=c("Additive","Dominant"), 
                  Test.Model=c("Additive","Dominant", "Recessive"))
ss.plot(ss)
```

For both additive and dominant models, a sample size of about 200 individuals is sufficient to detect an OR of 3 if we test for additive or dominant genetic effects.

#### Case study with continuous outcome

Similar as above, we calculate the sample size needed to detect an effect size of 30 with 70% or 80% power. The continuous phenotype used is still tamoxifen with SD=76.
```{r ss3,message=FALSE, warnings=FALSE}
# R codes- run in R Console
ss <- genpwr.calc(calc = "ss", model = "linear",
                  sd_y=67, k=NULL, Power=c(0.7,0.8),
                  MAF=seq(0.1, 0.5, 0.01), ES=30, Alpha=0.05,
                  True.Model=c("Additive","Dominant"), 
                  Test.Model=c("Additive","Dominant", "Recessive"))

ss.plot(ss)
```
#### Binary study with binary GxE interactions
```{r ss4,message=FALSE, warnings=FALSE}
# R codes- run in R Console
sse <- genpwr.calc(calc = "ss", model = "logistic",
                  Case.Rate=0.5, ge.interaction = "binary", Power=0.8,
                  MAF=seq(0.1,0.5,0.01), OR_G=3, 
                  OR_E=0.5, OR_GE=4, P_e = 0.4,  Alpha=0.05, 
                  True.Model=c("Additive","Dominant"), 
                  Test.Model=c("Additive","Dominant", "Recessive"))
ss.plot(sse)
```



