<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 6 GWAS with Binary Trait | CSHL course - GWAS Hands-on Analysis</title>
  <meta name="description" content="This is the course material for the CSHL hands-on session in GWAS and eQTL." />
  <meta name="generator" content="bookdown 0.41 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 6 GWAS with Binary Trait | CSHL course - GWAS Hands-on Analysis" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This is the course material for the CSHL hands-on session in GWAS and eQTL." />
  <meta name="github-repo" content="rstudio/cshl_gwas" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 6 GWAS with Binary Trait | CSHL course - GWAS Hands-on Analysis" />
  
  <meta name="twitter:description" content="This is the course material for the CSHL hands-on session in GWAS and eQTL." />
  

<meta name="author" content="Min Zhang, Danni Liu- University of California, Irvine; Purdue University" />


<meta name="date" content="2025-06-09" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="gwas-with-continuous-trait.html"/>
<link rel="next" href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">CSHL Course- GWAS and EQTL Hands-on Analysis</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Welcome to the GWAS and eQTL Workshop</a>
<ul>
<li class="chapter" data-level="1.0.1" data-path="index.html"><a href="index.html#overview"><i class="fa fa-check"></i><b>1.0.1</b> Overview</a></li>
<li class="chapter" data-level="1.0.2" data-path="index.html"><a href="index.html#folder-structure"><i class="fa fa-check"></i><b>1.0.2</b> Folder structure:</a></li>
<li class="chapter" data-level="1.0.3" data-path="index.html"><a href="index.html#programming-languages"><i class="fa fa-check"></i><b>1.0.3</b> Programming Languages:</a></li>
<li class="chapter" data-level="1.0.4" data-path="index.html"><a href="index.html#required-gwas-tools"><i class="fa fa-check"></i><b>1.0.4</b> Required GWAS Tools</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="an-overview-of-regression-with-toy-examples.html"><a href="an-overview-of-regression-with-toy-examples.html"><i class="fa fa-check"></i><b>2</b> An Overview of Regression with Toy Examples</a>
<ul>
<li class="chapter" data-level="2.1" data-path="an-overview-of-regression-with-toy-examples.html"><a href="an-overview-of-regression-with-toy-examples.html#environment-setting-and-packages"><i class="fa fa-check"></i><b>2.1</b> Environment Setting and Packages</a></li>
<li class="chapter" data-level="2.2" data-path="an-overview-of-regression-with-toy-examples.html"><a href="an-overview-of-regression-with-toy-examples.html#toy-example-data"><i class="fa fa-check"></i><b>2.2</b> Toy Example Data</a></li>
<li class="chapter" data-level="2.3" data-path="an-overview-of-regression-with-toy-examples.html"><a href="an-overview-of-regression-with-toy-examples.html#logistic-regression"><i class="fa fa-check"></i><b>2.3</b> Logistic Regression</a></li>
<li class="chapter" data-level="2.4" data-path="an-overview-of-regression-with-toy-examples.html"><a href="an-overview-of-regression-with-toy-examples.html#roc-curve-and-auc-for-hoxb13"><i class="fa fa-check"></i><b>2.4</b> ROC curve and AUC for HOXB13</a></li>
<li class="chapter" data-level="2.5" data-path="an-overview-of-regression-with-toy-examples.html"><a href="an-overview-of-regression-with-toy-examples.html#practice"><i class="fa fa-check"></i><b>2.5</b> Practice</a></li>
<li class="chapter" data-level="2.6" data-path="an-overview-of-regression-with-toy-examples.html"><a href="an-overview-of-regression-with-toy-examples.html#comparison-of-hoxb13-roc-curve-and-nhlh1-roc-curve"><i class="fa fa-check"></i><b>2.6</b> Comparison of HOXB13 ROC curve and NHLH1 ROC curve</a></li>
<li class="chapter" data-level="2.7" data-path="an-overview-of-regression-with-toy-examples.html"><a href="an-overview-of-regression-with-toy-examples.html#linear-regression"><i class="fa fa-check"></i><b>2.7</b> Linear Regression</a>
<ul>
<li class="chapter" data-level="2.7.1" data-path="an-overview-of-regression-with-toy-examples.html"><a href="an-overview-of-regression-with-toy-examples.html#linear-regression-of-hoxb13-vs.-il17br"><i class="fa fa-check"></i><b>2.7.1</b> Linear regression of HOXB13 vs. IL17BR</a></li>
</ul></li>
<li class="chapter" data-level="2.8" data-path="an-overview-of-regression-with-toy-examples.html"><a href="an-overview-of-regression-with-toy-examples.html#variable-selectionforward-backward-and-stepwise-selection"><i class="fa fa-check"></i><b>2.8</b> Variable selection–Forward, backward and stepwise selection</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html"><i class="fa fa-check"></i><b>3</b> GWAS Tools and Data Preprocessing Workflow with Toy Example</a>
<ul>
<li class="chapter" data-level="3.1" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#environment-setting"><i class="fa fa-check"></i><b>3.1</b> Environment Setting</a></li>
<li class="chapter" data-level="3.2" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#gwas-tools"><i class="fa fa-check"></i><b>3.2</b> GWAS Tools</a></li>
<li class="chapter" data-level="3.3" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#quality-control"><i class="fa fa-check"></i><b>3.3</b> Quality Control</a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#an-example-of-summary-statistics"><i class="fa fa-check"></i><b>3.3.1</b> An Example of Summary Statistics</a></li>
<li class="chapter" data-level="3.3.2" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#an-example-of-inclusion-thresholds-for-qc"><i class="fa fa-check"></i><b>3.3.2</b> An Example of Inclusion Thresholds for QC</a></li>
<li class="chapter" data-level="3.3.3" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#remove-closely-related-individuals"><i class="fa fa-check"></i><b>3.3.3</b> Remove Closely Related Individuals</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#an-imputation-example"><i class="fa fa-check"></i><b>3.4</b> An Imputation Example</a></li>
<li class="chapter" data-level="3.5" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#population-stratification-correction-using-plink"><i class="fa fa-check"></i><b>3.5</b> Population Stratification Correction Using PLINK</a>
<ul>
<li class="chapter" data-level="3.5.1" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#pca-on-whole-genome-data-using-plink"><i class="fa fa-check"></i><b>3.5.1</b> PCA on whole genome data using PLINK</a></li>
<li class="chapter" data-level="3.5.2" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#test-the-significance-of-principle-components-pcs"><i class="fa fa-check"></i><b>3.5.2</b> Test the Significance of Principle Components (PCs)</a></li>
</ul></li>
<li class="chapter" data-level="3.6" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#simple-association-test"><i class="fa fa-check"></i><b>3.6</b> Simple Association Test</a>
<ul>
<li class="chapter" data-level="3.6.1" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#quantitative-trait-association-test-for-continuous-trait"><i class="fa fa-check"></i><b>3.6.1</b> Quantitative Trait Association Test for Continuous Trait</a></li>
<li class="chapter" data-level="3.6.2" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#casecontrol-asssociation-test-for-binary-trait"><i class="fa fa-check"></i><b>3.6.2</b> Case/Control Asssociation Test for Binary Trait</a></li>
</ul></li>
<li class="chapter" data-level="3.7" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#quantitative-trait-association-test-for-continuous-trait-with-covariates"><i class="fa fa-check"></i><b>3.7</b> Quantitative Trait Association Test for Continuous Trait with Covariates</a>
<ul>
<li class="chapter" data-level="3.7.1" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#covariates-adjustments-using-r"><i class="fa fa-check"></i><b>3.7.1</b> Covariates adjustments (using R):</a></li>
<li class="chapter" data-level="3.7.2" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#association-test-using-plink"><i class="fa fa-check"></i><b>3.7.2</b> Association test using plink</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="gwas-case-study-data-preparation.html"><a href="gwas-case-study-data-preparation.html"><i class="fa fa-check"></i><b>4</b> GWAS Case Study-Data Preparation</a>
<ul>
<li class="chapter" data-level="4.1" data-path="gwas-case-study-data-preparation.html"><a href="gwas-case-study-data-preparation.html#environment-setting-and-packages-1"><i class="fa fa-check"></i><b>4.1</b> Environment setting and packages</a></li>
<li class="chapter" data-level="4.2" data-path="gwas-case-study-data-preparation.html"><a href="gwas-case-study-data-preparation.html#accessing-data"><i class="fa fa-check"></i><b>4.2</b> Accessing data</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="gwas-case-study-data-preparation.html"><a href="gwas-case-study-data-preparation.html#data-availability"><i class="fa fa-check"></i><b>4.2.1</b> Data availability</a></li>
<li class="chapter" data-level="4.2.2" data-path="gwas-case-study-data-preparation.html"><a href="gwas-case-study-data-preparation.html#data-files-documentation"><i class="fa fa-check"></i><b>4.2.2</b> Data files documentation</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="gwas-case-study-data-preparation.html"><a href="gwas-case-study-data-preparation.html#preprocessing-phenotype-data"><i class="fa fa-check"></i><b>4.3</b> Preprocessing Phenotype Data</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="gwas-case-study-data-preparation.html"><a href="gwas-case-study-data-preparation.html#extracting-phenotype-data"><i class="fa fa-check"></i><b>4.3.1</b> Extracting phenotype data</a></li>
<li class="chapter" data-level="4.3.2" data-path="gwas-case-study-data-preparation.html"><a href="gwas-case-study-data-preparation.html#summary-of-phenotypes-data"><i class="fa fa-check"></i><b>4.3.2</b> Summary of Phenotypes Data</a></li>
<li class="chapter" data-level="4.3.3" data-path="gwas-case-study-data-preparation.html"><a href="gwas-case-study-data-preparation.html#histogram-of-each-trait"><i class="fa fa-check"></i><b>4.3.3</b> Histogram of each trait</a></li>
<li class="chapter" data-level="4.3.4" data-path="gwas-case-study-data-preparation.html"><a href="gwas-case-study-data-preparation.html#handling-the-missing-values-in-the-phenotypes-and-covariates"><i class="fa fa-check"></i><b>4.3.4</b> Handling the missing values in the phenotypes and covariates</a></li>
<li class="chapter" data-level="4.3.5" data-path="gwas-case-study-data-preparation.html"><a href="gwas-case-study-data-preparation.html#checking-normality-assumptions-for-each-trait"><i class="fa fa-check"></i><b>4.3.5</b> Checking normality assumptions for each trait</a></li>
<li class="chapter" data-level="4.3.6" data-path="gwas-case-study-data-preparation.html"><a href="gwas-case-study-data-preparation.html#normalize-phenotype-data"><i class="fa fa-check"></i><b>4.3.6</b> Normalize phenotype data</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="gwas-case-study-data-preparation.html"><a href="gwas-case-study-data-preparation.html#covariates-adjustment"><i class="fa fa-check"></i><b>4.4</b> Covariates Adjustment</a></li>
<li class="chapter" data-level="4.5" data-path="gwas-case-study-data-preparation.html"><a href="gwas-case-study-data-preparation.html#prepare-plink-data"><i class="fa fa-check"></i><b>4.5</b> Prepare PLINK data</a>
<ul>
<li class="chapter" data-level="4.5.1" data-path="gwas-case-study-data-preparation.html"><a href="gwas-case-study-data-preparation.html#step-1-prepare-plink-ped-file"><i class="fa fa-check"></i><b>4.5.1</b> Step 1: prepare PLINK ped file</a></li>
<li class="chapter" data-level="4.5.2" data-path="gwas-case-study-data-preparation.html"><a href="gwas-case-study-data-preparation.html#step-2-prepare-plink-map-file"><i class="fa fa-check"></i><b>4.5.2</b> Step 2: prepare PLINK map file</a></li>
<li class="chapter" data-level="4.5.3" data-path="gwas-case-study-data-preparation.html"><a href="gwas-case-study-data-preparation.html#step-3-prepare-plink-alternate-phenotype-file-in-linux"><i class="fa fa-check"></i><b>4.5.3</b> Step 3 : prepare PLINK alternate phenotype file (in Linux)</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="gwas-with-continuous-trait.html"><a href="gwas-with-continuous-trait.html"><i class="fa fa-check"></i><b>5</b> GWAS with Continuous Trait</a>
<ul>
<li class="chapter" data-level="5.1" data-path="gwas-with-continuous-trait.html"><a href="gwas-with-continuous-trait.html#manhattan-plot-and-q-q-plot"><i class="fa fa-check"></i><b>5.1</b> Manhattan Plot and Q-Q Plot</a>
<ul>
<li class="chapter" data-level="5.1.1" data-path="gwas-with-continuous-trait.html"><a href="gwas-with-continuous-trait.html#generate-manhattan-plot"><i class="fa fa-check"></i><b>5.1.1</b> Generate manhattan plot</a></li>
<li class="chapter" data-level="5.1.2" data-path="gwas-with-continuous-trait.html"><a href="gwas-with-continuous-trait.html#generate-q-q-plot"><i class="fa fa-check"></i><b>5.1.2</b> Generate Q-Q plot</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="gwas-with-continuous-trait.html"><a href="gwas-with-continuous-trait.html#multiple-test-correction"><i class="fa fa-check"></i><b>5.2</b> Multiple-Test Correction</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="gwas-with-continuous-trait.html"><a href="gwas-with-continuous-trait.html#prepare-r-input-files-in-linux"><i class="fa fa-check"></i><b>5.2.1</b> Prepare R input files (in Linux)</a></li>
<li class="chapter" data-level="5.2.2" data-path="gwas-with-continuous-trait.html"><a href="gwas-with-continuous-trait.html#significant-snps-after-a-multiple-test-correction-of-fdr-using-r"><i class="fa fa-check"></i><b>5.2.2</b> Significant SNPs after a multiple-test correction of FDR (using R)</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="gwas-with-continuous-trait.html"><a href="gwas-with-continuous-trait.html#regional-association-plot"><i class="fa fa-check"></i><b>5.3</b> Regional Association Plot</a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="gwas-with-continuous-trait.html"><a href="gwas-with-continuous-trait.html#obtain-regional-snps-using-r"><i class="fa fa-check"></i><b>5.3.1</b> Obtain regional SNPs (using R)</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="gwas-with-continuous-trait.html"><a href="gwas-with-continuous-trait.html#distribution-of-z-endoxifen-ratio-for-rs8138080"><i class="fa fa-check"></i><b>5.4</b> Distribution of Z-endoxifen ratio for rs8138080</a>
<ul>
<li class="chapter" data-level="5.4.1" data-path="gwas-with-continuous-trait.html"><a href="gwas-with-continuous-trait.html#obtain-phenotype-data-z-endoxifen-ratio-using-r"><i class="fa fa-check"></i><b>5.4.1</b> Obtain phenotype data (Z-endoxifen ratio) (using R)</a></li>
<li class="chapter" data-level="5.4.2" data-path="gwas-with-continuous-trait.html"><a href="gwas-with-continuous-trait.html#obtain-genotype-for-significant-snps-using-r"><i class="fa fa-check"></i><b>5.4.2</b> Obtain genotype for significant SNPs (using R)</a></li>
<li class="chapter" data-level="5.4.3" data-path="gwas-with-continuous-trait.html"><a href="gwas-with-continuous-trait.html#create-box-plot-using-r"><i class="fa fa-check"></i><b>5.4.3</b> Create Box plot (using R)</a></li>
</ul></li>
<li class="chapter" data-level="5.5" data-path="gwas-with-continuous-trait.html"><a href="gwas-with-continuous-trait.html#proportion-of-variation-explained-by-rs8138080"><i class="fa fa-check"></i><b>5.5</b> Proportion of Variation Explained by rs8138080</a>
<ul>
<li class="chapter" data-level="5.5.1" data-path="gwas-with-continuous-trait.html"><a href="gwas-with-continuous-trait.html#recode-genotype-of-significant-snps-using-r"><i class="fa fa-check"></i><b>5.5.1</b> Recode genotype of significant SNPs (using R)</a></li>
<li class="chapter" data-level="5.5.2" data-path="gwas-with-continuous-trait.html"><a href="gwas-with-continuous-trait.html#calculate-marginal-r2-using-r"><i class="fa fa-check"></i><b>5.5.2</b> Calculate marginal <span class="math inline">\(R^2\)</span> (using R)</a></li>
</ul></li>
<li class="chapter" data-level="5.6" data-path="gwas-with-continuous-trait.html"><a href="gwas-with-continuous-trait.html#practice-time"><i class="fa fa-check"></i><b>5.6</b> PRACTICE TIME</a>
<ul>
<li class="chapter" data-level="5.6.1" data-path="gwas-with-continuous-trait.html"><a href="gwas-with-continuous-trait.html#practice-1"><i class="fa fa-check"></i><b>5.6.1</b> Practice 1</a></li>
<li class="chapter" data-level="5.6.2" data-path="gwas-with-continuous-trait.html"><a href="gwas-with-continuous-trait.html#practice-2"><i class="fa fa-check"></i><b>5.6.2</b> Practice 2</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="gwas-with-binary-trait.html"><a href="gwas-with-binary-trait.html"><i class="fa fa-check"></i><b>6</b> GWAS with Binary Trait</a>
<ul>
<li class="chapter" data-level="6.1" data-path="gwas-with-binary-trait.html"><a href="gwas-with-binary-trait.html#data-preparation"><i class="fa fa-check"></i><b>6.1</b> Data Preparation</a></li>
<li class="chapter" data-level="6.2" data-path="gwas-with-binary-trait.html"><a href="gwas-with-binary-trait.html#genome-wide-casecontrol-association-test-in-plink"><i class="fa fa-check"></i><b>6.2</b> Genome-Wide Case/Control Association Test (in PLINK)</a></li>
<li class="chapter" data-level="6.3" data-path="gwas-with-binary-trait.html"><a href="gwas-with-binary-trait.html#manhattan-plot-and-q-q-plot-1"><i class="fa fa-check"></i><b>6.3</b> Manhattan Plot and Q-Q Plot</a></li>
<li class="chapter" data-level="6.4" data-path="gwas-with-binary-trait.html"><a href="gwas-with-binary-trait.html#generate-manhattan-plot-1"><i class="fa fa-check"></i><b>6.4</b> Generate Manhattan plot</a></li>
<li class="chapter" data-level="6.5" data-path="gwas-with-binary-trait.html"><a href="gwas-with-binary-trait.html#generate-q-q-plot-1"><i class="fa fa-check"></i><b>6.5</b> Generate Q-Q plot</a></li>
<li class="chapter" data-level="6.6" data-path="gwas-with-binary-trait.html"><a href="gwas-with-binary-trait.html#multiple-test-correction-1"><i class="fa fa-check"></i><b>6.6</b> Multiple-Test Correction</a></li>
<li class="chapter" data-level="6.7" data-path="gwas-with-binary-trait.html"><a href="gwas-with-binary-trait.html#top-10-snps"><i class="fa fa-check"></i><b>6.7</b> Top 10 SNPs</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="gxg-and-gxe-interactions-power-and-sample-size-calculation.html"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html"><i class="fa fa-check"></i><b>7</b> GxG and GxE Interactions, Power and Sample Size Calculation</a>
<ul>
<li class="chapter" data-level="7.1" data-path="gxg-and-gxe-interactions-power-and-sample-size-calculation.html"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#prepare-files"><i class="fa fa-check"></i><b>7.1</b> Prepare Files</a></li>
<li class="chapter" data-level="7.2" data-path="gxg-and-gxe-interactions-power-and-sample-size-calculation.html"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#test-for-epistasis-gxg"><i class="fa fa-check"></i><b>7.2</b> Test for Epistasis (GxG)</a></li>
<li class="chapter" data-level="7.3" data-path="gxg-and-gxe-interactions-power-and-sample-size-calculation.html"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#gene-environment-interaction-gxe"><i class="fa fa-check"></i><b>7.3</b> Gene-Environment Interaction (GxE)</a></li>
<li class="chapter" data-level="7.4" data-path="gxg-and-gxe-interactions-power-and-sample-size-calculation.html"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#power-and-sample-size-calculation-for-gwas"><i class="fa fa-check"></i><b>7.4</b> Power and sample size calculation for GWAS</a>
<ul>
<li class="chapter" data-level="7.4.1" data-path="gxg-and-gxe-interactions-power-and-sample-size-calculation.html"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#power-calculation-with-binary-outcome"><i class="fa fa-check"></i><b>7.4.1</b> Power calculation with binary outcome:</a></li>
<li class="chapter" data-level="7.4.2" data-path="gxg-and-gxe-interactions-power-and-sample-size-calculation.html"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#sample-size-calculation"><i class="fa fa-check"></i><b>7.4.2</b> Sample size calculation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8" data-path="ldsc-and-gene-set-analysis.html"><a href="ldsc-and-gene-set-analysis.html"><i class="fa fa-check"></i><b>8</b> LDSC and Gene-Set Analysis</a>
<ul>
<li class="chapter" data-level="8.1" data-path="ldsc-and-gene-set-analysis.html"><a href="ldsc-and-gene-set-analysis.html#ldsc-linkage-disequilibrium-score-regression"><i class="fa fa-check"></i><b>8.1</b> LDSC (Linkage Disequilibrium Score Regression)</a></li>
<li class="chapter" data-level="8.2" data-path="ldsc-and-gene-set-analysis.html"><a href="ldsc-and-gene-set-analysis.html#magma-for-gene-level-analysis"><i class="fa fa-check"></i><b>8.2</b> MAGMA (for Gene-Level Analysis)</a>
<ul>
<li class="chapter" data-level="8.2.1" data-path="ldsc-and-gene-set-analysis.html"><a href="ldsc-and-gene-set-analysis.html#prepare-annotation-file"><i class="fa fa-check"></i><b>8.2.1</b> Prepare Annotation File</a></li>
<li class="chapter" data-level="8.2.2" data-path="ldsc-and-gene-set-analysis.html"><a href="ldsc-and-gene-set-analysis.html#part-1-gene-analysis"><i class="fa fa-check"></i><b>8.2.2</b> Part 1: Gene Analysis</a></li>
<li class="chapter" data-level="8.2.3" data-path="ldsc-and-gene-set-analysis.html"><a href="ldsc-and-gene-set-analysis.html#part-2-gene-set-analysis"><i class="fa fa-check"></i><b>8.2.3</b> Part 2: Gene-Set Analysis</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="9" data-path="perform-eqtl-analysis.html"><a href="perform-eqtl-analysis.html"><i class="fa fa-check"></i><b>9</b> Perform eQTL Analysis</a>
<ul>
<li class="chapter" data-level="9.1" data-path="perform-eqtl-analysis.html"><a href="perform-eqtl-analysis.html#filtering-snps-by-maf"><i class="fa fa-check"></i><b>9.1</b> Filtering SNPs by MAF</a></li>
<li class="chapter" data-level="9.2" data-path="perform-eqtl-analysis.html"><a href="perform-eqtl-analysis.html#gene-expression-profiling"><i class="fa fa-check"></i><b>9.2</b> Gene expression profiling</a></li>
<li class="chapter" data-level="9.3" data-path="perform-eqtl-analysis.html"><a href="perform-eqtl-analysis.html#covariates"><i class="fa fa-check"></i><b>9.3</b> Covariates</a></li>
<li class="chapter" data-level="9.4" data-path="perform-eqtl-analysis.html"><a href="perform-eqtl-analysis.html#using-principle-components-as-covariates"><i class="fa fa-check"></i><b>9.4</b> Using principle components as covariates</a></li>
<li class="chapter" data-level="9.5" data-path="perform-eqtl-analysis.html"><a href="perform-eqtl-analysis.html#eqtl-analysis"><i class="fa fa-check"></i><b>9.5</b> eQTL analysis</a>
<ul>
<li class="chapter" data-level="9.5.1" data-path="perform-eqtl-analysis.html"><a href="perform-eqtl-analysis.html#cis-eqtl-and-trans-eqtl-analysis"><i class="fa fa-check"></i><b>9.5.1</b> cis-eQTL and trans-eQTL analysis</a></li>
</ul></li>
<li class="chapter" data-level="9.6" data-path="perform-eqtl-analysis.html"><a href="perform-eqtl-analysis.html#multiple-testing-correction"><i class="fa fa-check"></i><b>9.6</b> Multiple testing correction</a></li>
<li class="chapter" data-level="9.7" data-path="perform-eqtl-analysis.html"><a href="perform-eqtl-analysis.html#gene-regional-manhanttan-plots"><i class="fa fa-check"></i><b>9.7</b> Gene regional Manhanttan plots</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">CSHL course - GWAS Hands-on Analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="gwas-with-binary-trait" class="section level1 hasAnchor" number="6">
<h1><span class="header-section-number">Chapter 6</span> GWAS with Binary Trait<a href="gwas-with-binary-trait.html#gwas-with-binary-trait" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>This section introduces a hands-on practice of a genome-wide association study with binary traits. The reference paper provided a criteria to define the binary groups. The groups were divided based on MR(metabolite ratio) values, such that groups are coded as 1 if MR &lt; 0.0146 and 0 if MR&gt;= 0.0146. We will adopt their citeria in this notebook to define the binary groups. Then we perform a genome-wide case/control association analysis for the binary traits.</p>
<p>Reference paper: Hennig EE, Piatkowska M, Goryca K, Pospiech E, Paziewska A, Karczmarski J, Kluska A, Brewczynska E, Ostrowski J “<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6722498/">Non-CYP2D6 variants selected by GWAS improved the prediction of impaired tamoxifen metabolism in breast cancer patients</a>.” J Clin Med. 2019 Aug; 8(8): 1087. doi: 10.3390/jcm8081087</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb1-1"><a href="gwas-with-binary-trait.html#cb1-1" tabindex="-1"></a><span class="co"># change the path below to where you saved &quot;cshl_gwas&quot; folder</span></span>
<span id="cb1-2"><a href="gwas-with-binary-trait.html#cb1-2" tabindex="-1"></a><span class="co">## run in R Terminal</span></span>
<span id="cb1-3"><a href="gwas-with-binary-trait.html#cb1-3" tabindex="-1"></a><span class="va">mydir</span><span class="op">=</span><span class="st">&quot;/path/to/cshl_gwas/Data/case_data_binary&quot;</span></span>
<span id="cb1-4"><a href="gwas-with-binary-trait.html#cb1-4" tabindex="-1"></a><span class="bu">cd</span> <span class="va">$mydir</span></span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="gwas-with-binary-trait.html#cb2-1" tabindex="-1"></a><span class="co"># R codes- run in R Console</span></span>
<span id="cb2-2"><a href="gwas-with-binary-trait.html#cb2-2" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">&quot;/path/to/cshl_gwas/Data/case_data_binary&quot;</span>)</span>
<span id="cb2-3"><a href="gwas-with-binary-trait.html#cb2-3" tabindex="-1"></a>mydir<span class="ot">&lt;-</span><span class="fu">getwd</span>()</span></code></pre></div>
<div id="data-preparation" class="section level2 hasAnchor" number="6.1">
<h2><span class="header-section-number">6.1</span> Data Preparation<a href="gwas-with-binary-trait.html#data-preparation" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<ol style="list-style-type: decimal">
<li>Phenotype data (using R):</li>
</ol>
<ul>
<li>We convert the continuous trait of <code>Z-endoxifen ratio</code> from the previous notebook to binary traits based on the cut-off value 0.0146:</li>
<li>Values greater than 0.0146 are coded as 1</li>
<li>Values less than 0.0146 are coded as 2</li>
</ul>
<ol start="2" style="list-style-type: decimal">
<li>Genotype data:</li>
</ol>
<ul>
<li>genotype.ped</li>
<li>genotype.map</li>
</ul>
<p>The following code load the imputed phenotype data <code>phenotype0.txt</code> and extract our target phenotype data of <code>Z-endoxifen ratio</code>. Then, we convert the continuous traits into binary traits, and save the binary group variable <code>binary_pheno0</code> to file <code>binary_phenotype0.txt</code>.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="gwas-with-binary-trait.html#cb3-1" tabindex="-1"></a><span class="co"># R codes- run in R Console</span></span>
<span id="cb3-2"><a href="gwas-with-binary-trait.html#cb3-2" tabindex="-1"></a>pheno0<span class="ot">=</span><span class="fu">fread</span>(<span class="st">&quot;phenotype0.txt&quot;</span>,<span class="at">header=</span>F) <span class="co">#read phenotype data</span></span>
<span id="cb3-3"><a href="gwas-with-binary-trait.html#cb3-3" tabindex="-1"></a>binary_pheno0<span class="ot">=</span>pheno0[,<span class="dv">3</span>]</span>
<span id="cb3-4"><a href="gwas-with-binary-trait.html#cb3-4" tabindex="-1"></a>binary_pheno0[binary_pheno0<span class="sc">&gt;=</span><span class="fl">0.0146</span>]<span class="ot">=</span><span class="dv">1</span> <span class="co">#code values greater than 0 as 1</span></span>
<span id="cb3-5"><a href="gwas-with-binary-trait.html#cb3-5" tabindex="-1"></a>binary_pheno0[binary_pheno0<span class="sc">&lt;</span><span class="fl">0.0146</span>]<span class="ot">=</span><span class="dv">2</span> <span class="co">#code values less than 0 as 2</span></span>
<span id="cb3-6"><a href="gwas-with-binary-trait.html#cb3-6" tabindex="-1"></a><span class="fu">write.table</span>(binary_pheno0,<span class="st">&quot;binary_phenotype0.txt&quot;</span>,<span class="at">quote=</span>F,<span class="at">row.names=</span>F,<span class="at">col.names=</span>F,<span class="at">sep=</span><span class="st">&quot; &quot;</span>) <span class="co">#save binary phenotype data</span></span>
<span id="cb3-7"><a href="gwas-with-binary-trait.html#cb3-7" tabindex="-1"></a><span class="fu">head</span>(binary_pheno0)</span></code></pre></div>
<pre><code>##       V3
##    &lt;num&gt;
## 1:     1
## 2:     1
## 3:     2
## 4:     2
## 5:     2
## 6:     2</code></pre>
<p>After the binary group is ready, we use the same procedure in section 5.3 of last notebook to prepare PLINK alternate phenotype file (in Linux):</p>
<p>Note: famID.txt and samID.txt are both required for PLINK step. They are already prepared in last notebook.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb5-1"><a href="gwas-with-binary-trait.html#cb5-1" tabindex="-1"></a><span class="co">## bash codes- run in terminal</span></span>
<span id="cb5-2"><a href="gwas-with-binary-trait.html#cb5-2" tabindex="-1"></a><span class="fu">paste</span> famID.txt samID.txt binary_phenotype0.txt <span class="op">&gt;</span> binary_phenotype.txt</span></code></pre></div>
<p>Take a look at the prepared phenotype data.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="gwas-with-binary-trait.html#cb6-1" tabindex="-1"></a><span class="co"># R codes- run in R Console</span></span>
<span id="cb6-2"><a href="gwas-with-binary-trait.html#cb6-2" tabindex="-1"></a>binary_pheno<span class="ot">&lt;-</span><span class="fu">read.table</span>(<span class="st">&quot;binary_phenotype.txt&quot;</span>)</span>
<span id="cb6-3"><a href="gwas-with-binary-trait.html#cb6-3" tabindex="-1"></a><span class="fu">head</span>(binary_pheno)</span></code></pre></div>
<pre><code>##   V1   V2 V3
## 1  1  3TX  1
## 2  2 23TX  1
## 3  3 41TX  2
## 4  4 54TX  2
## 5  5 76TX  2
## 6  6 94TX  2</code></pre>
</div>
<div id="genome-wide-casecontrol-association-test-in-plink" class="section level2 hasAnchor" number="6.2">
<h2><span class="header-section-number">6.2</span> Genome-Wide Case/Control Association Test (in PLINK)<a href="gwas-with-binary-trait.html#genome-wide-casecontrol-association-test-in-plink" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Similar as last notebook, we will still use PLINK to perform association test. The only change is the phenotype file.</p>
<p>The following code conducts the genome-wide association test by PLINK in bash code. This code will take several minutes to finish.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb8-1"><a href="gwas-with-binary-trait.html#cb8-1" tabindex="-1"></a><span class="co">## # both Mac and Windows can run in Terminal</span></span>
<span id="cb8-2"><a href="gwas-with-binary-trait.html#cb8-2" tabindex="-1"></a><span class="co"># windows users</span></span>
<span id="cb8-3"><a href="gwas-with-binary-trait.html#cb8-3" tabindex="-1"></a><span class="ex">../../Tools/PLINK/plink_win64_20231211/plink.exe</span> <span class="dt">\</span></span>
<span id="cb8-4"><a href="gwas-with-binary-trait.html#cb8-4" tabindex="-1"></a>--noweb <span class="at">--file</span> genotype_1k <span class="at">--assoc</span> <span class="at">--pheno</span> binary_phenotype.txt <span class="at">--adjust</span> <span class="at">--all-pheno</span> <span class="at">--out</span> younameit</span></code></pre></div>
<div class="sourceCode" id="cb9"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb9-1"><a href="gwas-with-binary-trait.html#cb9-1" tabindex="-1"></a><span class="co"># macOS users</span></span>
<span id="cb9-2"><a href="gwas-with-binary-trait.html#cb9-2" tabindex="-1"></a><span class="ex">../../Tools/PLINK/plink_mac_20231211/plink</span> <span class="dt">\</span></span>
<span id="cb9-3"><a href="gwas-with-binary-trait.html#cb9-3" tabindex="-1"></a>--noweb <span class="at">--file</span> genotype_1k <span class="at">--assoc</span> <span class="at">--pheno</span> binary_phenotype.txt <span class="at">--adjust</span> <span class="at">--all-pheno</span> <span class="at">--out</span> younameit</span></code></pre></div>
<pre><code>## PLINK v1.90b7.2 64-bit (11 Dec 2023)           www.cog-genomics.org/plink/1.9/
## (C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3
## Logging to younameit.log.
## Options in effect:
##   --adjust
##   --all-pheno
##   --assoc
##   --file genotype_1k
##   --noweb
##   --out younameit
##   --pheno binary_phenotype.txt
## 
## Note: --noweb has no effect since no web check is implemented yet.
## 16384 MB RAM detected; reserving 8192 MB for main workspace.
## Scanning .ped file... 0%1%1%2%2%3%3%4%4%5%5%6%6%7%7%8%8%9%9%10%10%11%11%12%13%13%14%14%15%15%16%16%17%17%18%18%19%19%20%20%21%21%22%22%23%23%24%24%25%26%26%27%27%28%28%29%29%30%30%31%31%32%32%33%33%34%34%35%35%36%36%37%38%38%39%39%40%40%41%41%42%42%43%43%44%44%45%45%46%46%47%47%48%48%49%49%50%51%51%52%52%53%53%54%54%55%55%56%56%57%57%58%58%59%59%60%60%61%61%62%63%63%64%64%65%65%66%66%67%67%68%68%69%69%70%70%71%71%72%72%73%73%74%74%75%76%76%77%77%78%78%79%79%80%80%81%81%82%82%83%83%84%84%85%85%86%86%87%88%88%89%89%90%90%91%91%92%92%93%93%94%94%95%95%96%96%97%97%98%98%99%100%.ped scan complete (for binary autoconversion).
## Performing single-pass .bed write (1000 variants, 192 people).
## 0%1%2%3%4%5%6%7%8%9%10%11%12%13%14%15%16%17%18%19%20%21%22%23%24%25%26%27%28%29%30%31%32%33%34%35%36%37%38%39%40%41%42%43%44%45%46%47%48%49%50%51%52%53%54%55%56%57%58%59%60%61%62%63%64%65%66%67%68%69%70%71%72%73%74%75%76%77%78%79%80%81%82%83%84%85%86%87%88%89%90%91%92%93%94%--file: younameit-temporary.bed + younameit-temporary.bim +
## younameit-temporary.fam written.
## 1000 variants loaded from .bim file.
## 192 people (0 males, 192 females) loaded from .fam.
## 192 phenotype values present after --pheno.
## Using 1 thread (no multithreaded calculations invoked).
## Before main variant filters, 192 founders and 0 nonfounders present.
## Calculating allele frequencies... 0%1%2%3%4%5%6%7%8%9%10%11%12%13%14%15%16%17%18%19%20%21%22%23%24%25%26%27%28%29%30%31%32%33%34%35%36%37%38%39%40%41%42%43%44%45%46%47%48%49%50%51%52%53%54%55%56%57%58%59%60%61%62%63%64%65%66%67%68%69%70%71%72%73%74%75%76%77%78%79%80%81%82%83%84%85%86%87%88%89%90%91%92%93%94%95%96%97%98%99% done.
## Warning: Nonmissing nonmale Y chromosome genotype(s) present; many commands
## treat these as missing.
## Total genotyping rate is 0.995331.
## 1000 variants and 192 people pass filters and QC.
## Among remaining phenotypes, 97 are cases and 95 are controls.
## 192 phenotype values present after --pheno.
## P1 has 97 cases and 95 controls.
## Writing C/C --assoc report to younameit.P1.assoc ... 0%11%16%20%24%27%43%50%53%56%61%64%68%70%72%73%76%78%79%82%84%85%94%96%99%done.
## --adjust: Genomic inflation est. lambda (based on median chisq) = 1.37231.
## 0%1%2%3%4%5%6%7%8%9%10%11%12%13%14%15%16%17%18%19%20%21%22%23%24%25%26%27%28%29%30%31%32%33%34%35%36%37%38%39%40%41%42%43%44%45%46%47%48%49%50%51%52%53%54%55%56%57%58%59%60%61%62%63%64%65%66%67%68%69%70%71%72%73%74%75%76%77%78%79%80%81%82%83%84%85%86%87%88%89%90%91%92%93%94%95%96%97%98%99%--adjust values (805 variants) written to younameit.P1.assoc.adjusted .</code></pre>
<ol style="list-style-type: decimal">
<li><strong>–noweb</strong> suppress checking for new version,</li>
<li><strong>–file</strong> specifies .ped and .map files,</li>
<li><strong>–assoc</strong> performs case/control or QTL association, you can also perform other tests by changing the options,</li>
<li><strong>–pheno</strong> specifies alternate phenotype,</li>
<li><strong>–adjust</strong> generates a file of adjusted significance values that correct for all tests performed and other metrics,</li>
<li><strong>–all-pheno</strong> performs association test for all phenotypes in binary_phenotype.tix file,</li>
<li><strong>–out</strong> specifies output filename.</li>
</ol>
<p>This will generate the files <strong>younameit.P1.assoc</strong> with fields as follows:</p>
<pre><code> CHR     Chromosome
 
 SNP     SNP ID
 
 BP      Physical position (base-pair)
 
 A1      Minor allele name (based on whole sample)
 
 F_A     Frequency of this allele in cases
 
 F_U     Frequency of this allele in controls
 
 A2      Major allele name
 
 CHISQ   Basic allelic test chi-square (1df)
 
 P       Asymptotic p-value for this test
 
 OR      Estimated odds ratio (for A1, i.e. A2 is reference)     </code></pre>
<p><strong>–adjust</strong> generates the file <strong>younameit.adjust</strong>, which contains the following fields</p>
<pre><code> CHR         Chromosome number
       
 SNP         SNP identifer
       
 UNADJ       Unadjusted p-value
       
 GC          Genomic-control corrected p-values
       
 BONF        Bonferroni single-step adjusted p-values
       
 HOLM        Holm (1979) step-down adjusted p-values
       
 SIDAK_SS    Sidak single-step adjusted p-values
       
 SIDAK_SD    Sidak step-down adjusted p-values
       
 FDR_BH      Benjamini &amp; Hochberg (1995) step-up FDR control
       
 FDR_BY      Benjamini &amp; Yekutieli (2001) step-up FDR control 
 </code></pre>
</div>
<div id="manhattan-plot-and-q-q-plot-1" class="section level2 hasAnchor" number="6.3">
<h2><span class="header-section-number">6.3</span> Manhattan Plot and Q-Q Plot<a href="gwas-with-binary-trait.html#manhattan-plot-and-q-q-plot-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Similar to the GWAS study for continuous traits, we generate Manhattan Plot and Q-Q Plot to visualize the results of GWAS test.</p>
<p>The Manhattan plot is a good way to visualize the resuls of GWAS test. For more details, please refer: <a href="https://en.wikipedia.org/wiki/Manhattan_plot" class="uri">https://en.wikipedia.org/wiki/Manhattan_plot</a></p>
<p>The Q-Q plot for GWAS is another way to visualize the signals. For more details, please refer:<a href="https://genome.sph.umich.edu/wiki/Code_Sample:_Generating_QQ_Plots_in_R" class="uri">https://genome.sph.umich.edu/wiki/Code_Sample:_Generating_QQ_Plots_in_R</a></p>
<p>qqman is an R package for creating Q-Q and manhattan plots from GWAS results. See the reference page <a href="http://www.gettinggeneticsdone.com/2014/05/qqman-r-package-for-qq-and-manhattan-plots-for-gwas-results.html" class="uri">http://www.gettinggeneticsdone.com/2014/05/qqman-r-package-for-qq-and-manhattan-plots-for-gwas-results.html</a>.</p>
<p>The qqman R package assumes you have columns named SNP, CHR, BP, and P, corresponding to the SNP name (rs number), chromosome number, base-pair position, and p-value.</p>
<p>The following code prepares qqman R input files (CHR, SNP, BP, P) (in Linux)</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb13-1"><a href="gwas-with-binary-trait.html#cb13-1" tabindex="-1"></a><span class="co">## bash codes- run in terminal</span></span>
<span id="cb13-2"><a href="gwas-with-binary-trait.html#cb13-2" tabindex="-1"></a><span class="fu">awk</span> <span class="st">&#39;{print $1,$2,$3,$9}&#39;</span> younameit.P1.assoc <span class="op">&gt;</span> P1.assoc</span></code></pre></div>
<p>Here is sample data for P1.assoc</p>
<pre><code>CHR     SNP        BP     P

 1      rs9701055  565433 0.757

 1      rs3094315  752566 0.7004

 1      rs2073813  753541 0.467

 1      rs3131969  754182 0.131</code></pre>
<p>Load association results. Loaded results are in <code>assoc</code>.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="gwas-with-binary-trait.html#cb15-1" tabindex="-1"></a><span class="co"># R codes- run in R Console</span></span>
<span id="cb15-2"><a href="gwas-with-binary-trait.html#cb15-2" tabindex="-1"></a>assoc<span class="ot">=</span><span class="fu">fread</span>(<span class="st">&quot;P1.assoc&quot;</span>, <span class="at">header=</span>T)</span>
<span id="cb15-3"><a href="gwas-with-binary-trait.html#cb15-3" tabindex="-1"></a>assoc<span class="ot">=</span>assoc[(assoc<span class="sc">$</span>CHR<span class="sc">!=</span><span class="dv">0</span>) <span class="sc">&amp;</span> (<span class="sc">!</span><span class="fu">is.na</span>(assoc<span class="sc">$</span>P)),]</span>
<span id="cb15-4"><a href="gwas-with-binary-trait.html#cb15-4" tabindex="-1"></a><span class="fu">head</span>(assoc)</span></code></pre></div>
<pre><code>##      CHR        SNP       BP       P
##    &lt;int&gt;     &lt;char&gt;    &lt;int&gt;   &lt;num&gt;
## 1:     1   rs307354  1264539 0.98080
## 2:     1 rs12121577 11794553 0.56050
## 3:     1  rs4845898 12339619 0.05366
## 4:     1   rs665691 22960354 0.88740
## 5:     1  rs7537875 24695151 0.71760
## 6:     1     rs8876 25687308 0.64030</code></pre>
</div>
<div id="generate-manhattan-plot-1" class="section level2 hasAnchor" number="6.4">
<h2><span class="header-section-number">6.4</span> Generate Manhattan plot<a href="gwas-with-binary-trait.html#generate-manhattan-plot-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We use the same commands as continuous trait to generate a Manhattan plot. The following code will take several minutes to finish.</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="gwas-with-binary-trait.html#cb17-1" tabindex="-1"></a><span class="co"># R codes- run in R Console</span></span>
<span id="cb17-2"><a href="gwas-with-binary-trait.html#cb17-2" tabindex="-1"></a><span class="fu">png</span>(<span class="at">filename=</span><span class="fu">paste0</span>(<span class="st">&quot;Manhattan_Plot_for_binary_z-endoxifen_ratio.png&quot;</span>), <span class="at">type=</span><span class="st">&quot;cairo&quot;</span>)</span></code></pre></div>
<pre><code>## Warning in grSoftVersion(): unable to load shared object &#39;/Library/Frameworks/R.framework/Resources/modules//R_X11.so&#39;:
##   dlopen(/Library/Frameworks/R.framework/Resources/modules//R_X11.so, 0x0006): Library not loaded: /opt/X11/lib/libSM.6.dylib
##   Referenced from: &lt;31EADEB5-0A17-3546-9944-9B3747071FE8&gt; /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/modules/R_X11.so
##   Reason: tried: &#39;/opt/X11/lib/libSM.6.dylib&#39; (no such file), &#39;/System/Volumes/Preboot/Cryptexes/OS/opt/X11/lib/libSM.6.dylib&#39; (no such file), &#39;/opt/X11/lib/libSM.6.dylib&#39; (no such file), &#39;/Library/Frameworks/R.framework/Resources/lib/libSM.6.dylib&#39; (no such file), &#39;/Library/Java/JavaVirtualMachines/jdk-11.0.18+10/Contents/Home/lib/server/libSM.6.dylib&#39; (no such file)</code></pre>
<pre><code>## Warning in cairoVersion(): unable to load shared object &#39;/Library/Frameworks/R.framework/Resources/library/grDevices/libs//cairo.so&#39;:
##   dlopen(/Library/Frameworks/R.framework/Resources/library/grDevices/libs//cairo.so, 0x0006): Library not loaded: /opt/X11/lib/libXrender.1.dylib
##   Referenced from: &lt;63619C6D-FE72-3544-BCEF-9C834A5E39D8&gt; /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/library/grDevices/libs/cairo.so
##   Reason: tried: &#39;/opt/X11/lib/libXrender.1.dylib&#39; (no such file), &#39;/System/Volumes/Preboot/Cryptexes/OS/opt/X11/lib/libXrender.1.dylib&#39; (no such file), &#39;/opt/X11/lib/libXrender.1.dylib&#39; (no such file), &#39;/Library/Frameworks/R.framework/Resources/lib/libXrender.1.dylib&#39; (no such file), &#39;/Library/Java/JavaVirtualMachines/jdk-11.0.18+10/Contents/Home/lib/server/libXrender.1.dylib&#39; (no such file)</code></pre>
<pre><code>## Warning in png(filename =
## paste0(&quot;Manhattan_Plot_for_binary_z-endoxifen_ratio.png&quot;), : failed to load
## cairo DLL</code></pre>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="gwas-with-binary-trait.html#cb21-1" tabindex="-1"></a><span class="fu">manhattan</span>(assoc, <span class="at">col=</span><span class="fu">c</span>(<span class="st">&quot;dimgray&quot;</span>, <span class="st">&quot;darkgoldenrod1&quot;</span>), <span class="at">suggestiveline=</span>F, <span class="at">genomewideline=</span>F)</span></code></pre></div>
<p><img src="cshl_gwas_files/figure-html/unnamed-chunk-11-1.png" width="672" /></p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="gwas-with-binary-trait.html#cb22-1" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code></pre></div>
<pre><code>## null device 
##           1</code></pre>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="gwas-with-binary-trait.html#cb24-1" tabindex="-1"></a><span class="co"># R codes- run in R Console</span></span>
<span id="cb24-2"><a href="gwas-with-binary-trait.html#cb24-2" tabindex="-1"></a>img <span class="ot">&lt;-</span> <span class="fu">readPNG</span>(<span class="st">&quot;Manhattan_Plot_for_binary_z-endoxifen_ratio.png&quot;</span>)</span>
<span id="cb24-3"><a href="gwas-with-binary-trait.html#cb24-3" tabindex="-1"></a>grid<span class="sc">::</span><span class="fu">grid.raster</span>(img)</span></code></pre></div>
<p><img src="cshl_gwas_files/figure-html/mahbiny-1.png" width="672" /></p>
<p>This plot is very similar compared with the Manhattan plot we got for continuous trait. They both contain a very significant SNP in chromosome 22, however, this SNP for binary trait is not so significant as for continuous trait. The y-axis value for continuous trait is about 20.</p>
</div>
<div id="generate-q-q-plot-1" class="section level2 hasAnchor" number="6.5">
<h2><span class="header-section-number">6.5</span> Generate Q-Q plot<a href="gwas-with-binary-trait.html#generate-q-q-plot-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Similarly, a Q-Q plot is generated with qq function to visualize the test results. This code will take several minutes to finish.</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="gwas-with-binary-trait.html#cb25-1" tabindex="-1"></a><span class="co"># R codes- run in R Console</span></span>
<span id="cb25-2"><a href="gwas-with-binary-trait.html#cb25-2" tabindex="-1"></a><span class="fu">png</span>(<span class="at">filename=</span><span class="fu">paste0</span>(<span class="st">&quot;Q-Q_Plot_for_binary_z-endoxifen_ratio.png&quot;</span>), <span class="at">type=</span><span class="st">&quot;cairo&quot;</span>)</span></code></pre></div>
<pre><code>## Warning in png(filename = paste0(&quot;Q-Q_Plot_for_binary_z-endoxifen_ratio.png&quot;),
## : failed to load cairo DLL</code></pre>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="gwas-with-binary-trait.html#cb27-1" tabindex="-1"></a><span class="fu">qq</span>(assoc<span class="sc">$</span>P)</span></code></pre></div>
<p><img src="cshl_gwas_files/figure-html/unnamed-chunk-12-1.png" width="672" /></p>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="gwas-with-binary-trait.html#cb28-1" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code></pre></div>
<pre><code>## null device 
##           1</code></pre>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="gwas-with-binary-trait.html#cb30-1" tabindex="-1"></a><span class="co"># R codes- run in R Console</span></span>
<span id="cb30-2"><a href="gwas-with-binary-trait.html#cb30-2" tabindex="-1"></a>img <span class="ot">&lt;-</span> <span class="fu">readPNG</span>(<span class="st">&quot;Q-Q_Plot_for_binary_z-endoxifen_ratio.png&quot;</span>)</span>
<span id="cb30-3"><a href="gwas-with-binary-trait.html#cb30-3" tabindex="-1"></a>grid<span class="sc">::</span><span class="fu">grid.raster</span>(img)</span></code></pre></div>
<p><img src="cshl_gwas_files/figure-html/mahbiny2-1.png" width="672" /></p>
<p>Q-Q plot still has a long tail, meaning that there are causal polymorphisms.</p>
</div>
<div id="multiple-test-correction-1" class="section level2 hasAnchor" number="6.6">
<h2><span class="header-section-number">6.6</span> Multiple-Test Correction<a href="gwas-with-binary-trait.html#multiple-test-correction-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Again, since we simultaneously test the significance of each SNP, we need to control the overall type-I error rate using false discovery rate (FDR).</p>
<p><strong>Prepare R input files (in Linux)</strong>
The following code extracts the CHR, SNP, UNADJ, FDR_BH from the PLINK output younameit.adjusted. The results are stored in the files: P1.assoc.adjusted.</p>
<div class="sourceCode" id="cb31"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb31-1"><a href="gwas-with-binary-trait.html#cb31-1" tabindex="-1"></a><span class="co">## bash codes- run in terminal</span></span>
<span id="cb31-2"><a href="gwas-with-binary-trait.html#cb31-2" tabindex="-1"></a><span class="fu">awk</span> <span class="st">&#39;{print $1,$2,$3,$9}&#39;</span> younameit.P1.assoc.adjusted <span class="op">&gt;</span> P1.assoc.adjusted</span></code></pre></div>
<p><strong>Significant SNPs after a multiple-test correction of FDR (using R)</strong>
The following code identifies the significant SNPs by the FDR adjusted p-values with cutoff 0.05.</p>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="gwas-with-binary-trait.html#cb32-1" tabindex="-1"></a><span class="co"># R codes- run in R Console</span></span>
<span id="cb32-2"><a href="gwas-with-binary-trait.html#cb32-2" tabindex="-1"></a>assoc.adjusted<span class="ot">=</span><span class="fu">fread</span>(<span class="st">&quot;P1.assoc.adjusted&quot;</span>)<span class="co">#load association results</span></span>
<span id="cb32-3"><a href="gwas-with-binary-trait.html#cb32-3" tabindex="-1"></a>sigidx<span class="ot">=</span><span class="fu">which</span>(assoc.adjusted<span class="sc">$</span>FDR_BH<span class="sc">&lt;</span><span class="fl">0.05</span>) <span class="co">#index of significant SNPs</span></span>
<span id="cb32-4"><a href="gwas-with-binary-trait.html#cb32-4" tabindex="-1"></a>sigSNP<span class="ot">=</span>assoc.adjusted[sigidx,] <span class="co">#extract association results for significant SNPs</span></span>
<span id="cb32-5"><a href="gwas-with-binary-trait.html#cb32-5" tabindex="-1"></a><span class="fu">write.table</span>(sigSNP, <span class="st">&quot;significant_SNPs_for_z-endoxifen_ratio.txt&quot;</span>, <span class="at">row.names=</span>F, <span class="at">col.names=</span>T, <span class="at">quote=</span>F, <span class="at">sep=</span><span class="st">&quot; &quot;</span>) <span class="co">#save results</span></span>
<span id="cb32-6"><a href="gwas-with-binary-trait.html#cb32-6" tabindex="-1"></a></span>
<span id="cb32-7"><a href="gwas-with-binary-trait.html#cb32-7" tabindex="-1"></a><span class="co">#view result</span></span>
<span id="cb32-8"><a href="gwas-with-binary-trait.html#cb32-8" tabindex="-1"></a><span class="fu">dim</span>(sigSNP)</span></code></pre></div>
<pre><code>## [1] 79  4</code></pre>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="gwas-with-binary-trait.html#cb34-1" tabindex="-1"></a><span class="fu">head</span>(sigSNP)</span></code></pre></div>
<pre><code>##      CHR         SNP     UNADJ    FDR_BH
##    &lt;int&gt;      &lt;char&gt;     &lt;num&gt;     &lt;num&gt;
## 1:    22      rs7245 1.172e-15 3.144e-13
## 2:    22   rs2284087 1.172e-15 3.144e-13
## 3:    22   rs5758589 1.172e-15 3.144e-13
## 4:    22  kgp7814563 2.593e-15 5.218e-13
## 5:    22 kgp11017363 1.228e-14 1.647e-12
## 6:    22   kgp137964 1.228e-14 1.647e-12</code></pre>
<p>There are 79 significant SNPs identified. However, remember that we have many SNPs with kgp identifiers, we still need to check if they have matched rs identifiers using UCSC Table Browser.</p>
<p>First we need to match genotype information such as position.</p>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="gwas-with-binary-trait.html#cb36-1" tabindex="-1"></a><span class="co"># R codes- run in R Console</span></span>
<span id="cb36-2"><a href="gwas-with-binary-trait.html#cb36-2" tabindex="-1"></a>map<span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="st">&quot;genotype_1k.map&quot;</span>,<span class="at">header=</span>F)</span>
<span id="cb36-3"><a href="gwas-with-binary-trait.html#cb36-3" tabindex="-1"></a><span class="fu">names</span>(map) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;CHR&quot;</span>,<span class="st">&quot;SNP&quot;</span>,<span class="st">&quot;GP&quot;</span>,<span class="st">&quot;BP&quot;</span>) <span class="co"># assign header</span></span>
<span id="cb36-4"><a href="gwas-with-binary-trait.html#cb36-4" tabindex="-1"></a>zr_result <span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="st">&quot;significant_SNPs_for_z-endoxifen_ratio.txt&quot;</span>)</span>
<span id="cb36-5"><a href="gwas-with-binary-trait.html#cb36-5" tabindex="-1"></a>idx <span class="ot">&lt;-</span> <span class="fu">match</span>(zr_result<span class="sc">$</span>SNP, map<span class="sc">$</span>SNP)<span class="co">#index of map that match with SNP in result</span></span>
<span id="cb36-6"><a href="gwas-with-binary-trait.html#cb36-6" tabindex="-1"></a>zr_result<span class="sc">$</span>BP <span class="ot">&lt;-</span> map<span class="sc">$</span>BP[idx]</span>
<span id="cb36-7"><a href="gwas-with-binary-trait.html#cb36-7" tabindex="-1"></a><span class="fu">head</span>(zr_result)</span></code></pre></div>
<pre><code>##      CHR         SNP     UNADJ    FDR_BH       BP
##    &lt;int&gt;      &lt;char&gt;     &lt;num&gt;     &lt;num&gt;    &lt;int&gt;
## 1:    22      rs7245 1.172e-15 3.144e-13 42481849
## 2:    22   rs2284087 1.172e-15 3.144e-13 42485671
## 3:    22   rs5758589 1.172e-15 3.144e-13 42518382
## 4:    22  kgp7814563 2.593e-15 5.218e-13 42465788
## 5:    22 kgp11017363 1.228e-14 1.647e-12 42348790
## 6:    22   kgp137964 1.228e-14 1.647e-12 42359204</code></pre>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb38-1"><a href="gwas-with-binary-trait.html#cb38-1" tabindex="-1"></a><span class="fu">dim</span>(zr_result) <span class="co">#get number of significant SNPs</span></span></code></pre></div>
<pre><code>## [1] 79  5</code></pre>
<p>Next, we will extract the SNPs with kgp identifier, and prepare the table that will be submitted to UCSC.</p>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb40-1"><a href="gwas-with-binary-trait.html#cb40-1" tabindex="-1"></a><span class="co"># R codes- run in R Console</span></span>
<span id="cb40-2"><a href="gwas-with-binary-trait.html#cb40-2" tabindex="-1"></a>kgp_snps <span class="ot">&lt;-</span> zr_result[<span class="fu">grepl</span>(<span class="st">&quot;kgp&quot;</span>,zr_result<span class="sc">$</span>SNP),]</span>
<span id="cb40-3"><a href="gwas-with-binary-trait.html#cb40-3" tabindex="-1"></a><span class="fu">head</span>(kgp_snps)</span></code></pre></div>
<pre><code>##      CHR         SNP     UNADJ    FDR_BH       BP
##    &lt;int&gt;      &lt;char&gt;     &lt;num&gt;     &lt;num&gt;    &lt;int&gt;
## 1:    22  kgp7814563 2.593e-15 5.218e-13 42465788
## 2:    22 kgp11017363 1.228e-14 1.647e-12 42348790
## 3:    22   kgp137964 1.228e-14 1.647e-12 42359204
## 4:    22  kgp9081934 3.649e-14 2.558e-12 42389839
## 5:    22 kgp12016588 3.649e-14 2.558e-12 42395960
## 6:    22 kgp11007658 3.649e-14 2.558e-12 42401749</code></pre>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb42-1"><a href="gwas-with-binary-trait.html#cb42-1" tabindex="-1"></a>kgp_snps<span class="sc">$</span>CHR<span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;chr&quot;</span>,kgp_snps<span class="sc">$</span>CHR)</span>
<span id="cb42-2"><a href="gwas-with-binary-trait.html#cb42-2" tabindex="-1"></a>kgp_snps<span class="sc">$</span>BP_1<span class="ot">&lt;-</span> kgp_snps<span class="sc">$</span>BP<span class="dv">-1</span></span>
<span id="cb42-3"><a href="gwas-with-binary-trait.html#cb42-3" tabindex="-1"></a><span class="fu">fwrite</span>(kgp_snps[,<span class="fu">c</span>(<span class="st">&quot;CHR&quot;</span>,<span class="st">&quot;BP_1&quot;</span>,<span class="st">&quot;BP&quot;</span>,<span class="st">&quot;SNP&quot;</span>)], <span class="st">&quot;kgp_snps.txt&quot;</span>, <span class="at">sep=</span><span class="st">&quot; &quot;</span>, <span class="at">quote=</span>F, <span class="at">row.name=</span>F, <span class="at">col.name=</span>F)</span></code></pre></div>
<p>After matching, we again got only 2 matched SNPs, same as in continuous study.</p>
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb43-1"><a href="gwas-with-binary-trait.html#cb43-1" tabindex="-1"></a><span class="co"># R codes- run in R Console</span></span>
<span id="cb43-2"><a href="gwas-with-binary-trait.html#cb43-2" tabindex="-1"></a>kgp_rs<span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="st">&quot;kgp_rs.txt&quot;</span>,<span class="at">header=</span>F)</span>
<span id="cb43-3"><a href="gwas-with-binary-trait.html#cb43-3" tabindex="-1"></a><span class="fu">names</span>(kgp_rs)<span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;CHR&quot;</span>,<span class="st">&quot;BP_1&quot;</span>,<span class="st">&quot;BP&quot;</span>,<span class="st">&quot;SNP&quot;</span>)<span class="co"># assign header</span></span>
<span id="cb43-4"><a href="gwas-with-binary-trait.html#cb43-4" tabindex="-1"></a>idx <span class="ot">&lt;-</span> <span class="fu">match</span>(kgp_rs<span class="sc">$</span>BP, kgp_snps<span class="sc">$</span>BP)<span class="co"># get index of these matched SNPs in results</span></span>
<span id="cb43-5"><a href="gwas-with-binary-trait.html#cb43-5" tabindex="-1"></a>kgp_match<span class="ot">&lt;-</span> kgp_snps[idx[<span class="sc">!</span><span class="fu">is.na</span>(idx)],]</span>
<span id="cb43-6"><a href="gwas-with-binary-trait.html#cb43-6" tabindex="-1"></a>kgp_match<span class="sc">$</span>SNP<span class="ot">&lt;-</span> kgp_rs<span class="sc">$</span>SNP[<span class="sc">!</span><span class="fu">is.na</span>(idx)] <span class="co"># replace kgp identifier with rs identifier</span></span>
<span id="cb43-7"><a href="gwas-with-binary-trait.html#cb43-7" tabindex="-1"></a>kgp_match</span></code></pre></div>
<pre><code>##       CHR         SNP     UNADJ    FDR_BH       BP     BP_1
##    &lt;char&gt;      &lt;char&gt;     &lt;num&gt;     &lt;num&gt;    &lt;int&gt;    &lt;num&gt;
## 1:  chr22  rs60133761 3.508e-11 6.567e-10 42668504 42668503
## 2:  chr22 rs140061401 3.649e-14 2.558e-12 42395960 42395959</code></pre>
<p>After combining with other rs identified SNPs, we got a total of 29 significant SNPs after multiple test correction.</p>
<div class="sourceCode" id="cb45"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb45-1"><a href="gwas-with-binary-trait.html#cb45-1" tabindex="-1"></a><span class="co"># R codes- run in R Console</span></span>
<span id="cb45-2"><a href="gwas-with-binary-trait.html#cb45-2" tabindex="-1"></a>rs_snps <span class="ot">&lt;-</span> zr_result[<span class="fu">grepl</span>(<span class="st">&quot;rs&quot;</span>,zr_result<span class="sc">$</span>SNP),]</span>
<span id="cb45-3"><a href="gwas-with-binary-trait.html#cb45-3" tabindex="-1"></a>kgp_match<span class="sc">$</span>CHR <span class="ot">&lt;-</span> <span class="st">&quot;22&quot;</span></span>
<span id="cb45-4"><a href="gwas-with-binary-trait.html#cb45-4" tabindex="-1"></a>rs_snps <span class="ot">&lt;-</span> <span class="fu">rbind</span>(rs_snps, kgp_match[,<span class="sc">-</span><span class="dv">6</span>])</span>
<span id="cb45-5"><a href="gwas-with-binary-trait.html#cb45-5" tabindex="-1"></a><span class="fu">fwrite</span>(rs_snps, <span class="st">&quot;significant_snps_with_rs_for_z-endoxifen-ratio.txt&quot;</span>, <span class="at">sep=</span><span class="st">&quot; &quot;</span>, <span class="at">quote=</span>F, <span class="at">row.name=</span>F, <span class="at">col.name=</span>T)</span>
<span id="cb45-6"><a href="gwas-with-binary-trait.html#cb45-6" tabindex="-1"></a><span class="fu">dim</span>(rs_snps)</span></code></pre></div>
<pre><code>## [1] 29  5</code></pre>
<div class="sourceCode" id="cb47"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb47-1"><a href="gwas-with-binary-trait.html#cb47-1" tabindex="-1"></a>rs_snps</span></code></pre></div>
<pre><code>##        CHR         SNP     UNADJ    FDR_BH       BP
##     &lt;char&gt;      &lt;char&gt;     &lt;num&gt;     &lt;num&gt;    &lt;int&gt;
##  1:     22      rs7245 1.172e-15 3.144e-13 42481849
##  2:     22   rs2284087 1.172e-15 3.144e-13 42485671
##  3:     22   rs5758589 1.172e-15 3.144e-13 42518382
##  4:     22   rs1063392 3.649e-14 2.558e-12 42454950
##  5:     22    rs133335 3.814e-14 2.558e-12 42416056
##  6:     22   rs5751247 2.015e-12 1.014e-10 42633054
##  7:     22   rs8138080 2.271e-12 1.076e-10 42396371
##  8:     22   rs5751191 3.974e-12 1.529e-10 42370991
##  9:     22   rs1062753 3.988e-12 1.529e-10 42392811
## 10:     22   rs4467371 5.090e-12 1.782e-10 42541328
## 11:     22    rs932376 6.210e-12 1.993e-10 42614362
## 12:     22   rs2011944 6.932e-12 1.993e-10 42569999
## 13:     22   rs4453786 9.750e-12 2.707e-10 42563308
## 14:     22   rs5751255 1.064e-11 2.856e-10 42648408
## 15:     22   rs5751221 2.500e-11 5.296e-10 42516366
## 16:     22   rs5758651 3.508e-11 6.567e-10 42609148
## 17:     22   rs1052717 6.599e-11 1.181e-09 42281429
## 18:     22   rs5751239 1.266e-10 2.080e-09 42592239
## 19:     22    rs133280 4.811e-10 7.172e-09 42255525
## 20:     22    rs134871 3.272e-09 4.704e-08 42652716
## 21:     22   rs8140869 6.890e-08 8.946e-07 42344297
## 22:     22    rs742086 1.797e-07 2.260e-06 42552253
## 23:     22   rs5758511 2.643e-07 3.273e-06 42336172
## 24:     22   rs6002673 4.638e-06 5.656e-05 42691296
## 25:     22   rs7285782 5.222e-05 6.182e-04 42225018
## 26:     22  rs17365991 8.325e-04 9.180e-03 41790179
## 27:     22   rs9607821 2.249e-03 2.321e-02 41983938
## 28:     22  rs60133761 3.508e-11 6.567e-10 42668504
## 29:     22 rs140061401 3.649e-14 2.558e-12 42395960
##        CHR         SNP     UNADJ    FDR_BH       BP</code></pre>
</div>
<div id="top-10-snps" class="section level2 hasAnchor" number="6.7">
<h2><span class="header-section-number">6.7</span> Top 10 SNPs<a href="gwas-with-binary-trait.html#top-10-snps" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Here, we present the Top 10 SNPs and p-values for both binary and continuous Z-endoxifen ratio trait.</p>
<div class="sourceCode" id="cb49"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb49-1"><a href="gwas-with-binary-trait.html#cb49-1" tabindex="-1"></a><span class="co"># R codes- run in R Console</span></span>
<span id="cb49-2"><a href="gwas-with-binary-trait.html#cb49-2" tabindex="-1"></a><span class="co"># binary trait</span></span>
<span id="cb49-3"><a href="gwas-with-binary-trait.html#cb49-3" tabindex="-1"></a>assoc_binary<span class="ot">=</span><span class="fu">read.table</span>(<span class="st">&quot;significant_snps_with_rs_for_z-endoxifen-ratio.txt&quot;</span>,<span class="at">header=</span>T) <span class="co">#load association results for binary_r_met</span></span>
<span id="cb49-4"><a href="gwas-with-binary-trait.html#cb49-4" tabindex="-1"></a>ordered_assoc_binary<span class="ot">=</span>assoc_binary[<span class="fu">order</span>(assoc_binary<span class="sc">$</span>FDR_BH),] <span class="co">#order association results based on p-value</span></span>
<span id="cb49-5"><a href="gwas-with-binary-trait.html#cb49-5" tabindex="-1"></a><span class="fu">head</span>(ordered_assoc_binary,<span class="dv">10</span>) <span class="co">#association results for the top 10 SNPs with the 10 smallest p-values</span></span></code></pre></div>
<pre><code>##    CHR         SNP     UNADJ    FDR_BH       BP
## 1   22      rs7245 1.172e-15 3.144e-13 42481849
## 2   22   rs2284087 1.172e-15 3.144e-13 42485671
## 3   22   rs5758589 1.172e-15 3.144e-13 42518382
## 4   22   rs1063392 3.649e-14 2.558e-12 42454950
## 5   22    rs133335 3.814e-14 2.558e-12 42416056
## 29  22 rs140061401 3.649e-14 2.558e-12 42395960
## 6   22   rs5751247 2.015e-12 1.014e-10 42633054
## 7   22   rs8138080 2.271e-12 1.076e-10 42396371
## 8   22   rs5751191 3.974e-12 1.529e-10 42370991
## 9   22   rs1062753 3.988e-12 1.529e-10 42392811</code></pre>
<div class="sourceCode" id="cb51"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb51-1"><a href="gwas-with-binary-trait.html#cb51-1" tabindex="-1"></a><span class="co"># R codes- run in R Console</span></span>
<span id="cb51-2"><a href="gwas-with-binary-trait.html#cb51-2" tabindex="-1"></a><span class="co"># continuous trait</span></span>
<span id="cb51-3"><a href="gwas-with-binary-trait.html#cb51-3" tabindex="-1"></a>assoc_cont<span class="ot">=</span><span class="fu">read.table</span>(<span class="st">&quot;../case_data/significant_snps_with_rs_for_z-endoxifen-ratio.txt&quot;</span>,<span class="at">header=</span>T) <span class="co">#load association results for binary_r_met</span></span>
<span id="cb51-4"><a href="gwas-with-binary-trait.html#cb51-4" tabindex="-1"></a>ordered_assoc_cont<span class="ot">=</span>assoc_cont[<span class="fu">order</span>(assoc_cont<span class="sc">$</span>FDR_BH),] <span class="co">#order association results based on p-value</span></span>
<span id="cb51-5"><a href="gwas-with-binary-trait.html#cb51-5" tabindex="-1"></a><span class="fu">head</span>(ordered_assoc_cont,<span class="dv">10</span>) <span class="co">#association results for the top 10 SNPs with the 10 smallest p-values</span></span></code></pre></div>
<pre><code>##    CHR       SNP     UNADJ    FDR_BH       BP
## 1   22 rs8138080 3.318e-20 1.335e-17 42396371
## 2   22 rs1062753 5.643e-19 9.347e-17 42392811
## 3   22 rs4467371 1.365e-18 1.570e-16 42541328
## 4   22    rs7245 4.102e-18 2.540e-16 42481849
## 5   22 rs2284087 4.102e-18 2.540e-16 42485671
## 6   22 rs5758589 4.102e-18 2.540e-16 42518382
## 7   22 rs5751247 6.923e-18 3.981e-16 42633054
## 8   22 rs2011944 8.301e-18 4.176e-16 42569999
## 9   22 rs4453786 1.068e-17 5.059e-16 42563308
## 10  22 rs5751221 3.812e-17 1.534e-15 42516366</code></pre>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="gwas-with-continuous-trait.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/05-gwas_part3.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["cshl_gwas.pdf", "cshl_gwas.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
