<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 5 GWAS with Continuous Trait | CSHL course - GWAS Hands-on Analysis</title>
  <meta name="description" content="This is the course material for the CSHL hands-on session in GWAS and eQTL." />
  <meta name="generator" content="bookdown 0.41 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 5 GWAS with Continuous Trait | CSHL course - GWAS Hands-on Analysis" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This is the course material for the CSHL hands-on session in GWAS and eQTL." />
  <meta name="github-repo" content="rstudio/cshl_gwas" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 5 GWAS with Continuous Trait | CSHL course - GWAS Hands-on Analysis" />
  
  <meta name="twitter:description" content="This is the course material for the CSHL hands-on session in GWAS and eQTL." />
  

<meta name="author" content="Min Zhang, Danni Liu- University of California, Irvine; Purdue University" />


<meta name="date" content="2025-06-09" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="gwas-case-study-data-preparation.html"/>
<link rel="next" href="gwas-with-binary-trait.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">CSHL Course- GWAS and EQTL Hands-on Analysis</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Welcome to the GWAS and eQTL Workshop</a>
<ul>
<li class="chapter" data-level="1.0.1" data-path="index.html"><a href="index.html#overview"><i class="fa fa-check"></i><b>1.0.1</b> Overview</a></li>
<li class="chapter" data-level="1.0.2" data-path="index.html"><a href="index.html#folder-structure"><i class="fa fa-check"></i><b>1.0.2</b> Folder structure:</a></li>
<li class="chapter" data-level="1.0.3" data-path="index.html"><a href="index.html#programming-languages"><i class="fa fa-check"></i><b>1.0.3</b> Programming Languages:</a></li>
<li class="chapter" data-level="1.0.4" data-path="index.html"><a href="index.html#required-gwas-tools"><i class="fa fa-check"></i><b>1.0.4</b> Required GWAS Tools</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="an-overview-of-regression-with-toy-examples.html"><a href="an-overview-of-regression-with-toy-examples.html"><i class="fa fa-check"></i><b>2</b> An Overview of Regression with Toy Examples</a>
<ul>
<li class="chapter" data-level="2.1" data-path="an-overview-of-regression-with-toy-examples.html"><a href="an-overview-of-regression-with-toy-examples.html#environment-setting-and-packages"><i class="fa fa-check"></i><b>2.1</b> Environment Setting and Packages</a></li>
<li class="chapter" data-level="2.2" data-path="an-overview-of-regression-with-toy-examples.html"><a href="an-overview-of-regression-with-toy-examples.html#toy-example-data"><i class="fa fa-check"></i><b>2.2</b> Toy Example Data</a></li>
<li class="chapter" data-level="2.3" data-path="an-overview-of-regression-with-toy-examples.html"><a href="an-overview-of-regression-with-toy-examples.html#logistic-regression"><i class="fa fa-check"></i><b>2.3</b> Logistic Regression</a></li>
<li class="chapter" data-level="2.4" data-path="an-overview-of-regression-with-toy-examples.html"><a href="an-overview-of-regression-with-toy-examples.html#roc-curve-and-auc-for-hoxb13"><i class="fa fa-check"></i><b>2.4</b> ROC curve and AUC for HOXB13</a></li>
<li class="chapter" data-level="2.5" data-path="an-overview-of-regression-with-toy-examples.html"><a href="an-overview-of-regression-with-toy-examples.html#practice"><i class="fa fa-check"></i><b>2.5</b> Practice</a></li>
<li class="chapter" data-level="2.6" data-path="an-overview-of-regression-with-toy-examples.html"><a href="an-overview-of-regression-with-toy-examples.html#comparison-of-hoxb13-roc-curve-and-nhlh1-roc-curve"><i class="fa fa-check"></i><b>2.6</b> Comparison of HOXB13 ROC curve and NHLH1 ROC curve</a></li>
<li class="chapter" data-level="2.7" data-path="an-overview-of-regression-with-toy-examples.html"><a href="an-overview-of-regression-with-toy-examples.html#linear-regression"><i class="fa fa-check"></i><b>2.7</b> Linear Regression</a>
<ul>
<li class="chapter" data-level="2.7.1" data-path="an-overview-of-regression-with-toy-examples.html"><a href="an-overview-of-regression-with-toy-examples.html#linear-regression-of-hoxb13-vs.-il17br"><i class="fa fa-check"></i><b>2.7.1</b> Linear regression of HOXB13 vs. IL17BR</a></li>
</ul></li>
<li class="chapter" data-level="2.8" data-path="an-overview-of-regression-with-toy-examples.html"><a href="an-overview-of-regression-with-toy-examples.html#variable-selectionforward-backward-and-stepwise-selection"><i class="fa fa-check"></i><b>2.8</b> Variable selection–Forward, backward and stepwise selection</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html"><i class="fa fa-check"></i><b>3</b> GWAS Tools and Data Preprocessing Workflow with Toy Example</a>
<ul>
<li class="chapter" data-level="3.1" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#environment-setting"><i class="fa fa-check"></i><b>3.1</b> Environment Setting</a></li>
<li class="chapter" data-level="3.2" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#gwas-tools"><i class="fa fa-check"></i><b>3.2</b> GWAS Tools</a></li>
<li class="chapter" data-level="3.3" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#quality-control"><i class="fa fa-check"></i><b>3.3</b> Quality Control</a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#an-example-of-summary-statistics"><i class="fa fa-check"></i><b>3.3.1</b> An Example of Summary Statistics</a></li>
<li class="chapter" data-level="3.3.2" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#an-example-of-inclusion-thresholds-for-qc"><i class="fa fa-check"></i><b>3.3.2</b> An Example of Inclusion Thresholds for QC</a></li>
<li class="chapter" data-level="3.3.3" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#remove-closely-related-individuals"><i class="fa fa-check"></i><b>3.3.3</b> Remove Closely Related Individuals</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#an-imputation-example"><i class="fa fa-check"></i><b>3.4</b> An Imputation Example</a></li>
<li class="chapter" data-level="3.5" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#population-stratification-correction-using-plink"><i class="fa fa-check"></i><b>3.5</b> Population Stratification Correction Using PLINK</a>
<ul>
<li class="chapter" data-level="3.5.1" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#pca-on-whole-genome-data-using-plink"><i class="fa fa-check"></i><b>3.5.1</b> PCA on whole genome data using PLINK</a></li>
<li class="chapter" data-level="3.5.2" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#test-the-significance-of-principle-components-pcs"><i class="fa fa-check"></i><b>3.5.2</b> Test the Significance of Principle Components (PCs)</a></li>
</ul></li>
<li class="chapter" data-level="3.6" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#simple-association-test"><i class="fa fa-check"></i><b>3.6</b> Simple Association Test</a>
<ul>
<li class="chapter" data-level="3.6.1" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#quantitative-trait-association-test-for-continuous-trait"><i class="fa fa-check"></i><b>3.6.1</b> Quantitative Trait Association Test for Continuous Trait</a></li>
<li class="chapter" data-level="3.6.2" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#casecontrol-asssociation-test-for-binary-trait"><i class="fa fa-check"></i><b>3.6.2</b> Case/Control Asssociation Test for Binary Trait</a></li>
</ul></li>
<li class="chapter" data-level="3.7" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#quantitative-trait-association-test-for-continuous-trait-with-covariates"><i class="fa fa-check"></i><b>3.7</b> Quantitative Trait Association Test for Continuous Trait with Covariates</a>
<ul>
<li class="chapter" data-level="3.7.1" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#covariates-adjustments-using-r"><i class="fa fa-check"></i><b>3.7.1</b> Covariates adjustments (using R):</a></li>
<li class="chapter" data-level="3.7.2" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#association-test-using-plink"><i class="fa fa-check"></i><b>3.7.2</b> Association test using plink</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="gwas-case-study-data-preparation.html"><a href="gwas-case-study-data-preparation.html"><i class="fa fa-check"></i><b>4</b> GWAS Case Study-Data Preparation</a>
<ul>
<li class="chapter" data-level="4.1" data-path="gwas-case-study-data-preparation.html"><a href="gwas-case-study-data-preparation.html#environment-setting-and-packages-1"><i class="fa fa-check"></i><b>4.1</b> Environment setting and packages</a></li>
<li class="chapter" data-level="4.2" data-path="gwas-case-study-data-preparation.html"><a href="gwas-case-study-data-preparation.html#accessing-data"><i class="fa fa-check"></i><b>4.2</b> Accessing data</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="gwas-case-study-data-preparation.html"><a href="gwas-case-study-data-preparation.html#data-availability"><i class="fa fa-check"></i><b>4.2.1</b> Data availability</a></li>
<li class="chapter" data-level="4.2.2" data-path="gwas-case-study-data-preparation.html"><a href="gwas-case-study-data-preparation.html#data-files-documentation"><i class="fa fa-check"></i><b>4.2.2</b> Data files documentation</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="gwas-case-study-data-preparation.html"><a href="gwas-case-study-data-preparation.html#preprocessing-phenotype-data"><i class="fa fa-check"></i><b>4.3</b> Preprocessing Phenotype Data</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="gwas-case-study-data-preparation.html"><a href="gwas-case-study-data-preparation.html#extracting-phenotype-data"><i class="fa fa-check"></i><b>4.3.1</b> Extracting phenotype data</a></li>
<li class="chapter" data-level="4.3.2" data-path="gwas-case-study-data-preparation.html"><a href="gwas-case-study-data-preparation.html#summary-of-phenotypes-data"><i class="fa fa-check"></i><b>4.3.2</b> Summary of Phenotypes Data</a></li>
<li class="chapter" data-level="4.3.3" data-path="gwas-case-study-data-preparation.html"><a href="gwas-case-study-data-preparation.html#histogram-of-each-trait"><i class="fa fa-check"></i><b>4.3.3</b> Histogram of each trait</a></li>
<li class="chapter" data-level="4.3.4" data-path="gwas-case-study-data-preparation.html"><a href="gwas-case-study-data-preparation.html#handling-the-missing-values-in-the-phenotypes-and-covariates"><i class="fa fa-check"></i><b>4.3.4</b> Handling the missing values in the phenotypes and covariates</a></li>
<li class="chapter" data-level="4.3.5" data-path="gwas-case-study-data-preparation.html"><a href="gwas-case-study-data-preparation.html#checking-normality-assumptions-for-each-trait"><i class="fa fa-check"></i><b>4.3.5</b> Checking normality assumptions for each trait</a></li>
<li class="chapter" data-level="4.3.6" data-path="gwas-case-study-data-preparation.html"><a href="gwas-case-study-data-preparation.html#normalize-phenotype-data"><i class="fa fa-check"></i><b>4.3.6</b> Normalize phenotype data</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="gwas-case-study-data-preparation.html"><a href="gwas-case-study-data-preparation.html#covariates-adjustment"><i class="fa fa-check"></i><b>4.4</b> Covariates Adjustment</a></li>
<li class="chapter" data-level="4.5" data-path="gwas-case-study-data-preparation.html"><a href="gwas-case-study-data-preparation.html#prepare-plink-data"><i class="fa fa-check"></i><b>4.5</b> Prepare PLINK data</a>
<ul>
<li class="chapter" data-level="4.5.1" data-path="gwas-case-study-data-preparation.html"><a href="gwas-case-study-data-preparation.html#step-1-prepare-plink-ped-file"><i class="fa fa-check"></i><b>4.5.1</b> Step 1: prepare PLINK ped file</a></li>
<li class="chapter" data-level="4.5.2" data-path="gwas-case-study-data-preparation.html"><a href="gwas-case-study-data-preparation.html#step-2-prepare-plink-map-file"><i class="fa fa-check"></i><b>4.5.2</b> Step 2: prepare PLINK map file</a></li>
<li class="chapter" data-level="4.5.3" data-path="gwas-case-study-data-preparation.html"><a href="gwas-case-study-data-preparation.html#step-3-prepare-plink-alternate-phenotype-file-in-linux"><i class="fa fa-check"></i><b>4.5.3</b> Step 3 : prepare PLINK alternate phenotype file (in Linux)</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="gwas-with-continuous-trait.html"><a href="gwas-with-continuous-trait.html"><i class="fa fa-check"></i><b>5</b> GWAS with Continuous Trait</a>
<ul>
<li class="chapter" data-level="5.1" data-path="gwas-with-continuous-trait.html"><a href="gwas-with-continuous-trait.html#manhattan-plot-and-q-q-plot"><i class="fa fa-check"></i><b>5.1</b> Manhattan Plot and Q-Q Plot</a>
<ul>
<li class="chapter" data-level="5.1.1" data-path="gwas-with-continuous-trait.html"><a href="gwas-with-continuous-trait.html#generate-manhattan-plot"><i class="fa fa-check"></i><b>5.1.1</b> Generate manhattan plot</a></li>
<li class="chapter" data-level="5.1.2" data-path="gwas-with-continuous-trait.html"><a href="gwas-with-continuous-trait.html#generate-q-q-plot"><i class="fa fa-check"></i><b>5.1.2</b> Generate Q-Q plot</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="gwas-with-continuous-trait.html"><a href="gwas-with-continuous-trait.html#multiple-test-correction"><i class="fa fa-check"></i><b>5.2</b> Multiple-Test Correction</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="gwas-with-continuous-trait.html"><a href="gwas-with-continuous-trait.html#prepare-r-input-files-in-linux"><i class="fa fa-check"></i><b>5.2.1</b> Prepare R input files (in Linux)</a></li>
<li class="chapter" data-level="5.2.2" data-path="gwas-with-continuous-trait.html"><a href="gwas-with-continuous-trait.html#significant-snps-after-a-multiple-test-correction-of-fdr-using-r"><i class="fa fa-check"></i><b>5.2.2</b> Significant SNPs after a multiple-test correction of FDR (using R)</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="gwas-with-continuous-trait.html"><a href="gwas-with-continuous-trait.html#regional-association-plot"><i class="fa fa-check"></i><b>5.3</b> Regional Association Plot</a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="gwas-with-continuous-trait.html"><a href="gwas-with-continuous-trait.html#obtain-regional-snps-using-r"><i class="fa fa-check"></i><b>5.3.1</b> Obtain regional SNPs (using R)</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="gwas-with-continuous-trait.html"><a href="gwas-with-continuous-trait.html#distribution-of-z-endoxifen-ratio-for-rs8138080"><i class="fa fa-check"></i><b>5.4</b> Distribution of Z-endoxifen ratio for rs8138080</a>
<ul>
<li class="chapter" data-level="5.4.1" data-path="gwas-with-continuous-trait.html"><a href="gwas-with-continuous-trait.html#obtain-phenotype-data-z-endoxifen-ratio-using-r"><i class="fa fa-check"></i><b>5.4.1</b> Obtain phenotype data (Z-endoxifen ratio) (using R)</a></li>
<li class="chapter" data-level="5.4.2" data-path="gwas-with-continuous-trait.html"><a href="gwas-with-continuous-trait.html#obtain-genotype-for-significant-snps-using-r"><i class="fa fa-check"></i><b>5.4.2</b> Obtain genotype for significant SNPs (using R)</a></li>
<li class="chapter" data-level="5.4.3" data-path="gwas-with-continuous-trait.html"><a href="gwas-with-continuous-trait.html#create-box-plot-using-r"><i class="fa fa-check"></i><b>5.4.3</b> Create Box plot (using R)</a></li>
</ul></li>
<li class="chapter" data-level="5.5" data-path="gwas-with-continuous-trait.html"><a href="gwas-with-continuous-trait.html#proportion-of-variation-explained-by-rs8138080"><i class="fa fa-check"></i><b>5.5</b> Proportion of Variation Explained by rs8138080</a>
<ul>
<li class="chapter" data-level="5.5.1" data-path="gwas-with-continuous-trait.html"><a href="gwas-with-continuous-trait.html#recode-genotype-of-significant-snps-using-r"><i class="fa fa-check"></i><b>5.5.1</b> Recode genotype of significant SNPs (using R)</a></li>
<li class="chapter" data-level="5.5.2" data-path="gwas-with-continuous-trait.html"><a href="gwas-with-continuous-trait.html#calculate-marginal-r2-using-r"><i class="fa fa-check"></i><b>5.5.2</b> Calculate marginal <span class="math inline">\(R^2\)</span> (using R)</a></li>
</ul></li>
<li class="chapter" data-level="5.6" data-path="gwas-with-continuous-trait.html"><a href="gwas-with-continuous-trait.html#practice-time"><i class="fa fa-check"></i><b>5.6</b> PRACTICE TIME</a>
<ul>
<li class="chapter" data-level="5.6.1" data-path="gwas-with-continuous-trait.html"><a href="gwas-with-continuous-trait.html#practice-1"><i class="fa fa-check"></i><b>5.6.1</b> Practice 1</a></li>
<li class="chapter" data-level="5.6.2" data-path="gwas-with-continuous-trait.html"><a href="gwas-with-continuous-trait.html#practice-2"><i class="fa fa-check"></i><b>5.6.2</b> Practice 2</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="gwas-with-binary-trait.html"><a href="gwas-with-binary-trait.html"><i class="fa fa-check"></i><b>6</b> GWAS with Binary Trait</a>
<ul>
<li class="chapter" data-level="6.1" data-path="gwas-with-binary-trait.html"><a href="gwas-with-binary-trait.html#data-preparation"><i class="fa fa-check"></i><b>6.1</b> Data Preparation</a></li>
<li class="chapter" data-level="6.2" data-path="gwas-with-binary-trait.html"><a href="gwas-with-binary-trait.html#genome-wide-casecontrol-association-test-in-plink"><i class="fa fa-check"></i><b>6.2</b> Genome-Wide Case/Control Association Test (in PLINK)</a></li>
<li class="chapter" data-level="6.3" data-path="gwas-with-binary-trait.html"><a href="gwas-with-binary-trait.html#manhattan-plot-and-q-q-plot-1"><i class="fa fa-check"></i><b>6.3</b> Manhattan Plot and Q-Q Plot</a></li>
<li class="chapter" data-level="6.4" data-path="gwas-with-binary-trait.html"><a href="gwas-with-binary-trait.html#generate-manhattan-plot-1"><i class="fa fa-check"></i><b>6.4</b> Generate Manhattan plot</a></li>
<li class="chapter" data-level="6.5" data-path="gwas-with-binary-trait.html"><a href="gwas-with-binary-trait.html#generate-q-q-plot-1"><i class="fa fa-check"></i><b>6.5</b> Generate Q-Q plot</a></li>
<li class="chapter" data-level="6.6" data-path="gwas-with-binary-trait.html"><a href="gwas-with-binary-trait.html#multiple-test-correction-1"><i class="fa fa-check"></i><b>6.6</b> Multiple-Test Correction</a></li>
<li class="chapter" data-level="6.7" data-path="gwas-with-binary-trait.html"><a href="gwas-with-binary-trait.html#top-10-snps"><i class="fa fa-check"></i><b>6.7</b> Top 10 SNPs</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="gxg-and-gxe-interactions-power-and-sample-size-calculation.html"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html"><i class="fa fa-check"></i><b>7</b> GxG and GxE Interactions, Power and Sample Size Calculation</a>
<ul>
<li class="chapter" data-level="7.1" data-path="gxg-and-gxe-interactions-power-and-sample-size-calculation.html"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#prepare-files"><i class="fa fa-check"></i><b>7.1</b> Prepare Files</a></li>
<li class="chapter" data-level="7.2" data-path="gxg-and-gxe-interactions-power-and-sample-size-calculation.html"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#test-for-epistasis-gxg"><i class="fa fa-check"></i><b>7.2</b> Test for Epistasis (GxG)</a></li>
<li class="chapter" data-level="7.3" data-path="gxg-and-gxe-interactions-power-and-sample-size-calculation.html"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#gene-environment-interaction-gxe"><i class="fa fa-check"></i><b>7.3</b> Gene-Environment Interaction (GxE)</a></li>
<li class="chapter" data-level="7.4" data-path="gxg-and-gxe-interactions-power-and-sample-size-calculation.html"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#power-and-sample-size-calculation-for-gwas"><i class="fa fa-check"></i><b>7.4</b> Power and sample size calculation for GWAS</a>
<ul>
<li class="chapter" data-level="7.4.1" data-path="gxg-and-gxe-interactions-power-and-sample-size-calculation.html"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#power-calculation-with-binary-outcome"><i class="fa fa-check"></i><b>7.4.1</b> Power calculation with binary outcome:</a></li>
<li class="chapter" data-level="7.4.2" data-path="gxg-and-gxe-interactions-power-and-sample-size-calculation.html"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#sample-size-calculation"><i class="fa fa-check"></i><b>7.4.2</b> Sample size calculation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8" data-path="ldsc-and-gene-set-analysis.html"><a href="ldsc-and-gene-set-analysis.html"><i class="fa fa-check"></i><b>8</b> LDSC and Gene-Set Analysis</a>
<ul>
<li class="chapter" data-level="8.1" data-path="ldsc-and-gene-set-analysis.html"><a href="ldsc-and-gene-set-analysis.html#ldsc-linkage-disequilibrium-score-regression"><i class="fa fa-check"></i><b>8.1</b> LDSC (Linkage Disequilibrium Score Regression)</a></li>
<li class="chapter" data-level="8.2" data-path="ldsc-and-gene-set-analysis.html"><a href="ldsc-and-gene-set-analysis.html#magma-for-gene-level-analysis"><i class="fa fa-check"></i><b>8.2</b> MAGMA (for Gene-Level Analysis)</a>
<ul>
<li class="chapter" data-level="8.2.1" data-path="ldsc-and-gene-set-analysis.html"><a href="ldsc-and-gene-set-analysis.html#prepare-annotation-file"><i class="fa fa-check"></i><b>8.2.1</b> Prepare Annotation File</a></li>
<li class="chapter" data-level="8.2.2" data-path="ldsc-and-gene-set-analysis.html"><a href="ldsc-and-gene-set-analysis.html#part-1-gene-analysis"><i class="fa fa-check"></i><b>8.2.2</b> Part 1: Gene Analysis</a></li>
<li class="chapter" data-level="8.2.3" data-path="ldsc-and-gene-set-analysis.html"><a href="ldsc-and-gene-set-analysis.html#part-2-gene-set-analysis"><i class="fa fa-check"></i><b>8.2.3</b> Part 2: Gene-Set Analysis</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="9" data-path="perform-eqtl-analysis.html"><a href="perform-eqtl-analysis.html"><i class="fa fa-check"></i><b>9</b> Perform eQTL Analysis</a>
<ul>
<li class="chapter" data-level="9.1" data-path="perform-eqtl-analysis.html"><a href="perform-eqtl-analysis.html#filtering-snps-by-maf"><i class="fa fa-check"></i><b>9.1</b> Filtering SNPs by MAF</a></li>
<li class="chapter" data-level="9.2" data-path="perform-eqtl-analysis.html"><a href="perform-eqtl-analysis.html#gene-expression-profiling"><i class="fa fa-check"></i><b>9.2</b> Gene expression profiling</a></li>
<li class="chapter" data-level="9.3" data-path="perform-eqtl-analysis.html"><a href="perform-eqtl-analysis.html#covariates"><i class="fa fa-check"></i><b>9.3</b> Covariates</a></li>
<li class="chapter" data-level="9.4" data-path="perform-eqtl-analysis.html"><a href="perform-eqtl-analysis.html#using-principle-components-as-covariates"><i class="fa fa-check"></i><b>9.4</b> Using principle components as covariates</a></li>
<li class="chapter" data-level="9.5" data-path="perform-eqtl-analysis.html"><a href="perform-eqtl-analysis.html#eqtl-analysis"><i class="fa fa-check"></i><b>9.5</b> eQTL analysis</a>
<ul>
<li class="chapter" data-level="9.5.1" data-path="perform-eqtl-analysis.html"><a href="perform-eqtl-analysis.html#cis-eqtl-and-trans-eqtl-analysis"><i class="fa fa-check"></i><b>9.5.1</b> cis-eQTL and trans-eQTL analysis</a></li>
</ul></li>
<li class="chapter" data-level="9.6" data-path="perform-eqtl-analysis.html"><a href="perform-eqtl-analysis.html#multiple-testing-correction"><i class="fa fa-check"></i><b>9.6</b> Multiple testing correction</a></li>
<li class="chapter" data-level="9.7" data-path="perform-eqtl-analysis.html"><a href="perform-eqtl-analysis.html#gene-regional-manhanttan-plots"><i class="fa fa-check"></i><b>9.7</b> Gene regional Manhanttan plots</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">CSHL course - GWAS Hands-on Analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="gwas-with-continuous-trait" class="section level1 hasAnchor" number="5">
<h1><span class="header-section-number">Chapter 5</span> GWAS with Continuous Trait<a href="gwas-with-continuous-trait.html#gwas-with-continuous-trait" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>After all files are prepared in the PLINK format, we are ready to perform association test. The following code conducts the Genome-Wide Association Test by PLINK. <strong>This code will take 1 minute to finish.</strong></p>
<div class="sourceCode" id="cb1"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb1-1"><a href="gwas-with-continuous-trait.html#cb1-1" tabindex="-1"></a><span class="co"># both Mac and Windows can run in Terminal</span></span>
<span id="cb1-2"><a href="gwas-with-continuous-trait.html#cb1-2" tabindex="-1"></a><span class="co"># windows users</span></span>
<span id="cb1-3"><a href="gwas-with-continuous-trait.html#cb1-3" tabindex="-1"></a><span class="ex">../../Tools/PLINK/plink_win64_20231211/plink.exe</span> <span class="dt">\</span></span>
<span id="cb1-4"><a href="gwas-with-continuous-trait.html#cb1-4" tabindex="-1"></a>--noweb <span class="at">--file</span> genotype_1k <span class="at">--assoc</span> <span class="at">--adjust</span> <span class="at">--pheno</span> phenotype_plink.txt <span class="dt">\</span></span>
<span id="cb1-5"><a href="gwas-with-continuous-trait.html#cb1-5" tabindex="-1"></a>--all-pheno <span class="at">--out</span> younameit</span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb2-1"><a href="gwas-with-continuous-trait.html#cb2-1" tabindex="-1"></a><span class="co"># macOS users</span></span>
<span id="cb2-2"><a href="gwas-with-continuous-trait.html#cb2-2" tabindex="-1"></a><span class="ex">../../Tools/PLINK/plink_mac_20231211/plink</span> <span class="dt">\</span></span>
<span id="cb2-3"><a href="gwas-with-continuous-trait.html#cb2-3" tabindex="-1"></a>--noweb <span class="at">--file</span> genotype_1k <span class="at">--assoc</span> <span class="at">--adjust</span> <span class="at">--pheno</span> phenotype_plink.txt <span class="dt">\</span></span>
<span id="cb2-4"><a href="gwas-with-continuous-trait.html#cb2-4" tabindex="-1"></a>--all-pheno <span class="at">--out</span> younameit</span></code></pre></div>
<pre><code>## PLINK v1.90b7.2 64-bit (11 Dec 2023)           www.cog-genomics.org/plink/1.9/
## (C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3
## Logging to younameit.log.
## Options in effect:
##   --adjust
##   --all-pheno
##   --assoc
##   --file genotype_1k
##   --noweb
##   --out younameit
##   --pheno phenotype_plink.txt
## 
## Note: --noweb has no effect since no web check is implemented yet.
## 16384 MB RAM detected; reserving 8192 MB for main workspace.
## Scanning .ped file... 0%1%1%2%2%3%3%4%4%5%5%6%6%7%7%8%8%9%9%10%10%11%11%12%13%13%14%14%15%15%16%16%17%17%18%18%19%19%20%20%21%21%22%22%23%23%24%24%25%26%26%27%27%28%28%29%29%30%30%31%31%32%32%33%33%34%34%35%35%36%36%37%38%38%39%39%40%40%41%41%42%42%43%43%44%44%45%45%46%46%47%47%48%48%49%49%50%51%51%52%52%53%53%54%54%55%55%56%56%57%57%58%58%59%59%60%60%61%61%62%63%63%64%64%65%65%66%66%67%67%68%68%69%69%70%70%71%71%72%72%73%73%74%74%75%76%76%77%77%78%78%79%79%80%80%81%81%82%82%83%83%84%84%85%85%86%86%87%88%88%89%89%90%90%91%91%92%92%93%93%94%94%95%95%96%96%97%97%98%98%99%100%.ped scan complete (for binary autoconversion).
## Performing single-pass .bed write (1000 variants, 192 people).
## 0%1%2%3%4%5%6%7%8%9%10%11%12%13%14%15%16%17%18%19%20%21%22%23%24%25%26%27%28%29%30%31%32%33%34%35%36%37%38%39%40%41%42%43%44%45%46%47%48%49%50%51%52%53%54%55%56%57%58%59%60%61%62%63%64%65%66%67%68%69%70%71%72%73%74%75%76%77%78%79%80%81%82%83%84%85%86%87%88%89%90%91%92%93%94%--file: younameit-temporary.bed + younameit-temporary.bim +
## younameit-temporary.fam written.
## 1000 variants loaded from .bim file.
## 192 people (0 males, 192 females) loaded from .fam.
## 192 phenotype values present after --pheno.
## Using 1 thread (no multithreaded calculations invoked).
## Before main variant filters, 192 founders and 0 nonfounders present.
## Calculating allele frequencies... 0%1%2%3%4%5%6%7%8%9%10%11%12%13%14%15%16%17%18%19%20%21%22%23%24%25%26%27%28%29%30%31%32%33%34%35%36%37%38%39%40%41%42%43%44%45%46%47%48%49%50%51%52%53%54%55%56%57%58%59%60%61%62%63%64%65%66%67%68%69%70%71%72%73%74%75%76%77%78%79%80%81%82%83%84%85%86%87%88%89%90%91%92%93%94%95%96%97%98%99% done.
## Warning: Nonmissing nonmale Y chromosome genotype(s) present; many commands
## treat these as missing.
## Total genotyping rate is 0.995331.
## 1000 variants and 192 people pass filters and QC.
## Phenotype data is quantitative.
## 192 phenotype values present after --pheno.
## Warning: QT --assoc doesn&#39;t handle X/Y/MT/haploid variants normally (try
## --linear).
## Writing QT --assoc report to younameit.P1.qassoc ... 0%11%16%20%24%27%43%50%53%56%61%64%68%70%72%73%76%78%79%82%84%85%94%96%99%done.
## --adjust: Genomic inflation est. lambda (based on median chisq) = 1.34576.
## 0%1%2%3%4%5%6%7%8%9%10%11%12%13%14%15%16%17%18%19%20%21%22%23%24%25%26%27%28%29%30%31%32%33%34%35%36%37%38%39%40%41%42%43%44%45%46%47%48%49%50%51%52%53%54%55%56%57%58%59%60%61%62%63%64%65%66%67%68%69%70%71%72%73%74%75%76%77%78%79%80%81%82%83%84%85%86%87%88%89%90%91%92%93%94%95%96%97%98%99%--adjust values (805 variants) written to younameit.P1.qassoc.adjusted .
## 192 phenotype values present after --pheno.
## Warning: QT --assoc doesn&#39;t handle X/Y/MT/haploid variants normally (try
## --linear).
## Writing QT --assoc report to younameit.P2.qassoc ... 0%11%16%20%24%27%43%50%53%56%61%64%68%70%72%73%76%78%79%82%84%85%94%96%99%done.
## --adjust: Genomic inflation est. lambda (based on median chisq) = 1.
## 0%1%2%3%4%5%6%7%8%9%10%11%12%13%14%15%16%17%18%19%20%21%22%23%24%25%26%27%28%29%30%31%32%33%34%35%36%37%38%39%40%41%42%43%44%45%46%47%48%49%50%51%52%53%54%55%56%57%58%59%60%61%62%63%64%65%66%67%68%69%70%71%72%73%74%75%76%77%78%79%80%81%82%83%84%85%86%87%88%89%90%91%92%93%94%95%96%97%98%99%--adjust values (805 variants) written to younameit.P2.qassoc.adjusted .
## 192 phenotype values present after --pheno.
## Warning: QT --assoc doesn&#39;t handle X/Y/MT/haploid variants normally (try
## --linear).
## Writing QT --assoc report to younameit.P3.qassoc ... 0%11%16%20%24%27%43%50%53%56%61%64%68%70%72%73%76%78%79%82%84%85%94%96%99%done.
## --adjust: Genomic inflation est. lambda (based on median chisq) = 1.37367.
## 0%1%2%3%4%5%6%7%8%9%10%11%12%13%14%15%16%17%18%19%20%21%22%23%24%25%26%27%28%29%30%31%32%33%34%35%36%37%38%39%40%41%42%43%44%45%46%47%48%49%50%51%52%53%54%55%56%57%58%59%60%61%62%63%64%65%66%67%68%69%70%71%72%73%74%75%76%77%78%79%80%81%82%83%84%85%86%87%88%89%90%91%92%93%94%95%96%97%98%99%--adjust values (805 variants) written to younameit.P3.qassoc.adjusted .</code></pre>
<ol style="list-style-type: decimal">
<li><strong>–file</strong> specifies .ped and .map files,</li>
<li><strong>–assoc</strong> performs case/control or QTL association,</li>
<li><strong>–adjust</strong> generates a file of adjusted significance values that correct for all tests performed and other metrics,</li>
<li><strong>–pheno</strong> specifies alternate phenotype,</li>
<li><strong>–all-pheno</strong> performs association for all phenotypes in file,</li>
<li><strong>–out</strong> specifies output filename.</li>
</ol>
<p>This will generate the files <strong>younameit.P1.qassoc</strong>, <strong>younameit.P2.qassoc</strong>, <strong>younameit.P3.qassoc</strong> for each of the phenotype with fields as follows:</p>
<pre><code>      CHR      Chromosome number
      
      SNP      SNP identifier
      
      BP       Physical position (base-pair)
      
      NMISS    Number of non-missing genotypes
      
      BETA     Regression coefficient
      
      SE       Standard error
      
      R2       Regression r-squared
      
      T        Wald test (based on t-distribtion)
      
      P        Wald test asymptotic p-value
      </code></pre>
<div id="manhattan-plot-and-q-q-plot" class="section level2 hasAnchor" number="5.1">
<h2><span class="header-section-number">5.1</span> Manhattan Plot and Q-Q Plot<a href="gwas-with-continuous-trait.html#manhattan-plot-and-q-q-plot" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The Manhattan plot is a good way to visualize the resuls of GWAS test. For more details, please refer: <a href="https://en.wikipedia.org/wiki/Manhattan_plot" class="uri">https://en.wikipedia.org/wiki/Manhattan_plot</a></p>
<p>The Q-Q plot for GWAS is another way to visualize the signals. For more details, please refer:<a href="https://genome.sph.umich.edu/wiki/Code_Sample:_Generating_QQ_Plots_in_R" class="uri">https://genome.sph.umich.edu/wiki/Code_Sample:_Generating_QQ_Plots_in_R</a></p>
<p>qqman is an R package for creating Q-Q and manhattan plots from GWAS results. See the reference page <a href="http://www.gettinggeneticsdone.com/2014/05/qqman-r-package-for-qq-and-manhattan-plots-for-gwas-results.html" class="uri">http://www.gettinggeneticsdone.com/2014/05/qqman-r-package-for-qq-and-manhattan-plots-for-gwas-results.html</a>.</p>
<p>The <code>qqman</code> R package assumes you have columns named SNP, CHR, BP, and P, corresponding to the SNP name (rs number), chromosome number, base-pair position, and p-value.</p>
<p>Here is what the data looks like:</p>
<pre><code>      SNP        CHR    BP         P 
      
      rs10495434 1      235800006  0.62220 
      
      rs6689417  1      46100028   0.06195 
      
      rs3897197  1      143700035  0.10700 
        
      rs2282450  1      202300047  0.47280 
      
      rs11208515 1      64900051   0.53430 
      </code></pre>
<p>So we need to extract these columns from our results files so they can be used with <code>qqman</code>. The resulted files are P1.qassoc, P2.qassoc and P3.qassoc for each of the 3 traits.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb6-1"><a href="gwas-with-continuous-trait.html#cb6-1" tabindex="-1"></a><span class="co"># bash codes- run in terminal</span></span>
<span id="cb6-2"><a href="gwas-with-continuous-trait.html#cb6-2" tabindex="-1"></a><span class="co">### Prepare qqman R input files (CHR, SNP, BP, P) (in Linux)</span></span>
<span id="cb6-3"><a href="gwas-with-continuous-trait.html#cb6-3" tabindex="-1"></a><span class="fu">awk</span> <span class="st">&#39;{print $1,$2,$3,$9}&#39;</span> younameit.P1.qassoc <span class="op">&gt;</span> P1.qassoc</span>
<span id="cb6-4"><a href="gwas-with-continuous-trait.html#cb6-4" tabindex="-1"></a><span class="fu">awk</span> <span class="st">&#39;{print $1,$2,$3,$9}&#39;</span> younameit.P2.qassoc <span class="op">&gt;</span> P2.qassoc</span>
<span id="cb6-5"><a href="gwas-with-continuous-trait.html#cb6-5" tabindex="-1"></a><span class="fu">awk</span> <span class="st">&#39;{print $1,$2,$3,$9}&#39;</span> younameit.P3.qassoc <span class="op">&gt;</span> P3.qassoc</span></code></pre></div>
<p>The codes below read in the association results as <code>qassoc1</code>, <code>qassoc2</code>, <code>qassoc3</code>. Here we only read in one of the trait as an example, as reading in all three will take a long time.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="gwas-with-continuous-trait.html#cb7-1" tabindex="-1"></a><span class="co"># R codes- run in R Console</span></span>
<span id="cb7-2"><a href="gwas-with-continuous-trait.html#cb7-2" tabindex="-1"></a>traits<span class="ot">=</span><span class="fu">c</span>(<span class="st">&quot;z-4-oh-Tam&quot;</span>,<span class="st">&quot;tamoxifen&quot;</span>,<span class="st">&quot;z-endoxifen_ratio&quot;</span>) <span class="co">#name of four traits</span></span>
<span id="cb7-3"><a href="gwas-with-continuous-trait.html#cb7-3" tabindex="-1"></a>traits<span class="ot">=</span><span class="fu">as.matrix</span>(traits) <span class="co">#create a matrix from traits</span></span>
<span id="cb7-4"><a href="gwas-with-continuous-trait.html#cb7-4" tabindex="-1"></a></span>
<span id="cb7-5"><a href="gwas-with-continuous-trait.html#cb7-5" tabindex="-1"></a><span class="co">#Load association results</span></span>
<span id="cb7-6"><a href="gwas-with-continuous-trait.html#cb7-6" tabindex="-1"></a>i<span class="ot">=</span><span class="dv">1</span> <span class="co">#first trait: Z-4-OH-TAM</span></span>
<span id="cb7-7"><a href="gwas-with-continuous-trait.html#cb7-7" tabindex="-1"></a>qassoc1<span class="ot">=</span><span class="fu">read.table</span>(<span class="fu">paste0</span>(<span class="st">&quot;P&quot;</span>, i, <span class="st">&quot;.qassoc&quot;</span>), <span class="at">header=</span>T)</span>
<span id="cb7-8"><a href="gwas-with-continuous-trait.html#cb7-8" tabindex="-1"></a>qassoc1<span class="ot">=</span>qassoc1[(qassoc1<span class="sc">$</span>CHR<span class="sc">!=</span><span class="dv">0</span>) <span class="sc">&amp;</span> (<span class="sc">!</span><span class="fu">is.na</span>(qassoc1<span class="sc">$</span>P)),]</span>
<span id="cb7-9"><a href="gwas-with-continuous-trait.html#cb7-9" tabindex="-1"></a><span class="fu">head</span>(qassoc1)</span></code></pre></div>
<pre><code>##    CHR        SNP       BP       P
## 10   1   rs307354  1264539 0.36230
## 13   1 rs12121577 11794553 0.92960
## 14   1  rs4845898 12339619 0.03024
## 16   1   rs665691 22960354 0.70330
## 18   1  rs7537875 24695151 0.45760
## 19   1     rs8876 25687308 0.88700</code></pre>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="gwas-with-continuous-trait.html#cb9-1" tabindex="-1"></a>i<span class="ot">=</span><span class="dv">2</span> <span class="co">#second trait: tamoxifen</span></span>
<span id="cb9-2"><a href="gwas-with-continuous-trait.html#cb9-2" tabindex="-1"></a>qassoc2<span class="ot">=</span><span class="fu">read.table</span>(<span class="fu">paste0</span>(<span class="st">&quot;P&quot;</span>, i, <span class="st">&quot;.qassoc&quot;</span>), <span class="at">header=</span>T)</span>
<span id="cb9-3"><a href="gwas-with-continuous-trait.html#cb9-3" tabindex="-1"></a>qassoc2<span class="ot">=</span>qassoc2[(qassoc2<span class="sc">$</span>CHR<span class="sc">!=</span><span class="dv">0</span>) <span class="sc">&amp;</span> (<span class="sc">!</span><span class="fu">is.na</span>(qassoc2<span class="sc">$</span>P)),]</span>
<span id="cb9-4"><a href="gwas-with-continuous-trait.html#cb9-4" tabindex="-1"></a><span class="fu">head</span>(qassoc2)</span></code></pre></div>
<pre><code>##    CHR        SNP       BP      P
## 10   1   rs307354  1264539 0.1601
## 13   1 rs12121577 11794553 0.4010
## 14   1  rs4845898 12339619 0.7168
## 16   1   rs665691 22960354 0.8944
## 18   1  rs7537875 24695151 0.2571
## 19   1     rs8876 25687308 0.3090</code></pre>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="gwas-with-continuous-trait.html#cb11-1" tabindex="-1"></a>i<span class="ot">=</span><span class="dv">3</span> <span class="co">#third trait: MR</span></span>
<span id="cb11-2"><a href="gwas-with-continuous-trait.html#cb11-2" tabindex="-1"></a>qassoc3<span class="ot">=</span><span class="fu">read.table</span>(<span class="fu">paste0</span>(<span class="st">&quot;P&quot;</span>, i, <span class="st">&quot;.qassoc&quot;</span>), <span class="at">header=</span>T)</span>
<span id="cb11-3"><a href="gwas-with-continuous-trait.html#cb11-3" tabindex="-1"></a>qassoc3<span class="ot">=</span>qassoc3[(qassoc3<span class="sc">$</span>CHR<span class="sc">!=</span><span class="dv">0</span>) <span class="sc">&amp;</span> (<span class="sc">!</span><span class="fu">is.na</span>(qassoc3<span class="sc">$</span>P)),]</span>
<span id="cb11-4"><a href="gwas-with-continuous-trait.html#cb11-4" tabindex="-1"></a><span class="fu">head</span>(qassoc3)</span></code></pre></div>
<pre><code>##    CHR        SNP       BP      P
## 10   1   rs307354  1264539 0.8006
## 13   1 rs12121577 11794553 0.6506
## 14   1  rs4845898 12339619 0.1082
## 16   1   rs665691 22960354 0.8454
## 18   1  rs7537875 24695151 0.8902
## 19   1     rs8876 25687308 0.9103</code></pre>
<div id="generate-manhattan-plot" class="section level3 hasAnchor" number="5.1.1">
<h3><span class="header-section-number">5.1.1</span> Generate manhattan plot<a href="gwas-with-continuous-trait.html#generate-manhattan-plot" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>We are ready to generate the Manhattan plots. Manhattan plots are generated using <code>manhattan</code> function, and are saved as png files.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="gwas-with-continuous-trait.html#cb13-1" tabindex="-1"></a><span class="co"># R codes- run in R Console</span></span>
<span id="cb13-2"><a href="gwas-with-continuous-trait.html#cb13-2" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code></pre></div>
<pre><code>## null device 
##           1</code></pre>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="gwas-with-continuous-trait.html#cb15-1" tabindex="-1"></a>traits<span class="ot">=</span><span class="fu">c</span>(<span class="st">&quot;z-4-oh-Tam&quot;</span>,<span class="st">&quot;tamoxifen&quot;</span>,<span class="st">&quot;z-endoxifen_ratio&quot;</span>) <span class="co">#name of four traits</span></span>
<span id="cb15-2"><a href="gwas-with-continuous-trait.html#cb15-2" tabindex="-1"></a></span>
<span id="cb15-3"><a href="gwas-with-continuous-trait.html#cb15-3" tabindex="-1"></a><span class="co"># trait 1: Z-4-OH-TAM</span></span>
<span id="cb15-4"><a href="gwas-with-continuous-trait.html#cb15-4" tabindex="-1"></a>i<span class="ot">=</span><span class="dv">1</span></span>
<span id="cb15-5"><a href="gwas-with-continuous-trait.html#cb15-5" tabindex="-1"></a><span class="fu">png</span>(<span class="at">filename=</span><span class="fu">paste0</span>(<span class="st">&quot;Manhattan_Plot_for_&quot;</span>, traits[i], <span class="st">&quot;.png&quot;</span>), <span class="at">type=</span><span class="st">&quot;cairo&quot;</span>)</span></code></pre></div>
<pre><code>## Warning in grSoftVersion(): unable to load shared object &#39;/Library/Frameworks/R.framework/Resources/modules//R_X11.so&#39;:
##   dlopen(/Library/Frameworks/R.framework/Resources/modules//R_X11.so, 0x0006): Library not loaded: /opt/X11/lib/libSM.6.dylib
##   Referenced from: &lt;31EADEB5-0A17-3546-9944-9B3747071FE8&gt; /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/modules/R_X11.so
##   Reason: tried: &#39;/opt/X11/lib/libSM.6.dylib&#39; (no such file), &#39;/System/Volumes/Preboot/Cryptexes/OS/opt/X11/lib/libSM.6.dylib&#39; (no such file), &#39;/opt/X11/lib/libSM.6.dylib&#39; (no such file), &#39;/Library/Frameworks/R.framework/Resources/lib/libSM.6.dylib&#39; (no such file), &#39;/Library/Java/JavaVirtualMachines/jdk-11.0.18+10/Contents/Home/lib/server/libSM.6.dylib&#39; (no such file)</code></pre>
<pre><code>## Warning in cairoVersion(): unable to load shared object &#39;/Library/Frameworks/R.framework/Resources/library/grDevices/libs//cairo.so&#39;:
##   dlopen(/Library/Frameworks/R.framework/Resources/library/grDevices/libs//cairo.so, 0x0006): Library not loaded: /opt/X11/lib/libXrender.1.dylib
##   Referenced from: &lt;63619C6D-FE72-3544-BCEF-9C834A5E39D8&gt; /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/library/grDevices/libs/cairo.so
##   Reason: tried: &#39;/opt/X11/lib/libXrender.1.dylib&#39; (no such file), &#39;/System/Volumes/Preboot/Cryptexes/OS/opt/X11/lib/libXrender.1.dylib&#39; (no such file), &#39;/opt/X11/lib/libXrender.1.dylib&#39; (no such file), &#39;/Library/Frameworks/R.framework/Resources/lib/libXrender.1.dylib&#39; (no such file), &#39;/Library/Java/JavaVirtualMachines/jdk-11.0.18+10/Contents/Home/lib/server/libXrender.1.dylib&#39; (no such file)</code></pre>
<pre><code>## Warning in png(filename = paste0(&quot;Manhattan_Plot_for_&quot;, traits[i], &quot;.png&quot;), :
## failed to load cairo DLL</code></pre>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="gwas-with-continuous-trait.html#cb19-1" tabindex="-1"></a><span class="fu">manhattan</span>(qassoc1, <span class="at">col=</span><span class="fu">c</span>(<span class="st">&quot;dimgray&quot;</span>, <span class="st">&quot;darkgoldenrod1&quot;</span>), <span class="at">suggestiveline=</span>F, <span class="at">genomewideline=</span>F)</span>
<span id="cb19-2"><a href="gwas-with-continuous-trait.html#cb19-2" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code></pre></div>
<pre><code>## null device 
##           1</code></pre>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="gwas-with-continuous-trait.html#cb21-1" tabindex="-1"></a><span class="co"># trait 2: tamoxifen</span></span>
<span id="cb21-2"><a href="gwas-with-continuous-trait.html#cb21-2" tabindex="-1"></a>i<span class="ot">=</span><span class="dv">2</span></span>
<span id="cb21-3"><a href="gwas-with-continuous-trait.html#cb21-3" tabindex="-1"></a><span class="fu">png</span>(<span class="at">filename=</span><span class="fu">paste0</span>(<span class="st">&quot;Manhattan_Plot_for_&quot;</span>, traits[i], <span class="st">&quot;.png&quot;</span>), <span class="at">type=</span><span class="st">&quot;cairo&quot;</span>)</span></code></pre></div>
<pre><code>## Warning in png(filename = paste0(&quot;Manhattan_Plot_for_&quot;, traits[i], &quot;.png&quot;), :
## failed to load cairo DLL</code></pre>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="gwas-with-continuous-trait.html#cb23-1" tabindex="-1"></a><span class="fu">manhattan</span>(qassoc2, <span class="at">col=</span><span class="fu">c</span>(<span class="st">&quot;dimgray&quot;</span>, <span class="st">&quot;darkgoldenrod1&quot;</span>), <span class="at">suggestiveline=</span>F, <span class="at">genomewideline=</span>F)</span></code></pre></div>
<p><img src="cshl_gwas_files/figure-html/mahattan2-1.png" width="672" /></p>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="gwas-with-continuous-trait.html#cb24-1" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code></pre></div>
<pre><code>## null device 
##           1</code></pre>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="gwas-with-continuous-trait.html#cb26-1" tabindex="-1"></a><span class="co"># trait 3: z-endoxifen ratio</span></span>
<span id="cb26-2"><a href="gwas-with-continuous-trait.html#cb26-2" tabindex="-1"></a>i<span class="ot">=</span><span class="dv">3</span></span>
<span id="cb26-3"><a href="gwas-with-continuous-trait.html#cb26-3" tabindex="-1"></a><span class="fu">png</span>(<span class="at">filename=</span><span class="fu">paste0</span>(<span class="st">&quot;Manhattan_Plot_for_&quot;</span>, traits[i], <span class="st">&quot;.png&quot;</span>), <span class="at">type=</span><span class="st">&quot;cairo&quot;</span>)</span></code></pre></div>
<pre><code>## Warning in png(filename = paste0(&quot;Manhattan_Plot_for_&quot;, traits[i], &quot;.png&quot;), :
## failed to load cairo DLL</code></pre>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="gwas-with-continuous-trait.html#cb28-1" tabindex="-1"></a><span class="fu">manhattan</span>(qassoc3, <span class="at">col=</span><span class="fu">c</span>(<span class="st">&quot;dimgray&quot;</span>, <span class="st">&quot;darkgoldenrod1&quot;</span>), <span class="at">suggestiveline=</span>F, <span class="at">genomewideline=</span>F)</span></code></pre></div>
<p><img src="cshl_gwas_files/figure-html/mahattan3-1.png" width="672" /></p>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="gwas-with-continuous-trait.html#cb29-1" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code></pre></div>
<pre><code>## null device 
##           1</code></pre>
<p>We will directly read the Manhattan plots from saved files. From these manhattan plots, we can observe several significant SNPs or peaks that stand out over others. Specifically, there is one extremely significant SNP for z-endoxifen ratio in chromosome 22. (chr26 here means mitochondrial DNA.)</p>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="gwas-with-continuous-trait.html#cb31-1" tabindex="-1"></a><span class="co"># R codes- run in R Console</span></span>
<span id="cb31-2"><a href="gwas-with-continuous-trait.html#cb31-2" tabindex="-1"></a>img <span class="ot">&lt;-</span> <span class="fu">readPNG</span>(<span class="st">&quot;Manhattan_Plot_for_z-4-oh-Tam.png&quot;</span>)</span>
<span id="cb31-3"><a href="gwas-with-continuous-trait.html#cb31-3" tabindex="-1"></a>grid<span class="sc">::</span><span class="fu">grid.raster</span>(img)</span></code></pre></div>
<p><img src="cshl_gwas_files/figure-html/mah1-1.png" width="672" /></p>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="gwas-with-continuous-trait.html#cb32-1" tabindex="-1"></a><span class="co"># R codes- run in R Console</span></span>
<span id="cb32-2"><a href="gwas-with-continuous-trait.html#cb32-2" tabindex="-1"></a>img <span class="ot">&lt;-</span> <span class="fu">readPNG</span>(<span class="st">&quot;Manhattan_Plot_for_tamoxifen.png&quot;</span>)</span>
<span id="cb32-3"><a href="gwas-with-continuous-trait.html#cb32-3" tabindex="-1"></a>grid<span class="sc">::</span><span class="fu">grid.raster</span>(img)</span></code></pre></div>
<p><img src="cshl_gwas_files/figure-html/mah2-1.png" width="672" /></p>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="gwas-with-continuous-trait.html#cb33-1" tabindex="-1"></a><span class="co"># R codes- run in R Console</span></span>
<span id="cb33-2"><a href="gwas-with-continuous-trait.html#cb33-2" tabindex="-1"></a>img <span class="ot">&lt;-</span> <span class="fu">readPNG</span>(<span class="st">&quot;Manhattan_Plot_for_z-endoxifen_ratio.png&quot;</span>)</span>
<span id="cb33-3"><a href="gwas-with-continuous-trait.html#cb33-3" tabindex="-1"></a>grid<span class="sc">::</span><span class="fu">grid.raster</span>(img)</span></code></pre></div>
<p><img src="cshl_gwas_files/figure-html/mah3-1.png" width="672" /></p>
</div>
<div id="generate-q-q-plot" class="section level3 hasAnchor" number="5.1.2">
<h3><span class="header-section-number">5.1.2</span> Generate Q-Q plot<a href="gwas-with-continuous-trait.html#generate-q-q-plot" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>A Quantile-Quantile (QQ) plot (in general) plots the observed quantiles of one distribution versus another OR plots the
observed quantiles of a distribution versus the quantiles of the ideal distribution (in this case is normal distritbution). In GWAS we use a QQ plot to plot the quantile distribution of observed p-values (on the y-axis) versus the quantile distribution of expected p-values.</p>
<p>In an ideal GWAS case where there are <strong>NO</strong> causal polymorphisms, your QQ plot will be a straight line with a slope. However, in an ideal GWAS case where there <strong>ARE</strong> causal polymorphisms, your QQ plot will be <strong>a line with a tail!</strong></p>
<p>The codes below generate the QQ plot with <code>qq</code> function in <code>qqman</code> package.</p>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="gwas-with-continuous-trait.html#cb34-1" tabindex="-1"></a><span class="co"># R codes- run in R Console</span></span>
<span id="cb34-2"><a href="gwas-with-continuous-trait.html#cb34-2" tabindex="-1"></a>traits<span class="ot">=</span><span class="fu">c</span>(<span class="st">&quot;z-4-oh-Tam&quot;</span>,<span class="st">&quot;tamoxifen&quot;</span>,<span class="st">&quot;z-endoxifen_ratio&quot;</span>) <span class="co">#name of four traits</span></span>
<span id="cb34-3"><a href="gwas-with-continuous-trait.html#cb34-3" tabindex="-1"></a></span>
<span id="cb34-4"><a href="gwas-with-continuous-trait.html#cb34-4" tabindex="-1"></a><span class="co"># trait 1: Z-4-OH-TAM</span></span>
<span id="cb34-5"><a href="gwas-with-continuous-trait.html#cb34-5" tabindex="-1"></a>i<span class="ot">=</span><span class="dv">1</span></span>
<span id="cb34-6"><a href="gwas-with-continuous-trait.html#cb34-6" tabindex="-1"></a><span class="fu">png</span>(<span class="at">filename=</span><span class="fu">paste0</span>(<span class="st">&quot;Q-Q_Plot_for_&quot;</span>, traits[i], <span class="st">&quot;.png&quot;</span>), <span class="at">type=</span><span class="st">&quot;cairo&quot;</span>)</span></code></pre></div>
<pre><code>## Warning in png(filename = paste0(&quot;Q-Q_Plot_for_&quot;, traits[i], &quot;.png&quot;), type =
## &quot;cairo&quot;): failed to load cairo DLL</code></pre>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="gwas-with-continuous-trait.html#cb36-1" tabindex="-1"></a><span class="fu">qq</span>(qassoc1<span class="sc">$</span>P)</span></code></pre></div>
<p><img src="cshl_gwas_files/figure-html/unnamed-chunk-6-1.png" width="672" /></p>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="gwas-with-continuous-trait.html#cb37-1" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code></pre></div>
<pre><code>## null device 
##           1</code></pre>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb39-1"><a href="gwas-with-continuous-trait.html#cb39-1" tabindex="-1"></a><span class="co"># trait 2: tamoxifen</span></span>
<span id="cb39-2"><a href="gwas-with-continuous-trait.html#cb39-2" tabindex="-1"></a>i<span class="ot">=</span><span class="dv">2</span></span>
<span id="cb39-3"><a href="gwas-with-continuous-trait.html#cb39-3" tabindex="-1"></a><span class="fu">png</span>(<span class="at">filename=</span><span class="fu">paste0</span>(<span class="st">&quot;Q-Q_Plot_for_&quot;</span>, traits[i], <span class="st">&quot;.png&quot;</span>), <span class="at">type=</span><span class="st">&quot;cairo&quot;</span>)</span></code></pre></div>
<pre><code>## Warning in png(filename = paste0(&quot;Q-Q_Plot_for_&quot;, traits[i], &quot;.png&quot;), type =
## &quot;cairo&quot;): failed to load cairo DLL</code></pre>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb41-1"><a href="gwas-with-continuous-trait.html#cb41-1" tabindex="-1"></a><span class="fu">qq</span>(qassoc2<span class="sc">$</span>P)</span>
<span id="cb41-2"><a href="gwas-with-continuous-trait.html#cb41-2" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code></pre></div>
<pre><code>## null device 
##           1</code></pre>
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb43-1"><a href="gwas-with-continuous-trait.html#cb43-1" tabindex="-1"></a><span class="co"># trait 3: z-endoxifen ratio</span></span>
<span id="cb43-2"><a href="gwas-with-continuous-trait.html#cb43-2" tabindex="-1"></a>i<span class="ot">=</span><span class="dv">3</span></span>
<span id="cb43-3"><a href="gwas-with-continuous-trait.html#cb43-3" tabindex="-1"></a><span class="fu">png</span>(<span class="at">filename=</span><span class="fu">paste0</span>(<span class="st">&quot;Q-Q_Plot_for_&quot;</span>, traits[i], <span class="st">&quot;.png&quot;</span>), <span class="at">type=</span><span class="st">&quot;cairo&quot;</span>)</span></code></pre></div>
<pre><code>## Warning in png(filename = paste0(&quot;Q-Q_Plot_for_&quot;, traits[i], &quot;.png&quot;), type =
## &quot;cairo&quot;): failed to load cairo DLL</code></pre>
<div class="sourceCode" id="cb45"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb45-1"><a href="gwas-with-continuous-trait.html#cb45-1" tabindex="-1"></a><span class="fu">qq</span>(qassoc3<span class="sc">$</span>P)</span>
<span id="cb45-2"><a href="gwas-with-continuous-trait.html#cb45-2" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code></pre></div>
<pre><code>## null device 
##           1</code></pre>
<p>Then, we read the generated Q-Q plot for each of the three traits. Plot of Z-4-OH-TAM shows a slight deviation at the tail. Tamoxifen has a little more deviation, and the points that forms a straight short line at the tail are very likely to have some correlation, such as they are from the same chromosome. However, plot pf Z-endoxifen ratio shows a even better result with a long tail, meaning that we might expect many significant markers associated with Z-endoxifen ratio when we look at the test results.</p>
<div class="sourceCode" id="cb47"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb47-1"><a href="gwas-with-continuous-trait.html#cb47-1" tabindex="-1"></a><span class="co"># R codes- run in R Console</span></span>
<span id="cb47-2"><a href="gwas-with-continuous-trait.html#cb47-2" tabindex="-1"></a>img1 <span class="ot">&lt;-</span> <span class="fu">readPNG</span>(<span class="st">&quot;Q-Q_Plot_for_z-4-oh-Tam.png&quot;</span>)</span>
<span id="cb47-3"><a href="gwas-with-continuous-trait.html#cb47-3" tabindex="-1"></a>grid<span class="sc">::</span><span class="fu">grid.raster</span>(img1)</span></code></pre></div>
<p><img src="cshl_gwas_files/figure-html/qq1-1.png" width="672" /></p>
<div class="sourceCode" id="cb48"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb48-1"><a href="gwas-with-continuous-trait.html#cb48-1" tabindex="-1"></a><span class="co"># R codes- run in R Console</span></span>
<span id="cb48-2"><a href="gwas-with-continuous-trait.html#cb48-2" tabindex="-1"></a>img2 <span class="ot">&lt;-</span> <span class="fu">readPNG</span>(<span class="st">&quot;Q-Q_Plot_for_tamoxifen.png&quot;</span>)</span>
<span id="cb48-3"><a href="gwas-with-continuous-trait.html#cb48-3" tabindex="-1"></a>grid<span class="sc">::</span><span class="fu">grid.raster</span>(img2)</span></code></pre></div>
<p><img src="cshl_gwas_files/figure-html/qq2-1.png" width="672" /></p>
<div class="sourceCode" id="cb49"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb49-1"><a href="gwas-with-continuous-trait.html#cb49-1" tabindex="-1"></a><span class="co"># R codes- run in R Console</span></span>
<span id="cb49-2"><a href="gwas-with-continuous-trait.html#cb49-2" tabindex="-1"></a>img3 <span class="ot">&lt;-</span> <span class="fu">readPNG</span>(<span class="st">&quot;Q-Q_Plot_for_z-endoxifen_ratio.png&quot;</span>)</span>
<span id="cb49-3"><a href="gwas-with-continuous-trait.html#cb49-3" tabindex="-1"></a>grid<span class="sc">::</span><span class="fu">grid.raster</span>(img3)</span></code></pre></div>
<p><img src="cshl_gwas_files/figure-html/qq3-1.png" width="672" /></p>
</div>
</div>
<div id="multiple-test-correction" class="section level2 hasAnchor" number="5.2">
<h2><span class="header-section-number">5.2</span> Multiple-Test Correction<a href="gwas-with-continuous-trait.html#multiple-test-correction" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>In the Association test, we simultaneously performed tests on each SNP, which means we performed millions of test at the same time. Therefore, it is necessary that we carry out a multiple test correction procedure that control the overall type-I error rate using false discovery rate (FDR). Pleas refer to <a href="https://en.wikipedia.org/wiki/Multiple_comparisons_problem#:~:text=Multiple%20testing%20correction%20refers%20to,be%20more%20stringent%20than%20%CE%B1.">this page</a> for multiple-test correction.</p>
<div id="prepare-r-input-files-in-linux" class="section level3 hasAnchor" number="5.2.1">
<h3><span class="header-section-number">5.2.1</span> Prepare R input files (in Linux)<a href="gwas-with-continuous-trait.html#prepare-r-input-files-in-linux" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>PLINK already performed this correction using Bonferroni correction method, when we specified <code>--adjust</code> in the PLINK command in section 6. The following code extracts the <code>chr</code>, <code>snp</code>, <code>unadj</code>, <code>fdr_bh</code> from the PLINK output.
The results are stored in the files: <strong>P(1,2,3).qassoc.adjusted</strong>.</p>
<div class="sourceCode" id="cb50"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb50-1"><a href="gwas-with-continuous-trait.html#cb50-1" tabindex="-1"></a><span class="co"># bash codes- run in terminal</span></span>
<span id="cb50-2"><a href="gwas-with-continuous-trait.html#cb50-2" tabindex="-1"></a><span class="fu">awk</span> <span class="st">&#39;{print $1,$2,$3,$9}&#39;</span> younameit.P1.qassoc.adjusted <span class="op">&gt;</span> P1.qassoc.adjusted</span>
<span id="cb50-3"><a href="gwas-with-continuous-trait.html#cb50-3" tabindex="-1"></a><span class="fu">awk</span> <span class="st">&#39;{print $1,$2,$3,$9}&#39;</span> younameit.P2.qassoc.adjusted <span class="op">&gt;</span> P2.qassoc.adjusted</span>
<span id="cb50-4"><a href="gwas-with-continuous-trait.html#cb50-4" tabindex="-1"></a><span class="fu">awk</span> <span class="st">&#39;{print $1,$2,$3,$9}&#39;</span> younameit.P3.qassoc.adjusted <span class="op">&gt;</span> P3.qassoc.adjusted</span></code></pre></div>
</div>
<div id="significant-snps-after-a-multiple-test-correction-of-fdr-using-r" class="section level3 hasAnchor" number="5.2.2">
<h3><span class="header-section-number">5.2.2</span> Significant SNPs after a multiple-test correction of FDR (using R)<a href="gwas-with-continuous-trait.html#significant-snps-after-a-multiple-test-correction-of-fdr-using-r" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The following code identifies the significant SNPs with the adjusted p-values and a cutoff 0.05. SNPs with adjusted p-values &lt; 0.05 are significant SNPs associated with the corresponding trait. The resulted significant SNP lists are save as “significant_SNPsfor(trait).txt”.</p>
<div class="sourceCode" id="cb51"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb51-1"><a href="gwas-with-continuous-trait.html#cb51-1" tabindex="-1"></a><span class="co"># R codes- run in R Console</span></span>
<span id="cb51-2"><a href="gwas-with-continuous-trait.html#cb51-2" tabindex="-1"></a>traits<span class="ot">=</span><span class="fu">c</span>(<span class="st">&quot;z-4-oh-Tam&quot;</span>,<span class="st">&quot;tamoxifen&quot;</span>,<span class="st">&quot;z-endoxifen_ratio&quot;</span>)</span>
<span id="cb51-3"><a href="gwas-with-continuous-trait.html#cb51-3" tabindex="-1"></a>traits<span class="ot">=</span><span class="fu">as.matrix</span>(traits)</span>
<span id="cb51-4"><a href="gwas-with-continuous-trait.html#cb51-4" tabindex="-1"></a></span>
<span id="cb51-5"><a href="gwas-with-continuous-trait.html#cb51-5" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>){</span>
<span id="cb51-6"><a href="gwas-with-continuous-trait.html#cb51-6" tabindex="-1"></a>  qassoc.adjusted<span class="ot">=</span><span class="fu">read.table</span>(<span class="fu">paste0</span>(<span class="st">&quot;P&quot;</span>, i, <span class="st">&quot;.qassoc.adjusted&quot;</span>), <span class="at">header=</span>T) <span class="co">#load association results</span></span>
<span id="cb51-7"><a href="gwas-with-continuous-trait.html#cb51-7" tabindex="-1"></a>  sigidx<span class="ot">=</span><span class="fu">which</span>(qassoc.adjusted<span class="sc">$</span>FDR_BH<span class="sc">&lt;</span><span class="fl">0.05</span>) <span class="co">#index of significant SNPs (FDR_BH&lt;0.05)</span></span>
<span id="cb51-8"><a href="gwas-with-continuous-trait.html#cb51-8" tabindex="-1"></a>  sigSNP<span class="ot">=</span>qassoc.adjusted[sigidx,] <span class="co">#extract association results for significant SNPs</span></span>
<span id="cb51-9"><a href="gwas-with-continuous-trait.html#cb51-9" tabindex="-1"></a>  <span class="fu">write.table</span>(sigSNP, <span class="fu">paste0</span>(<span class="st">&quot;significant_SNPs_for_&quot;</span>, traits[i],<span class="st">&quot;.txt&quot;</span>), <span class="at">row.names=</span>F, <span class="at">col.names=</span>T, <span class="at">quote=</span>F, <span class="at">sep=</span><span class="st">&quot; &quot;</span>) <span class="co">#save results</span></span>
<span id="cb51-10"><a href="gwas-with-continuous-trait.html#cb51-10" tabindex="-1"></a>}  </span></code></pre></div>
<p>As a result, if you open the result files of each trait, you can observe that only “Z-endoxifen ratio” still have significant SNPs after multiple correction. Let’s read in the results and match it with SNP annotation, so it can be used for plotting in next step.</p>
<div class="sourceCode" id="cb52"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb52-1"><a href="gwas-with-continuous-trait.html#cb52-1" tabindex="-1"></a><span class="co"># R codes- run in R Console</span></span>
<span id="cb52-2"><a href="gwas-with-continuous-trait.html#cb52-2" tabindex="-1"></a><span class="fu">library</span>(data.table)</span>
<span id="cb52-3"><a href="gwas-with-continuous-trait.html#cb52-3" tabindex="-1"></a>map<span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="st">&quot;genotype_1k.map&quot;</span>,<span class="at">header=</span>F)</span>
<span id="cb52-4"><a href="gwas-with-continuous-trait.html#cb52-4" tabindex="-1"></a><span class="fu">names</span>(map) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;CHR&quot;</span>,<span class="st">&quot;SNP&quot;</span>,<span class="st">&quot;GP&quot;</span>,<span class="st">&quot;BP&quot;</span>) <span class="co"># assign header</span></span>
<span id="cb52-5"><a href="gwas-with-continuous-trait.html#cb52-5" tabindex="-1"></a>zr_result <span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="st">&quot;significant_SNPs_for_z-endoxifen_ratio.txt&quot;</span>)</span>
<span id="cb52-6"><a href="gwas-with-continuous-trait.html#cb52-6" tabindex="-1"></a>idx <span class="ot">&lt;-</span> <span class="fu">match</span>(zr_result<span class="sc">$</span>SNP, map<span class="sc">$</span>SNP)<span class="co">#index of map that match with SNP in result</span></span>
<span id="cb52-7"><a href="gwas-with-continuous-trait.html#cb52-7" tabindex="-1"></a>zr_result<span class="sc">$</span>BP <span class="ot">&lt;-</span> map<span class="sc">$</span>BP[idx]</span>
<span id="cb52-8"><a href="gwas-with-continuous-trait.html#cb52-8" tabindex="-1"></a><span class="fu">head</span>(zr_result)</span></code></pre></div>
<pre><code>##      CHR         SNP     UNADJ    FDR_BH       BP
##    &lt;int&gt;      &lt;char&gt;     &lt;num&gt;     &lt;num&gt;    &lt;int&gt;
## 1:    22 kgp10581167 2.153e-20 1.335e-17 42397026
## 2:    22   rs8138080 3.318e-20 1.335e-17 42396371
## 3:    22 kgp12372278 5.256e-19 9.347e-17 42573139
## 4:    22   rs1062753 5.643e-19 9.347e-17 42392811
## 5:    22  kgp5009187 6.121e-19 9.347e-17 42580933
## 6:    22   kgp276463 6.967e-19 9.347e-17 42517922</code></pre>
<div class="sourceCode" id="cb54"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb54-1"><a href="gwas-with-continuous-trait.html#cb54-1" tabindex="-1"></a><span class="fu">dim</span>(zr_result) <span class="co">#get number of significant SNPs</span></span></code></pre></div>
<pre><code>## [1] 82  5</code></pre>
<p>There are a total of 82 significant SNPs after multiple adjusting. However, many SNP IDs are started with kgpXXXX. This kgp identifier is created by Illumina and there is no direct mapping from kpg identifier to rs identifier. I used a GitHub tutorial here to match the kgp identifier with rs identifier based on SNP position. This mapping is done using the <a href="http://genome.ucsc.edu/cgi-bin/hgTables">UCSC Table Browser GRCh38</a>.</p>
<p>The codes below extract the SNPs with kgp identifier using <code>grepl</code> function, and modified the table into UCSC required format. The resulted SNP list is saved in <code>kgp_snps.txt</code>.</p>
<div class="sourceCode" id="cb56"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb56-1"><a href="gwas-with-continuous-trait.html#cb56-1" tabindex="-1"></a><span class="co"># R codes- run in R Console</span></span>
<span id="cb56-2"><a href="gwas-with-continuous-trait.html#cb56-2" tabindex="-1"></a>kgp_snps <span class="ot">&lt;-</span> zr_result[<span class="fu">grepl</span>(<span class="st">&quot;kgp&quot;</span>,zr_result<span class="sc">$</span>SNP),]</span>
<span id="cb56-3"><a href="gwas-with-continuous-trait.html#cb56-3" tabindex="-1"></a><span class="fu">head</span>(kgp_snps)</span></code></pre></div>
<pre><code>##      CHR         SNP     UNADJ    FDR_BH       BP
##    &lt;int&gt;      &lt;char&gt;     &lt;num&gt;     &lt;num&gt;    &lt;int&gt;
## 1:    22 kgp10581167 2.153e-20 1.335e-17 42397026
## 2:    22 kgp12372278 5.256e-19 9.347e-17 42573139
## 3:    22  kgp5009187 6.121e-19 9.347e-17 42580933
## 4:    22   kgp276463 6.967e-19 9.347e-17 42517922
## 5:    22  kgp5109906 1.799e-18 1.810e-16 42567451
## 6:    22  kgp7814563 2.489e-18 2.226e-16 42465788</code></pre>
<div class="sourceCode" id="cb58"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb58-1"><a href="gwas-with-continuous-trait.html#cb58-1" tabindex="-1"></a>kgp_snps<span class="sc">$</span>CHR<span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;chr&quot;</span>,kgp_snps<span class="sc">$</span>CHR)</span>
<span id="cb58-2"><a href="gwas-with-continuous-trait.html#cb58-2" tabindex="-1"></a>kgp_snps<span class="sc">$</span>BP_1<span class="ot">&lt;-</span> kgp_snps<span class="sc">$</span>BP<span class="dv">-1</span></span>
<span id="cb58-3"><a href="gwas-with-continuous-trait.html#cb58-3" tabindex="-1"></a><span class="fu">fwrite</span>(kgp_snps[,<span class="fu">c</span>(<span class="st">&quot;CHR&quot;</span>,<span class="st">&quot;BP_1&quot;</span>,<span class="st">&quot;BP&quot;</span>,<span class="st">&quot;SNP&quot;</span>)], <span class="st">&quot;kgp_snps.txt&quot;</span>, <span class="at">sep=</span><span class="st">&quot; &quot;</span>, <span class="at">quote=</span>F, <span class="at">row.name=</span>F, <span class="at">col.name=</span>F)</span>
<span id="cb58-4"><a href="gwas-with-continuous-trait.html#cb58-4" tabindex="-1"></a><span class="fu">dim</span>(kgp_snps)</span></code></pre></div>
<pre><code>## [1] 53  6</code></pre>
<p>There are 53 significant SNPs with kgp identifier, a large number! However, after mapping using UCSC, we only find 2 SNPs with corresponding rs identifiers. The matched SNPs is saved in file <code>kgp_rs.txt</code>. Now, we match these 4 SNPs with the p-values in the original results.</p>
<div class="sourceCode" id="cb60"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb60-1"><a href="gwas-with-continuous-trait.html#cb60-1" tabindex="-1"></a><span class="co"># R codes- run in R Console</span></span>
<span id="cb60-2"><a href="gwas-with-continuous-trait.html#cb60-2" tabindex="-1"></a>kgp_rs<span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="st">&quot;kgp_rs.txt&quot;</span>,<span class="at">header=</span>F)</span>
<span id="cb60-3"><a href="gwas-with-continuous-trait.html#cb60-3" tabindex="-1"></a><span class="fu">names</span>(kgp_rs)<span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;CHR&quot;</span>,<span class="st">&quot;BP_1&quot;</span>,<span class="st">&quot;BP&quot;</span>,<span class="st">&quot;SNP&quot;</span>)<span class="co"># assign header</span></span>
<span id="cb60-4"><a href="gwas-with-continuous-trait.html#cb60-4" tabindex="-1"></a>idx <span class="ot">&lt;-</span> <span class="fu">match</span>(kgp_rs<span class="sc">$</span>BP, kgp_snps<span class="sc">$</span>BP)<span class="co"># get index of these matched SNPs in results</span></span>
<span id="cb60-5"><a href="gwas-with-continuous-trait.html#cb60-5" tabindex="-1"></a>kgp_match<span class="ot">&lt;-</span> kgp_snps[idx[<span class="sc">!</span><span class="fu">is.na</span>(idx)],]</span>
<span id="cb60-6"><a href="gwas-with-continuous-trait.html#cb60-6" tabindex="-1"></a>kgp_match<span class="sc">$</span>SNP<span class="ot">&lt;-</span> kgp_rs<span class="sc">$</span>SNP[<span class="sc">!</span><span class="fu">is.na</span>(idx)] <span class="co"># replace kgp identifier with rs identifier</span></span>
<span id="cb60-7"><a href="gwas-with-continuous-trait.html#cb60-7" tabindex="-1"></a>kgp_match</span></code></pre></div>
<pre><code>##       CHR         SNP     UNADJ    FDR_BH       BP     BP_1
##    &lt;char&gt;      &lt;char&gt;     &lt;num&gt;     &lt;num&gt;    &lt;int&gt;    &lt;num&gt;
## 1:  chr22  rs60133761 5.314e-17 1.711e-15 42668504 42668503
## 2:  chr22 rs140061401 9.004e-17 2.265e-15 42395960 42395959</code></pre>
<p>Then, these 2 match SNPs are combined with rs SNPs in the results. Print the results and look at the dimension.</p>
<p>There are 31 significant SNPs in total. The most significant SNP is “rs8138080”.</p>
<div class="sourceCode" id="cb62"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb62-1"><a href="gwas-with-continuous-trait.html#cb62-1" tabindex="-1"></a><span class="co"># R codes- run in R Console</span></span>
<span id="cb62-2"><a href="gwas-with-continuous-trait.html#cb62-2" tabindex="-1"></a>rs_snps <span class="ot">&lt;-</span> zr_result[<span class="fu">grepl</span>(<span class="st">&quot;rs&quot;</span>,zr_result<span class="sc">$</span>SNP),]</span>
<span id="cb62-3"><a href="gwas-with-continuous-trait.html#cb62-3" tabindex="-1"></a>kgp_match<span class="sc">$</span>CHR <span class="ot">&lt;-</span> <span class="st">&quot;22&quot;</span></span>
<span id="cb62-4"><a href="gwas-with-continuous-trait.html#cb62-4" tabindex="-1"></a>rs_snps <span class="ot">&lt;-</span> <span class="fu">rbind</span>(rs_snps, kgp_match[,<span class="sc">-</span><span class="dv">6</span>])</span>
<span id="cb62-5"><a href="gwas-with-continuous-trait.html#cb62-5" tabindex="-1"></a><span class="fu">fwrite</span>(rs_snps, <span class="st">&quot;significant_snps_with_rs_for_z-endoxifen-ratio.txt&quot;</span>, <span class="at">sep=</span><span class="st">&quot; &quot;</span>, <span class="at">quote=</span>F, <span class="at">row.name=</span>F, <span class="at">col.name=</span>T)</span>
<span id="cb62-6"><a href="gwas-with-continuous-trait.html#cb62-6" tabindex="-1"></a><span class="fu">dim</span>(rs_snps)</span></code></pre></div>
<pre><code>## [1] 31  5</code></pre>
<div class="sourceCode" id="cb64"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb64-1"><a href="gwas-with-continuous-trait.html#cb64-1" tabindex="-1"></a>rs_snps</span></code></pre></div>
<pre><code>##        CHR         SNP     UNADJ    FDR_BH        BP
##     &lt;char&gt;      &lt;char&gt;     &lt;num&gt;     &lt;num&gt;     &lt;int&gt;
##  1:     22   rs8138080 3.318e-20 1.335e-17  42396371
##  2:     22   rs1062753 5.643e-19 9.347e-17  42392811
##  3:     22   rs4467371 1.365e-18 1.570e-16  42541328
##  4:     22      rs7245 4.102e-18 2.540e-16  42481849
##  5:     22   rs2284087 4.102e-18 2.540e-16  42485671
##  6:     22   rs5758589 4.102e-18 2.540e-16  42518382
##  7:     22   rs5751247 6.923e-18 3.981e-16  42633054
##  8:     22   rs2011944 8.301e-18 4.176e-16  42569999
##  9:     22   rs4453786 1.068e-17 5.059e-16  42563308
## 10:     22   rs5751221 3.812e-17 1.534e-15  42516366
## 11:     22   rs5758651 5.314e-17 1.711e-15  42609148
## 12:     22    rs133335 5.540e-17 1.715e-15  42416056
## 13:     22   rs1063392 9.004e-17 2.265e-15  42454950
## 14:     22    rs932376 4.397e-16 1.041e-14  42614362
## 15:     22   rs8140869 1.996e-13 3.492e-12  42344297
## 16:     22   rs5751191 2.773e-13 4.749e-12  42370991
## 17:     22   rs5751255 5.269e-13 8.656e-12  42648408
## 18:     22   rs5751239 2.071e-12 3.146e-11  42592239
## 19:     22   rs5758511 2.764e-12 3.974e-11  42336172
## 20:     22   rs1052717 7.077e-11 9.822e-10  42281429
## 21:     22    rs133280 8.991e-11 1.206e-09  42255525
## 22:     22    rs134871 1.635e-10 2.157e-09  42652716
## 23:     22    rs742086 1.773e-10 2.302e-09  42552253
## 24:     22   rs6002673 1.695e-07 2.100e-06  42691296
## 25:     22   rs7285782 7.006e-06 8.173e-05  42225018
## 26:     22  rs17365991 1.373e-05 1.579e-04  41790179
## 27:     22   rs9607821 5.431e-05 5.908e-04  41983938
## 28:      8   rs1487173 2.726e-03 2.709e-02 122405332
## 29:      8   rs1042381 2.954e-03 2.900e-02  97614661
## 30:     22  rs60133761 5.314e-17 1.711e-15  42668504
## 31:     22 rs140061401 9.004e-17 2.265e-15  42395960
##        CHR         SNP     UNADJ    FDR_BH        BP</code></pre>
</div>
</div>
<div id="regional-association-plot" class="section level2 hasAnchor" number="5.3">
<h2><span class="header-section-number">5.3</span> Regional Association Plot<a href="gwas-with-continuous-trait.html#regional-association-plot" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><code>make.fancy.locus.plot</code> is an R function for highlighting the statistical strength of an association in the context of the association results for surrounding markers, gene annotations, estimated recombination rates and pairwise correlations between the surrounding markers and the putative associated variant, see the reference page <a href="http://www.broadinstitute.org/diabetes/scandinavs/figures.html" class="uri">http://www.broadinstitute.org/diabetes/scandinavs/figures.html</a></p>
<p>You need provide a file that contains the following data for every SNP across the region of interest: position, p-value, a label to indicate whether a SNP is “typed” or “imputed”, and the r-squared between that SNP and the putative associated variant.</p>
<p>All SNPs in this file will be plotted with their corresponding P-values (as -log10 values) as a function of chromosomal position. SNPs that are “typed” are plotted as diamonds; “imputed” SNPs are plotted as circles. Estimated recombination rates are plotted to reflect the local LD structure around the associated SNP and their correlated proxies (bright red indicating highly correlated, faint red indicating weakly correlated).</p>
<div id="obtain-regional-snps-using-r" class="section level3 hasAnchor" number="5.3.1">
<h3><span class="header-section-number">5.3.1</span> Obtain regional SNPs (using R)<a href="gwas-with-continuous-trait.html#obtain-regional-snps-using-r" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>As most of the significant SNPs are located within chromosome 22, we could take a closer look at this specific chromosome.</p>
<div class="sourceCode" id="cb66"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb66-1"><a href="gwas-with-continuous-trait.html#cb66-1" tabindex="-1"></a><span class="co"># R codes- run in R Console</span></span>
<span id="cb66-2"><a href="gwas-with-continuous-trait.html#cb66-2" tabindex="-1"></a>qassoc<span class="ot">=</span><span class="fu">read.table</span>(<span class="st">&quot;younameit.P3.qassoc&quot;</span>, <span class="at">header=</span>T, <span class="at">stringsAsFactors=</span>F) <span class="co">#load association results</span></span>
<span id="cb66-3"><a href="gwas-with-continuous-trait.html#cb66-3" tabindex="-1"></a>idx<span class="ot">=</span><span class="fu">which</span>(qassoc<span class="sc">$</span>CHR<span class="sc">==</span><span class="dv">22</span> <span class="sc">&amp;</span> qassoc<span class="sc">$</span>BP<span class="sc">&gt;=</span><span class="dv">4200000</span> <span class="sc">&amp;</span> qassoc<span class="sc">$</span>BP<span class="sc">&lt;=</span><span class="dv">52000000</span> <span class="sc">&amp;</span> qassoc<span class="sc">$</span>SNP<span class="sc">!=</span><span class="st">&quot;---&quot;</span> <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">is.na</span>(qassoc<span class="sc">$</span>P)) <span class="co">#index of regional SNPs</span></span>
<span id="cb66-4"><a href="gwas-with-continuous-trait.html#cb66-4" tabindex="-1"></a>TYPE<span class="ot">=</span><span class="fu">rep</span>(<span class="st">&quot;typed&quot;</span>, <span class="fu">length</span>(idx)) <span class="co">#all SNPs are typed SNPs</span></span>
<span id="cb66-5"><a href="gwas-with-continuous-trait.html#cb66-5" tabindex="-1"></a>region<span class="ot">=</span><span class="fu">data.frame</span>(qassoc<span class="sc">$</span>SNP[idx], qassoc<span class="sc">$</span>BP[idx], qassoc<span class="sc">$</span>P[idx], TYPE, qassoc<span class="sc">$</span>R2[idx]) <span class="co">#information for regional SNPs</span></span>
<span id="cb66-6"><a href="gwas-with-continuous-trait.html#cb66-6" tabindex="-1"></a><span class="fu">write.table</span>(region, <span class="st">&quot;regional_SNPs.txt&quot;</span>, <span class="at">row.names=</span>F, <span class="at">col.names=</span><span class="fu">c</span>(<span class="st">&quot;SNP&quot;</span>, <span class="st">&quot;POS&quot;</span>, <span class="st">&quot;PVAL&quot;</span>, <span class="st">&quot;TYPE&quot;</span>, <span class="st">&quot;RSQR&quot;</span>), <span class="at">quote=</span>F, <span class="at">sep=</span><span class="st">&quot; &quot;</span>)</span>
<span id="cb66-7"><a href="gwas-with-continuous-trait.html#cb66-7" tabindex="-1"></a><span class="fu">head</span>(region)</span></code></pre></div>
<pre><code>##   qassoc.SNP.idx. qassoc.BP.idx. qassoc.P.idx.  TYPE qassoc.R2.idx.
## 1      rs12166151       24479193     6.178e-01 typed      1.313e-03
## 2        rs229526       37581422     9.326e-01 typed      3.793e-05
## 3          rs6572       37621445     2.994e-01 typed      5.696e-03
## 4       rs2239774       37637653     5.653e-02 typed      1.921e-02
## 5     kgp11471870       41547154     2.899e-06 typed      1.091e-01
## 6      kgp3940114       41627386     6.354e-05 typed      8.094e-02</code></pre>
<p>The regional association plot is generated by <code>make.fancy.locus.plot</code>. The arguments in this function is:
1. specify which SNP is the key association (“rs8138080”)
2. provide a header title (“rs8138080”)
3. specify which chromosome it is (“22”)
4. provide the locus of the SNP
5. specify the range of p-value
6. provide the adjusted p-value for the top SNP (1.709e-15)(The blue diamond will represent this data point with the red diamond representing the initial P-value (present in the result file).</p>
<p><strong>The following codes might generate several warning messages and you can safely ignore them.</strong></p>
<p>The souce code “<strong>regional_association_plot.r</strong>”, the estimated recombination rate from HapMap and the gene annotations from the UCSC genome browser (using Build 35 coordinates) should be available in the same folder.</p>
<div class="sourceCode" id="cb68"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb68-1"><a href="gwas-with-continuous-trait.html#cb68-1" tabindex="-1"></a><span class="co"># R codes- run in R Console</span></span>
<span id="cb68-2"><a href="gwas-with-continuous-trait.html#cb68-2" tabindex="-1"></a><span class="fu">source</span>(<span class="st">&quot;regional_association_plot.r&quot;</span>) <span class="co">#load source function</span></span>
<span id="cb68-3"><a href="gwas-with-continuous-trait.html#cb68-3" tabindex="-1"></a>locus<span class="ot">=</span><span class="fu">read.table</span>(<span class="st">&quot;regional_SNPs.txt&quot;</span>, <span class="at">header=</span>T, <span class="at">row.names=</span><span class="dv">1</span>) <span class="co">#read information of regional SNPs</span></span>
<span id="cb68-4"><a href="gwas-with-continuous-trait.html#cb68-4" tabindex="-1"></a><span class="fu">png</span>(<span class="st">&quot;assocplot_rs8138080.png&quot;</span>)</span>
<span id="cb68-5"><a href="gwas-with-continuous-trait.html#cb68-5" tabindex="-1"></a><span class="fu">make.fancy.locus.plot</span>(<span class="st">&quot;rs8138080&quot;</span>, <span class="st">&quot;rs8138080&quot;</span>, <span class="st">&quot;22&quot;</span>, locus, <span class="dv">25</span>, <span class="fl">1.709e-15</span>) </span></code></pre></div>
<pre><code>##        START     STOP   SIZE STRAND      GENE
## 1   31993616 32640964 647348      -     LARGE
## 2   31233093 31727237 494144      -      SYN3
## 3   45479067 45890241 411174      +  TBC1D22A
## 4   24462673 24751561 288888      +    MYO18B
## 5   42249554 42533019 283465      -  FLJ23588
## 6   39130791 39357190 226399      -      MKL1
## 7   29415418 29628361 212943      +     OSBP2
## 8   24890033 25100988 210955      +     SEZ6L
## 9   45077249 45253586 176337      -    CELSR1
## 10  44388223 44561366 173143      +    ATXN10
## 11  42720073 42890002 169929      +     PARVB
## 14  30474622 30627554 152932      +    DEPDC5
## 15  38463549 38614294 150745      -    ENTHD1
## 16  43431005 43579199 148194      + LOC553158
## 18  41590274 41735649 145375      -   PACSIN2
## 19  28603752 28748423 144671      +     MTMR3
## 20  41924495 42063785 139290      -    SCUBE1
## 21  35284603 35423403 138800      -    CACNG2
## 24  43597578 43726118 128540      -    PHF21B
## 25  30216814 30339087 122273      +      SFI1
## 26  38291257 38410238 118981      +   CACNA1I
## 27  35001826 35108481 106655      -      MYH9
## 31  44219255 44317551  98296      +     FBLN1
## 32  34464148 34560928  96780      -      RBM9
## 33  28801006 28897614  96608      +   HORMAD2
## 34  28324118 28419137  95019      +       NF2
## 36  39813106 39900578  87472      +     EP300
## 40  40553605 40626873  73268      +    SREBF2
## 42  39577593 39648378  70785      +   XPNPEP3
## 43  38621585 38691884  70299      +     GRAP2
## 44  36832002 36902261  70259      -    PLA2G6
## 45  27793659 27863040  69381      +   KREMEN1
## 46  29932803 30000618  67815      +     LIMK2
## 47  26572211 26639809  67598      -    PITPNB
## 48  30763812 30831199  67387      +    SLC5A1
## 51  31521361 31583581  62220      +     TIMP3
## 52  28048222 28109123  60901      -     AP1B1
## 53  41517029 41577798  60769      -   ARFGAP3
## 54  43393473 43454098  60625      +      PRR5
## 55  40189628 40249492  59864      +      ACO2
## 56  40022066 40080651  58585      +    ZC3H7B
## 59  40880516 40935943  55427      -     TCF20
## 62  27408284 27462376  54092      -     CHEK2
## 63  45400826 45454671  53845      -      CERK
## 64  36939804 36993496  53692      -   C22orf5
## 65  27926506 27980140  53634      +     EMID1
## 66  26468819 26522040  53221      -       MN1
## 68  37239453 37290689  51236      -      DMC1
## 69  30159908 30210258  50350      - EIF4ENIF1
## 70  41760142 41809927  49785      -     TTLL1
## 71  39490136 39539892  49756      -  SLC25A17
## 72  28509155 28558808  49653      -     ASCC2
## 73  41103470 41152894  49424      -     NFAM1
## 74  43908659 43957187  48528      -   C22orf9
## 75  34020414 34068527  48113      +      TOM1
## 78  29880732 29927554  46822      +    RNF185
## 79  28228710 28274198  45488      -     THOC5
## 82  31134843 31177927  43084      -     BPIL2
## 83  40341794 40384552  42758      +     XRCC6
## 85  29647153 29688741  41588      -     MORC2
## 86  42676201 42717312  41111      +    SAMM50
## 88  39966115 40006716  40601      -   RANGAP1
## 90  39091094 39130793  39699      +    RUTBC3
## 91  25246269 25285924  39655      -     TPST2
## 92  36569869 36609137  39268      +  EIF3S6IP
## 93  33978042 34016353  38311      +    HMG2L1
## 95  35785978 35824193  38215      -   TMPRSS6
## 96  42545289 42583257  37968      -   SULT4A1
## 97  38120258 38157632  37374      + MAP3K7IP1
## 98  30939018 30975873  36855      -    SLC5A4
## 99  45013330 45050112  36782      +     GTSE1
## 101 30396795 30433285  36490      -  C22orf30
## 103 28451499 28487523  36024      -     ZMAT5
## 104 36626845 36662525  35680      +   MICALL1
## 105 38070453 38106080  35627      +    SYNGR1
## 106 29012533 29047448  34915      -  TBC1D10A
## 107 30120092 30154725  34633      +      DRG1
## 111 36002923 36035880  32957      +     PSCD4
## 112 25171999 25204374  32375      -      HPS4
## 113 27988836 28021059  32223      +     EWSR1
## 114 39545102 39577187  32085      -      ST13
## 118 41304224 41335460  31236      -   POLDIP3
## 119 44026351 44057180  30829      +   FAM118A
## 121 41339318 41369877  30559      -    CYB5R3
## 123 40719291 40748975  29684      +    WBP2NL
## 125 37146831 37175703  28872      -     KCNJ4
## 126 36210900 36239710  28810      -    CARD10
## 127 41412625 41441374  28749      -    A4GALT
## 129 36466747 36495016  28269      +    TRIOBP
## 135 29117486 29144382  26896      +   SEC14L2
## 138 37011197 37037913  26716      -    CSNK1E
## 139 28275353 28301880  26527      -  NIPSNAP1
## 140 37426468 37452744  26276      +    GTPBP1
## 141 39925812 39951774  25962      +   L3MBTL2
## 143 34860876 34886725  25849      -     APOL3
## 145 36805409 36831152  25743      -  BAIAP2L2
## 146 42901757 42927435  25678      +     PARVG
## 147 44802401 44828037  25636      +  FLJ27365
## 148 40521177 40546803  25626      +   CCDC134
## 149 37538956 37564517  25561      -     NPTXR
## 152 36329002 36354070  25068      +      GGA1
## 153 29052532 29077467  24935      -     SF3A1
## 156 31108122 31132787  24665      -  C22orf28
## 157 34120681 34145048  24367      +      MCM5
## 158 31195263 31219369  24106      +     FBXO7
## 159 35846381 35870462  24081      -     IL2RB
## 161 42644519 42668349  23830      +    PNPLA3
## 163 29801858 29825163  23305      +      SMTN
## 164 37203942 37226816  22874      -     DDX17
## 165 45051840 45073755  21915      +      TRMU
## 166 37851280 37873038  21758      -      CBX7
## 167 39671884 39693168  21284      +      RBX1
## 168 37944218 37965490  21272      -     PDGFB
## 169 33786683 33807932  21249      +       ISX
## 170 40697495 40718723  21228      +     SEPT3
## 171 37455234 37476447  21213      -    UNC84B
## 172 43880262 43901242  20980      +     NUP50
## 174 25212450 25232991  20541      -    TFIP11
## 175 41887126 41907605  20479      -    TTLL12
## 176 41274422 41294886  20464      +    SERHL2
## 177 30046343 30066803  20460      -     PATZ1
## 178 38730191 38750543  20352      +    FAM83F
## 179 39067062 39087251  20189      +      ADSL
## 180 40321435 40341416  19981      - D15Wsu75e
## 181 29327714 29347601  19887      +      TCN2
## 183 35945811 35964805  18994      -      RAC2
## 184 41831254 41850215  18961      +       BIK
## 185 35208795 35227569  18774      -   FOXRED2
## 186 35521245 35540017  18772      -     PVALB
## 188 44130116 44148839  18723      +     RIBC2
## 189 25319915 25338545  18630      -    CRYBB1
## 190 35231397 35249983  18586      -    EIF3S7
## 191 36777902 36796207  18305      +     PICK1
## 193 35478754 35496614  17860      -     RABL4
## 194 40102462 40119828  17366      +       TEF
## 195 36189602 36206885  17283      -      MFNG
## 196 37377157 37394353  17196      +      CBY1
## 197 35581544 35598558  17014      +      NCF4
## 199 29209453 29226236  16783      -   SEC14L4
## 201 35711672 35728339  16667      -  C22orf33
## 203 35642721 35659318  16597      +    CSF2RB
## 204 34327310 34343901  16591      -        MB
## 205 27493260 27509832  16572      +   CCDC117
## 209 38222783 38238636  15853      +    SMCR7L
## 210 37499013 37514661  15648      -     DNAL4
## 211 27462596 27478048  15452      +      HSCB
## 212 37188582 37203945  15363      +    KDELR3
## 213 29297166 29312448  15282      -      PES1
## 217 40394437 40409413  14976      -    NHP2L1
## 219 36361877 36376545  14668      +    SH3BP1
## 220 37406126 37420769  14643      -     JOSD1
## 221 35187592 35202187  14595      -      TXN2
## 223 36922497 36937012  14515      +      MAFF
## 224 34973623 34988077  14454      +     APOL1
## 225 25149855 25164125  14270      +    ASPHD2
## 226 36674217 36688306  14089      +    POLR2F
## 227 34946754 34960500  13746      -     APOL2
## 231 40153991 40167527  13536      -      TOB2
## 233 36551362 36564803  13441      -   ANKRD54
## 235 37761172 37774414  13242      +  APOBEC3F
## 236 30665032 30678143  13111      +     YWHAH
## 237 34101640 34114748  13108      +     HMOX1
## 238 40297411 40310313  12902      -      PMM1
## 239 29179772 29192571  12799      -   SEC14L3
## 240 34261851 34274543  12692      +     RASD2
## 241 40778884 40791273  12389      -      NAGA
## 242 30339034 30351364  12330      -      PISD
## 243 42600459 42612758  12299      -    PNPLA5
## 244 36692819 36705039  12220      -     SOX10
## 245 41221093 41233064  11971      +     SERHL
## 247 35772115 35783930  11815      +    KCTD17
## 248 29843514 29855236  11722      +    PIB5PA
## 249 41872050 41883746  11696      +      TSPO
## 250 34438418 34450030  11612      +     APOL5
## 252 28440897 28452372  11475      +     CABP7
## 254 28200772 28211829  11057      +      NEFH
## 255 30002133 30013028  10895      -  MGC17330
## 256 44001425 44012291  10866      +     UPK3A
## 257 29356866 29367660  10794      +   SLC35E4
## 259 37797635 37808247  10612      +  APOBEC3G
## 260 46337310 46347837  10527      -  FLJ46257
## 261 41853385 41863894  10509      -      MCAT
## 263 37702904 37713283  10379      +  APOBEC3B
## 264 29275177 29285430  10253      -   GAL3ST1
## 265 35740201 35750362  10161      +      MPST
## 266 36664032 36674103  10071      -  C22orf23
## 267 36290753 36300524   9771      -    LGALS2
## 268 30870074 30879797   9723      - LOC150297
## 274 40180221 40189208   8987      -     PHF5A
## 275 36528454 36537403   8949      +      GCAT
## 276 36280999 36289908   8909      +  CDC42EP1
## 280 25342481 25351186   8705      +    CRYBA4
## 282 35731406 35739991   8585      -       TST
## 283 29099358 29107856   8498      - LOC200312
## 285 40659241 40667648   8407      -     CENPM
## 287 36379237 36387439   8202      +      PDXP
## 288 34347539 34355682   8143      - LOC284912
## 289 35900705 35908830   8125      -   C1QTNF6
## 291 27980397 27988468   8071      -    RHBDD3
## 292 25204399 25212451   8052      + LOC402055
## 293 37584749 37592758   8009      -      CBX6
## 295 44972078 44979738   7660      -    PKDREJ
## 297 40411046 40418640   7594      +  FLJ23584
## 298 36088500 36096079   7579      -    LRRC62
## 301 41233072 41240306   7234      -    CGI-96
## 303 36406866 36413978   7112      +     NOL12
## 305 38033389 38040128   6739      -      RPL3
## 306 37817789 37824500   6711      +  APOBEC3H
## 309 28960996 28967302   6306      -       LIF
## 310 44960429 44966712   6283      - LOC150383
## 312 35926744 35932853   6109      -     SSTR3
## 313 27515102 27521114   6012      -      XBP1
## 315 29382592 29388426   5834      -    DUSP18
## 318 28027550 28033328   5778      +    GAS2L1
## 319 29855346 29861023   5677      -    PLA2G3
## 320 37678122 37683688   5566      +  APOBEC3A
## 321 43208986 43214542   5556      -    LDOC1L
## 323 40806030 40811388   5358      -    NDUFA6
## 324 40794754 40799941   5187      +   FAM109B
## 325 36798644 36803649   5005      -   SLC16A8
## 329 40800198 40804787   4589      +  C22orf32
## 330 37734764 37739323   4559      +  APOBEC3C
## 333 40847000 40851379   4379      -    CYP2D6
## 336 36396142 36400307   4165      +    LGALS1
## 337 29642949 29647091   4142      +  C22orf27
## 339 28983374 28987394   4020      -       OSM
## 341 28159125 28162998   3873      +     RFPL1
## 343 38249598 38253360   3762      -  RPS19BP1
## 344 44766857 44770543   3686      -  C22orf26
## 345 39399736 39403318   3582      +     MCHR1
## 349 29146273 29149595   3322      +     MTP18
## 350 31078407 31081702   3295      +     RFPL3
## 351 27779213 27782461   3248      -  C22orf31
## 354 43285755 43288866   3111      + LOC388910
## 355 28487911 28490956   3045      +      UCRC
## 356 30910979 30914020   3041      -     RFPL2
## 358 28033475 28036302   2827      -   RASL10A
## 359 37402453 37405263   2810      +    TOMM22
## 360 29825316 29828105   2789      -      SELM
## 361 39580052 39582630   2578      -    DNAJB7
## 366 28748776 28751048   2272      + LOC400924
## 367 38241068 38243191   2123      +      ATF4
## 369 36543888 36546002   2114      +     GALR3
## 374 40645535 40647282   1747      - TNFRSF13C</code></pre>
<pre><code>## Warning in arrows(max(genes.in.locus[i, ]$START, min.pos), -offset,
## min(genes.in.locus[i, : zero-length arrow is of indeterminate angle and so
## skipped
## Warning in arrows(max(genes.in.locus[i, ]$START, min.pos), -offset,
## min(genes.in.locus[i, : zero-length arrow is of indeterminate angle and so
## skipped</code></pre>
<pre><code>## Warning in arrows(max(genes.in.locus[i, ]$START, min.pos), -offset,
## min(genes.in.locus[i, : zero-length arrow is of indeterminate angle and so
## skipped</code></pre>
<pre><code>## Warning in arrows(max(genes.in.locus[i, ]$START, min.pos), -offset,
## min(genes.in.locus[i, : zero-length arrow is of indeterminate angle and so
## skipped
## Warning in arrows(max(genes.in.locus[i, ]$START, min.pos), -offset,
## min(genes.in.locus[i, : zero-length arrow is of indeterminate angle and so
## skipped</code></pre>
<pre><code>## Warning in arrows(max(genes.in.locus[i, ]$START, min.pos), -offset,
## min(genes.in.locus[i, : zero-length arrow is of indeterminate angle and so
## skipped</code></pre>
<pre><code>## Warning in arrows(max(genes.in.locus[i, ]$START, min.pos), -offset,
## min(genes.in.locus[i, : zero-length arrow is of indeterminate angle and so
## skipped</code></pre>
<pre><code>## Warning in arrows(max(genes.in.locus[i, ]$START, min.pos), -offset,
## min(genes.in.locus[i, : zero-length arrow is of indeterminate angle and so
## skipped
## Warning in arrows(max(genes.in.locus[i, ]$START, min.pos), -offset,
## min(genes.in.locus[i, : zero-length arrow is of indeterminate angle and so
## skipped</code></pre>
<pre><code>## Warning in arrows(max(genes.in.locus[i, ]$START, min.pos), -offset,
## min(genes.in.locus[i, : zero-length arrow is of indeterminate angle and so
## skipped
## Warning in arrows(max(genes.in.locus[i, ]$START, min.pos), -offset,
## min(genes.in.locus[i, : zero-length arrow is of indeterminate angle and so
## skipped
## Warning in arrows(max(genes.in.locus[i, ]$START, min.pos), -offset,
## min(genes.in.locus[i, : zero-length arrow is of indeterminate angle and so
## skipped</code></pre>
<pre><code>## Warning in arrows(max(genes.in.locus[i, ]$START, min.pos), -offset,
## min(genes.in.locus[i, : zero-length arrow is of indeterminate angle and so
## skipped</code></pre>
<pre><code>## Warning in arrows(max(genes.in.locus[i, ]$START, min.pos), -offset,
## min(genes.in.locus[i, : zero-length arrow is of indeterminate angle and so
## skipped
## Warning in arrows(max(genes.in.locus[i, ]$START, min.pos), -offset,
## min(genes.in.locus[i, : zero-length arrow is of indeterminate angle and so
## skipped</code></pre>
<pre><code>## Warning in arrows(max(genes.in.locus[i, ]$START, min.pos), -offset,
## min(genes.in.locus[i, : zero-length arrow is of indeterminate angle and so
## skipped
## Warning in arrows(max(genes.in.locus[i, ]$START, min.pos), -offset,
## min(genes.in.locus[i, : zero-length arrow is of indeterminate angle and so
## skipped</code></pre>
<pre><code>## Warning in arrows(max(genes.in.locus[i, ]$START, min.pos), -offset,
## min(genes.in.locus[i, : zero-length arrow is of indeterminate angle and so
## skipped</code></pre>
<pre><code>## Warning in arrows(max(genes.in.locus[i, ]$START, min.pos), -offset,
## min(genes.in.locus[i, : zero-length arrow is of indeterminate angle and so
## skipped
## Warning in arrows(max(genes.in.locus[i, ]$START, min.pos), -offset,
## min(genes.in.locus[i, : zero-length arrow is of indeterminate angle and so
## skipped</code></pre>
<pre><code>## Warning in arrows(max(genes.in.locus[i, ]$START, min.pos), -offset,
## min(genes.in.locus[i, : zero-length arrow is of indeterminate angle and so
## skipped
## Warning in arrows(max(genes.in.locus[i, ]$START, min.pos), -offset,
## min(genes.in.locus[i, : zero-length arrow is of indeterminate angle and so
## skipped
## Warning in arrows(max(genes.in.locus[i, ]$START, min.pos), -offset,
## min(genes.in.locus[i, : zero-length arrow is of indeterminate angle and so
## skipped</code></pre>
<pre><code>## Warning in arrows(max(genes.in.locus[i, ]$START, min.pos), -offset,
## min(genes.in.locus[i, : zero-length arrow is of indeterminate angle and so
## skipped</code></pre>
<div class="sourceCode" id="cb84"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb84-1"><a href="gwas-with-continuous-trait.html#cb84-1" tabindex="-1"></a><span class="co">#specify which SNP is the key association (&quot;rs8138080&quot;), </span></span>
<span id="cb84-2"><a href="gwas-with-continuous-trait.html#cb84-2" tabindex="-1"></a><span class="co"># provide a header title (&quot;rs8138080&quot;), </span></span>
<span id="cb84-3"><a href="gwas-with-continuous-trait.html#cb84-3" tabindex="-1"></a><span class="co"># specify which chromosome it is (&quot;22&quot;), </span></span>
<span id="cb84-4"><a href="gwas-with-continuous-trait.html#cb84-4" tabindex="-1"></a><span class="co"># provide the locus of the SNP, </span></span>
<span id="cb84-5"><a href="gwas-with-continuous-trait.html#cb84-5" tabindex="-1"></a><span class="co"># specify the range of p-value, </span></span>
<span id="cb84-6"><a href="gwas-with-continuous-trait.html#cb84-6" tabindex="-1"></a><span class="co"># provide the replication or joint p-value (4.335e-14)</span></span>
<span id="cb84-7"><a href="gwas-with-continuous-trait.html#cb84-7" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code></pre></div>
<pre><code>## quartz_off_screen 
##                 2</code></pre>
<div class="sourceCode" id="cb86"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb86-1"><a href="gwas-with-continuous-trait.html#cb86-1" tabindex="-1"></a><span class="co"># R codes- run in R Console</span></span>
<span id="cb86-2"><a href="gwas-with-continuous-trait.html#cb86-2" tabindex="-1"></a>img_fancy <span class="ot">&lt;-</span> <span class="fu">readPNG</span>(<span class="st">&quot;assocplot_rs8138080.png&quot;</span>)</span>
<span id="cb86-3"><a href="gwas-with-continuous-trait.html#cb86-3" tabindex="-1"></a>grid<span class="sc">::</span><span class="fu">grid.raster</span>(img_fancy)</span></code></pre></div>
<p><img src="cshl_gwas_files/figure-html/regmah-1.png" width="672" /></p>
<p>There are definitely a lot more ways to generate a regional association plot. We can use the <code>manhattan</code> function in the <code>qqman</code> package by input a subset of the results. <code>QCGWAS</code> package also has a <code>plot_regional</code> function where you can specify the chromosome or zoom-in region. If you are familiar with R, you can also write your own functions with ggplot or base plot to enable more flexibility in customizing the plots. Below is an example for the <code>plot_regional</code> function.</p>
<p>We need to combine the SNP information and test results in P3.assoc and P3.assoc.adjusted. We use the <code>match</code> function to match the SNP ID in these two files.</p>
<div class="sourceCode" id="cb87"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb87-1"><a href="gwas-with-continuous-trait.html#cb87-1" tabindex="-1"></a><span class="co"># R codes- run in R Console</span></span>
<span id="cb87-2"><a href="gwas-with-continuous-trait.html#cb87-2" tabindex="-1"></a><span class="co"># read in, the SNP doesn&#39;t match</span></span>
<span id="cb87-3"><a href="gwas-with-continuous-trait.html#cb87-3" tabindex="-1"></a>qassoc<span class="ot">=</span><span class="fu">fread</span>(<span class="fu">paste0</span>(<span class="st">&quot;P3.qassoc&quot;</span>), <span class="at">header=</span>T)</span>
<span id="cb87-4"><a href="gwas-with-continuous-trait.html#cb87-4" tabindex="-1"></a>qassoc.adjusted<span class="ot">=</span><span class="fu">fread</span>(<span class="st">&quot;P3.qassoc.adjusted&quot;</span>, <span class="at">header=</span>T)</span>
<span id="cb87-5"><a href="gwas-with-continuous-trait.html#cb87-5" tabindex="-1"></a><span class="fu">head</span>(qassoc.adjusted)</span></code></pre></div>
<pre><code>##      CHR         SNP     UNADJ    FDR_BH
##    &lt;int&gt;      &lt;char&gt;     &lt;num&gt;     &lt;num&gt;
## 1:    22 kgp10581167 2.153e-20 1.335e-17
## 2:    22   rs8138080 3.318e-20 1.335e-17
## 3:    22 kgp12372278 5.256e-19 9.347e-17
## 4:    22   rs1062753 5.643e-19 9.347e-17
## 5:    22  kgp5009187 6.121e-19 9.347e-17
## 6:    22   kgp276463 6.967e-19 9.347e-17</code></pre>
<div class="sourceCode" id="cb89"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb89-1"><a href="gwas-with-continuous-trait.html#cb89-1" tabindex="-1"></a><span class="fu">head</span>(qassoc)</span></code></pre></div>
<pre><code>##      CHR        SNP    BP      P
##    &lt;int&gt;     &lt;char&gt; &lt;int&gt;  &lt;num&gt;
## 1:     0  rs2698846     0     NA
## 2:     0  rs2542903     0     NA
## 3:     0  rs1059667     0     NA
## 4:     0 rs35144699     0 0.5454
## 5:     0 rs35425675     0 0.5770
## 6:     0  rs3876241     0     NA</code></pre>
<div class="sourceCode" id="cb91"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb91-1"><a href="gwas-with-continuous-trait.html#cb91-1" tabindex="-1"></a><span class="co"># R codes- run in R Console</span></span>
<span id="cb91-2"><a href="gwas-with-continuous-trait.html#cb91-2" tabindex="-1"></a><span class="co"># match and merge data</span></span>
<span id="cb91-3"><a href="gwas-with-continuous-trait.html#cb91-3" tabindex="-1"></a>qassoc<span class="ot">&lt;-</span>qassoc[<span class="sc">!</span><span class="fu">is.na</span>(qassoc<span class="sc">$</span>P),]</span>
<span id="cb91-4"><a href="gwas-with-continuous-trait.html#cb91-4" tabindex="-1"></a>idx<span class="ot">&lt;-</span><span class="fu">match</span>(qassoc.adjusted<span class="sc">$</span>SNP,qassoc<span class="sc">$</span>SNP)</span>
<span id="cb91-5"><a href="gwas-with-continuous-trait.html#cb91-5" tabindex="-1"></a>qassoc_sub<span class="ot">&lt;-</span> qassoc[idx,]</span>
<span id="cb91-6"><a href="gwas-with-continuous-trait.html#cb91-6" tabindex="-1"></a>qassoc.adjusted<span class="sc">$</span>BP<span class="ot">&lt;-</span>qassoc_sub<span class="sc">$</span>BP</span>
<span id="cb91-7"><a href="gwas-with-continuous-trait.html#cb91-7" tabindex="-1"></a>qassoc.adjusted<span class="sc">$</span>P<span class="ot">&lt;-</span>qassoc_sub<span class="sc">$</span>P</span></code></pre></div>
<p>Next, we are able to call the <code>plot_regional</code> function to generate regional plot for our specified chromosome number and region. This function uses the base plot function in R instead of ggplot. The <code>col</code> option enables a conditional coloring by highlighting the significant SNPs as yellow. Finally, the function directly saves the figure to PNG under current directory in “regional_association_plot.png”. You can specify the cutoff value for the SNPs you want to label in the plot with <code>name_cutoff_p</code>. By default, this option is NULL and there is no label.</p>
<div class="sourceCode" id="cb92"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb92-1"><a href="gwas-with-continuous-trait.html#cb92-1" tabindex="-1"></a><span class="co"># R codes- run in R Console</span></span>
<span id="cb92-2"><a href="gwas-with-continuous-trait.html#cb92-2" tabindex="-1"></a><span class="co"># generate plot</span></span>
<span id="cb92-3"><a href="gwas-with-continuous-trait.html#cb92-3" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code></pre></div>
<pre><code>## null device 
##           1</code></pre>
<div class="sourceCode" id="cb94"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb94-1"><a href="gwas-with-continuous-trait.html#cb94-1" tabindex="-1"></a><span class="fu">colnames</span>(qassoc.adjusted)<span class="ot">&lt;-</span><span class="fu">c</span>(<span class="st">&quot;CHR&quot;</span>,<span class="st">&quot;MARKER&quot;</span>,<span class="st">&quot;PVALUE&quot;</span>,<span class="st">&quot;FDR_BH&quot;</span>,<span class="st">&quot;POSITION&quot;</span>,<span class="st">&quot;P2&quot;</span>)</span>
<span id="cb94-2"><a href="gwas-with-continuous-trait.html#cb94-2" tabindex="-1"></a><span class="co"># label the most significant SNP</span></span>
<span id="cb94-3"><a href="gwas-with-continuous-trait.html#cb94-3" tabindex="-1"></a><span class="co"># color all significant SNPs</span></span>
<span id="cb94-4"><a href="gwas-with-continuous-trait.html#cb94-4" tabindex="-1"></a><span class="fu">plot_regional</span>(qassoc.adjusted, <span class="at">chr=</span><span class="dv">22</span>, <span class="at">start_pos=</span><span class="dv">41000000</span>,<span class="at">end_pos=</span><span class="dv">51000000</span>,</span>
<span id="cb94-5"><a href="gwas-with-continuous-trait.html#cb94-5" tabindex="-1"></a>              <span class="at">name_cutoff_p=</span><span class="fu">min</span>(qassoc.adjusted<span class="sc">$</span>PVALUE),</span>
<span id="cb94-6"><a href="gwas-with-continuous-trait.html#cb94-6" tabindex="-1"></a>             <span class="at">col=</span><span class="fu">ifelse</span>( qassoc.adjusted<span class="sc">$</span>FDR_BH<span class="sc">&lt;=</span><span class="fl">0.05</span> , <span class="st">&quot;darkgoldenrod1&quot;</span>,<span class="st">&quot;dimgray&quot;</span>))</span>
<span id="cb94-7"><a href="gwas-with-continuous-trait.html#cb94-7" tabindex="-1"></a><span class="co">#plot_regional(qassoc.region, chr=22, start_pos=41000000,end_pos=51000000,name_cutoff_p=NULL) #no label </span></span></code></pre></div>
<div class="sourceCode" id="cb95"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb95-1"><a href="gwas-with-continuous-trait.html#cb95-1" tabindex="-1"></a><span class="co"># R codes- run in R Console</span></span>
<span id="cb95-2"><a href="gwas-with-continuous-trait.html#cb95-2" tabindex="-1"></a>regional_plot <span class="ot">&lt;-</span> <span class="fu">readPNG</span>(<span class="st">&quot;regional_association_plot.png&quot;</span>)</span>
<span id="cb95-3"><a href="gwas-with-continuous-trait.html#cb95-3" tabindex="-1"></a>grid<span class="sc">::</span><span class="fu">grid.raster</span>(regional_plot)</span></code></pre></div>
<p><img src="cshl_gwas_files/figure-html/reg11-1.png" width="672" /></p>
</div>
</div>
<div id="distribution-of-z-endoxifen-ratio-for-rs8138080" class="section level2 hasAnchor" number="5.4">
<h2><span class="header-section-number">5.4</span> Distribution of Z-endoxifen ratio for rs8138080<a href="gwas-with-continuous-trait.html#distribution-of-z-endoxifen-ratio-for-rs8138080" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>After an overview of the regional association plot, it might also be intersting to look at the distribution of phenotype on a specific SNP. Here, we pick the most significant SNP rs8138080 as an example.</p>
<div id="obtain-phenotype-data-z-endoxifen-ratio-using-r" class="section level3 hasAnchor" number="5.4.1">
<h3><span class="header-section-number">5.4.1</span> Obtain phenotype data (Z-endoxifen ratio) (using R)<a href="gwas-with-continuous-trait.html#obtain-phenotype-data-z-endoxifen-ratio-using-r" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The imputed phenotype is in file phenotype0.txt.</p>
<div class="sourceCode" id="cb96"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb96-1"><a href="gwas-with-continuous-trait.html#cb96-1" tabindex="-1"></a><span class="co"># R codes- run in R Console</span></span>
<span id="cb96-2"><a href="gwas-with-continuous-trait.html#cb96-2" tabindex="-1"></a>pheno<span class="ot">=</span><span class="fu">read.table</span>(<span class="st">&quot;phenotype0.txt&quot;</span>) <span class="co">#read phenotype data</span></span>
<span id="cb96-3"><a href="gwas-with-continuous-trait.html#cb96-3" tabindex="-1"></a>zr<span class="ot">=</span>pheno[,<span class="dv">3</span>] <span class="co">#obtain phenotype data of Z-endoxifen ratio (Col 3)</span></span>
<span id="cb96-4"><a href="gwas-with-continuous-trait.html#cb96-4" tabindex="-1"></a><span class="fu">write.table</span>(zr,<span class="st">&quot;z_endoxifen_ratio.txt&quot;</span>,<span class="at">row.names=</span>F,<span class="at">col.names=</span>F,<span class="at">quote=</span>F,<span class="at">sep=</span><span class="st">&quot; &quot;</span>) <span class="co">#save phenotype data of Z-endoxifen ratio</span></span></code></pre></div>
</div>
<div id="obtain-genotype-for-significant-snps-using-r" class="section level3 hasAnchor" number="5.4.2">
<h3><span class="header-section-number">5.4.2</span> Obtain genotype for significant SNPs (using R)<a href="gwas-with-continuous-trait.html#obtain-genotype-for-significant-snps-using-r" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The following code finds index for the significant SNPs rs8138080 in the genotype.map file and then extract the genotype from the genotype.ped file. Note that the number of columns in PED file doesn’t match that in MAP file. PED file has 6 column of family information in front, and there are 2 alleles for each SNP.</p>
<div class="sourceCode" id="cb97"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb97-1"><a href="gwas-with-continuous-trait.html#cb97-1" tabindex="-1"></a><span class="co"># R codes- run in R Console</span></span>
<span id="cb97-2"><a href="gwas-with-continuous-trait.html#cb97-2" tabindex="-1"></a><span class="co"># index of significant SNPs in PED file (using R)</span></span>
<span id="cb97-3"><a href="gwas-with-continuous-trait.html#cb97-3" tabindex="-1"></a>map<span class="ot">=</span><span class="fu">fread</span>(<span class="st">&quot;genotype_1k.map&quot;</span>,<span class="at">header=</span>F) <span class="co">#read MAP file</span></span>
<span id="cb97-4"><a href="gwas-with-continuous-trait.html#cb97-4" tabindex="-1"></a><span class="fu">names</span>(map)<span class="ot">=</span><span class="fu">c</span>(<span class="st">&quot;CHR&quot;</span>, <span class="st">&quot;SNP&quot;</span>, <span class="st">&quot;GP&quot;</span>, <span class="st">&quot;BP&quot;</span>) <span class="co">#specify names of MAP file</span></span>
<span id="cb97-5"><a href="gwas-with-continuous-trait.html#cb97-5" tabindex="-1"></a>index<span class="ot">=</span><span class="fu">which</span>(map<span class="sc">$</span>SNP <span class="sc">%in%</span> <span class="st">&quot;rs8138080&quot;</span>) <span class="co">#index of rs8138080 in MAP file</span></span>
<span id="cb97-6"><a href="gwas-with-continuous-trait.html#cb97-6" tabindex="-1"></a>genoidx<span class="ot">=</span><span class="fu">sort</span>(<span class="fu">c</span>(<span class="dv">2</span><span class="sc">*</span>index<span class="sc">+</span><span class="dv">5</span>, <span class="dv">2</span><span class="sc">*</span>index<span class="sc">+</span><span class="dv">6</span>)) <span class="co">#index of rs8138080 alleles in PED file</span></span>
<span id="cb97-7"><a href="gwas-with-continuous-trait.html#cb97-7" tabindex="-1"></a>genoidx</span></code></pre></div>
<pre><code>## [1] 1901 1902</code></pre>
<div class="sourceCode" id="cb99"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb99-1"><a href="gwas-with-continuous-trait.html#cb99-1" tabindex="-1"></a><span class="co"># 1901 1902 # this is the index of rs8138080 alleles in PED file</span></span></code></pre></div>
<div class="sourceCode" id="cb100"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb100-1"><a href="gwas-with-continuous-trait.html#cb100-1" tabindex="-1"></a><span class="co"># bash codes- run in terminal</span></span>
<span id="cb100-2"><a href="gwas-with-continuous-trait.html#cb100-2" tabindex="-1"></a><span class="co"># extract genotype for rs8138080</span></span>
<span id="cb100-3"><a href="gwas-with-continuous-trait.html#cb100-3" tabindex="-1"></a><span class="fu">awk</span> <span class="st">&#39;{print $1901 $1902}&#39;</span> <span class="op">&lt;</span> genotype_1k.ped <span class="op">&gt;</span> rs8138080.txt</span></code></pre></div>
</div>
<div id="create-box-plot-using-r" class="section level3 hasAnchor" number="5.4.3">
<h3><span class="header-section-number">5.4.3</span> Create Box plot (using R)<a href="gwas-with-continuous-trait.html#create-box-plot-using-r" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The following code generate the box plot for the phenotype z-endoxifen ratio by the genotypes of SNP rs8138080.</p>
<div class="sourceCode" id="cb101"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb101-1"><a href="gwas-with-continuous-trait.html#cb101-1" tabindex="-1"></a><span class="co"># R codes- run in R Console</span></span>
<span id="cb101-2"><a href="gwas-with-continuous-trait.html#cb101-2" tabindex="-1"></a>pheno<span class="ot">=</span><span class="fu">read.table</span>(<span class="st">&quot;z_endoxifen_ratio.txt&quot;</span>) <span class="co">#read phenotype data of Z-endoxifen ratio</span></span>
<span id="cb101-3"><a href="gwas-with-continuous-trait.html#cb101-3" tabindex="-1"></a>geno<span class="ot">=</span><span class="fu">read.table</span>(<span class="st">&quot;rs8138080.txt&quot;</span>, <span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>) <span class="co">#read genotype data of rs8138080</span></span>
<span id="cb101-4"><a href="gwas-with-continuous-trait.html#cb101-4" tabindex="-1"></a>pheno<span class="ot">=</span><span class="fu">as.matrix</span>(pheno) <span class="co">#create a matrix from the phenotype data</span></span>
<span id="cb101-5"><a href="gwas-with-continuous-trait.html#cb101-5" tabindex="-1"></a>geno<span class="ot">=</span><span class="fu">as.matrix</span>(geno) <span class="co">#create a matrix from the genotype data</span></span>
<span id="cb101-6"><a href="gwas-with-continuous-trait.html#cb101-6" tabindex="-1"></a><span class="fu">table</span>(geno) <span class="co">#number of individuals having AA, AG and GG</span></span></code></pre></div>
<pre><code>## geno
## 00 AA AG GG 
##  1 20 85 86</code></pre>
<div class="sourceCode" id="cb103"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb103-1"><a href="gwas-with-continuous-trait.html#cb103-1" tabindex="-1"></a><span class="fu">boxplot</span>(pheno<span class="sc">~</span>geno, <span class="at">xlab=</span><span class="st">&quot;Genotype of SNP rs8138080&quot;</span>, <span class="at">ylab=</span><span class="st">&quot;Z-endoxifen Ratio&quot;</span>) <span class="co">#create Box Plot</span></span></code></pre></div>
<p><img src="cshl_gwas_files/figure-html/box-1.png" width="672" /></p>
</div>
</div>
<div id="proportion-of-variation-explained-by-rs8138080" class="section level2 hasAnchor" number="5.5">
<h2><span class="header-section-number">5.5</span> Proportion of Variation Explained by rs8138080<a href="gwas-with-continuous-trait.html#proportion-of-variation-explained-by-rs8138080" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Marginal <span class="math inline">\(R^2\)</span> reveals the coefficient of determination of a full regression model that contains one or more SNPs. Here we calculate the marginal <span class="math inline">\(R^2\)</span> to assess proportion of variation that can be explained by SNP rs8138080.</p>
<div id="recode-genotype-of-significant-snps-using-r" class="section level3 hasAnchor" number="5.5.1">
<h3><span class="header-section-number">5.5.1</span> Recode genotype of significant SNPs (using R)<a href="gwas-with-continuous-trait.html#recode-genotype-of-significant-snps-using-r" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>We wll first recode the genotype into numbers:</p>
<ul>
<li>AA as “0”</li>
<li>AG as “1”</li>
<li>GG as “2”</li>
</ul>
<div class="sourceCode" id="cb104"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb104-1"><a href="gwas-with-continuous-trait.html#cb104-1" tabindex="-1"></a><span class="co"># R codes- run in R Console</span></span>
<span id="cb104-2"><a href="gwas-with-continuous-trait.html#cb104-2" tabindex="-1"></a>geno<span class="ot">=</span><span class="fu">read.table</span>(<span class="st">&quot;rs8138080.txt&quot;</span>) <span class="co">#read genotype data of rs8138080</span></span>
<span id="cb104-3"><a href="gwas-with-continuous-trait.html#cb104-3" tabindex="-1"></a>geno<span class="ot">=</span><span class="fu">as.matrix</span>(geno) <span class="co">#create a matrix from the genotype data</span></span>
<span id="cb104-4"><a href="gwas-with-continuous-trait.html#cb104-4" tabindex="-1"></a>geno[geno<span class="sc">==</span><span class="st">&quot;GG&quot;</span>]<span class="ot">=</span><span class="dv">2</span> <span class="co">#replace GG with 2</span></span>
<span id="cb104-5"><a href="gwas-with-continuous-trait.html#cb104-5" tabindex="-1"></a>geno[geno<span class="sc">==</span><span class="st">&quot;AG&quot;</span><span class="sc">|</span>geno<span class="sc">==</span><span class="st">&quot;GA&quot;</span>]<span class="ot">=</span><span class="dv">1</span> <span class="co">#replace AG and GA with 1</span></span>
<span id="cb104-6"><a href="gwas-with-continuous-trait.html#cb104-6" tabindex="-1"></a>geno[geno<span class="sc">==</span><span class="st">&quot;AA&quot;</span>]<span class="ot">=</span><span class="dv">0</span> <span class="co">#replace AA with 0</span></span>
<span id="cb104-7"><a href="gwas-with-continuous-trait.html#cb104-7" tabindex="-1"></a><span class="fu">write.table</span>(geno, <span class="st">&quot;rs8138080_geno.txt&quot;</span>, <span class="at">row.names=</span>F, <span class="at">col.names=</span>F, <span class="at">quote=</span>F, <span class="at">sep=</span><span class="st">&quot; &quot;</span>) <span class="co">#save recoded genotype data</span></span></code></pre></div>
</div>
<div id="calculate-marginal-r2-using-r" class="section level3 hasAnchor" number="5.5.2">
<h3><span class="header-section-number">5.5.2</span> Calculate marginal <span class="math inline">\(R^2\)</span> (using R)<a href="gwas-with-continuous-trait.html#calculate-marginal-r2-using-r" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The <span class="math inline">\(R^2\)</span> value is included in the linear regression results.</p>
<div class="sourceCode" id="cb105"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb105-1"><a href="gwas-with-continuous-trait.html#cb105-1" tabindex="-1"></a><span class="co"># R codes- run in R Console</span></span>
<span id="cb105-2"><a href="gwas-with-continuous-trait.html#cb105-2" tabindex="-1"></a>pheno<span class="ot">=</span><span class="fu">read.table</span>(<span class="st">&quot;z_endoxifen_ratio.txt&quot;</span>) <span class="co">#read phenotype data </span></span>
<span id="cb105-3"><a href="gwas-with-continuous-trait.html#cb105-3" tabindex="-1"></a>geno<span class="ot">=</span><span class="fu">read.table</span>(<span class="st">&quot;rs8138080_geno.txt&quot;</span>) <span class="co">#read genotype data of rs8138080</span></span>
<span id="cb105-4"><a href="gwas-with-continuous-trait.html#cb105-4" tabindex="-1"></a>pheno<span class="ot">=</span><span class="fu">as.matrix</span>(pheno) <span class="co">#create a matrix from the phenotype data</span></span>
<span id="cb105-5"><a href="gwas-with-continuous-trait.html#cb105-5" tabindex="-1"></a>geno<span class="ot">=</span><span class="fu">as.matrix</span>(geno) <span class="co">#create a matrix from the genotype data</span></span>
<span id="cb105-6"><a href="gwas-with-continuous-trait.html#cb105-6" tabindex="-1"></a>fit<span class="ot">=</span><span class="fu">lm</span>(pheno<span class="sc">~</span>geno) <span class="co">#fit a linear regression of phenotype against genotype</span></span>
<span id="cb105-7"><a href="gwas-with-continuous-trait.html#cb105-7" tabindex="-1"></a><span class="fu">summary</span>(fit)<span class="sc">$</span>r.squared <span class="co">#R squared of the regression</span></span></code></pre></div>
<pre><code>## [1] 0.3582235</code></pre>
<p>The <span class="math inline">\(R^2\)</span> is 0.374 in this case, meaning that about 37.4% of variation is explained by rs8138080.</p>
</div>
</div>
<div id="practice-time" class="section level2 hasAnchor" number="5.6">
<h2><span class="header-section-number">5.6</span> PRACTICE TIME<a href="gwas-with-continuous-trait.html#practice-time" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="practice-1" class="section level3 hasAnchor" number="5.6.1">
<h3><span class="header-section-number">5.6.1</span> Practice 1<a href="gwas-with-continuous-trait.html#practice-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>Calculate the marginal <span class="math inline">\(R^2\)</span> for other significant SNPs, how large is it compared with the most significant SNP <code>rs8138080</code>? Write several sentences regarding your results.</strong></p>
<p><strong>Instruction</strong>: The first 4 significant SNPs besides rs8138080, with p-values from smallest to largest, are , <code>rs4467371</code>,<code>rs2011944</code>, <code>rs1062753</code>, <code>rs4453786</code>. The genotype data for these 4 SNPs is already prepared for you.(<code>rsXXXX_geno.txt</code>,replace <code>XXXX</code> with SNP ID.) Following the codes in <strong>section 11.2</strong>, read in the data and perform linear regression of phenotype against genotype, then calculate their <span class="math inline">\(R^2\)</span>. Report their values.</p>
<div class="sourceCode" id="cb107"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb107-1"><a href="gwas-with-continuous-trait.html#cb107-1" tabindex="-1"></a><span class="co"># R codes- run in R Console</span></span>
<span id="cb107-2"><a href="gwas-with-continuous-trait.html#cb107-2" tabindex="-1"></a><span class="co"># write your code here</span></span></code></pre></div>
</div>
<div id="practice-2" class="section level3 hasAnchor" number="5.6.2">
<h3><span class="header-section-number">5.6.2</span> Practice 2<a href="gwas-with-continuous-trait.html#practice-2" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>In Practice 1, we calculated the <span class="math inline">\(R^2\)</span>, proportion of variation, explained by single SNP. What will happen if we include all these 5 most significant SNP in the linear regression model? How will this affect <span class="math inline">\(R^2\)</span>? Write a few sentences to describe what you observe.</strong></p>
<p><strong>Instructions</strong>: The codes to read in genotype for all 5 SNPs is already provided. <code>geno</code> is a combined genotype data with 5 columns, with each represents a SNP. Repeat the last two lines in section 11.2 to calculate the <span class="math inline">\(R^2\)</span>.</p>
<div class="sourceCode" id="cb108"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb108-1"><a href="gwas-with-continuous-trait.html#cb108-1" tabindex="-1"></a><span class="co"># R codes- run in R Console</span></span>
<span id="cb108-2"><a href="gwas-with-continuous-trait.html#cb108-2" tabindex="-1"></a><span class="co"># here is the code to read in phenotype and combine genotype data for all 5 SNPs</span></span>
<span id="cb108-3"><a href="gwas-with-continuous-trait.html#cb108-3" tabindex="-1"></a>pheno<span class="ot">=</span><span class="fu">read.table</span>(<span class="st">&quot;z_endoxifen_ratio.txt&quot;</span>) <span class="co">#read phenotype data </span></span>
<span id="cb108-4"><a href="gwas-with-continuous-trait.html#cb108-4" tabindex="-1"></a>geno1<span class="ot">=</span><span class="fu">read.table</span>(<span class="st">&quot;rs8138080_geno.txt&quot;</span>) <span class="co">#read genotype data of rs8138080</span></span>
<span id="cb108-5"><a href="gwas-with-continuous-trait.html#cb108-5" tabindex="-1"></a>geno2<span class="ot">=</span><span class="fu">read.table</span>(<span class="st">&quot;rs4467371_geno.txt&quot;</span>)</span>
<span id="cb108-6"><a href="gwas-with-continuous-trait.html#cb108-6" tabindex="-1"></a>geno3<span class="ot">=</span><span class="fu">read.table</span>(<span class="st">&quot;rs2011944_geno.txt&quot;</span>)</span>
<span id="cb108-7"><a href="gwas-with-continuous-trait.html#cb108-7" tabindex="-1"></a>geno4<span class="ot">=</span><span class="fu">read.table</span>(<span class="st">&quot;rs1062753_geno.txt&quot;</span>)</span>
<span id="cb108-8"><a href="gwas-with-continuous-trait.html#cb108-8" tabindex="-1"></a>geno5<span class="ot">=</span><span class="fu">read.table</span>(<span class="st">&quot;rs4453786_geno.txt&quot;</span>)</span>
<span id="cb108-9"><a href="gwas-with-continuous-trait.html#cb108-9" tabindex="-1"></a>geno<span class="ot">=</span><span class="fu">cbind</span>(geno1, geno2, geno3, geno4, geno5)</span>
<span id="cb108-10"><a href="gwas-with-continuous-trait.html#cb108-10" tabindex="-1"></a>pheno<span class="ot">=</span><span class="fu">as.matrix</span>(pheno) <span class="co">#create a matrix from the phenotype data</span></span>
<span id="cb108-11"><a href="gwas-with-continuous-trait.html#cb108-11" tabindex="-1"></a>geno<span class="ot">=</span><span class="fu">as.matrix</span>(geno) <span class="co">#create a matrix from the genotype data</span></span></code></pre></div>
<div class="sourceCode" id="cb109"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb109-1"><a href="gwas-with-continuous-trait.html#cb109-1" tabindex="-1"></a><span class="co"># R codes- run in R Console</span></span>
<span id="cb109-2"><a href="gwas-with-continuous-trait.html#cb109-2" tabindex="-1"></a><span class="do">## write your code here to fit linear regression and calculate R^2</span></span></code></pre></div>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="gwas-case-study-data-preparation.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="gwas-with-binary-trait.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/04-gwas_part2.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["cshl_gwas.pdf", "cshl_gwas.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
