<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 3 GWAS Tools and Data Preprocessing Workflow with Toy Example | CSHL course - GWAS Hands-on Analysis</title>
  <meta name="description" content="This is the course material for the CSHL hands-on session in GWAS and eQTL." />
  <meta name="generator" content="bookdown 0.41 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 3 GWAS Tools and Data Preprocessing Workflow with Toy Example | CSHL course - GWAS Hands-on Analysis" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This is the course material for the CSHL hands-on session in GWAS and eQTL." />
  <meta name="github-repo" content="rstudio/cshl_gwas" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 3 GWAS Tools and Data Preprocessing Workflow with Toy Example | CSHL course - GWAS Hands-on Analysis" />
  
  <meta name="twitter:description" content="This is the course material for the CSHL hands-on session in GWAS and eQTL." />
  

<meta name="author" content="Min Zhang, Danni Liu- University of California, Irvine; Purdue University" />


<meta name="date" content="2025-06-09" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="an-overview-of-regression-with-toy-examples.html"/>
<link rel="next" href="gwas-case-study-data-preparation.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">CSHL Course- GWAS and EQTL Hands-on Analysis</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Welcome to the GWAS and eQTL Workshop</a>
<ul>
<li class="chapter" data-level="1.0.1" data-path="index.html"><a href="index.html#overview"><i class="fa fa-check"></i><b>1.0.1</b> Overview</a></li>
<li class="chapter" data-level="1.0.2" data-path="index.html"><a href="index.html#folder-structure"><i class="fa fa-check"></i><b>1.0.2</b> Folder structure:</a></li>
<li class="chapter" data-level="1.0.3" data-path="index.html"><a href="index.html#programming-languages"><i class="fa fa-check"></i><b>1.0.3</b> Programming Languages:</a></li>
<li class="chapter" data-level="1.0.4" data-path="index.html"><a href="index.html#required-gwas-tools"><i class="fa fa-check"></i><b>1.0.4</b> Required GWAS Tools</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="an-overview-of-regression-with-toy-examples.html"><a href="an-overview-of-regression-with-toy-examples.html"><i class="fa fa-check"></i><b>2</b> An Overview of Regression with Toy Examples</a>
<ul>
<li class="chapter" data-level="2.1" data-path="an-overview-of-regression-with-toy-examples.html"><a href="an-overview-of-regression-with-toy-examples.html#environment-setting-and-packages"><i class="fa fa-check"></i><b>2.1</b> Environment Setting and Packages</a></li>
<li class="chapter" data-level="2.2" data-path="an-overview-of-regression-with-toy-examples.html"><a href="an-overview-of-regression-with-toy-examples.html#toy-example-data"><i class="fa fa-check"></i><b>2.2</b> Toy Example Data</a></li>
<li class="chapter" data-level="2.3" data-path="an-overview-of-regression-with-toy-examples.html"><a href="an-overview-of-regression-with-toy-examples.html#logistic-regression"><i class="fa fa-check"></i><b>2.3</b> Logistic Regression</a></li>
<li class="chapter" data-level="2.4" data-path="an-overview-of-regression-with-toy-examples.html"><a href="an-overview-of-regression-with-toy-examples.html#roc-curve-and-auc-for-hoxb13"><i class="fa fa-check"></i><b>2.4</b> ROC curve and AUC for HOXB13</a></li>
<li class="chapter" data-level="2.5" data-path="an-overview-of-regression-with-toy-examples.html"><a href="an-overview-of-regression-with-toy-examples.html#practice"><i class="fa fa-check"></i><b>2.5</b> Practice</a></li>
<li class="chapter" data-level="2.6" data-path="an-overview-of-regression-with-toy-examples.html"><a href="an-overview-of-regression-with-toy-examples.html#comparison-of-hoxb13-roc-curve-and-nhlh1-roc-curve"><i class="fa fa-check"></i><b>2.6</b> Comparison of HOXB13 ROC curve and NHLH1 ROC curve</a></li>
<li class="chapter" data-level="2.7" data-path="an-overview-of-regression-with-toy-examples.html"><a href="an-overview-of-regression-with-toy-examples.html#linear-regression"><i class="fa fa-check"></i><b>2.7</b> Linear Regression</a>
<ul>
<li class="chapter" data-level="2.7.1" data-path="an-overview-of-regression-with-toy-examples.html"><a href="an-overview-of-regression-with-toy-examples.html#linear-regression-of-hoxb13-vs.-il17br"><i class="fa fa-check"></i><b>2.7.1</b> Linear regression of HOXB13 vs. IL17BR</a></li>
</ul></li>
<li class="chapter" data-level="2.8" data-path="an-overview-of-regression-with-toy-examples.html"><a href="an-overview-of-regression-with-toy-examples.html#variable-selectionforward-backward-and-stepwise-selection"><i class="fa fa-check"></i><b>2.8</b> Variable selection–Forward, backward and stepwise selection</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html"><i class="fa fa-check"></i><b>3</b> GWAS Tools and Data Preprocessing Workflow with Toy Example</a>
<ul>
<li class="chapter" data-level="3.1" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#environment-setting"><i class="fa fa-check"></i><b>3.1</b> Environment Setting</a></li>
<li class="chapter" data-level="3.2" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#gwas-tools"><i class="fa fa-check"></i><b>3.2</b> GWAS Tools</a></li>
<li class="chapter" data-level="3.3" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#quality-control"><i class="fa fa-check"></i><b>3.3</b> Quality Control</a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#an-example-of-summary-statistics"><i class="fa fa-check"></i><b>3.3.1</b> An Example of Summary Statistics</a></li>
<li class="chapter" data-level="3.3.2" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#an-example-of-inclusion-thresholds-for-qc"><i class="fa fa-check"></i><b>3.3.2</b> An Example of Inclusion Thresholds for QC</a></li>
<li class="chapter" data-level="3.3.3" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#remove-closely-related-individuals"><i class="fa fa-check"></i><b>3.3.3</b> Remove Closely Related Individuals</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#an-imputation-example"><i class="fa fa-check"></i><b>3.4</b> An Imputation Example</a></li>
<li class="chapter" data-level="3.5" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#population-stratification-correction-using-plink"><i class="fa fa-check"></i><b>3.5</b> Population Stratification Correction Using PLINK</a>
<ul>
<li class="chapter" data-level="3.5.1" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#pca-on-whole-genome-data-using-plink"><i class="fa fa-check"></i><b>3.5.1</b> PCA on whole genome data using PLINK</a></li>
<li class="chapter" data-level="3.5.2" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#test-the-significance-of-principle-components-pcs"><i class="fa fa-check"></i><b>3.5.2</b> Test the Significance of Principle Components (PCs)</a></li>
</ul></li>
<li class="chapter" data-level="3.6" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#simple-association-test"><i class="fa fa-check"></i><b>3.6</b> Simple Association Test</a>
<ul>
<li class="chapter" data-level="3.6.1" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#quantitative-trait-association-test-for-continuous-trait"><i class="fa fa-check"></i><b>3.6.1</b> Quantitative Trait Association Test for Continuous Trait</a></li>
<li class="chapter" data-level="3.6.2" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#casecontrol-asssociation-test-for-binary-trait"><i class="fa fa-check"></i><b>3.6.2</b> Case/Control Asssociation Test for Binary Trait</a></li>
</ul></li>
<li class="chapter" data-level="3.7" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#quantitative-trait-association-test-for-continuous-trait-with-covariates"><i class="fa fa-check"></i><b>3.7</b> Quantitative Trait Association Test for Continuous Trait with Covariates</a>
<ul>
<li class="chapter" data-level="3.7.1" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#covariates-adjustments-using-r"><i class="fa fa-check"></i><b>3.7.1</b> Covariates adjustments (using R):</a></li>
<li class="chapter" data-level="3.7.2" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#association-test-using-plink"><i class="fa fa-check"></i><b>3.7.2</b> Association test using plink</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="gwas-case-study-data-preparation.html"><a href="gwas-case-study-data-preparation.html"><i class="fa fa-check"></i><b>4</b> GWAS Case Study-Data Preparation</a>
<ul>
<li class="chapter" data-level="4.1" data-path="gwas-case-study-data-preparation.html"><a href="gwas-case-study-data-preparation.html#environment-setting-and-packages-1"><i class="fa fa-check"></i><b>4.1</b> Environment setting and packages</a></li>
<li class="chapter" data-level="4.2" data-path="gwas-case-study-data-preparation.html"><a href="gwas-case-study-data-preparation.html#accessing-data"><i class="fa fa-check"></i><b>4.2</b> Accessing data</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="gwas-case-study-data-preparation.html"><a href="gwas-case-study-data-preparation.html#data-availability"><i class="fa fa-check"></i><b>4.2.1</b> Data availability</a></li>
<li class="chapter" data-level="4.2.2" data-path="gwas-case-study-data-preparation.html"><a href="gwas-case-study-data-preparation.html#data-files-documentation"><i class="fa fa-check"></i><b>4.2.2</b> Data files documentation</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="gwas-case-study-data-preparation.html"><a href="gwas-case-study-data-preparation.html#preprocessing-phenotype-data"><i class="fa fa-check"></i><b>4.3</b> Preprocessing Phenotype Data</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="gwas-case-study-data-preparation.html"><a href="gwas-case-study-data-preparation.html#extracting-phenotype-data"><i class="fa fa-check"></i><b>4.3.1</b> Extracting phenotype data</a></li>
<li class="chapter" data-level="4.3.2" data-path="gwas-case-study-data-preparation.html"><a href="gwas-case-study-data-preparation.html#summary-of-phenotypes-data"><i class="fa fa-check"></i><b>4.3.2</b> Summary of Phenotypes Data</a></li>
<li class="chapter" data-level="4.3.3" data-path="gwas-case-study-data-preparation.html"><a href="gwas-case-study-data-preparation.html#histogram-of-each-trait"><i class="fa fa-check"></i><b>4.3.3</b> Histogram of each trait</a></li>
<li class="chapter" data-level="4.3.4" data-path="gwas-case-study-data-preparation.html"><a href="gwas-case-study-data-preparation.html#handling-the-missing-values-in-the-phenotypes-and-covariates"><i class="fa fa-check"></i><b>4.3.4</b> Handling the missing values in the phenotypes and covariates</a></li>
<li class="chapter" data-level="4.3.5" data-path="gwas-case-study-data-preparation.html"><a href="gwas-case-study-data-preparation.html#checking-normality-assumptions-for-each-trait"><i class="fa fa-check"></i><b>4.3.5</b> Checking normality assumptions for each trait</a></li>
<li class="chapter" data-level="4.3.6" data-path="gwas-case-study-data-preparation.html"><a href="gwas-case-study-data-preparation.html#normalize-phenotype-data"><i class="fa fa-check"></i><b>4.3.6</b> Normalize phenotype data</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="gwas-case-study-data-preparation.html"><a href="gwas-case-study-data-preparation.html#covariates-adjustment"><i class="fa fa-check"></i><b>4.4</b> Covariates Adjustment</a></li>
<li class="chapter" data-level="4.5" data-path="gwas-case-study-data-preparation.html"><a href="gwas-case-study-data-preparation.html#prepare-plink-data"><i class="fa fa-check"></i><b>4.5</b> Prepare PLINK data</a>
<ul>
<li class="chapter" data-level="4.5.1" data-path="gwas-case-study-data-preparation.html"><a href="gwas-case-study-data-preparation.html#step-1-prepare-plink-ped-file"><i class="fa fa-check"></i><b>4.5.1</b> Step 1: prepare PLINK ped file</a></li>
<li class="chapter" data-level="4.5.2" data-path="gwas-case-study-data-preparation.html"><a href="gwas-case-study-data-preparation.html#step-2-prepare-plink-map-file"><i class="fa fa-check"></i><b>4.5.2</b> Step 2: prepare PLINK map file</a></li>
<li class="chapter" data-level="4.5.3" data-path="gwas-case-study-data-preparation.html"><a href="gwas-case-study-data-preparation.html#step-3-prepare-plink-alternate-phenotype-file-in-linux"><i class="fa fa-check"></i><b>4.5.3</b> Step 3 : prepare PLINK alternate phenotype file (in Linux)</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="gwas-with-continuous-trait.html"><a href="gwas-with-continuous-trait.html"><i class="fa fa-check"></i><b>5</b> GWAS with Continuous Trait</a>
<ul>
<li class="chapter" data-level="5.1" data-path="gwas-with-continuous-trait.html"><a href="gwas-with-continuous-trait.html#manhattan-plot-and-q-q-plot"><i class="fa fa-check"></i><b>5.1</b> Manhattan Plot and Q-Q Plot</a>
<ul>
<li class="chapter" data-level="5.1.1" data-path="gwas-with-continuous-trait.html"><a href="gwas-with-continuous-trait.html#generate-manhattan-plot"><i class="fa fa-check"></i><b>5.1.1</b> Generate manhattan plot</a></li>
<li class="chapter" data-level="5.1.2" data-path="gwas-with-continuous-trait.html"><a href="gwas-with-continuous-trait.html#generate-q-q-plot"><i class="fa fa-check"></i><b>5.1.2</b> Generate Q-Q plot</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="gwas-with-continuous-trait.html"><a href="gwas-with-continuous-trait.html#multiple-test-correction"><i class="fa fa-check"></i><b>5.2</b> Multiple-Test Correction</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="gwas-with-continuous-trait.html"><a href="gwas-with-continuous-trait.html#prepare-r-input-files-in-linux"><i class="fa fa-check"></i><b>5.2.1</b> Prepare R input files (in Linux)</a></li>
<li class="chapter" data-level="5.2.2" data-path="gwas-with-continuous-trait.html"><a href="gwas-with-continuous-trait.html#significant-snps-after-a-multiple-test-correction-of-fdr-using-r"><i class="fa fa-check"></i><b>5.2.2</b> Significant SNPs after a multiple-test correction of FDR (using R)</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="gwas-with-continuous-trait.html"><a href="gwas-with-continuous-trait.html#regional-association-plot"><i class="fa fa-check"></i><b>5.3</b> Regional Association Plot</a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="gwas-with-continuous-trait.html"><a href="gwas-with-continuous-trait.html#obtain-regional-snps-using-r"><i class="fa fa-check"></i><b>5.3.1</b> Obtain regional SNPs (using R)</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="gwas-with-continuous-trait.html"><a href="gwas-with-continuous-trait.html#distribution-of-z-endoxifen-ratio-for-rs8138080"><i class="fa fa-check"></i><b>5.4</b> Distribution of Z-endoxifen ratio for rs8138080</a>
<ul>
<li class="chapter" data-level="5.4.1" data-path="gwas-with-continuous-trait.html"><a href="gwas-with-continuous-trait.html#obtain-phenotype-data-z-endoxifen-ratio-using-r"><i class="fa fa-check"></i><b>5.4.1</b> Obtain phenotype data (Z-endoxifen ratio) (using R)</a></li>
<li class="chapter" data-level="5.4.2" data-path="gwas-with-continuous-trait.html"><a href="gwas-with-continuous-trait.html#obtain-genotype-for-significant-snps-using-r"><i class="fa fa-check"></i><b>5.4.2</b> Obtain genotype for significant SNPs (using R)</a></li>
<li class="chapter" data-level="5.4.3" data-path="gwas-with-continuous-trait.html"><a href="gwas-with-continuous-trait.html#create-box-plot-using-r"><i class="fa fa-check"></i><b>5.4.3</b> Create Box plot (using R)</a></li>
</ul></li>
<li class="chapter" data-level="5.5" data-path="gwas-with-continuous-trait.html"><a href="gwas-with-continuous-trait.html#proportion-of-variation-explained-by-rs8138080"><i class="fa fa-check"></i><b>5.5</b> Proportion of Variation Explained by rs8138080</a>
<ul>
<li class="chapter" data-level="5.5.1" data-path="gwas-with-continuous-trait.html"><a href="gwas-with-continuous-trait.html#recode-genotype-of-significant-snps-using-r"><i class="fa fa-check"></i><b>5.5.1</b> Recode genotype of significant SNPs (using R)</a></li>
<li class="chapter" data-level="5.5.2" data-path="gwas-with-continuous-trait.html"><a href="gwas-with-continuous-trait.html#calculate-marginal-r2-using-r"><i class="fa fa-check"></i><b>5.5.2</b> Calculate marginal <span class="math inline">\(R^2\)</span> (using R)</a></li>
</ul></li>
<li class="chapter" data-level="5.6" data-path="gwas-with-continuous-trait.html"><a href="gwas-with-continuous-trait.html#practice-time"><i class="fa fa-check"></i><b>5.6</b> PRACTICE TIME</a>
<ul>
<li class="chapter" data-level="5.6.1" data-path="gwas-with-continuous-trait.html"><a href="gwas-with-continuous-trait.html#practice-1"><i class="fa fa-check"></i><b>5.6.1</b> Practice 1</a></li>
<li class="chapter" data-level="5.6.2" data-path="gwas-with-continuous-trait.html"><a href="gwas-with-continuous-trait.html#practice-2"><i class="fa fa-check"></i><b>5.6.2</b> Practice 2</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="gwas-with-binary-trait.html"><a href="gwas-with-binary-trait.html"><i class="fa fa-check"></i><b>6</b> GWAS with Binary Trait</a>
<ul>
<li class="chapter" data-level="6.1" data-path="gwas-with-binary-trait.html"><a href="gwas-with-binary-trait.html#data-preparation"><i class="fa fa-check"></i><b>6.1</b> Data Preparation</a></li>
<li class="chapter" data-level="6.2" data-path="gwas-with-binary-trait.html"><a href="gwas-with-binary-trait.html#genome-wide-casecontrol-association-test-in-plink"><i class="fa fa-check"></i><b>6.2</b> Genome-Wide Case/Control Association Test (in PLINK)</a></li>
<li class="chapter" data-level="6.3" data-path="gwas-with-binary-trait.html"><a href="gwas-with-binary-trait.html#manhattan-plot-and-q-q-plot-1"><i class="fa fa-check"></i><b>6.3</b> Manhattan Plot and Q-Q Plot</a></li>
<li class="chapter" data-level="6.4" data-path="gwas-with-binary-trait.html"><a href="gwas-with-binary-trait.html#generate-manhattan-plot-1"><i class="fa fa-check"></i><b>6.4</b> Generate Manhattan plot</a></li>
<li class="chapter" data-level="6.5" data-path="gwas-with-binary-trait.html"><a href="gwas-with-binary-trait.html#generate-q-q-plot-1"><i class="fa fa-check"></i><b>6.5</b> Generate Q-Q plot</a></li>
<li class="chapter" data-level="6.6" data-path="gwas-with-binary-trait.html"><a href="gwas-with-binary-trait.html#multiple-test-correction-1"><i class="fa fa-check"></i><b>6.6</b> Multiple-Test Correction</a></li>
<li class="chapter" data-level="6.7" data-path="gwas-with-binary-trait.html"><a href="gwas-with-binary-trait.html#top-10-snps"><i class="fa fa-check"></i><b>6.7</b> Top 10 SNPs</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="gxg-and-gxe-interactions-power-and-sample-size-calculation.html"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html"><i class="fa fa-check"></i><b>7</b> GxG and GxE Interactions, Power and Sample Size Calculation</a>
<ul>
<li class="chapter" data-level="7.1" data-path="gxg-and-gxe-interactions-power-and-sample-size-calculation.html"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#prepare-files"><i class="fa fa-check"></i><b>7.1</b> Prepare Files</a></li>
<li class="chapter" data-level="7.2" data-path="gxg-and-gxe-interactions-power-and-sample-size-calculation.html"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#test-for-epistasis-gxg"><i class="fa fa-check"></i><b>7.2</b> Test for Epistasis (GxG)</a></li>
<li class="chapter" data-level="7.3" data-path="gxg-and-gxe-interactions-power-and-sample-size-calculation.html"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#gene-environment-interaction-gxe"><i class="fa fa-check"></i><b>7.3</b> Gene-Environment Interaction (GxE)</a></li>
<li class="chapter" data-level="7.4" data-path="gxg-and-gxe-interactions-power-and-sample-size-calculation.html"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#power-and-sample-size-calculation-for-gwas"><i class="fa fa-check"></i><b>7.4</b> Power and sample size calculation for GWAS</a>
<ul>
<li class="chapter" data-level="7.4.1" data-path="gxg-and-gxe-interactions-power-and-sample-size-calculation.html"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#power-calculation-with-binary-outcome"><i class="fa fa-check"></i><b>7.4.1</b> Power calculation with binary outcome:</a></li>
<li class="chapter" data-level="7.4.2" data-path="gxg-and-gxe-interactions-power-and-sample-size-calculation.html"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#sample-size-calculation"><i class="fa fa-check"></i><b>7.4.2</b> Sample size calculation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8" data-path="ldsc-and-gene-set-analysis.html"><a href="ldsc-and-gene-set-analysis.html"><i class="fa fa-check"></i><b>8</b> LDSC and Gene-Set Analysis</a>
<ul>
<li class="chapter" data-level="8.1" data-path="ldsc-and-gene-set-analysis.html"><a href="ldsc-and-gene-set-analysis.html#ldsc-linkage-disequilibrium-score-regression"><i class="fa fa-check"></i><b>8.1</b> LDSC (Linkage Disequilibrium Score Regression)</a></li>
<li class="chapter" data-level="8.2" data-path="ldsc-and-gene-set-analysis.html"><a href="ldsc-and-gene-set-analysis.html#magma-for-gene-level-analysis"><i class="fa fa-check"></i><b>8.2</b> MAGMA (for Gene-Level Analysis)</a>
<ul>
<li class="chapter" data-level="8.2.1" data-path="ldsc-and-gene-set-analysis.html"><a href="ldsc-and-gene-set-analysis.html#prepare-annotation-file"><i class="fa fa-check"></i><b>8.2.1</b> Prepare Annotation File</a></li>
<li class="chapter" data-level="8.2.2" data-path="ldsc-and-gene-set-analysis.html"><a href="ldsc-and-gene-set-analysis.html#part-1-gene-analysis"><i class="fa fa-check"></i><b>8.2.2</b> Part 1: Gene Analysis</a></li>
<li class="chapter" data-level="8.2.3" data-path="ldsc-and-gene-set-analysis.html"><a href="ldsc-and-gene-set-analysis.html#part-2-gene-set-analysis"><i class="fa fa-check"></i><b>8.2.3</b> Part 2: Gene-Set Analysis</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="9" data-path="perform-eqtl-analysis.html"><a href="perform-eqtl-analysis.html"><i class="fa fa-check"></i><b>9</b> Perform eQTL Analysis</a>
<ul>
<li class="chapter" data-level="9.1" data-path="perform-eqtl-analysis.html"><a href="perform-eqtl-analysis.html#filtering-snps-by-maf"><i class="fa fa-check"></i><b>9.1</b> Filtering SNPs by MAF</a></li>
<li class="chapter" data-level="9.2" data-path="perform-eqtl-analysis.html"><a href="perform-eqtl-analysis.html#gene-expression-profiling"><i class="fa fa-check"></i><b>9.2</b> Gene expression profiling</a></li>
<li class="chapter" data-level="9.3" data-path="perform-eqtl-analysis.html"><a href="perform-eqtl-analysis.html#covariates"><i class="fa fa-check"></i><b>9.3</b> Covariates</a></li>
<li class="chapter" data-level="9.4" data-path="perform-eqtl-analysis.html"><a href="perform-eqtl-analysis.html#using-principle-components-as-covariates"><i class="fa fa-check"></i><b>9.4</b> Using principle components as covariates</a></li>
<li class="chapter" data-level="9.5" data-path="perform-eqtl-analysis.html"><a href="perform-eqtl-analysis.html#eqtl-analysis"><i class="fa fa-check"></i><b>9.5</b> eQTL analysis</a>
<ul>
<li class="chapter" data-level="9.5.1" data-path="perform-eqtl-analysis.html"><a href="perform-eqtl-analysis.html#cis-eqtl-and-trans-eqtl-analysis"><i class="fa fa-check"></i><b>9.5.1</b> cis-eQTL and trans-eQTL analysis</a></li>
</ul></li>
<li class="chapter" data-level="9.6" data-path="perform-eqtl-analysis.html"><a href="perform-eqtl-analysis.html#multiple-testing-correction"><i class="fa fa-check"></i><b>9.6</b> Multiple testing correction</a></li>
<li class="chapter" data-level="9.7" data-path="perform-eqtl-analysis.html"><a href="perform-eqtl-analysis.html#gene-regional-manhanttan-plots"><i class="fa fa-check"></i><b>9.7</b> Gene regional Manhanttan plots</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">CSHL course - GWAS Hands-on Analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="gwas-tools-and-data-preprocessing-workflow-with-toy-example" class="section level1 hasAnchor" number="3">
<h1><span class="header-section-number">Chapter 3</span> GWAS Tools and Data Preprocessing Workflow with Toy Example<a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#gwas-tools-and-data-preprocessing-workflow-with-toy-example" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>This notebook demonstrates a standard approach to conduct the genome wide association study (GWAS) using toy example datasets. The procedures consist of four steps:</p>
<ul>
<li>Quality Control</li>
<li>Imputation</li>
<li>Population Stratification Correction</li>
<li>Simple Association Test</li>
</ul>
<p>The data used for this example is a toy genotype data containing 5 random SNPs and a random phenotype data with covariates.</p>
<p>We use both bash and R codes in this notebook. The type of programming language is labeled at the top of each code cell.</p>
<blockquote>
<p><span style="color: #0064a4;"> Special notes for Windows users: Windows has two command-line shells, Command shell and Windows Powershell. Command shell is specific to Windows system and can be run from “Command Prompt” App, but the language it uses is NOT the same as Bash! Powershell App is the app to run Bash commands in Windows. Therefore if you are Windows users and use RStudio for this course, please refer to Chapter 1 for setting up RStudio Terminal for running bash codes or ask for help.
</span></p>
</blockquote>
<div id="environment-setting" class="section level2 hasAnchor" number="3.1">
<h2><span class="header-section-number">3.1</span> Environment Setting<a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#environment-setting" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Before we can run any analysis, we need to set the working directory path to the downloaded “cshl_gwas” folder for both Terminal and Console in RStudio. Below are the codes to setup paths at both places. These paths have to be setup correctly at first, otherwise we can’t locate the correct input files.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb1-1" tabindex="-1"></a><span class="co"># change the paths to where you saved &quot;cshl_gwas&quot;</span></span>
<span id="cb1-2"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb1-2" tabindex="-1"></a><span class="do">## run in Terminal</span></span>
<span id="cb1-3"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb1-3" tabindex="-1"></a>mydir<span class="ot">=</span><span class="st">&quot;/path/to/cshl_gwas&quot;</span></span>
<span id="cb1-4"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb1-4" tabindex="-1"></a>cd <span class="sc">$</span>mydir</span>
<span id="cb1-5"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb1-5" tabindex="-1"></a></span>
<span id="cb1-6"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb1-6" tabindex="-1"></a><span class="co"># R codes- run in R Console</span></span>
<span id="cb1-7"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb1-7" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">&quot;/path/to/cshl_gwas&quot;</span>)</span></code></pre></div>
</div>
<div id="gwas-tools" class="section level2 hasAnchor" number="3.2">
<h2><span class="header-section-number">3.2</span> GWAS Tools<a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#gwas-tools" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We will use two main program tools in this notebook, <strong>PLINK</strong> (version 1.9) and <strong>IMPUTE2</strong>.</p>
<p><strong>PLINK</strong> (version 1.9) is used for quality control and perform association test for large scale genotype data. You can find the .zip files suitable for your laptop system in “cshl_gwas/Tools/PLINK” and unzip it to call the <code>plink</code> program in the folder.</p>
<p><strong>IMPUTE2</strong> is used for genotype imputation and haplotype phasing. You can download its program using this link. IMPUTE2 program is only available to run in Linux or macOS system. You can find the zipped installation file in “cshl_gwas/Tools/IMPUTE2”. If you want to perform imputation for your own data, you also need to download the required reference genome from the above link. But in this notebook will only use the example genome.</p>
<p>We will also introduce a tool called <strong>MAGMA</strong> for gene-set analysis in later chapters.</p>
<blockquote>
<p><span style="color: #0064a4;">
Important: If you want to run these programs on your laptop, please make sure you have unzipped the installation packages in “./cshl_gwas/Tools” that is suitable for your laptop systerm. We will directly call PLINK and IMPUTE2 programs using abusolute or relative paths.
</span></p>
</blockquote>
<p>Instructions on installing these two tools can be found on in Chapter 1.</p>
<p>After you cloned the repository “cshl_gwas” folder, you can find the <code>Tools</code> folder contatining installation packages for different systems. Then, you can unzip the zip files and call the programs by specifying its absolute path. For example, for macOS users, you can unzip the installer by double clicking, and call PLINK using <code>/path/to/cshl_gwas/Tools/PLINK/plink_mac_20231211/plink</code>; for Windows users, you need to unzip the package “plink_win64.zip” with right click-&gt;“Extract All…”. Then you can call PLINK with <code>/path/to/cshl_gwas/Tools/PLINK/plink_win64_20231211/plink.exe</code>.</p>
<p>You should modify the above <code>/path/to/cshl_gwas</code> to where your “cshl_gwas” folder is.</p>
<blockquote>
<p><span style="color: #0064a4;">
However, macOS users need to go through additional steps to authorize PLINK, IMPUTE2 and MAGMA programs before you can call them! But you can perform these authorization steps only when you need to call the programs. See below.
</span></p>
</blockquote>
<p>When you try to call PLINK, IMPUTE2 and MAGMA with the above commands, you might get a pop-up window with the following message “plink” cannot be opened because the developer cannot be verified. Select “Cancel” and we will authorize PLINK first. Please go to macOS “System Settings”-&gt;“Privacy &amp; Security”, and scroll down until you see a message “plink” was blocked from use because it is not from an identified developer. Click “Allow Anyway” and you will be prompt to enter your Mac username and password. Click “Unlock” and then “Allow Anyway” again, and the previous message will disappear. Then try to run the same codes again in RStudio Terminal. You will get another pop-up window saying macOS cannot verify the developer of “plink”. Are you sure you want to open it?, select “Open” then you can see PLINK or IMPUTE2 commands options show up in Terminal.</p>
</div>
<div id="quality-control" class="section level2 hasAnchor" number="3.3">
<h2><span class="header-section-number">3.3</span> Quality Control<a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#quality-control" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>PLINK will generate a number of standard summary statistics that are useful for quality control (e.g. missing genotype rate, minor allele frequency, Hardy-Weinberg equilibrium test and non-Mendelian transmission rate). These can also be used as thresholds for subsequent analyses. For more detailed information about plink, please refer its website <a href="https://www.cog-genomics.org/plink/1.9/" class="uri">https://www.cog-genomics.org/plink/1.9/</a>.</p>
<p>The following is a list of some common options that can be used to filter out individuals or SNPs on the basis of the summary statistic measures:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb2-1" tabindex="-1"></a>tabl <span class="ot">&lt;-</span> <span class="st">&quot;</span></span>
<span id="cb2-2"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb2-2" tabindex="-1"></a><span class="st">| Feature inclusion criteria |  As summary statistic | As inclusion criteria |</span></span>
<span id="cb2-3"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb2-3" tabindex="-1"></a><span class="st">|----------------------------|-----------------------|-----------------------|</span></span>
<span id="cb2-4"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb2-4" tabindex="-1"></a><span class="st">| Missingness per individual |`--missing`            |`--mind N`.            |</span></span>
<span id="cb2-5"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb2-5" tabindex="-1"></a><span class="st">| Missingness per marker       | --missing             | --geno N              |</span></span>
<span id="cb2-6"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb2-6" tabindex="-1"></a><span class="st">| Allele frequency           | --freq                  | --maf N               |</span></span>
<span id="cb2-7"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb2-7" tabindex="-1"></a><span class="st">| Hardy-Weinberg equilibrium |--hardy                  | --hwe N               |</span></span>
<span id="cb2-8"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb2-8" tabindex="-1"></a><span class="st">&quot;</span></span>
<span id="cb2-9"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb2-9" tabindex="-1"></a><span class="fu">cat</span>(tabl) <span class="co"># output the table in a format good for HTML/PDF/docx conversion</span></span></code></pre></div>
<table>
<colgroup>
<col width="37%" />
<col width="31%" />
<col width="31%" />
</colgroup>
<thead>
<tr class="header">
<th>Feature inclusion criteria</th>
<th>As summary statistic</th>
<th>As inclusion criteria</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Missingness per individual</td>
<td><code>--missing</code></td>
<td><code>--mind N</code>.</td>
</tr>
<tr class="even">
<td>Missingness per marker</td>
<td>–missing</td>
<td>–geno N</td>
</tr>
<tr class="odd">
<td>Allele frequency</td>
<td>–freq</td>
<td>–maf N</td>
</tr>
<tr class="even">
<td>Hardy-Weinberg equilibrium</td>
<td>–hardy</td>
<td>–hwe N</td>
</tr>
</tbody>
</table>
<div id="an-example-of-summary-statistics" class="section level3 hasAnchor" number="3.3.1">
<h3><span class="header-section-number">3.3.1</span> An Example of Summary Statistics<a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#an-example-of-summary-statistics" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="input-files" class="section level4 hasAnchor" number="3.3.1.1">
<h4><span class="header-section-number">3.3.1.1</span> Input files<a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#input-files" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Input files: in folder <code>./Data/QC</code>– We will use a random selected 5 SNPs as an example data:</p>
<ul>
<li>5SNPs.ped</li>
<li>5SNPs.map</li>
</ul>
<p>The PED file is a white-space delimited file.It contains no header line, and one line per sample with 6+2V fields where V is the number of variants. The first six columns are mandatory:</p>
<pre><code> Family ID
 Individual ID
 Paternal ID
 Maternal ID
 Sex (1=male; 2=female; other=unknown)
 Phenotype</code></pre>
<p>The seventh and eighth fields are allele calls for the first variant in the .map file (‘0’ = no call); the 9th and 10th are allele calls for the second variant; and so on.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb4-1"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb4-1" tabindex="-1"></a><span class="co"># bash codes- run in Terminal</span></span>
<span id="cb4-2"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb4-2" tabindex="-1"></a><span class="co"># Windows user need to run in Powershell App </span></span>
<span id="cb4-3"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb4-3" tabindex="-1"></a><span class="fu">head</span> ./Data/QC/5SNPs.ped</span></code></pre></div>
<pre><code>## 1 206-001 0 0 1 0 A A G G G G G G A A
## 2 206-003 0 0 1 0 A A G G G G G G A A
## 3 206-004 0 0 1 0 A A G G G G G G A A
## 4 206-014 0 0 1 0 A A G G G G G G A A
## 5 206-015 0 0 1 0 A G A G A G A G A A
## 6 206-016 0 0 1 0 A G A G A G A G A A
## 7 206-018 0 0 1 0 A A G G G G G G A A
## 8 206-019 0 0 1 0 A A A G A G A G A G
## 9 206-021 0 0 1 0 A A G G G G G G A A
## 10 206-023 0 0 1 0 A G A G A G A G A A</code></pre>
<p>By default, each line of the MAP file describes a single marker and must contain exactly 4 columns:</p>
<pre><code> chromosome (1-22, X, Y or 0 if unplaced)
 rs# or snp identifier
 Genetic distance (morgans)
 Base-pair position (bp units)
 </code></pre>
<div class="sourceCode" id="cb7"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb7-1"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb7-1" tabindex="-1"></a><span class="fu">head</span> ./Data/QC/5SNPs.map</span></code></pre></div>
<pre><code>## 1 rs3094315 0 752566
## 1 rs2073813 0 753541
## 1 rs3131969 0 754182
## 1 rs3131967 0 754334
## 1 rs77598327 0 757120</code></pre>
<p>Three types of summary statistics that we are going to calculate using <code>PLINK</code>:</p>
<ol style="list-style-type: decimal">
<li>missingness checking</li>
<li>allele frequencies</li>
<li>hardy-weinberg</li>
</ol>
<p>All results generated from the following commands are saved in folder <code>./Data/QC</code>. You can open the files in this folder to view the data structure.</p>
</div>
<div id="calculating-missing-rate" class="section level4 hasAnchor" number="3.3.1.2">
<h4><span class="header-section-number">3.3.1.2</span> Calculating missing rate<a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#calculating-missing-rate" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>The following commands compute the <strong>missingness</strong> for each SNP and each subject. (The output will print all calculation progress generated by <code>PLINK</code>.)</p>
<ol style="list-style-type: decimal">
<li><strong>–noweb</strong>, suppress the connection to internet for version check.</li>
<li><strong>–file</strong>, specify the input files.</li>
<li><strong>–missing</strong>, calculate the number of missing and missing rate.</li>
<li><strong>–out</strong>, specify the output file name.</li>
</ol>
<div class="sourceCode" id="cb9"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb9-1"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb9-1" tabindex="-1"></a><span class="co"># both Mac and Windows can run in Terminal</span></span>
<span id="cb9-2"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb9-2" tabindex="-1"></a><span class="co"># modify the path to &quot;plink&quot; program</span></span>
<span id="cb9-3"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb9-3" tabindex="-1"></a></span>
<span id="cb9-4"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb9-4" tabindex="-1"></a><span class="co"># for windows users</span></span>
<span id="cb9-5"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb9-5" tabindex="-1"></a><span class="ex">./Tools/PLINK/plink_win64_20231211/plink.exe</span> <span class="dt">\</span></span>
<span id="cb9-6"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb9-6" tabindex="-1"></a>--noweb <span class="at">--file</span> ./Data/QC/5SNPs <span class="at">--missing</span> <span class="at">--out</span> ./Data/QC/younameit</span></code></pre></div>
<div class="sourceCode" id="cb10"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb10-1"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb10-1" tabindex="-1"></a><span class="co"># for macoS users</span></span>
<span id="cb10-2"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb10-2" tabindex="-1"></a><span class="ex">./Tools/PLINK/plink_mac_20231211/plink</span> <span class="dt">\</span></span>
<span id="cb10-3"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb10-3" tabindex="-1"></a>--noweb <span class="at">--file</span> ./Data/QC/5SNPs <span class="at">--missing</span> <span class="at">--out</span> ./Data/QC/younameit</span></code></pre></div>
<pre><code>## PLINK v1.90b7.2 64-bit (11 Dec 2023)           www.cog-genomics.org/plink/1.9/
## (C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3
## Logging to ./Data/QC/younameit.log.
## Options in effect:
##   --file ./Data/QC/5SNPs
##   --missing
##   --noweb
##   --out ./Data/QC/younameit
## 
## Note: --noweb has no effect since no web check is implemented yet.
## 16384 MB RAM detected; reserving 8192 MB for main workspace.
## Scanning .ped file... 0%1%1%1%1%2%2%2%3%3%3%3%4%4%4%5%5%5%5%6%6%6%7%7%7%7%8%8%8%9%9%9%9%10%10%10%11%11%11%11%12%12%12%13%13%13%13%14%14%14%15%15%15%15%16%16%16%17%17%17%17%18%18%18%19%19%19%19%20%20%20%21%21%21%21%22%22%22%23%23%23%23%24%24%24%25%25%25%25%26%26%26%27%27%27%27%28%28%28%29%29%29%29%30%30%30%31%31%31%32%32%32%32%33%33%33%34%34%34%34%35%35%35%36%36%36%37%37%37%37%38%38%38%39%39%39%39%40%40%40%41%41%41%41%42%42%42%43%43%43%44%44%44%44%45%45%45%46%46%46%46%47%47%47%48%48%48%49%49%49%49%50%50%50%51%51%51%51%52%52%52%53%53%53%53%54%54%54%55%55%55%56%56%56%56%57%57%57%58%58%58%58%59%59%59%60%60%60%61%61%61%61%62%62%62%63%63%63%63%64%64%64%65%65%65%66%66%66%66%67%67%67%68%68%68%68%69%69%69%70%70%70%70%71%71%71%72%72%72%73%73%73%73%74%74%74%75%75%75%75%76%76%76%77%77%77%78%78%78%78%79%79%79%80%80%80%80%81%81%81%82%82%82%83%83%83%83%84%84%84%85%85%85%85%86%86%86%87%87%87%87%88%88%88%89%89%89%90%90%90%90%91%91%91%92%92%92%92%93%93%93%94%94%94%95%95%95%95%96%96%96%97%97%97%97%98%98%98%99%99%99%100%.ped scan complete (for binary autoconversion).
## Performing single-pass .bed write (5 variants, 344 people).
## 0%1%2%3%4%5%6%7%8%9%10%11%12%13%14%15%16%17%18%19%20%21%22%23%24%25%26%27%28%29%30%31%32%33%34%35%36%37%38%39%40%41%42%43%44%45%46%47%48%49%50%51%52%53%54%55%56%57%58%59%60%61%62%63%64%65%66%67%68%69%70%71%72%73%74%75%76%77%78%79%80%81%82%83%84%85%86%87%88%89%90%91%92%93%94%--file: ./Data/QC/younameit-temporary.bed + ./Data/QC/younameit-temporary.bim +
## ./Data/QC/younameit-temporary.fam written.
## 5 variants loaded from .bim file.
## 344 people (281 males, 0 females, 63 ambiguous) loaded from .fam.
## Ambiguous sex IDs written to ./Data/QC/younameit.nosex .
## Using 1 thread (no multithreaded calculations invoked).
## Before main variant filters, 344 founders and 0 nonfounders present.
## Calculating allele frequencies... 0%1%2%3%4%5%6%7%8%9%10%11%12%13%14%15%16%17%18%19%20%21%22%23%24%25%26%27%28%29%30%31%32%33%34%35%36%37%38%39%40%41%42%43%44%45%46%47%48%49%50%51%52%53%54%55%56%57%58%59%60%61%62%63%64%65%66%67%68%69%70%71%72%73%74%75%76%77%78%79%80%81%82%83%84%85%86%87%88%89%90%91%92%93%94%95%96%97%98%99% done.
## Total genotyping rate is 0.976744.
## --missing: Sample missing data report written to ./Data/QC/younameit.imiss, and
## variant-based missing data report written to ./Data/QC/younameit.lmiss.</code></pre>
<p>This option creates two output files in ‘results’ folder:</p>
<ol style="list-style-type: decimal">
<li><strong>“younameit.imiss”</strong>, missingness by individual.</li>
<li><strong>“younameit.lmiss”</strong>, missingness by SNP.</li>
</ol>
<p>For individuals “.imiss” file, the format is:</p>
<pre><code> FID                Family ID
 IID                Individual ID
 MISS_PHENO         Missing phenotype? (Y/N)
 N_MISS             Number of missing SNPs
 N_GENO             Number of non-obligatory missing genotypes
 F_MISS             Proportion of missing SNPs</code></pre>
<div class="sourceCode" id="cb13"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb13-1"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb13-1" tabindex="-1"></a><span class="co"># bash codes- run in Terminal</span></span>
<span id="cb13-2"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb13-2" tabindex="-1"></a><span class="co"># Windows user need to run in Powershell App </span></span>
<span id="cb13-3"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb13-3" tabindex="-1"></a><span class="fu">head</span> ./Data/QC/younameit.imiss</span></code></pre></div>
<pre><code>##  FID       IID MISS_PHENO   N_MISS   N_GENO   F_MISS
##    1   206-001          Y        0        5        0
##    2   206-003          Y        0        5        0
##    3   206-004          Y        0        5        0
##    4   206-014          Y        0        5        0
##    5   206-015          Y        0        5        0
##    6   206-016          Y        0        5        0
##    7   206-018          Y        0        5        0
##    8   206-019          Y        0        5        0
##    9   206-021          Y        0        5        0</code></pre>
<p>For SNP “lmiss” file, the format is:</p>
<pre><code> SNP                SNP identifier
 CHR                Chromosome number
 N_MISS             Number of individuals missing this SNP
 N_GENO             Number of non-obligatory missing genotypes
 F_MISS             Proportion of sample missing for this SNP</code></pre>
<div class="sourceCode" id="cb16"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb16-1"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb16-1" tabindex="-1"></a><span class="co"># bash codes- run in terminal</span></span>
<span id="cb16-2"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb16-2" tabindex="-1"></a><span class="co"># Windows user need to run in Powershell App </span></span>
<span id="cb16-3"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb16-3" tabindex="-1"></a><span class="fu">head</span> ./Data/QC/younameit.lmiss</span></code></pre></div>
<pre><code>##  CHR         SNP   N_MISS   N_GENO   F_MISS
##    1   rs3094315        2      344 0.005814
##    1   rs2073813       14      344   0.0407
##    1   rs3131969       11      344  0.03198
##    1   rs3131967        3      344 0.008721
##    1  rs77598327       10      344  0.02907</code></pre>
</div>
<div id="calculating-allel-frequencies" class="section level4 hasAnchor" number="3.3.1.3">
<h4><span class="header-section-number">3.3.1.3</span> Calculating allel frequencies<a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#calculating-allel-frequencies" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>The following command computes the <strong>allele frequencies</strong>. (The output will print all calculation progress generated by <code>PLINK</code>.)
1. <strong>–freq</strong>, calculate the allele frequencies.</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb18-1"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb18-1" tabindex="-1"></a><span class="co"># both Mac and Windows can run in Terminal</span></span>
<span id="cb18-2"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb18-2" tabindex="-1"></a><span class="co"># modify the path to &quot;plink&quot; program</span></span>
<span id="cb18-3"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb18-3" tabindex="-1"></a></span>
<span id="cb18-4"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb18-4" tabindex="-1"></a><span class="co"># Windows user</span></span>
<span id="cb18-5"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb18-5" tabindex="-1"></a><span class="ex">./Tools/PLINK/plink_win64_20231211/plink.exe</span> <span class="dt">\</span></span>
<span id="cb18-6"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb18-6" tabindex="-1"></a>--noweb <span class="at">--file</span> ./Data/QC/5SNPs <span class="at">--freq</span> <span class="at">--out</span> ./Data/QC/younameit</span></code></pre></div>
<div class="sourceCode" id="cb19"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb19-1"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb19-1" tabindex="-1"></a><span class="co"># macOS user</span></span>
<span id="cb19-2"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb19-2" tabindex="-1"></a><span class="ex">./Tools/PLINK/plink_mac_20231211/plink</span> <span class="dt">\</span></span>
<span id="cb19-3"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb19-3" tabindex="-1"></a>--noweb <span class="at">--file</span> ./Data/QC/5SNPs <span class="at">--freq</span> <span class="at">--out</span> ./Data/QC/younameit</span></code></pre></div>
<pre><code>## PLINK v1.90b7.2 64-bit (11 Dec 2023)           www.cog-genomics.org/plink/1.9/
## (C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3
## Logging to ./Data/QC/younameit.log.
## Options in effect:
##   --file ./Data/QC/5SNPs
##   --freq
##   --noweb
##   --out ./Data/QC/younameit
## 
## Note: --noweb has no effect since no web check is implemented yet.
## 16384 MB RAM detected; reserving 8192 MB for main workspace.
## Scanning .ped file... 0%1%1%1%1%2%2%2%3%3%3%3%4%4%4%5%5%5%5%6%6%6%7%7%7%7%8%8%8%9%9%9%9%10%10%10%11%11%11%11%12%12%12%13%13%13%13%14%14%14%15%15%15%15%16%16%16%17%17%17%17%18%18%18%19%19%19%19%20%20%20%21%21%21%21%22%22%22%23%23%23%23%24%24%24%25%25%25%25%26%26%26%27%27%27%27%28%28%28%29%29%29%29%30%30%30%31%31%31%32%32%32%32%33%33%33%34%34%34%34%35%35%35%36%36%36%37%37%37%37%38%38%38%39%39%39%39%40%40%40%41%41%41%41%42%42%42%43%43%43%44%44%44%44%45%45%45%46%46%46%46%47%47%47%48%48%48%49%49%49%49%50%50%50%51%51%51%51%52%52%52%53%53%53%53%54%54%54%55%55%55%56%56%56%56%57%57%57%58%58%58%58%59%59%59%60%60%60%61%61%61%61%62%62%62%63%63%63%63%64%64%64%65%65%65%66%66%66%66%67%67%67%68%68%68%68%69%69%69%70%70%70%70%71%71%71%72%72%72%73%73%73%73%74%74%74%75%75%75%75%76%76%76%77%77%77%78%78%78%78%79%79%79%80%80%80%80%81%81%81%82%82%82%83%83%83%83%84%84%84%85%85%85%85%86%86%86%87%87%87%87%88%88%88%89%89%89%90%90%90%90%91%91%91%92%92%92%92%93%93%93%94%94%94%95%95%95%95%96%96%96%97%97%97%97%98%98%98%99%99%99%100%.ped scan complete (for binary autoconversion).
## Performing single-pass .bed write (5 variants, 344 people).
## 0%1%2%3%4%5%6%7%8%9%10%11%12%13%14%15%16%17%18%19%20%21%22%23%24%25%26%27%28%29%30%31%32%33%34%35%36%37%38%39%40%41%42%43%44%45%46%47%48%49%50%51%52%53%54%55%56%57%58%59%60%61%62%63%64%65%66%67%68%69%70%71%72%73%74%75%76%77%78%79%80%81%82%83%84%85%86%87%88%89%90%91%92%93%94%--file: ./Data/QC/younameit-temporary.bed + ./Data/QC/younameit-temporary.bim +
## ./Data/QC/younameit-temporary.fam written.
## 5 variants loaded from .bim file.
## 344 people (281 males, 0 females, 63 ambiguous) loaded from .fam.
## Ambiguous sex IDs written to ./Data/QC/younameit.nosex .
## Using 1 thread (no multithreaded calculations invoked).
## Before main variant filters, 344 founders and 0 nonfounders present.
## Calculating allele frequencies... 0%1%2%3%4%5%6%7%8%9%10%11%12%13%14%15%16%17%18%19%20%21%22%23%24%25%26%27%28%29%30%31%32%33%34%35%36%37%38%39%40%41%42%43%44%45%46%47%48%49%50%51%52%53%54%55%56%57%58%59%60%61%62%63%64%65%66%67%68%69%70%71%72%73%74%75%76%77%78%79%80%81%82%83%84%85%86%87%88%89%90%91%92%93%94%95%96%97%98%99% done.
## Total genotyping rate is 0.976744.
## --freq: Allele frequencies (founders only) written to ./Data/QC/younameit.frq .</code></pre>
<p>This command will create a output file <strong>“younameit.frq”</strong> with six columns:</p>
<pre><code> CHR       Chromosome
 
 SNP       SNP identifier
 
 A1        Allele 1 code (minor allele)
 
 A2        Allele 2 code (major allele)
 
 MAF       Minor allele frequency
 
 NCHROBS   Non-missing allele count</code></pre>
<div class="sourceCode" id="cb22"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb22-1"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb22-1" tabindex="-1"></a><span class="co"># bash codes- run in terminal</span></span>
<span id="cb22-2"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb22-2" tabindex="-1"></a><span class="co"># Windows user need to run in Powershell App</span></span>
<span id="cb22-3"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb22-3" tabindex="-1"></a><span class="fu">head</span> ./Data/QC/younameit.frq</span></code></pre></div>
<pre><code>##  CHR         SNP   A1   A2          MAF  NCHROBS
##    1   rs3094315    G    A      0.07749      684
##    1   rs2073813    A    G       0.2242      660
##    1   rs3131969    A    G       0.2387      666
##    1   rs3131967    A    G       0.2082      682
##    1  rs77598327    G    A       0.1228      668</code></pre>
</div>
<div id="hardy-weinberg-equilibrium" class="section level4 hasAnchor" number="3.3.1.4">
<h4><span class="header-section-number">3.3.1.4</span> Hardy-Weinberg Equilibrium<a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#hardy-weinberg-equilibrium" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>The following command computes the <strong>Hardy-Weiberg exact test statistic</strong>. (The output will print all calculation progress generated by <code>PLINK</code>.)
1. <strong>–hardy</strong>, perform hardy-weinberg test.</p>
<div class="sourceCode" id="cb24"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb24-1"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb24-1" tabindex="-1"></a><span class="co"># both Mac and Windows can run in Terminal</span></span>
<span id="cb24-2"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb24-2" tabindex="-1"></a><span class="co"># modify the path to &quot;plink&quot; program</span></span>
<span id="cb24-3"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb24-3" tabindex="-1"></a></span>
<span id="cb24-4"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb24-4" tabindex="-1"></a><span class="co"># Windows user</span></span>
<span id="cb24-5"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb24-5" tabindex="-1"></a><span class="ex">./Tools/PLINK/plink_win64_20231211/plink.exe</span> <span class="dt">\</span></span>
<span id="cb24-6"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb24-6" tabindex="-1"></a>--noweb <span class="at">--file</span> ./Data/QC/5SNPs <span class="at">--hardy</span> <span class="at">--out</span> ./Data/QC/younameit</span></code></pre></div>
<div class="sourceCode" id="cb25"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb25-1"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb25-1" tabindex="-1"></a><span class="co"># macOS user</span></span>
<span id="cb25-2"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb25-2" tabindex="-1"></a><span class="ex">./Tools/PLINK/plink_mac_20231211/plink</span> <span class="dt">\</span></span>
<span id="cb25-3"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb25-3" tabindex="-1"></a>--noweb <span class="at">--file</span> ./Data/QC/5SNPs <span class="at">--hardy</span> <span class="at">--out</span> ./Data/QC/younameit</span></code></pre></div>
<pre><code>## PLINK v1.90b7.2 64-bit (11 Dec 2023)           www.cog-genomics.org/plink/1.9/
## (C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3
## Logging to ./Data/QC/younameit.log.
## Options in effect:
##   --file ./Data/QC/5SNPs
##   --hardy
##   --noweb
##   --out ./Data/QC/younameit
## 
## Note: --noweb has no effect since no web check is implemented yet.
## 16384 MB RAM detected; reserving 8192 MB for main workspace.
## Scanning .ped file... 0%1%1%1%1%2%2%2%3%3%3%3%4%4%4%5%5%5%5%6%6%6%7%7%7%7%8%8%8%9%9%9%9%10%10%10%11%11%11%11%12%12%12%13%13%13%13%14%14%14%15%15%15%15%16%16%16%17%17%17%17%18%18%18%19%19%19%19%20%20%20%21%21%21%21%22%22%22%23%23%23%23%24%24%24%25%25%25%25%26%26%26%27%27%27%27%28%28%28%29%29%29%29%30%30%30%31%31%31%32%32%32%32%33%33%33%34%34%34%34%35%35%35%36%36%36%37%37%37%37%38%38%38%39%39%39%39%40%40%40%41%41%41%41%42%42%42%43%43%43%44%44%44%44%45%45%45%46%46%46%46%47%47%47%48%48%48%49%49%49%49%50%50%50%51%51%51%51%52%52%52%53%53%53%53%54%54%54%55%55%55%56%56%56%56%57%57%57%58%58%58%58%59%59%59%60%60%60%61%61%61%61%62%62%62%63%63%63%63%64%64%64%65%65%65%66%66%66%66%67%67%67%68%68%68%68%69%69%69%70%70%70%70%71%71%71%72%72%72%73%73%73%73%74%74%74%75%75%75%75%76%76%76%77%77%77%78%78%78%78%79%79%79%80%80%80%80%81%81%81%82%82%82%83%83%83%83%84%84%84%85%85%85%85%86%86%86%87%87%87%87%88%88%88%89%89%89%90%90%90%90%91%91%91%92%92%92%92%93%93%93%94%94%94%95%95%95%95%96%96%96%97%97%97%97%98%98%98%99%99%99%100%.ped scan complete (for binary autoconversion).
## Performing single-pass .bed write (5 variants, 344 people).
## 0%1%2%3%4%5%6%7%8%9%10%11%12%13%14%15%16%17%18%19%20%21%22%23%24%25%26%27%28%29%30%31%32%33%34%35%36%37%38%39%40%41%42%43%44%45%46%47%48%49%50%51%52%53%54%55%56%57%58%59%60%61%62%63%64%65%66%67%68%69%70%71%72%73%74%75%76%77%78%79%80%81%82%83%84%85%86%87%88%89%90%91%92%93%94%--file: ./Data/QC/younameit-temporary.bed + ./Data/QC/younameit-temporary.bim +
## ./Data/QC/younameit-temporary.fam written.
## 5 variants loaded from .bim file.
## 344 people (281 males, 0 females, 63 ambiguous) loaded from .fam.
## Ambiguous sex IDs written to ./Data/QC/younameit.nosex .
## Using 1 thread (no multithreaded calculations invoked).
## Before main variant filters, 344 founders and 0 nonfounders present.
## Calculating allele frequencies... 0%1%2%3%4%5%6%7%8%9%10%11%12%13%14%15%16%17%18%19%20%21%22%23%24%25%26%27%28%29%30%31%32%33%34%35%36%37%38%39%40%41%42%43%44%45%46%47%48%49%50%51%52%53%54%55%56%57%58%59%60%61%62%63%64%65%66%67%68%69%70%71%72%73%74%75%76%77%78%79%80%81%82%83%84%85%86%87%88%89%90%91%92%93%94%95%96%97%98%99% done.
## Total genotyping rate is 0.976744.
## --hardy: Writing Hardy-Weinberg report (founders only) to
## ./Data/QC/younameit.hwe ... 0%0%1%2%3%4%5%6%7%8%9%10%11%12%13%14%15%16%17%18%19%20%21%22%23%24%25%26%27%28%29%30%31%32%33%34%35%36%37%38%39%40%41%42%43%44%45%46%47%48%49%50%51%52%53%54%55%56%57%58%59%60%61%62%63%64%65%66%67%68%69%70%71%72%73%74%75%76%77%78%79%80%81%82%83%84%85%86%87%88%89%90%91%92%93%94%95%96%97%98%99%done.</code></pre>
<p>This command will create an output file <strong>“younameit.hwe”</strong> with the following columns:</p>
<pre><code> CHR             Chromosome
 
 SNP             SNP identifier
 
 TEST            Code indicating whether the sample is a case or control
 
 A1              Minor allele code
 
 A2              Major allele code
 
 GENO            Genotype counts: 11/12/22 
 
 O(HET)          Observed heterozygosity
 
 E(HET)          Expected heterozygosity
 
 P               P-value for Hardy-Weinberg test</code></pre>
<div class="sourceCode" id="cb28"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb28-1"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb28-1" tabindex="-1"></a><span class="co"># bash codes- run in terminal</span></span>
<span id="cb28-2"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb28-2" tabindex="-1"></a><span class="co"># Windows user need to run in Powershell App</span></span>
<span id="cb28-3"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb28-3" tabindex="-1"></a><span class="fu">head</span> ./Data/QC/younameit.hwe</span></code></pre></div>
<pre><code>##  CHR         SNP     TEST   A1   A2                 GENO   O(HET)   E(HET)            P 
##    1   rs3094315  ALL(NP)    G    A             3/47/292   0.1374    0.143       0.4398
##    1   rs2073813  ALL(NP)    A    G           18/112/200   0.3394   0.3479       0.6366
##    1   rs3131969  ALL(NP)    A    G           24/111/198   0.3333   0.3635       0.1323
##    1   rs3131967  ALL(NP)    A    G           12/118/211    0.346   0.3297       0.4149
##    1  rs77598327  ALL(NP)    G    A             2/78/254   0.2335   0.2154       0.1984</code></pre>
</div>
</div>
<div id="an-example-of-inclusion-thresholds-for-qc" class="section level3 hasAnchor" number="3.3.2">
<h3><span class="header-section-number">3.3.2</span> An Example of Inclusion Thresholds for QC<a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#an-example-of-inclusion-thresholds-for-qc" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Input files: in folder <code>./QC</code></p>
<ul>
<li><p>5SNPs.ped</p></li>
<li><p>5SNPs.map</p></li>
</ul>
<p>The following command performs <strong>multiple quality controls</strong>. (The output will print all calculation progress generated by <code>PLINK</code>.)
1. <strong>–mind</strong> 0.1, exclude the individuals with missing rate larger than 0.1. You can change 0.1 to other desired values.
2. <strong>–geno</strong> 0.1, exclude the SNPs with missing rate larger than 0.1. You can change 0.1 to other desiered values.
3. <strong>–maf</strong> 0.05, exclude the SNPs with minor allele frequency (MAF) smaller than 0.05. You can change 0.05 to other desiered values.
4. <strong>–hwe</strong> 0.001, exclude the SNPs with P-values of Hardy-Weinberg test smaller than 0.001. You can change it to otehr desired values.
5. <strong>–recode</strong>, generate a new dataset.</p>
<div class="sourceCode" id="cb30"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb30-1"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb30-1" tabindex="-1"></a><span class="co"># both Mac and Windows can run in Terminal</span></span>
<span id="cb30-2"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb30-2" tabindex="-1"></a><span class="co"># modify the path to &quot;plink&quot; program</span></span>
<span id="cb30-3"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb30-3" tabindex="-1"></a></span>
<span id="cb30-4"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb30-4" tabindex="-1"></a><span class="co"># Windows user</span></span>
<span id="cb30-5"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb30-5" tabindex="-1"></a><span class="ex">./Tools/PLINK/plink_win64_20231211/plink.exe</span> <span class="dt">\</span></span>
<span id="cb30-6"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb30-6" tabindex="-1"></a>--noweb <span class="at">--file</span> ./Data/QC/5SNPs <span class="dt">\</span></span>
<span id="cb30-7"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb30-7" tabindex="-1"></a>--mind 0.1 <span class="at">--geno</span> 0.1 <span class="at">--maf</span> 0.05 <span class="at">--hwe</span> 0.001 <span class="at">--recode</span> <span class="at">--out</span> ./Data/QC/younameit</span></code></pre></div>
<div class="sourceCode" id="cb31"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb31-1"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb31-1" tabindex="-1"></a><span class="co"># macOS user</span></span>
<span id="cb31-2"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb31-2" tabindex="-1"></a><span class="ex">./Tools/PLINK/plink_mac_20231211/plink</span> <span class="dt">\</span></span>
<span id="cb31-3"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb31-3" tabindex="-1"></a>--noweb <span class="at">--file</span> ./Data/QC/5SNPs <span class="dt">\</span></span>
<span id="cb31-4"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb31-4" tabindex="-1"></a>--mind 0.1 <span class="at">--geno</span> 0.1 <span class="at">--maf</span> 0.05 <span class="at">--hwe</span> 0.001 <span class="at">--recode</span> <span class="at">--out</span> ./Data/QC/younameit</span></code></pre></div>
<pre><code>## PLINK v1.90b7.2 64-bit (11 Dec 2023)           www.cog-genomics.org/plink/1.9/
## (C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3
## Logging to ./Data/QC/younameit.log.
## Options in effect:
##   --file ./Data/QC/5SNPs
##   --geno 0.1
##   --hwe 0.001
##   --maf 0.05
##   --mind 0.1
##   --noweb
##   --out ./Data/QC/younameit
##   --recode
## 
## Note: --noweb has no effect since no web check is implemented yet.
## 16384 MB RAM detected; reserving 8192 MB for main workspace.
## Scanning .ped file... 0%1%1%1%1%2%2%2%3%3%3%3%4%4%4%5%5%5%5%6%6%6%7%7%7%7%8%8%8%9%9%9%9%10%10%10%11%11%11%11%12%12%12%13%13%13%13%14%14%14%15%15%15%15%16%16%16%17%17%17%17%18%18%18%19%19%19%19%20%20%20%21%21%21%21%22%22%22%23%23%23%23%24%24%24%25%25%25%25%26%26%26%27%27%27%27%28%28%28%29%29%29%29%30%30%30%31%31%31%32%32%32%32%33%33%33%34%34%34%34%35%35%35%36%36%36%37%37%37%37%38%38%38%39%39%39%39%40%40%40%41%41%41%41%42%42%42%43%43%43%44%44%44%44%45%45%45%46%46%46%46%47%47%47%48%48%48%49%49%49%49%50%50%50%51%51%51%51%52%52%52%53%53%53%53%54%54%54%55%55%55%56%56%56%56%57%57%57%58%58%58%58%59%59%59%60%60%60%61%61%61%61%62%62%62%63%63%63%63%64%64%64%65%65%65%66%66%66%66%67%67%67%68%68%68%68%69%69%69%70%70%70%70%71%71%71%72%72%72%73%73%73%73%74%74%74%75%75%75%75%76%76%76%77%77%77%78%78%78%78%79%79%79%80%80%80%80%81%81%81%82%82%82%83%83%83%83%84%84%84%85%85%85%85%86%86%86%87%87%87%87%88%88%88%89%89%89%90%90%90%90%91%91%91%92%92%92%92%93%93%93%94%94%94%95%95%95%95%96%96%96%97%97%97%97%98%98%98%99%99%99%100%.ped scan complete (for binary autoconversion).
## Performing single-pass .bed write (5 variants, 344 people).
## 0%1%2%3%4%5%6%7%8%9%10%11%12%13%14%15%16%17%18%19%20%21%22%23%24%25%26%27%28%29%30%31%32%33%34%35%36%37%38%39%40%41%42%43%44%45%46%47%48%49%50%51%52%53%54%55%56%57%58%59%60%61%62%63%64%65%66%67%68%69%70%71%72%73%74%75%76%77%78%79%80%81%82%83%84%85%86%87%88%89%90%91%92%93%94%--file: ./Data/QC/younameit-temporary.bed + ./Data/QC/younameit-temporary.bim +
## ./Data/QC/younameit-temporary.fam written.
## 5 variants loaded from .bim file.
## 344 people (281 males, 0 females, 63 ambiguous) loaded from .fam.
## Ambiguous sex IDs written to ./Data/QC/younameit.nosex .
## 32 people removed due to missing genotype data (--mind).
## IDs written to ./Data/QC/younameit.irem .
## Using 1 thread (no multithreaded calculations invoked).
## Before main variant filters, 312 founders and 0 nonfounders present.
## Calculating allele frequencies... 0%1%2%3%4%5%6%7%8%9%10%11%12%13%14%15%16%17%18%19%20%21%22%23%24%25%26%27%28%29%30%31%32%33%34%35%36%37%38%39%40%41%42%43%44%45%46%47%48%49%50%51%52%53%54%55%56%57%58%59%60%61%62%63%64%65%66%67%68%69%70%71%72%73%74%75%76%77%78%79%80%81%82%83%84%85%86%87%88%89%90%91%92%93%94%95%96%97%98%99% done.
## Total genotyping rate in remaining samples is exactly 1.
## 0 variants removed due to missing genotype data (--geno).
## --hwe: 0 variants removed due to Hardy-Weinberg exact test.
## 0 variants removed due to minor allele threshold(s)
## (--maf/--max-maf/--mac/--max-mac).
## 5 variants and 312 people pass filters and QC.
## Note: No phenotypes present.
## --recode ped to ./Data/QC/younameit.ped + ./Data/QC/younameit.map ... 0%1%2%3%4%5%6%7%8%9%10%11%12%13%14%15%16%17%18%19%20%21%22%23%24%25%26%27%28%29%30%31%32%33%34%35%36%37%38%39%40%41%42%43%44%45%46%47%48%49%50%51%52%53%54%55%56%57%58%59%60%61%62%63%64%65%66%67%68%69%70%71%72%73%74%75%76%77%78%79%80%81%82%83%84%85%86%87%88%89%90%91%92%93%94%95%96%97%98%99%done.</code></pre>
<p>The output files will be <strong>“younameit.map”</strong> and <strong>“younameit.ped”</strong> which have been quality controlled by the criteria you specified.</p>
</div>
<div id="remove-closely-related-individuals" class="section level3 hasAnchor" number="3.3.3">
<h3><span class="header-section-number">3.3.3</span> Remove Closely Related Individuals<a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#remove-closely-related-individuals" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Input files: in folder <code>.QC</code></p>
<ul>
<li><p>5SNPs.ped</p></li>
<li><p>5SNPs.map</p></li>
</ul>
<p>The following command will calculate <strong>indentical by descent (IBS) distances</strong> between all individuals.
1. <strong>–genome</strong>, calculate IBS</p>
<div class="sourceCode" id="cb33"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb33-1"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb33-1" tabindex="-1"></a><span class="co"># both Mac and Windows can run in Terminal</span></span>
<span id="cb33-2"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb33-2" tabindex="-1"></a><span class="co"># modify the path to &quot;plink&quot; program</span></span>
<span id="cb33-3"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb33-3" tabindex="-1"></a></span>
<span id="cb33-4"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb33-4" tabindex="-1"></a><span class="co"># Windows user</span></span>
<span id="cb33-5"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb33-5" tabindex="-1"></a><span class="ex">./Tools/PLINK/plink_win64_20231211/plink.exe</span> <span class="dt">\</span></span>
<span id="cb33-6"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb33-6" tabindex="-1"></a>--noweb <span class="at">--file</span> ./Data/QC/5SNPs <span class="at">--genome</span> <span class="at">--out</span> ./Data/QC/younameit</span></code></pre></div>
<div class="sourceCode" id="cb34"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb34-1"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb34-1" tabindex="-1"></a><span class="co"># macOS user</span></span>
<span id="cb34-2"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb34-2" tabindex="-1"></a><span class="ex">./Tools/PLINK/plink_mac_20231211/plink</span> <span class="dt">\</span></span>
<span id="cb34-3"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb34-3" tabindex="-1"></a>--noweb <span class="at">--file</span> ./Data/QC/5SNPs <span class="at">--genome</span> <span class="at">--out</span> ./Data/QC/younameit</span></code></pre></div>
<pre><code>## PLINK v1.90b7.2 64-bit (11 Dec 2023)           www.cog-genomics.org/plink/1.9/
## (C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3
## Logging to ./Data/QC/younameit.log.
## Options in effect:
##   --file ./Data/QC/5SNPs
##   --genome
##   --noweb
##   --out ./Data/QC/younameit
## 
## Note: --noweb has no effect since no web check is implemented yet.
## 16384 MB RAM detected; reserving 8192 MB for main workspace.
## Scanning .ped file... 0%1%1%1%1%2%2%2%3%3%3%3%4%4%4%5%5%5%5%6%6%6%7%7%7%7%8%8%8%9%9%9%9%10%10%10%11%11%11%11%12%12%12%13%13%13%13%14%14%14%15%15%15%15%16%16%16%17%17%17%17%18%18%18%19%19%19%19%20%20%20%21%21%21%21%22%22%22%23%23%23%23%24%24%24%25%25%25%25%26%26%26%27%27%27%27%28%28%28%29%29%29%29%30%30%30%31%31%31%32%32%32%32%33%33%33%34%34%34%34%35%35%35%36%36%36%37%37%37%37%38%38%38%39%39%39%39%40%40%40%41%41%41%41%42%42%42%43%43%43%44%44%44%44%45%45%45%46%46%46%46%47%47%47%48%48%48%49%49%49%49%50%50%50%51%51%51%51%52%52%52%53%53%53%53%54%54%54%55%55%55%56%56%56%56%57%57%57%58%58%58%58%59%59%59%60%60%60%61%61%61%61%62%62%62%63%63%63%63%64%64%64%65%65%65%66%66%66%66%67%67%67%68%68%68%68%69%69%69%70%70%70%70%71%71%71%72%72%72%73%73%73%73%74%74%74%75%75%75%75%76%76%76%77%77%77%78%78%78%78%79%79%79%80%80%80%80%81%81%81%82%82%82%83%83%83%83%84%84%84%85%85%85%85%86%86%86%87%87%87%87%88%88%88%89%89%89%90%90%90%90%91%91%91%92%92%92%92%93%93%93%94%94%94%95%95%95%95%96%96%96%97%97%97%97%98%98%98%99%99%99%100%.ped scan complete (for binary autoconversion).
## Performing single-pass .bed write (5 variants, 344 people).
## 0%1%2%3%4%5%6%7%8%9%10%11%12%13%14%15%16%17%18%19%20%21%22%23%24%25%26%27%28%29%30%31%32%33%34%35%36%37%38%39%40%41%42%43%44%45%46%47%48%49%50%51%52%53%54%55%56%57%58%59%60%61%62%63%64%65%66%67%68%69%70%71%72%73%74%75%76%77%78%79%80%81%82%83%84%85%86%87%88%89%90%91%92%93%94%--file: ./Data/QC/younameit-temporary.bed + ./Data/QC/younameit-temporary.bim +
## ./Data/QC/younameit-temporary.fam written.
## 5 variants loaded from .bim file.
## 344 people (281 males, 0 females, 63 ambiguous) loaded from .fam.
## Ambiguous sex IDs written to ./Data/QC/younameit.nosex .
## Using up to 8 threads (change this with --threads).
## Before main variant filters, 344 founders and 0 nonfounders present.
## Calculating allele frequencies... 0%1%2%3%4%5%6%7%8%9%10%11%12%13%14%15%16%17%18%19%20%21%22%23%24%25%26%27%28%29%30%31%32%33%34%35%36%37%38%39%40%41%42%43%44%45%46%47%48%49%50%51%52%53%54%55%56%57%58%59%60%61%62%63%64%65%66%67%68%69%70%71%72%73%74%75%76%77%78%79%80%81%82%83%84%85%86%87%88%89%90%91%92%93%94%95%96%97%98%99% done.
## Total genotyping rate is 0.976744.
## 5 variants and 344 people pass filters and QC.
## Note: No phenotypes present.
## 5 markers complete.IBD calculations complete.  
## Writing... 1%Writing... 4%Writing... 6%Writing... 8%Writing... 11%Writing... 13%Writing... 15%Writing... 17%Writing... 20%Writing... 22%Writing... 24%Writing... 26%Writing... 29%Writing... 31%Writing... 33%Writing... 36%Writing... 38%Writing... 40%Writing... 42%Writing... 45%Writing... 47%Writing... 49%Writing... 51%Writing... 54%Writing... 56%Writing... 58%Writing... 61%Writing... 63%Writing... 65%Writing... 67%Writing... 70%Writing... 72%Writing... 74%Writing... 77%Writing... 79%Writing... 81%Writing... 83%Writing... 86%Writing... 88%Writing... 90%Writing... 92%Writing... 95%Writing... 97%Writing... 99%Finished writing ./Data/QC/younameit.genome .</code></pre>
<p>This will create the file <strong>younameit.genome</strong> which has the following fields:</p>
<pre><code> FID1      Family ID for first individual
 
 IID1      Individual ID for first individual
 
 FID2      Family ID for second individual
 
 IID2      Individual ID for second individual
 
 RT        Relationship type given PED file
 
 EZ        Expected IBD sharing given PED file
 
 Z0        P(IBD=0)
 
 Z1        P(IBD=1)
 
 Z2        P(IBD=2)
 
 PI_HAT    P(IBD=2)+0.5*P(IBD=1) ( proportion IBD )
 
 PHE       Pairwise phenotypic code (1,0,-1 = AA, AU and UU pairs)
 
 DST       IBS distance (IBS2 + 0.5*IBS1) / ( N SNP pairs )
 
 PPC       IBS binomial test
 
 RATIO     Of HETHET : IBS 0 SNPs (expected value is 2)</code></pre>
<p>Scan the output file <strong>“younameit.genome”</strong> for any individuals with high <strong>PI_HAT</strong> values (e.g. greater than 0.05).</p>
<p>Optionally, remove one member of the pair if close relatives are identified.</p>
<div class="sourceCode" id="cb37"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb37-1"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb37-1" tabindex="-1"></a><span class="co"># bash codes- run in terminal</span></span>
<span id="cb37-2"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb37-2" tabindex="-1"></a><span class="co"># Windows user need to run in Powershell App</span></span>
<span id="cb37-3"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb37-3" tabindex="-1"></a><span class="fu">head</span> ./Data/QC/younameit.genome</span></code></pre></div>
<pre><code>##  FID1     IID1 FID2     IID2 RT    EZ      Z0      Z1      Z2  PI_HAT PHE       DST     PPC   RATIO
##    1  206-001   2  206-003 UN    NA  0.0000  0.0000  1.0000  1.0000  -1  1.000000      NA      NA
##    1  206-001   3  206-004 UN    NA  0.0000  0.0000  1.0000  1.0000  -1  1.000000      NA      NA
##    1  206-001   4  206-014 UN    NA  0.0000  0.0000  1.0000  1.0000  -1  1.000000      NA      NA
##    1  206-001   5  206-015 UN    NA  0.0000  1.0000  0.0000  0.5000  -1  0.600000      NA      NA
##    1  206-001   6  206-016 UN    NA  0.0000  1.0000  0.0000  0.5000  -1  0.600000      NA      NA
##    1  206-001   7  206-018 UN    NA  0.0000  0.0000  1.0000  1.0000  -1  1.000000      NA      NA
##    1  206-001   8  206-019 UN    NA  0.0000  1.0000  0.0000  0.5000  -1  0.600000      NA      NA
##    1  206-001   9  206-021 UN    NA  0.0000  0.0000  1.0000  1.0000  -1  1.000000      NA      NA
##    1  206-001  10  206-023 UN    NA  0.0000  1.0000  0.0000  0.5000  -1  0.600000      NA      NA</code></pre>
</div>
</div>
<div id="an-imputation-example" class="section level2 hasAnchor" number="3.4">
<h2><span class="header-section-number">3.4</span> An Imputation Example<a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#an-imputation-example" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>In many cases, raw sequence data may have missing values. Then, an imputation step will be neccessary. <strong>IMPUTE2</strong> is a popular software package for imputation. Below is a simple example for conducting imputation by IMPUTE2. For more information about IMPUTE2, please refer its website <a href="http://mathgen.stats.ox.ac.uk/impute/impute_v2.html" class="uri">http://mathgen.stats.ox.ac.uk/impute/impute_v2.html</a> and the associated publication <a href="http://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1000529" class="uri">http://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1000529</a></p>
<p>Here is the example provided by <strong>IMPUTE2 package</strong>. The link is <a href="https://mathgen.stats.ox.ac.uk/impute/impute_v2.html#ex1" class="uri">https://mathgen.stats.ox.ac.uk/impute/impute_v2.html#ex1</a></p>
<p>Files used for example are saved in folder <code>./Data/impute2_example</code>.</p>
<div class="sourceCode" id="cb39"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb39-1"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb39-1" tabindex="-1"></a><span class="co"># bash codes- run in terminal</span></span>
<span id="cb39-2"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb39-2" tabindex="-1"></a><span class="co"># IMPUTE2 only works for macOS or Linux users, Windows users can&#39;t run it</span></span>
<span id="cb39-3"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb39-3" tabindex="-1"></a></span>
<span id="cb39-4"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb39-4" tabindex="-1"></a><span class="co"># macOS user only</span></span>
<span id="cb39-5"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb39-5" tabindex="-1"></a><span class="co"># modify the path to &quot;impute2&quot; program</span></span>
<span id="cb39-6"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb39-6" tabindex="-1"></a><span class="ex">./Tools/IMPUTE2/impute_v2.3.2_MacOSX_Intel/impute2</span> <span class="dt">\</span></span>
<span id="cb39-7"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb39-7" tabindex="-1"></a>-m ./Data/impute2_example/example.chr22.map <span class="dt">\</span></span>
<span id="cb39-8"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb39-8" tabindex="-1"></a>-h ./Data/impute2_example/example.chr22.1kG.haps <span class="dt">\</span></span>
<span id="cb39-9"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb39-9" tabindex="-1"></a>-l ./Data/impute2_example/example.chr22.1kG.legend <span class="dt">\</span></span>
<span id="cb39-10"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb39-10" tabindex="-1"></a>-g ./Data/impute2_example/example.chr22.study.gens <span class="dt">\</span></span>
<span id="cb39-11"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb39-11" tabindex="-1"></a>-strand_g ./Data/impute2_example/example.chr22.study.strand <span class="dt">\</span></span>
<span id="cb39-12"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb39-12" tabindex="-1"></a>-int 20.4e6 20.5e6 <span class="at">-Ne</span> 20000 <span class="dt">\</span></span>
<span id="cb39-13"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb39-13" tabindex="-1"></a>-o ./Data/impute2_example/example.chr22.one.phased.impute2</span></code></pre></div>
<pre><code>## 
## ======================
##  IMPUTE version 2.3.2 
## ======================
## 
## Copyright 2008 Bryan Howie, Peter Donnelly, and Jonathan Marchini
## Please see the LICENCE file included with this program for conditions of use.
## 
## The seed for the random number generator is 1867707983.
## 
## Command-line input: ./Tools/IMPUTE2/impute_v2.3.2_MacOSX_Intel/impute2 -m ./Data/impute2_example/example.chr22.map -h ./Data/impute2_example/example.chr22.1kG.haps -l ./Data/impute2_example/example.chr22.1kG.legend -g ./Data/impute2_example/example.chr22.study.gens -strand_g ./Data/impute2_example/example.chr22.study.strand -int 20.4e6 20.5e6 -Ne 20000 -o ./Data/impute2_example/example.chr22.one.phased.impute2
## 
## ---------------------------------
##  Nomenclature and data structure 
## ---------------------------------
## 
##      Panel 0: phased reference haplotypes
##      Panel 2: unphased study genotypes
## 
## For optimal results, each successive panel (0,1,2) should contain a subset of the SNPs in the previous panel. When the data structure deviates from this ideal configuration, IMPUTE2 tries to use as much of the available information as possible; see documentation for details.
## 
## -------------
##  Input files 
## -------------
## 
##          Panel 0 haplotypes: ./Data/impute2_example/example.chr22.1kG.haps
##          Panel 0 hap legend: ./Data/impute2_example/example.chr22.1kG.legend
##           Panel 2 genotypes: ./Data/impute2_example/example.chr22.study.gens
##         Panel 2 strand info: ./Data/impute2_example/example.chr22.study.strand
##                 genetic map: ./Data/impute2_example/example.chr22.map
## 
## --------------
##  Output files 
## --------------
## 
##                 main output: ./Data/impute2_example/example.chr22.one.phased.impute2
##                 SNP QC info: ./Data/impute2_example/example.chr22.one.phased.impute2_info
##              sample QC info: ./Data/impute2_example/example.chr22.one.phased.impute2_info_by_sample
##                 run summary: ./Data/impute2_example/example.chr22.one.phased.impute2_summary
##                 warning log: ./Data/impute2_example/example.chr22.one.phased.impute2_warnings
## 
## -----------------
##  Data processing 
## -----------------
## 
## -reading genetic map from -m file
##  --filename=[./Data/impute2_example/example.chr22.map]
##  --read 262 SNPs in the analysis interval+buffer region
## 
## -reading strand info for Panel 2 from -strand_g file
##  --filename=[./Data/impute2_example/example.chr22.study.strand]
##  --read strand info for 33 SNPs in the analysis region
## 
## -reading Panel 2 genotypes from -g file
##  --filename=[./Data/impute2_example/example.chr22.study.gens]
##  --detected 250 individuals
##  --read 33 SNPs in the analysis interval+buffer region
## 
## -using -strand_g file to align Panel 2 allele labels
##  --flipped strand at 14 out of 33 SNPs
## 
## -reading Panel 0 haplotypes from -h and -l files
##  --filename=[./Data/impute2_example/example.chr22.1kG.haps]
##  --filename=[./Data/impute2_example/example.chr22.1kG.legend]
##  --detected 112 haplotypes
##  --read 790 SNPs in the analysis interval+buffer region
## 
## -removing SNPs that violate the hierarchical data requirements
##  --no SNPs removed
## 
## -removing reference-only SNPs from buffer region
##  --removed 538 SNPs
## 
## -checking strand alignment between Panel 2 and Panel 0 by allele labels
##  --flipped strand due to allele mismatch at 0 out of 33 SNPs in Panel 2
## 
## -aligning allele labels between panels
## 
## -removing non-aligned genotyped SNPs
##  --removed 0 out of 27 SNPs with data in multiple panels
## 
## --------------
##  Data summary 
## --------------
## 
## [type 0 = SNP in Panel 0 only]
## [type 1 = SNP in Panel 1]
## [type 2 = SNP in Panel 2 and all ref panels]
## [type 3 = SNP in Panel 2 only]
## 
## -Upstream buffer region
##  --0 type 0 SNPs
##  --0 type 1 SNPs
##  --10 type 2 SNPs
##  --2 type 3 SNPs
##  --12 total SNPs
## 
## -Downstream buffer region
##  --0 type 0 SNPs
##  --0 type 1 SNPs
##  --5 type 2 SNPs
##  --1 type 3 SNPs
##  --6 total SNPs
## 
## -Analysis region (as defined by -int argument)
##  --225 type 0 SNPs
##  --0 type 1 SNPs
##  --12 type 2 SNPs
##  --3 type 3 SNPs
##  --240 total SNPs
## 
## -Output file
##  --225 type 0 SNPs
##  --0 type 1 SNPs
##  --12 type 2 SNPs
##  --3 type 3 SNPs
## 
## -In total, 258 SNPs will be used in the analysis, including 27 Panel 2 SNPs
## 
## -making initial haplotype guesses for Panel 2 by phasing hets at random and imputing missing genotypes from allele freqs
## 
## -setting storage space
## -setting mutation matrices
## -setting switch rates
## 
## ----------------
##  Run parameters 
## ----------------
## 
##         reference haplotypes: 112 [Panel 0]
##            study individuals: 250 [Panel 2]
##            sequence interval: [20400000,20500000]
##                       buffer: 250 kb
##                           Ne: 20000
##            input call thresh: 0.900
##      burn-in MCMC iterations: 10
##        total MCMC iterations: 30 (20 used for inference)
##       HMM states for phasing: 80 [Panel 2]
##    HMM states for imputation: 112 [Panel 0-&gt;2]
## 
## ---------
##  Run log 
## ---------
## 
## MCMC iteration [1/30]
## 
## MCMC iteration [2/30]
## 
## MCMC iteration [3/30]
## 
## RESETTING PARAMETERS FOR &quot;SURROGATE FAMILY&quot; MODELING
## -setting mutation matrices
## -setting switch rates
## 
## MCMC iteration [4/30]
## 
## MCMC iteration [5/30]
## 
## MCMC iteration [6/30]
## 
## MCMC iteration [7/30]
## 
## MCMC iteration [8/30]
## 
## MCMC iteration [9/30]
## 
## MCMC iteration [10/30]
## 
## MCMC iteration [11/30]
## 
## MCMC iteration [12/30]
## 
## MCMC iteration [13/30]
## 
## MCMC iteration [14/30]
## 
## MCMC iteration [15/30]
## 
## MCMC iteration [16/30]
## 
## MCMC iteration [17/30]
## 
## MCMC iteration [18/30]
## 
## MCMC iteration [19/30]
## 
## MCMC iteration [20/30]
## 
## MCMC iteration [21/30]
## 
## MCMC iteration [22/30]
## 
## MCMC iteration [23/30]
## 
## MCMC iteration [24/30]
## 
## MCMC iteration [25/30]
## 
## MCMC iteration [26/30]
## 
## MCMC iteration [27/30]
## 
## MCMC iteration [28/30]
## 
## MCMC iteration [29/30]
## 
## MCMC iteration [30/30]
## 
## 
## diploid sampling success rate: 0.988
## 
## haploid sampling success rate: (no haploid sampling performed)
## 
## 
## --------------------------------
##  Imputation accuracy assessment 
## --------------------------------
## 
## The table below is based on an internal cross-validation that is performed during each IMPUTE2 run. For this analysis, the program masks the genotypes of one variant at a time in the study data (Panel 2) and imputes the masked genotypes by using the remaining study and reference data. The imputed genotypes are then compared with the original genotypes to produce the concordance statistics shown in the table. You can learn more about this procedure and the contents of the table at http://mathgen.stats.ox.ac.uk/impute/concordance_table_description.html.
## 
## In the current analysis, IMPUTE2 masked, imputed, and evaluated 2985 genotypes that were called with high confidence (maximum probability &gt;= 0.90) in the Panel 2 input file (-g or -known_haps_g).
## 
## When the masked study genotypes were imputed with reference data from Panel 0, the concordance between original and imputed genotypes was as follows:
## 
##   Interval  #Genotypes %Concordance         Interval  %Called %Concordance
##   [0.0-0.1]          0          0.0         [ &gt;= 0.0]   100.0         97.5
##   [0.1-0.2]          0          0.0         [ &gt;= 0.1]   100.0         97.5
##   [0.2-0.3]          0          0.0         [ &gt;= 0.2]   100.0         97.5
##   [0.3-0.4]          0          0.0         [ &gt;= 0.3]   100.0         97.5
##   [0.4-0.5]          0          0.0         [ &gt;= 0.4]   100.0         97.5
##   [0.5-0.6]         14         92.9         [ &gt;= 0.5]   100.0         97.5
##   [0.6-0.7]          4         75.0         [ &gt;= 0.6]    99.5         97.5
##   [0.7-0.8]          4        100.0         [ &gt;= 0.7]    99.4         97.5
##   [0.8-0.9]         46         80.4         [ &gt;= 0.8]    99.3         97.5
##   [0.9-1.0]       2917         97.8         [ &gt;= 0.9]    97.7         97.8
## 
## Have a nice day!</code></pre>
</div>
<div id="population-stratification-correction-using-plink" class="section level2 hasAnchor" number="3.5">
<h2><span class="header-section-number">3.5</span> Population Stratification Correction Using PLINK<a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#population-stratification-correction-using-plink" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><code>PLINK</code>(only version 1.9 and after) can perform population stratefication using Principal Component Analysis. PCA explicitly model variations along continuous axes of PCs. Top principal components are generally used as covariates in the resulting correction to help correct for population stratification, minimizing spurious associations while maximizing power to detect true associations.</p>
<div id="pca-on-whole-genome-data-using-plink" class="section level3 hasAnchor" number="3.5.1">
<h3><span class="header-section-number">3.5.1</span> PCA on whole genome data using PLINK<a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#pca-on-whole-genome-data-using-plink" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>PCA can be performed with <code>--pca</code>. By default, <code>--pca</code> extracts the top 20 principal components. But we can change the number of PCs by specifying in the commands. The following codes perform PCA and obtain the first 50 PCs.</p>
<div class="sourceCode" id="cb41"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb41-1"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb41-1" tabindex="-1"></a><span class="co"># both Mac and Windows can run in Terminal</span></span>
<span id="cb41-2"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb41-2" tabindex="-1"></a><span class="co"># modify the path to &quot;plink&quot; program</span></span>
<span id="cb41-3"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb41-3" tabindex="-1"></a></span>
<span id="cb41-4"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb41-4" tabindex="-1"></a><span class="co"># Windows user</span></span>
<span id="cb41-5"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb41-5" tabindex="-1"></a><span class="ex">./Tools/PLINK/plink_win64_20231211/plink.exe</span> <span class="dt">\</span></span>
<span id="cb41-6"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb41-6" tabindex="-1"></a>--noweb <span class="at">--file</span> ./Data/PC/1ksnp <span class="at">--pca</span> 50 <span class="at">--out</span> ./Data/PC/younameit</span></code></pre></div>
<div class="sourceCode" id="cb42"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb42-1"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb42-1" tabindex="-1"></a><span class="co"># macOS user</span></span>
<span id="cb42-2"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb42-2" tabindex="-1"></a><span class="ex">./Tools/PLINK/plink_mac_20231211/plink</span> <span class="dt">\</span></span>
<span id="cb42-3"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb42-3" tabindex="-1"></a>--noweb <span class="at">--file</span> ./Data/PC/1ksnp <span class="at">--pca</span> 50 <span class="at">--out</span> ./Data/PC/younameit</span></code></pre></div>
<pre><code>## PLINK v1.90b7.2 64-bit (11 Dec 2023)           www.cog-genomics.org/plink/1.9/
## (C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3
## Logging to ./Data/PC/younameit.log.
## Options in effect:
##   --file ./Data/PC/1ksnp
##   --noweb
##   --out ./Data/PC/younameit
##   --pca 50
## 
## Note: --noweb has no effect since no web check is implemented yet.
## 16384 MB RAM detected; reserving 8192 MB for main workspace.
## Scanning .ped file... 0%1%1%1%2%2%2%2%3%3%3%4%4%4%4%5%5%5%6%6%6%6%7%7%7%8%8%8%9%9%9%9%10%10%10%11%11%11%11%12%12%12%13%13%13%13%14%14%14%15%15%15%15%16%16%16%17%17%17%18%18%18%18%19%19%19%20%20%20%20%21%21%21%22%22%22%22%23%23%23%24%24%24%24%25%25%25%26%26%26%27%27%27%27%28%28%28%29%29%29%29%30%30%30%31%31%31%31%32%32%32%33%33%33%34%34%34%34%35%35%35%36%36%36%36%37%37%37%38%38%38%38%39%39%39%40%40%40%40%41%41%41%42%42%42%43%43%43%43%44%44%44%45%45%45%45%46%46%46%47%47%47%47%48%48%48%49%49%49%49%50%50%50%51%51%51%52%52%52%52%53%53%53%54%54%54%54%55%55%55%56%56%56%56%57%57%57%58%58%58%59%59%59%59%60%60%60%61%61%61%61%62%62%62%63%63%63%63%64%64%64%65%65%65%65%66%66%66%67%67%67%68%68%68%68%69%69%69%70%70%70%70%71%71%71%72%72%72%72%73%73%73%74%74%74%74%75%75%75%76%76%76%77%77%77%77%78%78%78%79%79%79%79%80%80%80%81%81%81%81%82%82%82%83%83%83%84%84%84%84%85%85%85%86%86%86%86%87%87%87%88%88%88%88%89%89%89%90%90%90%90%91%91%91%92%92%92%93%93%93%93%94%94%94%95%95%95%95%96%96%96%97%97%97%97%98%98%98%99%99%99%100%.ped scan complete (for binary autoconversion).
## Performing single-pass .bed write (1000 variants, 344 people).
## 0%1%2%3%4%5%6%7%8%9%10%11%12%13%14%15%16%17%18%19%20%21%22%23%24%25%26%27%28%29%30%31%32%33%34%35%36%37%38%39%40%41%42%43%44%45%46%47%48%49%50%51%52%53%54%55%56%57%58%59%60%61%62%63%64%65%66%67%68%69%70%71%72%73%74%75%76%77%78%79%80%81%82%83%84%85%86%87%88%89%90%91%92%93%94%--file: ./Data/PC/younameit-temporary.bed + ./Data/PC/younameit-temporary.bim +
## ./Data/PC/younameit-temporary.fam written.
## 1000 variants loaded from .bim file.
## 344 people (281 males, 0 females, 63 ambiguous) loaded from .fam.
## Ambiguous sex IDs written to ./Data/PC/younameit.nosex .
## Using up to 8 threads (change this with --threads).
## Before main variant filters, 344 founders and 0 nonfounders present.
## Calculating allele frequencies... 0%1%2%3%4%5%6%7%8%9%10%11%12%13%14%15%16%17%18%19%20%21%22%23%24%25%26%27%28%29%30%31%32%33%34%35%36%37%38%39%40%41%42%43%44%45%46%47%48%49%50%51%52%53%54%55%56%57%58%59%60%61%62%63%64%65%66%67%68%69%70%71%72%73%74%75%76%77%78%79%80%81%82%83%84%85%86%87%88%89%90%91%92%93%94%95%96%97%98%99% done.
## Total genotyping rate is 0.998599.
## 1000 variants and 344 people pass filters and QC.
## Note: No phenotypes present.
## 60 markers complete.120 markers complete.180 markers complete.240 markers complete.300 markers complete.360 markers complete.420 markers complete.480 markers complete.540 markers complete.600 markers complete.660 markers complete.720 markers complete.780 markers complete.840 markers complete.900 markers complete.960 markers complete.1000 markers complete.Relationship matrix calculation complete.
## [extracting eigenvalues and eigenvectors]--pca: Results saved to ./Data/PC/younameit.eigenval and
## ./Data/PC/younameit.eigenvec .</code></pre>
<p>The main outputs are <strong>younameit.eigenval</strong> and <strong>younameit.eigenvec</strong>, which store the eigenvalues and eigenvectors of the top 50 PCs. With these results, we can visualize population structure by generating scree plot and PC plots in R.</p>
<p>A <strong>scree plot</strong> displays the variance explained by each component. We look for the “elbow” of the plot, i.e., point where line bends. Could do formal test on derivative of scree line, but common sense approach often works fine.</p>
<p>We read in eigenvalues from the PCA results file <strong>younameit.eigenval</strong>, calculate the proportion of variance explained by each PC and generate elbow plot using <code>ggplot</code>.</p>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb44-1"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb44-1" tabindex="-1"></a><span class="co"># R codes- run in R Console</span></span>
<span id="cb44-2"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb44-2" tabindex="-1"></a>eigenval <span class="ot">&lt;-</span> <span class="fu">scan</span>(<span class="st">&quot;./Data/PC/younameit.eigenval&quot;</span>)</span>
<span id="cb44-3"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb44-3" tabindex="-1"></a></span>
<span id="cb44-4"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb44-4" tabindex="-1"></a><span class="co"># first convert to percentage variance explained</span></span>
<span id="cb44-5"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb44-5" tabindex="-1"></a>pve <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">PC =</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(eigenval), <span class="at">pve =</span> eigenval<span class="sc">/</span><span class="fu">sum</span>(eigenval)<span class="sc">*</span><span class="dv">100</span>)</span>
<span id="cb44-6"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb44-6" tabindex="-1"></a><span class="co"># make plot</span></span>
<span id="cb44-7"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb44-7" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb44-8"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb44-8" tabindex="-1"></a>a <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(pve, <span class="fu">aes</span>(PC, pve)) <span class="sc">+</span> <span class="fu">geom_point</span>()<span class="sc">+</span><span class="fu">geom_line</span>()</span>
<span id="cb44-9"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb44-9" tabindex="-1"></a>a <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">&quot;Percentage variance explained&quot;</span>) <span class="sc">+</span> <span class="fu">theme_light</span>()</span></code></pre></div>
<p><img src="cshl_gwas_files/figure-html/screeplot-1.png" width="672" /></p>
<div class="sourceCode" id="cb45"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb45-1"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb45-1" tabindex="-1"></a><span class="co"># calculate the cumulative sum of the percentage variance explained</span></span>
<span id="cb45-2"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb45-2" tabindex="-1"></a><span class="fu">cumsum</span>(pve<span class="sc">$</span>pve)</span></code></pre></div>
<pre><code>##  [1]   4.179504   8.190458  11.831947  15.246016  18.367473  21.416914
##  [7]  24.363319  27.243624  30.015504  32.729433  35.321589  37.769462
## [13]  40.133494  42.468229  44.751649  46.905330  49.045935  51.152252
## [19]  53.220206  55.225766  57.144462  59.045020  60.923933  62.732881
## [25]  64.517038  66.267024  67.983592  69.651952  71.313039  72.956064
## [31]  74.552253  76.119765  77.651627  79.162127  80.637654  82.069785
## [37]  83.487819  84.899302  86.282899  87.632580  88.942650  90.247754
## [43]  91.532766  92.794932  94.050488  95.280154  96.484373  97.677171
## [49]  98.846007 100.000000</code></pre>
<p>A <strong>PC plot</strong> is a graphical representation to visualize the data in terms of the principal components. We usually look at the projection to the first several PCs which explain most of the variation. To generate PC plots, we will read in the eigenvectors from <strong>younameit.eigenvec</strong> and plot PC1 vs PC2, and PC3 vs PC4.</p>
<div class="sourceCode" id="cb47"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb47-1"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb47-1" tabindex="-1"></a><span class="co"># R codes- run in R Console</span></span>
<span id="cb47-2"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb47-2" tabindex="-1"></a><span class="co"># read in eigenvectors</span></span>
<span id="cb47-3"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb47-3" tabindex="-1"></a>pca <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">&quot;./Data/PC/younameit.eigenvec&quot;</span>, <span class="at">header=</span>F,<span class="at">row.names =</span> <span class="dv">1</span>)</span>
<span id="cb47-4"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb47-4" tabindex="-1"></a><span class="co"># set names</span></span>
<span id="cb47-5"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb47-5" tabindex="-1"></a><span class="fu">names</span>(pca)[<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="st">&quot;ind&quot;</span></span>
<span id="cb47-6"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb47-6" tabindex="-1"></a><span class="fu">names</span>(pca)[<span class="dv">2</span><span class="sc">:</span><span class="fu">ncol</span>(pca)] <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;PC&quot;</span>, <span class="dv">1</span><span class="sc">:</span>(<span class="fu">ncol</span>(pca)<span class="sc">-</span><span class="dv">1</span>))</span>
<span id="cb47-7"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb47-7" tabindex="-1"></a><span class="co"># sort out the individual species and pops</span></span>
<span id="cb47-8"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb47-8" tabindex="-1"></a></span>
<span id="cb47-9"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb47-9" tabindex="-1"></a><span class="co"># plot pca</span></span>
<span id="cb47-10"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb47-10" tabindex="-1"></a>pc12 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(pca, <span class="fu">aes</span>(PC1, PC2)) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb47-11"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb47-11" tabindex="-1"></a>            <span class="fu">scale_colour_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">&quot;red&quot;</span>, <span class="st">&quot;blue&quot;</span>)) <span class="sc">+</span></span>
<span id="cb47-12"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb47-12" tabindex="-1"></a>            <span class="fu">coord_equal</span>() <span class="sc">+</span> <span class="fu">theme_light</span>() <span class="sc">+</span></span>
<span id="cb47-13"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb47-13" tabindex="-1"></a>            <span class="fu">xlab</span>(<span class="fu">paste0</span>(<span class="st">&quot;PC1 (&quot;</span>, <span class="fu">signif</span>(pve<span class="sc">$</span>pve[<span class="dv">1</span>], <span class="dv">3</span>), <span class="st">&quot;%)&quot;</span>)) <span class="sc">+</span>    </span>
<span id="cb47-14"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb47-14" tabindex="-1"></a>            <span class="fu">ylab</span>(<span class="fu">paste0</span>(<span class="st">&quot;PC2 (&quot;</span>, <span class="fu">signif</span>(pve<span class="sc">$</span>pve[<span class="dv">2</span>], <span class="dv">3</span>), <span class="st">&quot;%)&quot;</span>))</span>
<span id="cb47-15"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb47-15" tabindex="-1"></a>pc34 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(pca, <span class="fu">aes</span>(PC3, PC4)) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb47-16"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb47-16" tabindex="-1"></a>            <span class="fu">scale_colour_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">&quot;red&quot;</span>, <span class="st">&quot;blue&quot;</span>)) <span class="sc">+</span></span>
<span id="cb47-17"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb47-17" tabindex="-1"></a>            <span class="fu">coord_equal</span>() <span class="sc">+</span> <span class="fu">theme_light</span>() <span class="sc">+</span></span>
<span id="cb47-18"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb47-18" tabindex="-1"></a>            <span class="fu">xlab</span>(<span class="fu">paste0</span>(<span class="st">&quot;PC3 (&quot;</span>, <span class="fu">signif</span>(pve<span class="sc">$</span>pve[<span class="dv">3</span>], <span class="dv">3</span>), <span class="st">&quot;%)&quot;</span>)) <span class="sc">+</span>    </span>
<span id="cb47-19"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb47-19" tabindex="-1"></a>            <span class="fu">ylab</span>(<span class="fu">paste0</span>(<span class="st">&quot;PC4 (&quot;</span>, <span class="fu">signif</span>(pve<span class="sc">$</span>pve[<span class="dv">4</span>], <span class="dv">3</span>), <span class="st">&quot;%)&quot;</span>))</span>
<span id="cb47-20"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb47-20" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb47-21"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb47-21" tabindex="-1"></a>pc12 <span class="sc">+</span> pc34 <span class="sc">+</span><span class="fu">plot_layout</span>(<span class="at">ncol=</span><span class="dv">2</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:PCplot"></span>
<img src="cshl_gwas_files/figure-html/PCplot-1.png" alt="PC plot of genotype" width="150%" />
<p class="caption">
Figure 3.1: PC plot of genotype
</p>
</div>
<p>If the data contains categorical variables, such as case/control, we can also visualize the PCs by coloring different groups. We don’t have case/control in this example data, so I generate a random case/control variable to use as an example. The codes below randomly assign 30% of the samples as cases and rest as controls.</p>
<div class="sourceCode" id="cb48"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb48-1"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb48-1" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">123</span>)</span>
<span id="cb48-2"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb48-2" tabindex="-1"></a>group<span class="ot">&lt;-</span><span class="fu">rep</span>(<span class="st">&quot;control&quot;</span>,<span class="fu">nrow</span>(pca))</span>
<span id="cb48-3"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb48-3" tabindex="-1"></a>case_idx<span class="ot">&lt;-</span><span class="fu">sample</span>(<span class="fu">nrow</span>(pca),<span class="fl">0.3</span><span class="sc">*</span><span class="fu">nrow</span>(pca))</span>
<span id="cb48-4"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb48-4" tabindex="-1"></a>group[case_idx]<span class="ot">&lt;-</span><span class="st">&quot;case&quot;</span></span>
<span id="cb48-5"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb48-5" tabindex="-1"></a></span>
<span id="cb48-6"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb48-6" tabindex="-1"></a><span class="co"># plot pca by case/control</span></span>
<span id="cb48-7"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb48-7" tabindex="-1"></a>pc12 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(pca, <span class="fu">aes</span>(PC1, PC2,<span class="at">col =</span> group)) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb48-8"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb48-8" tabindex="-1"></a>            <span class="fu">scale_colour_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">&quot;red&quot;</span>, <span class="st">&quot;blue&quot;</span>)) <span class="sc">+</span></span>
<span id="cb48-9"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb48-9" tabindex="-1"></a>            <span class="fu">coord_equal</span>() <span class="sc">+</span> <span class="fu">theme_light</span>() <span class="sc">+</span></span>
<span id="cb48-10"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb48-10" tabindex="-1"></a>            <span class="fu">xlab</span>(<span class="fu">paste0</span>(<span class="st">&quot;PC1 (&quot;</span>, <span class="fu">signif</span>(pve<span class="sc">$</span>pve[<span class="dv">1</span>], <span class="dv">3</span>), <span class="st">&quot;%)&quot;</span>)) <span class="sc">+</span>    </span>
<span id="cb48-11"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb48-11" tabindex="-1"></a>            <span class="fu">ylab</span>(<span class="fu">paste0</span>(<span class="st">&quot;PC2 (&quot;</span>, <span class="fu">signif</span>(pve<span class="sc">$</span>pve[<span class="dv">2</span>], <span class="dv">3</span>), <span class="st">&quot;%)&quot;</span>))</span>
<span id="cb48-12"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb48-12" tabindex="-1"></a>pc34 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(pca, <span class="fu">aes</span>(PC3, PC4,<span class="at">col =</span> group)) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb48-13"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb48-13" tabindex="-1"></a>            <span class="fu">scale_colour_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">&quot;red&quot;</span>, <span class="st">&quot;blue&quot;</span>)) <span class="sc">+</span></span>
<span id="cb48-14"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb48-14" tabindex="-1"></a>            <span class="fu">coord_equal</span>() <span class="sc">+</span> <span class="fu">theme_light</span>() <span class="sc">+</span></span>
<span id="cb48-15"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb48-15" tabindex="-1"></a>            <span class="fu">xlab</span>(<span class="fu">paste0</span>(<span class="st">&quot;PC3 (&quot;</span>, <span class="fu">signif</span>(pve<span class="sc">$</span>pve[<span class="dv">3</span>], <span class="dv">3</span>), <span class="st">&quot;%)&quot;</span>)) <span class="sc">+</span>    </span>
<span id="cb48-16"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb48-16" tabindex="-1"></a>            <span class="fu">ylab</span>(<span class="fu">paste0</span>(<span class="st">&quot;PC4 (&quot;</span>, <span class="fu">signif</span>(pve<span class="sc">$</span>pve[<span class="dv">4</span>], <span class="dv">3</span>), <span class="st">&quot;%)&quot;</span>))</span>
<span id="cb48-17"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb48-17" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb48-18"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb48-18" tabindex="-1"></a>pc12 <span class="sc">+</span> pc34 <span class="sc">+</span><span class="fu">plot_layout</span>(<span class="at">ncol=</span><span class="dv">2</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:PCplot2"></span>
<img src="cshl_gwas_files/figure-html/PCplot2-1.png" alt="PC plot of genotype, colored by case/control" width="150%" />
<p class="caption">
Figure 3.2: PC plot of genotype, colored by case/control
</p>
</div>
</div>
<div id="test-the-significance-of-principle-components-pcs" class="section level3 hasAnchor" number="3.5.2">
<h3><span class="header-section-number">3.5.2</span> Test the Significance of Principle Components (PCs)<a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#test-the-significance-of-principle-components-pcs" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The following commands employ linear regression to test the significance of PCs. P values can be found in the last column of the table: <code>Pr(&gt;|t|)</code>.</p>
<ul>
<li>phenotype file: r_met.txt</li>
</ul>
<div class="sourceCode" id="cb49"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb49-1"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb49-1" tabindex="-1"></a><span class="co"># R codes- run in R Console</span></span>
<span id="cb49-2"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb49-2" tabindex="-1"></a>y<span class="ot">=</span><span class="fu">read.table</span>(<span class="st">&quot;./Data/PC/r_met.txt&quot;</span>) <span class="co">#read phenotype data</span></span>
<span id="cb49-3"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb49-3" tabindex="-1"></a>y<span class="ot">=</span><span class="fu">as.matrix</span>(y) <span class="co">#create a matrix from phenotype data</span></span>
<span id="cb49-4"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb49-4" tabindex="-1"></a>pca<span class="ot">=</span><span class="fu">as.matrix</span>(pca) <span class="co">#create a matrix from pc data</span></span>
<span id="cb49-5"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb49-5" tabindex="-1"></a>pval<span class="ot">&lt;-</span><span class="cn">NULL</span></span>
<span id="cb49-6"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb49-6" tabindex="-1"></a>rsq<span class="ot">&lt;-</span><span class="cn">NULL</span></span>
<span id="cb49-7"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb49-7" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">2</span><span class="sc">:</span><span class="fu">ncol</span>(pca)){</span>
<span id="cb49-8"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb49-8" tabindex="-1"></a>  fit<span class="ot">=</span><span class="fu">lm</span>(y<span class="sc">~</span><span class="fu">as.numeric</span>(pca[,i]))</span>
<span id="cb49-9"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb49-9" tabindex="-1"></a>  pval[i<span class="dv">-1</span>]<span class="ot">&lt;-</span><span class="fu">summary</span>(fit)<span class="sc">$</span>coefficients[<span class="dv">2</span>,<span class="dv">4</span>]</span>
<span id="cb49-10"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb49-10" tabindex="-1"></a>  rsq[i<span class="dv">-1</span>]<span class="ot">&lt;-</span><span class="fu">summary</span>(fit)<span class="sc">$</span>r.squared</span>
<span id="cb49-11"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb49-11" tabindex="-1"></a>}</span>
<span id="cb49-12"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb49-12" tabindex="-1"></a><span class="fu">head</span>(pval)</span></code></pre></div>
<pre><code>## [1] 0.9170677 0.5859599 0.3174120 0.4024963 0.6975409 0.1113872</code></pre>
<div class="sourceCode" id="cb51"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb51-1"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb51-1" tabindex="-1"></a><span class="fu">head</span>(rsq)</span></code></pre></div>
<pre><code>## [1] 3.174955e-05 8.684299e-04 2.922747e-03 2.050179e-03 4.421935e-04
## [6] 7.393974e-03</code></pre>
</div>
</div>
<div id="simple-association-test" class="section level2 hasAnchor" number="3.6">
<h2><span class="header-section-number">3.6</span> Simple Association Test<a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#simple-association-test" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>PLINK is a free, open-source whole genome association analysis toolset, designed to perform a range of basic, large-scale analyses in a computationally efficient manner, see <a href="http://zzz.bwh.harvard.edu/plink/" class="uri">http://zzz.bwh.harvard.edu/plink/</a>.</p>
<p>The input files include PED file, MAP file, alternate phenotype file, and optional covariate file.</p>
<p>The PED file is a white-space (space or tab) delimited file, the first six columns are mandatory:</p>
<pre><code>Col 1: Family ID 

Col 2: Individual ID 

Col 3: Paternal ID 

Col 4: Maternal ID 

Col 5: Sex (1=male; 2=female; other character =unknown) 

Col 6: Phenotype (The missing phenotype value for quantitative traits is, by default, -9)

Col 7(*): Genotype (The missing genotype value is denoted as 0, by default)</code></pre>
<p>Example:</p>
<pre><code>FAM001 1 0 0 1 3.4 A A G G A C C C 

FAM001 2 0 0 1 2.5 A A A G 0 0 A C</code></pre>
<p>The MAP file has four columns:</p>
<pre><code>Col 1: Chromosome (1-22, X, Y or 0 if unplaced)

Col 2: rs# or SNP identifier 

Col 3: Genetic distance (morgans) 

Col 4: Base-pair position (bp units)</code></pre>
<p>To specify an alternate phenotype for analysis, i.e. other than the one in the .ped file, use the option –pheno. The alternate phenotype file has three columns:</p>
<pre><code> Col 1: Family ID
 
 Col 2: Individual ID
 
 Col 3: Phenotype</code></pre>
<p>Example:</p>
<pre><code> F1     1110     2.30   22.22  2  
 
 F2     2202     34.12  18.23  1  
 </code></pre>
<p>The phenotype can be either a quantitative trait or an affection status column. PLINK will automatically detect which type it is based on whether a value other than 0, 1, 2 or the missing genotype code is observed.</p>
<p>(All files and results in this section are saved in <code>./PC</code> folder.)</p>
<div id="quantitative-trait-association-test-for-continuous-trait" class="section level3 hasAnchor" number="3.6.1">
<h3><span class="header-section-number">3.6.1</span> Quantitative Trait Association Test for Continuous Trait<a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#quantitative-trait-association-test-for-continuous-trait" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Below is an example that demonstrates how to use PLINK to perform a quantitative trait association test for a toy dataset with a continuous trait and 5 SNPs:</p>
<ul>
<li>Genotype data:
<ul>
<li>5SNPs.ped</li>
<li>5SNPs.map</li>
</ul></li>
<li>Phenotype data:
<ul>
<li>conty.txt</li>
</ul></li>
</ul>
<p>The following commands perform the <strong>simple association test</strong></p>
<ol style="list-style-type: decimal">
<li><strong>–assoc</strong>, perform case/control or QTL association. You can also use <strong>–fisher</strong> to do Fisher’s exact (allelic) test or <strong>–model</strong> to do Cochran-Armitage and full-model C/C association.</li>
<li><strong>–pheno</strong>, specify the phenotype file.</li>
</ol>
<div class="sourceCode" id="cb58"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb58-1"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb58-1" tabindex="-1"></a><span class="co"># both Mac and Windows can run in Terminal</span></span>
<span id="cb58-2"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb58-2" tabindex="-1"></a><span class="co"># Windows user</span></span>
<span id="cb58-3"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb58-3" tabindex="-1"></a><span class="ex">./Tools/PLINK/plink_win64_20231211/plink.exe</span> <span class="dt">\</span></span>
<span id="cb58-4"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb58-4" tabindex="-1"></a>--noweb <span class="at">--file</span> ./Data/Assoc/5SNPs <span class="at">--assoc</span> <span class="at">--pheno</span> ./Data/Assoc/conty.txt <span class="at">--out</span> ./Data/Assoc/younameit.conty</span></code></pre></div>
<div class="sourceCode" id="cb59"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb59-1"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb59-1" tabindex="-1"></a><span class="co"># macOS user</span></span>
<span id="cb59-2"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb59-2" tabindex="-1"></a><span class="ex">./Tools/PLINK/plink_mac_20231211/plink</span> <span class="dt">\</span></span>
<span id="cb59-3"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb59-3" tabindex="-1"></a>--noweb <span class="at">--file</span> ./Data/Assoc/5SNPs <span class="at">--assoc</span> <span class="at">--pheno</span> ./Data/Assoc/conty.txt <span class="at">--out</span> ./Data/Assoc/younameit.conty</span></code></pre></div>
<pre><code>## PLINK v1.90b7.2 64-bit (11 Dec 2023)           www.cog-genomics.org/plink/1.9/
## (C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3
## Logging to ./Data/Assoc/younameit.conty.log.
## Options in effect:
##   --assoc
##   --file ./Data/Assoc/5SNPs
##   --noweb
##   --out ./Data/Assoc/younameit.conty
##   --pheno ./Data/Assoc/conty.txt
## 
## Note: --noweb has no effect since no web check is implemented yet.
## 16384 MB RAM detected; reserving 8192 MB for main workspace.
## Scanning .ped file... 0%1%1%1%1%2%2%2%3%3%3%3%4%4%4%5%5%5%5%6%6%6%7%7%7%7%8%8%8%9%9%9%9%10%10%10%11%11%11%11%12%12%12%13%13%13%13%14%14%14%15%15%15%15%16%16%16%17%17%17%17%18%18%18%19%19%19%19%20%20%20%21%21%21%21%22%22%22%23%23%23%23%24%24%24%25%25%25%25%26%26%26%27%27%27%27%28%28%28%29%29%29%29%30%30%30%31%31%31%32%32%32%32%33%33%33%34%34%34%34%35%35%35%36%36%36%37%37%37%37%38%38%38%39%39%39%39%40%40%40%41%41%41%41%42%42%42%43%43%43%44%44%44%44%45%45%45%46%46%46%46%47%47%47%48%48%48%49%49%49%49%50%50%50%51%51%51%51%52%52%52%53%53%53%53%54%54%54%55%55%55%56%56%56%56%57%57%57%58%58%58%58%59%59%59%60%60%60%61%61%61%61%62%62%62%63%63%63%63%64%64%64%65%65%65%66%66%66%66%67%67%67%68%68%68%68%69%69%69%70%70%70%70%71%71%71%72%72%72%73%73%73%73%74%74%74%75%75%75%75%76%76%76%77%77%77%78%78%78%78%79%79%79%80%80%80%80%81%81%81%82%82%82%83%83%83%83%84%84%84%85%85%85%85%86%86%86%87%87%87%87%88%88%88%89%89%89%90%90%90%90%91%91%91%92%92%92%92%93%93%93%94%94%94%95%95%95%95%96%96%96%97%97%97%97%98%98%98%99%99%99%100%.ped scan complete (for binary autoconversion).
## Performing single-pass .bed write (5 variants, 344 people).
## 0%1%2%3%4%5%6%7%8%9%10%11%12%13%14%15%16%17%18%19%20%21%22%23%24%25%26%27%28%29%30%31%32%33%34%35%36%37%38%39%40%41%42%43%44%45%46%47%48%49%50%51%52%53%54%55%56%57%58%59%60%61%62%63%64%65%66%67%68%69%70%71%72%73%74%75%76%77%78%79%80%81%82%83%84%85%86%87%88%89%90%91%92%93%94%--file: ./Data/Assoc/younameit.conty-temporary.bed +
## ./Data/Assoc/younameit.conty-temporary.bim +
## ./Data/Assoc/younameit.conty-temporary.fam written.
## 5 variants loaded from .bim file.
## 344 people (281 males, 0 females, 63 ambiguous) loaded from .fam.
## Ambiguous sex IDs written to ./Data/Assoc/younameit.conty.nosex .
## 344 phenotype values present after --pheno.
## Warning: Ignoring phenotypes of missing-sex samples.  If you don&#39;t want those
## phenotypes to be ignored, use the --allow-no-sex flag.
## Using 1 thread (no multithreaded calculations invoked).
## Before main variant filters, 344 founders and 0 nonfounders present.
## Calculating allele frequencies... 0%1%2%3%4%5%6%7%8%9%10%11%12%13%14%15%16%17%18%19%20%21%22%23%24%25%26%27%28%29%30%31%32%33%34%35%36%37%38%39%40%41%42%43%44%45%46%47%48%49%50%51%52%53%54%55%56%57%58%59%60%61%62%63%64%65%66%67%68%69%70%71%72%73%74%75%76%77%78%79%80%81%82%83%84%85%86%87%88%89%90%91%92%93%94%95%96%97%98%99% done.
## Total genotyping rate is 0.976744.
## 5 variants and 344 people pass filters and QC.
## Phenotype data is quantitative.
## Writing QT --assoc report to ./Data/Assoc/younameit.conty.qassoc ... 0%done.</code></pre>
<p>This will generate the output file <strong>“younameit.conty.qassoc”</strong> with following columns:</p>
<pre><code>CHR      Chromosome number

SNP      SNP identifier

BP       Physical position (base-pair)

NMISS    Number of non-missing genotypes

BETA     Regression coefficient

SE       Standard error

R2       Regression r-squared

T        Wald test (based on t-distribution)

P        Wald test asymptotic p-value</code></pre>
<div class="sourceCode" id="cb62"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb62-1"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb62-1" tabindex="-1"></a><span class="co"># bash codes- run in terminal</span></span>
<span id="cb62-2"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb62-2" tabindex="-1"></a><span class="co"># Windows user need to run in Powershell App</span></span>
<span id="cb62-3"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb62-3" tabindex="-1"></a><span class="fu">head</span> ./Data/Assoc/younameit.conty.qassoc</span></code></pre></div>
<pre><code>##  CHR         SNP         BP    NMISS       BETA         SE         R2        T            P 
##    1   rs3094315     752566      280    0.07487     0.1485  0.0009132   0.5041       0.6146 
##    1   rs2073813     753541      272    0.07533     0.1007   0.002068   0.7481       0.4551 
##    1   rs3131969     754182      271     0.1038    0.09773   0.004176    1.062       0.2892 
##    1   rs3131967     754334      280    0.06321     0.1045   0.001313   0.6046        0.546 
##    1  rs77598327     757120      275    0.08091     0.1341   0.001332   0.6035       0.5467</code></pre>
</div>
<div id="casecontrol-asssociation-test-for-binary-trait" class="section level3 hasAnchor" number="3.6.2">
<h3><span class="header-section-number">3.6.2</span> Case/Control Asssociation Test for Binary Trait<a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#casecontrol-asssociation-test-for-binary-trait" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Below is an example that demonstrates how to use PLINK to perform a case/control association test for a binary trait and 5 SNPs:</p>
<ul>
<li><p>Genotype data:</p>
<ul>
<li><p>5SNPs.ped</p></li>
<li><p>5SNPs.map</p></li>
</ul></li>
<li><p>Phenotype data:</p>
<ul>
<li>biny.txt</li>
</ul></li>
</ul>
<p>The following commands perform the <strong>simple association test</strong></p>
<ol style="list-style-type: decimal">
<li><strong>–assoc</strong>, perform case/control or QTL association. You can also use <strong>–fisher</strong> to do Fisher’s exact (allelic) test or <strong>–model</strong> to do Cochran-Armitage and full-model C/C association.</li>
<li><strong>–pheno</strong>, specify the phenotype file.</li>
</ol>
<div class="sourceCode" id="cb64"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb64-1"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb64-1" tabindex="-1"></a><span class="co"># both Mac and Windows can run in Terminal</span></span>
<span id="cb64-2"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb64-2" tabindex="-1"></a><span class="co"># Windows user</span></span>
<span id="cb64-3"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb64-3" tabindex="-1"></a><span class="ex">./Tools/PLINK/plink_win64_20231211/plink.exe</span> <span class="dt">\</span></span>
<span id="cb64-4"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb64-4" tabindex="-1"></a>--noweb <span class="at">--file</span> ./Data/Assoc/5SNPs <span class="at">--assoc</span> <span class="at">--pheno</span> ./Data/Assoc/biny.txt <span class="at">--out</span> ./Data/Assoc/younameit.biny</span></code></pre></div>
<div class="sourceCode" id="cb65"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb65-1"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb65-1" tabindex="-1"></a><span class="co"># macOS user</span></span>
<span id="cb65-2"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb65-2" tabindex="-1"></a><span class="ex">./Tools/PLINK/plink_mac_20231211/plink</span> <span class="dt">\</span></span>
<span id="cb65-3"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb65-3" tabindex="-1"></a>--noweb <span class="at">--file</span> ./Data/Assoc/5SNPs <span class="at">--assoc</span> <span class="at">--pheno</span> ./Data/Assoc/biny.txt <span class="at">--out</span> ./Data/Assoc/younameit.biny</span></code></pre></div>
<pre><code>## PLINK v1.90b7.2 64-bit (11 Dec 2023)           www.cog-genomics.org/plink/1.9/
## (C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3
## Logging to ./Data/Assoc/younameit.biny.log.
## Options in effect:
##   --assoc
##   --file ./Data/Assoc/5SNPs
##   --noweb
##   --out ./Data/Assoc/younameit.biny
##   --pheno ./Data/Assoc/biny.txt
## 
## Note: --noweb has no effect since no web check is implemented yet.
## 16384 MB RAM detected; reserving 8192 MB for main workspace.
## Scanning .ped file... 0%1%1%1%1%2%2%2%3%3%3%3%4%4%4%5%5%5%5%6%6%6%7%7%7%7%8%8%8%9%9%9%9%10%10%10%11%11%11%11%12%12%12%13%13%13%13%14%14%14%15%15%15%15%16%16%16%17%17%17%17%18%18%18%19%19%19%19%20%20%20%21%21%21%21%22%22%22%23%23%23%23%24%24%24%25%25%25%25%26%26%26%27%27%27%27%28%28%28%29%29%29%29%30%30%30%31%31%31%32%32%32%32%33%33%33%34%34%34%34%35%35%35%36%36%36%37%37%37%37%38%38%38%39%39%39%39%40%40%40%41%41%41%41%42%42%42%43%43%43%44%44%44%44%45%45%45%46%46%46%46%47%47%47%48%48%48%49%49%49%49%50%50%50%51%51%51%51%52%52%52%53%53%53%53%54%54%54%55%55%55%56%56%56%56%57%57%57%58%58%58%58%59%59%59%60%60%60%61%61%61%61%62%62%62%63%63%63%63%64%64%64%65%65%65%66%66%66%66%67%67%67%68%68%68%68%69%69%69%70%70%70%70%71%71%71%72%72%72%73%73%73%73%74%74%74%75%75%75%75%76%76%76%77%77%77%78%78%78%78%79%79%79%80%80%80%80%81%81%81%82%82%82%83%83%83%83%84%84%84%85%85%85%85%86%86%86%87%87%87%87%88%88%88%89%89%89%90%90%90%90%91%91%91%92%92%92%92%93%93%93%94%94%94%95%95%95%95%96%96%96%97%97%97%97%98%98%98%99%99%99%100%.ped scan complete (for binary autoconversion).
## Performing single-pass .bed write (5 variants, 344 people).
## 0%1%2%3%4%5%6%7%8%9%10%11%12%13%14%15%16%17%18%19%20%21%22%23%24%25%26%27%28%29%30%31%32%33%34%35%36%37%38%39%40%41%42%43%44%45%46%47%48%49%50%51%52%53%54%55%56%57%58%59%60%61%62%63%64%65%66%67%68%69%70%71%72%73%74%75%76%77%78%79%80%81%82%83%84%85%86%87%88%89%90%91%92%93%94%--file: ./Data/Assoc/younameit.biny-temporary.bed +
## ./Data/Assoc/younameit.biny-temporary.bim +
## ./Data/Assoc/younameit.biny-temporary.fam written.
## 5 variants loaded from .bim file.
## 344 people (281 males, 0 females, 63 ambiguous) loaded from .fam.
## Ambiguous sex IDs written to ./Data/Assoc/younameit.biny.nosex .
## 344 phenotype values present after --pheno.
## Warning: Ignoring phenotypes of missing-sex samples.  If you don&#39;t want those
## phenotypes to be ignored, use the --allow-no-sex flag.
## Using 1 thread (no multithreaded calculations invoked).
## Before main variant filters, 344 founders and 0 nonfounders present.
## Calculating allele frequencies... 0%1%2%3%4%5%6%7%8%9%10%11%12%13%14%15%16%17%18%19%20%21%22%23%24%25%26%27%28%29%30%31%32%33%34%35%36%37%38%39%40%41%42%43%44%45%46%47%48%49%50%51%52%53%54%55%56%57%58%59%60%61%62%63%64%65%66%67%68%69%70%71%72%73%74%75%76%77%78%79%80%81%82%83%84%85%86%87%88%89%90%91%92%93%94%95%96%97%98%99% done.
## Total genotyping rate is 0.976744.
## 5 variants and 344 people pass filters and QC.
## Among remaining phenotypes, 137 are cases and 144 are controls.  (63 phenotypes
## are missing.)
## Writing C/C --assoc report to ./Data/Assoc/younameit.biny.assoc ...
## 0%done.</code></pre>
<p>The output is <strong>“younameit.biny.assoc”</strong>, which contains the fields:</p>
<pre><code> CHR     Chromosome
 SNP     SNP ID
 BP      Physical position (base-pair)
 A1      Minor allele name (based on whole sample)
 F_A     Frequency of this allele in cases
 F_U     Frequency of this allele in controls
 A2      Major allele name
 CHISQ   Basic allelic test chi-square (1df)
 P       Asymptotic p-value for this test
 OR      Estimated odds ratio (for A1, i.e. A2 is reference)
 </code></pre>
<div class="sourceCode" id="cb68"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb68-1"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb68-1" tabindex="-1"></a><span class="co"># bash codes- run in terminal</span></span>
<span id="cb68-2"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb68-2" tabindex="-1"></a><span class="co"># Windows user need to run in Powershell App</span></span>
<span id="cb68-3"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb68-3" tabindex="-1"></a><span class="fu">head</span> ./Data/Assoc/younameit.biny.assoc</span></code></pre></div>
<pre><code>##  CHR         SNP         BP   A1      F_A      F_U   A2        CHISQ            P           OR 
##    1   rs3094315     752566    G  0.07664  0.09091    A       0.3705       0.5428         0.83 
##    1   rs2073813     753541    A   0.2239   0.2391    G       0.1777       0.6734       0.9178 
##    1   rs3131969     754182    A   0.2176   0.2607    G        1.383       0.2396       0.7884 
##    1   rs3131967     754334    A   0.1949   0.2326    G        1.186       0.2761       0.7983 
##    1  rs77598327     757120    G   0.1111   0.1393    A       0.9945       0.3186       0.7724</code></pre>
</div>
</div>
<div id="quantitative-trait-association-test-for-continuous-trait-with-covariates" class="section level2 hasAnchor" number="3.7">
<h2><span class="header-section-number">3.7</span> Quantitative Trait Association Test for Continuous Trait with Covariates<a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#quantitative-trait-association-test-for-continuous-trait-with-covariates" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Below is an example that demonstrates how to use PLINK to perform a quantitative trait test for a continuous trait and 5 SNPs with 3 covariates:</p>
<ol style="list-style-type: decimal">
<li>Genotype data:</li>
</ol>
<ul>
<li>5SNPs.ped</li>
<li>5SNPs.map</li>
</ul>
<ol start="2" style="list-style-type: decimal">
<li>Phenotype data:</li>
</ol>
<ul>
<li>conty_pheno.txt</li>
</ul>
<ol start="3" style="list-style-type: decimal">
<li>Covariates:</li>
</ol>
<ul>
<li>conty_covar.txt</li>
</ul>
<p>Before the association test, we adjust the phenotypes to remove the effects of covariates. For the continous phenotype, we employ linear regression to adjust the phenotype. Then, we use the residual as the new phenotype in the association test.</p>
<div id="covariates-adjustments-using-r" class="section level3 hasAnchor" number="3.7.1">
<h3><span class="header-section-number">3.7.1</span> Covariates adjustments (using R):<a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#covariates-adjustments-using-r" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb70"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb70-1"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb70-1" tabindex="-1"></a><span class="do">## R codes- run in R Console</span></span>
<span id="cb70-2"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb70-2" tabindex="-1"></a><span class="co"># Read phenotype data and covariates data</span></span>
<span id="cb70-3"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb70-3" tabindex="-1"></a>pheno<span class="ot">=</span><span class="fu">read.table</span>(<span class="st">&quot;./Data/Assoc/conty_pheno.txt&quot;</span>) <span class="co">#read phenotype data</span></span>
<span id="cb70-4"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb70-4" tabindex="-1"></a>covar<span class="ot">=</span><span class="fu">read.table</span>(<span class="st">&quot;./Data/Assoc/conty_covar.txt&quot;</span>) <span class="co">#read covariates data</span></span>
<span id="cb70-5"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb70-5" tabindex="-1"></a>pheno<span class="ot">=</span><span class="fu">as.matrix</span>(pheno) <span class="co">#create a matrix from the phenotype data </span></span>
<span id="cb70-6"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb70-6" tabindex="-1"></a>covar<span class="ot">=</span><span class="fu">as.matrix</span>(covar) <span class="co">#create a matrix from the covariates data</span></span>
<span id="cb70-7"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb70-7" tabindex="-1"></a>n<span class="ot">=</span><span class="fu">dim</span>(pheno)[<span class="dv">1</span>] <span class="co">#sample size</span></span>
<span id="cb70-8"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb70-8" tabindex="-1"></a>p<span class="ot">=</span><span class="fu">dim</span>(pheno)[<span class="dv">2</span>] <span class="co">#number of traits</span></span></code></pre></div>
<p>Covariates adjustment is done by fitting a linear regression model and obtain residuals as new phenotype data for association test.</p>
<p>The following codes construct the linear model for each of the traits using <code>lm</code> function.The resulted residual data is saved for analysis.</p>
<div class="sourceCode" id="cb71"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb71-1"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb71-1" tabindex="-1"></a><span class="co"># R codes- run in R Console</span></span>
<span id="cb71-2"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb71-2" tabindex="-1"></a>fit<span class="ot">=</span><span class="fu">list</span>() <span class="co">#create an empty list</span></span>
<span id="cb71-3"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb71-3" tabindex="-1"></a>residpheno<span class="ot">=</span><span class="fu">matrix</span>(<span class="dv">0</span>, n, p) <span class="co">#create an empty matrix</span></span>
<span id="cb71-4"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb71-4" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>p){</span>
<span id="cb71-5"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb71-5" tabindex="-1"></a>  fit[[i]]<span class="ot">=</span><span class="fu">lm</span>(pheno[,i]<span class="sc">~</span>covar) <span class="co">#for each trait, fit a linear regression model of phenotype against covariates</span></span>
<span id="cb71-6"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb71-6" tabindex="-1"></a>  residpheno[,i]<span class="ot">=</span><span class="fu">resid</span>(fit[[i]]) <span class="co">#obtain residual phenotype data</span></span>
<span id="cb71-7"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb71-7" tabindex="-1"></a>}</span>
<span id="cb71-8"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb71-8" tabindex="-1"></a><span class="fu">write.table</span>(residpheno, <span class="st">&quot;./Data/Assoc/resid_phenotype.txt&quot;</span>, <span class="at">row.names=</span>F, <span class="at">col.names=</span>F, <span class="at">quote=</span>F, <span class="at">sep=</span><span class="st">&quot;&quot;</span>) <span class="co">#save residual phenotype data</span></span></code></pre></div>
<p>This code combines the above residual data and sample ID information to contruct the phenotype data for PLINK.</p>
<div class="sourceCode" id="cb72"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb72-1"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb72-1" tabindex="-1"></a><span class="co"># prepare phenotype data for PLINK by combining sample ID information</span></span>
<span id="cb72-2"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb72-2" tabindex="-1"></a>famID<span class="ot">&lt;-</span><span class="fu">read.table</span>(<span class="st">&quot;./Data/Assoc/famID.txt&quot;</span>)</span>
<span id="cb72-3"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb72-3" tabindex="-1"></a>sampleID<span class="ot">&lt;-</span><span class="fu">read.table</span>(<span class="st">&quot;./Data/Assoc/sampleID.txt&quot;</span>)</span>
<span id="cb72-4"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb72-4" tabindex="-1"></a>df<span class="ot">&lt;-</span><span class="fu">cbind</span>(famID,sampleID,residpheno)</span>
<span id="cb72-5"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb72-5" tabindex="-1"></a><span class="fu">write.table</span>(df, <span class="st">&quot;./Data/Assoc/resid_phenotype_assoc.txt&quot;</span>, <span class="at">row.names=</span>F, <span class="at">col.names=</span>F, <span class="at">quote=</span>F, <span class="at">sep=</span><span class="st">&quot; &quot;</span>)</span></code></pre></div>
</div>
<div id="association-test-using-plink" class="section level3 hasAnchor" number="3.7.2">
<h3><span class="header-section-number">3.7.2</span> Association test using plink<a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#association-test-using-plink" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>After adjusting the phenotype using a set of covariates, we use the residual as new phenotype for the association test. <code>PLINK</code> was used again to perform assiciation test.</p>
<div class="sourceCode" id="cb73"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb73-1"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb73-1" tabindex="-1"></a><span class="co"># both Mac and Windows can run in Terminal</span></span>
<span id="cb73-2"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb73-2" tabindex="-1"></a><span class="co"># Windows user</span></span>
<span id="cb73-3"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb73-3" tabindex="-1"></a><span class="ex">./Tools/PLINK/plink_win64_20231211/plink.exe</span> <span class="dt">\</span></span>
<span id="cb73-4"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb73-4" tabindex="-1"></a>--noweb <span class="at">--file</span> ./Data/Assoc/5SNPs <span class="at">--assoc</span> <span class="at">--pheno</span> ./Data/Assoc/resid_phenotype_assoc.txt <span class="at">--out</span> ./Data/Assoc/younameit_resid</span></code></pre></div>
<div class="sourceCode" id="cb74"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb74-1"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb74-1" tabindex="-1"></a><span class="co"># macOS user</span></span>
<span id="cb74-2"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb74-2" tabindex="-1"></a><span class="ex">./Tools/PLINK/plink_mac_20231211/plink</span> <span class="dt">\</span></span>
<span id="cb74-3"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb74-3" tabindex="-1"></a>--noweb <span class="at">--file</span> ./Data/Assoc/5SNPs <span class="at">--assoc</span> <span class="at">--pheno</span> ./Data/Assoc/resid_phenotype_assoc.txt <span class="dt">\</span></span>
<span id="cb74-4"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb74-4" tabindex="-1"></a>--out ./Data/Assoc/younameit_resid</span></code></pre></div>
<pre><code>## PLINK v1.90b7.2 64-bit (11 Dec 2023)           www.cog-genomics.org/plink/1.9/
## (C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3
## Logging to ./Data/Assoc/younameit_resid.log.
## Options in effect:
##   --assoc
##   --file ./Data/Assoc/5SNPs
##   --noweb
##   --out ./Data/Assoc/younameit_resid
##   --pheno ./Data/Assoc/resid_phenotype_assoc.txt
## 
## Note: --noweb has no effect since no web check is implemented yet.
## 16384 MB RAM detected; reserving 8192 MB for main workspace.
## Scanning .ped file... 0%1%1%1%1%2%2%2%3%3%3%3%4%4%4%5%5%5%5%6%6%6%7%7%7%7%8%8%8%9%9%9%9%10%10%10%11%11%11%11%12%12%12%13%13%13%13%14%14%14%15%15%15%15%16%16%16%17%17%17%17%18%18%18%19%19%19%19%20%20%20%21%21%21%21%22%22%22%23%23%23%23%24%24%24%25%25%25%25%26%26%26%27%27%27%27%28%28%28%29%29%29%29%30%30%30%31%31%31%32%32%32%32%33%33%33%34%34%34%34%35%35%35%36%36%36%37%37%37%37%38%38%38%39%39%39%39%40%40%40%41%41%41%41%42%42%42%43%43%43%44%44%44%44%45%45%45%46%46%46%46%47%47%47%48%48%48%49%49%49%49%50%50%50%51%51%51%51%52%52%52%53%53%53%53%54%54%54%55%55%55%56%56%56%56%57%57%57%58%58%58%58%59%59%59%60%60%60%61%61%61%61%62%62%62%63%63%63%63%64%64%64%65%65%65%66%66%66%66%67%67%67%68%68%68%68%69%69%69%70%70%70%70%71%71%71%72%72%72%73%73%73%73%74%74%74%75%75%75%75%76%76%76%77%77%77%78%78%78%78%79%79%79%80%80%80%80%81%81%81%82%82%82%83%83%83%83%84%84%84%85%85%85%85%86%86%86%87%87%87%87%88%88%88%89%89%89%90%90%90%90%91%91%91%92%92%92%92%93%93%93%94%94%94%95%95%95%95%96%96%96%97%97%97%97%98%98%98%99%99%99%100%.ped scan complete (for binary autoconversion).
## Performing single-pass .bed write (5 variants, 344 people).
## 0%1%2%3%4%5%6%7%8%9%10%11%12%13%14%15%16%17%18%19%20%21%22%23%24%25%26%27%28%29%30%31%32%33%34%35%36%37%38%39%40%41%42%43%44%45%46%47%48%49%50%51%52%53%54%55%56%57%58%59%60%61%62%63%64%65%66%67%68%69%70%71%72%73%74%75%76%77%78%79%80%81%82%83%84%85%86%87%88%89%90%91%92%93%94%--file: ./Data/Assoc/younameit_resid-temporary.bed +
## ./Data/Assoc/younameit_resid-temporary.bim +
## ./Data/Assoc/younameit_resid-temporary.fam written.
## 5 variants loaded from .bim file.
## 344 people (281 males, 0 females, 63 ambiguous) loaded from .fam.
## Ambiguous sex IDs written to ./Data/Assoc/younameit_resid.nosex .
## 344 phenotype values present after --pheno.
## Warning: Ignoring phenotypes of missing-sex samples.  If you don&#39;t want those
## phenotypes to be ignored, use the --allow-no-sex flag.
## Using 1 thread (no multithreaded calculations invoked).
## Before main variant filters, 344 founders and 0 nonfounders present.
## Calculating allele frequencies... 0%1%2%3%4%5%6%7%8%9%10%11%12%13%14%15%16%17%18%19%20%21%22%23%24%25%26%27%28%29%30%31%32%33%34%35%36%37%38%39%40%41%42%43%44%45%46%47%48%49%50%51%52%53%54%55%56%57%58%59%60%61%62%63%64%65%66%67%68%69%70%71%72%73%74%75%76%77%78%79%80%81%82%83%84%85%86%87%88%89%90%91%92%93%94%95%96%97%98%99% done.
## Total genotyping rate is 0.976744.
## 5 variants and 344 people pass filters and QC.
## Phenotype data is quantitative.
## Writing QT --assoc report to ./Data/Assoc/younameit_resid.qassoc ... 0%done.</code></pre>
<div class="sourceCode" id="cb76"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb76-1"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb76-1" tabindex="-1"></a><span class="co"># bash codes- run in terminal</span></span>
<span id="cb76-2"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb76-2" tabindex="-1"></a><span class="co"># Windows user need to run in Powershell App</span></span>
<span id="cb76-3"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb76-3" tabindex="-1"></a><span class="co">### check association results</span></span>
<span id="cb76-4"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#cb76-4" tabindex="-1"></a><span class="fu">head</span> ./Data/Assoc/younameit_resid.qassoc</span></code></pre></div>
<pre><code>##  CHR         SNP         BP    NMISS       BETA         SE         R2        T            P 
##    1   rs3094315     752566      280     0.0706      0.148  0.0008178    0.477       0.6337 
##    1   rs2073813     753541      272    0.07739     0.1003   0.002198   0.7713       0.4412 
##    1   rs3131969     754182      271     0.1062     0.0974     0.0044     1.09       0.2766 
##    1   rs3131967     754334      280    0.06205     0.1042   0.001274   0.5954       0.5521 
##    1  rs77598327     757120      275     0.0816     0.1336   0.001366    0.611       0.5417</code></pre>
<p>This output file <strong>“younameit.qassoc”</strong> has the following columns:</p>
<pre><code>CHR      Chromosome number

SNP      SNP identifier

BP       Physical position (base-pair)

NMISS    Number of non-missing genotypes

BETA     Regression coefficient

SE       Standard error

R2       Regression r-squared

T        Wald test (based on t-distribution)

P        Wald test asymptotic p-value</code></pre>
<p>In the next section, we will start analyzing real data from a case study. The procedures might not be exactly the same here, as real data require more proprocessing steps before it can be used for association study.</p>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="an-overview-of-regression-with-toy-examples.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="gwas-case-study-data-preparation.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/02-gwastool.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["cshl_gwas.pdf", "cshl_gwas.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
