<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 7 GxG and GxE Interactions, Power and Sample Size Calculation | CSHL course - GWAS Hands-on Analysis</title>
  <meta name="description" content="This is the course material for the CSHL hands-on session in GWAS and eQTL." />
  <meta name="generator" content="bookdown 0.41 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 7 GxG and GxE Interactions, Power and Sample Size Calculation | CSHL course - GWAS Hands-on Analysis" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This is the course material for the CSHL hands-on session in GWAS and eQTL." />
  <meta name="github-repo" content="rstudio/cshl_gwas" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 7 GxG and GxE Interactions, Power and Sample Size Calculation | CSHL course - GWAS Hands-on Analysis" />
  
  <meta name="twitter:description" content="This is the course material for the CSHL hands-on session in GWAS and eQTL." />
  

<meta name="author" content="Min Zhang, Danni Liu- University of California, Irvine; Purdue University" />


<meta name="date" content="2025-06-09" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="gwas-with-binary-trait.html"/>
<link rel="next" href="ldsc-and-gene-set-analysis.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">CSHL Course- GWAS and EQTL Hands-on Analysis</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Welcome to the GWAS and eQTL Workshop</a>
<ul>
<li class="chapter" data-level="1.0.1" data-path="index.html"><a href="index.html#overview"><i class="fa fa-check"></i><b>1.0.1</b> Overview</a></li>
<li class="chapter" data-level="1.0.2" data-path="index.html"><a href="index.html#folder-structure"><i class="fa fa-check"></i><b>1.0.2</b> Folder structure:</a></li>
<li class="chapter" data-level="1.0.3" data-path="index.html"><a href="index.html#programming-languages"><i class="fa fa-check"></i><b>1.0.3</b> Programming Languages:</a></li>
<li class="chapter" data-level="1.0.4" data-path="index.html"><a href="index.html#required-gwas-tools"><i class="fa fa-check"></i><b>1.0.4</b> Required GWAS Tools</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="an-overview-of-regression-with-toy-examples.html"><a href="an-overview-of-regression-with-toy-examples.html"><i class="fa fa-check"></i><b>2</b> An Overview of Regression with Toy Examples</a>
<ul>
<li class="chapter" data-level="2.1" data-path="an-overview-of-regression-with-toy-examples.html"><a href="an-overview-of-regression-with-toy-examples.html#environment-setting-and-packages"><i class="fa fa-check"></i><b>2.1</b> Environment Setting and Packages</a></li>
<li class="chapter" data-level="2.2" data-path="an-overview-of-regression-with-toy-examples.html"><a href="an-overview-of-regression-with-toy-examples.html#toy-example-data"><i class="fa fa-check"></i><b>2.2</b> Toy Example Data</a></li>
<li class="chapter" data-level="2.3" data-path="an-overview-of-regression-with-toy-examples.html"><a href="an-overview-of-regression-with-toy-examples.html#logistic-regression"><i class="fa fa-check"></i><b>2.3</b> Logistic Regression</a></li>
<li class="chapter" data-level="2.4" data-path="an-overview-of-regression-with-toy-examples.html"><a href="an-overview-of-regression-with-toy-examples.html#roc-curve-and-auc-for-hoxb13"><i class="fa fa-check"></i><b>2.4</b> ROC curve and AUC for HOXB13</a></li>
<li class="chapter" data-level="2.5" data-path="an-overview-of-regression-with-toy-examples.html"><a href="an-overview-of-regression-with-toy-examples.html#practice"><i class="fa fa-check"></i><b>2.5</b> Practice</a></li>
<li class="chapter" data-level="2.6" data-path="an-overview-of-regression-with-toy-examples.html"><a href="an-overview-of-regression-with-toy-examples.html#comparison-of-hoxb13-roc-curve-and-nhlh1-roc-curve"><i class="fa fa-check"></i><b>2.6</b> Comparison of HOXB13 ROC curve and NHLH1 ROC curve</a></li>
<li class="chapter" data-level="2.7" data-path="an-overview-of-regression-with-toy-examples.html"><a href="an-overview-of-regression-with-toy-examples.html#linear-regression"><i class="fa fa-check"></i><b>2.7</b> Linear Regression</a>
<ul>
<li class="chapter" data-level="2.7.1" data-path="an-overview-of-regression-with-toy-examples.html"><a href="an-overview-of-regression-with-toy-examples.html#linear-regression-of-hoxb13-vs.-il17br"><i class="fa fa-check"></i><b>2.7.1</b> Linear regression of HOXB13 vs. IL17BR</a></li>
</ul></li>
<li class="chapter" data-level="2.8" data-path="an-overview-of-regression-with-toy-examples.html"><a href="an-overview-of-regression-with-toy-examples.html#variable-selectionforward-backward-and-stepwise-selection"><i class="fa fa-check"></i><b>2.8</b> Variable selection–Forward, backward and stepwise selection</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html"><i class="fa fa-check"></i><b>3</b> GWAS Tools and Data Preprocessing Workflow with Toy Example</a>
<ul>
<li class="chapter" data-level="3.1" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#environment-setting"><i class="fa fa-check"></i><b>3.1</b> Environment Setting</a></li>
<li class="chapter" data-level="3.2" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#gwas-tools"><i class="fa fa-check"></i><b>3.2</b> GWAS Tools</a></li>
<li class="chapter" data-level="3.3" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#quality-control"><i class="fa fa-check"></i><b>3.3</b> Quality Control</a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#an-example-of-summary-statistics"><i class="fa fa-check"></i><b>3.3.1</b> An Example of Summary Statistics</a></li>
<li class="chapter" data-level="3.3.2" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#an-example-of-inclusion-thresholds-for-qc"><i class="fa fa-check"></i><b>3.3.2</b> An Example of Inclusion Thresholds for QC</a></li>
<li class="chapter" data-level="3.3.3" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#remove-closely-related-individuals"><i class="fa fa-check"></i><b>3.3.3</b> Remove Closely Related Individuals</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#an-imputation-example"><i class="fa fa-check"></i><b>3.4</b> An Imputation Example</a></li>
<li class="chapter" data-level="3.5" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#population-stratification-correction-using-plink"><i class="fa fa-check"></i><b>3.5</b> Population Stratification Correction Using PLINK</a>
<ul>
<li class="chapter" data-level="3.5.1" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#pca-on-whole-genome-data-using-plink"><i class="fa fa-check"></i><b>3.5.1</b> PCA on whole genome data using PLINK</a></li>
<li class="chapter" data-level="3.5.2" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#test-the-significance-of-principle-components-pcs"><i class="fa fa-check"></i><b>3.5.2</b> Test the Significance of Principle Components (PCs)</a></li>
</ul></li>
<li class="chapter" data-level="3.6" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#simple-association-test"><i class="fa fa-check"></i><b>3.6</b> Simple Association Test</a>
<ul>
<li class="chapter" data-level="3.6.1" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#quantitative-trait-association-test-for-continuous-trait"><i class="fa fa-check"></i><b>3.6.1</b> Quantitative Trait Association Test for Continuous Trait</a></li>
<li class="chapter" data-level="3.6.2" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#casecontrol-asssociation-test-for-binary-trait"><i class="fa fa-check"></i><b>3.6.2</b> Case/Control Asssociation Test for Binary Trait</a></li>
</ul></li>
<li class="chapter" data-level="3.7" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#quantitative-trait-association-test-for-continuous-trait-with-covariates"><i class="fa fa-check"></i><b>3.7</b> Quantitative Trait Association Test for Continuous Trait with Covariates</a>
<ul>
<li class="chapter" data-level="3.7.1" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#covariates-adjustments-using-r"><i class="fa fa-check"></i><b>3.7.1</b> Covariates adjustments (using R):</a></li>
<li class="chapter" data-level="3.7.2" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#association-test-using-plink"><i class="fa fa-check"></i><b>3.7.2</b> Association test using plink</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="gwas-case-study-data-preparation.html"><a href="gwas-case-study-data-preparation.html"><i class="fa fa-check"></i><b>4</b> GWAS Case Study-Data Preparation</a>
<ul>
<li class="chapter" data-level="4.1" data-path="gwas-case-study-data-preparation.html"><a href="gwas-case-study-data-preparation.html#environment-setting-and-packages-1"><i class="fa fa-check"></i><b>4.1</b> Environment setting and packages</a></li>
<li class="chapter" data-level="4.2" data-path="gwas-case-study-data-preparation.html"><a href="gwas-case-study-data-preparation.html#accessing-data"><i class="fa fa-check"></i><b>4.2</b> Accessing data</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="gwas-case-study-data-preparation.html"><a href="gwas-case-study-data-preparation.html#data-availability"><i class="fa fa-check"></i><b>4.2.1</b> Data availability</a></li>
<li class="chapter" data-level="4.2.2" data-path="gwas-case-study-data-preparation.html"><a href="gwas-case-study-data-preparation.html#data-files-documentation"><i class="fa fa-check"></i><b>4.2.2</b> Data files documentation</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="gwas-case-study-data-preparation.html"><a href="gwas-case-study-data-preparation.html#preprocessing-phenotype-data"><i class="fa fa-check"></i><b>4.3</b> Preprocessing Phenotype Data</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="gwas-case-study-data-preparation.html"><a href="gwas-case-study-data-preparation.html#extracting-phenotype-data"><i class="fa fa-check"></i><b>4.3.1</b> Extracting phenotype data</a></li>
<li class="chapter" data-level="4.3.2" data-path="gwas-case-study-data-preparation.html"><a href="gwas-case-study-data-preparation.html#summary-of-phenotypes-data"><i class="fa fa-check"></i><b>4.3.2</b> Summary of Phenotypes Data</a></li>
<li class="chapter" data-level="4.3.3" data-path="gwas-case-study-data-preparation.html"><a href="gwas-case-study-data-preparation.html#histogram-of-each-trait"><i class="fa fa-check"></i><b>4.3.3</b> Histogram of each trait</a></li>
<li class="chapter" data-level="4.3.4" data-path="gwas-case-study-data-preparation.html"><a href="gwas-case-study-data-preparation.html#handling-the-missing-values-in-the-phenotypes-and-covariates"><i class="fa fa-check"></i><b>4.3.4</b> Handling the missing values in the phenotypes and covariates</a></li>
<li class="chapter" data-level="4.3.5" data-path="gwas-case-study-data-preparation.html"><a href="gwas-case-study-data-preparation.html#checking-normality-assumptions-for-each-trait"><i class="fa fa-check"></i><b>4.3.5</b> Checking normality assumptions for each trait</a></li>
<li class="chapter" data-level="4.3.6" data-path="gwas-case-study-data-preparation.html"><a href="gwas-case-study-data-preparation.html#normalize-phenotype-data"><i class="fa fa-check"></i><b>4.3.6</b> Normalize phenotype data</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="gwas-case-study-data-preparation.html"><a href="gwas-case-study-data-preparation.html#covariates-adjustment"><i class="fa fa-check"></i><b>4.4</b> Covariates Adjustment</a></li>
<li class="chapter" data-level="4.5" data-path="gwas-case-study-data-preparation.html"><a href="gwas-case-study-data-preparation.html#prepare-plink-data"><i class="fa fa-check"></i><b>4.5</b> Prepare PLINK data</a>
<ul>
<li class="chapter" data-level="4.5.1" data-path="gwas-case-study-data-preparation.html"><a href="gwas-case-study-data-preparation.html#step-1-prepare-plink-ped-file"><i class="fa fa-check"></i><b>4.5.1</b> Step 1: prepare PLINK ped file</a></li>
<li class="chapter" data-level="4.5.2" data-path="gwas-case-study-data-preparation.html"><a href="gwas-case-study-data-preparation.html#step-2-prepare-plink-map-file"><i class="fa fa-check"></i><b>4.5.2</b> Step 2: prepare PLINK map file</a></li>
<li class="chapter" data-level="4.5.3" data-path="gwas-case-study-data-preparation.html"><a href="gwas-case-study-data-preparation.html#step-3-prepare-plink-alternate-phenotype-file-in-linux"><i class="fa fa-check"></i><b>4.5.3</b> Step 3 : prepare PLINK alternate phenotype file (in Linux)</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="gwas-with-continuous-trait.html"><a href="gwas-with-continuous-trait.html"><i class="fa fa-check"></i><b>5</b> GWAS with Continuous Trait</a>
<ul>
<li class="chapter" data-level="5.1" data-path="gwas-with-continuous-trait.html"><a href="gwas-with-continuous-trait.html#manhattan-plot-and-q-q-plot"><i class="fa fa-check"></i><b>5.1</b> Manhattan Plot and Q-Q Plot</a>
<ul>
<li class="chapter" data-level="5.1.1" data-path="gwas-with-continuous-trait.html"><a href="gwas-with-continuous-trait.html#generate-manhattan-plot"><i class="fa fa-check"></i><b>5.1.1</b> Generate manhattan plot</a></li>
<li class="chapter" data-level="5.1.2" data-path="gwas-with-continuous-trait.html"><a href="gwas-with-continuous-trait.html#generate-q-q-plot"><i class="fa fa-check"></i><b>5.1.2</b> Generate Q-Q plot</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="gwas-with-continuous-trait.html"><a href="gwas-with-continuous-trait.html#multiple-test-correction"><i class="fa fa-check"></i><b>5.2</b> Multiple-Test Correction</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="gwas-with-continuous-trait.html"><a href="gwas-with-continuous-trait.html#prepare-r-input-files-in-linux"><i class="fa fa-check"></i><b>5.2.1</b> Prepare R input files (in Linux)</a></li>
<li class="chapter" data-level="5.2.2" data-path="gwas-with-continuous-trait.html"><a href="gwas-with-continuous-trait.html#significant-snps-after-a-multiple-test-correction-of-fdr-using-r"><i class="fa fa-check"></i><b>5.2.2</b> Significant SNPs after a multiple-test correction of FDR (using R)</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="gwas-with-continuous-trait.html"><a href="gwas-with-continuous-trait.html#regional-association-plot"><i class="fa fa-check"></i><b>5.3</b> Regional Association Plot</a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="gwas-with-continuous-trait.html"><a href="gwas-with-continuous-trait.html#obtain-regional-snps-using-r"><i class="fa fa-check"></i><b>5.3.1</b> Obtain regional SNPs (using R)</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="gwas-with-continuous-trait.html"><a href="gwas-with-continuous-trait.html#distribution-of-z-endoxifen-ratio-for-rs8138080"><i class="fa fa-check"></i><b>5.4</b> Distribution of Z-endoxifen ratio for rs8138080</a>
<ul>
<li class="chapter" data-level="5.4.1" data-path="gwas-with-continuous-trait.html"><a href="gwas-with-continuous-trait.html#obtain-phenotype-data-z-endoxifen-ratio-using-r"><i class="fa fa-check"></i><b>5.4.1</b> Obtain phenotype data (Z-endoxifen ratio) (using R)</a></li>
<li class="chapter" data-level="5.4.2" data-path="gwas-with-continuous-trait.html"><a href="gwas-with-continuous-trait.html#obtain-genotype-for-significant-snps-using-r"><i class="fa fa-check"></i><b>5.4.2</b> Obtain genotype for significant SNPs (using R)</a></li>
<li class="chapter" data-level="5.4.3" data-path="gwas-with-continuous-trait.html"><a href="gwas-with-continuous-trait.html#create-box-plot-using-r"><i class="fa fa-check"></i><b>5.4.3</b> Create Box plot (using R)</a></li>
</ul></li>
<li class="chapter" data-level="5.5" data-path="gwas-with-continuous-trait.html"><a href="gwas-with-continuous-trait.html#proportion-of-variation-explained-by-rs8138080"><i class="fa fa-check"></i><b>5.5</b> Proportion of Variation Explained by rs8138080</a>
<ul>
<li class="chapter" data-level="5.5.1" data-path="gwas-with-continuous-trait.html"><a href="gwas-with-continuous-trait.html#recode-genotype-of-significant-snps-using-r"><i class="fa fa-check"></i><b>5.5.1</b> Recode genotype of significant SNPs (using R)</a></li>
<li class="chapter" data-level="5.5.2" data-path="gwas-with-continuous-trait.html"><a href="gwas-with-continuous-trait.html#calculate-marginal-r2-using-r"><i class="fa fa-check"></i><b>5.5.2</b> Calculate marginal <span class="math inline">\(R^2\)</span> (using R)</a></li>
</ul></li>
<li class="chapter" data-level="5.6" data-path="gwas-with-continuous-trait.html"><a href="gwas-with-continuous-trait.html#practice-time"><i class="fa fa-check"></i><b>5.6</b> PRACTICE TIME</a>
<ul>
<li class="chapter" data-level="5.6.1" data-path="gwas-with-continuous-trait.html"><a href="gwas-with-continuous-trait.html#practice-1"><i class="fa fa-check"></i><b>5.6.1</b> Practice 1</a></li>
<li class="chapter" data-level="5.6.2" data-path="gwas-with-continuous-trait.html"><a href="gwas-with-continuous-trait.html#practice-2"><i class="fa fa-check"></i><b>5.6.2</b> Practice 2</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="gwas-with-binary-trait.html"><a href="gwas-with-binary-trait.html"><i class="fa fa-check"></i><b>6</b> GWAS with Binary Trait</a>
<ul>
<li class="chapter" data-level="6.1" data-path="gwas-with-binary-trait.html"><a href="gwas-with-binary-trait.html#data-preparation"><i class="fa fa-check"></i><b>6.1</b> Data Preparation</a></li>
<li class="chapter" data-level="6.2" data-path="gwas-with-binary-trait.html"><a href="gwas-with-binary-trait.html#genome-wide-casecontrol-association-test-in-plink"><i class="fa fa-check"></i><b>6.2</b> Genome-Wide Case/Control Association Test (in PLINK)</a></li>
<li class="chapter" data-level="6.3" data-path="gwas-with-binary-trait.html"><a href="gwas-with-binary-trait.html#manhattan-plot-and-q-q-plot-1"><i class="fa fa-check"></i><b>6.3</b> Manhattan Plot and Q-Q Plot</a></li>
<li class="chapter" data-level="6.4" data-path="gwas-with-binary-trait.html"><a href="gwas-with-binary-trait.html#generate-manhattan-plot-1"><i class="fa fa-check"></i><b>6.4</b> Generate Manhattan plot</a></li>
<li class="chapter" data-level="6.5" data-path="gwas-with-binary-trait.html"><a href="gwas-with-binary-trait.html#generate-q-q-plot-1"><i class="fa fa-check"></i><b>6.5</b> Generate Q-Q plot</a></li>
<li class="chapter" data-level="6.6" data-path="gwas-with-binary-trait.html"><a href="gwas-with-binary-trait.html#multiple-test-correction-1"><i class="fa fa-check"></i><b>6.6</b> Multiple-Test Correction</a></li>
<li class="chapter" data-level="6.7" data-path="gwas-with-binary-trait.html"><a href="gwas-with-binary-trait.html#top-10-snps"><i class="fa fa-check"></i><b>6.7</b> Top 10 SNPs</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="gxg-and-gxe-interactions-power-and-sample-size-calculation.html"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html"><i class="fa fa-check"></i><b>7</b> GxG and GxE Interactions, Power and Sample Size Calculation</a>
<ul>
<li class="chapter" data-level="7.1" data-path="gxg-and-gxe-interactions-power-and-sample-size-calculation.html"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#prepare-files"><i class="fa fa-check"></i><b>7.1</b> Prepare Files</a></li>
<li class="chapter" data-level="7.2" data-path="gxg-and-gxe-interactions-power-and-sample-size-calculation.html"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#test-for-epistasis-gxg"><i class="fa fa-check"></i><b>7.2</b> Test for Epistasis (GxG)</a></li>
<li class="chapter" data-level="7.3" data-path="gxg-and-gxe-interactions-power-and-sample-size-calculation.html"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#gene-environment-interaction-gxe"><i class="fa fa-check"></i><b>7.3</b> Gene-Environment Interaction (GxE)</a></li>
<li class="chapter" data-level="7.4" data-path="gxg-and-gxe-interactions-power-and-sample-size-calculation.html"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#power-and-sample-size-calculation-for-gwas"><i class="fa fa-check"></i><b>7.4</b> Power and sample size calculation for GWAS</a>
<ul>
<li class="chapter" data-level="7.4.1" data-path="gxg-and-gxe-interactions-power-and-sample-size-calculation.html"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#power-calculation-with-binary-outcome"><i class="fa fa-check"></i><b>7.4.1</b> Power calculation with binary outcome:</a></li>
<li class="chapter" data-level="7.4.2" data-path="gxg-and-gxe-interactions-power-and-sample-size-calculation.html"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#sample-size-calculation"><i class="fa fa-check"></i><b>7.4.2</b> Sample size calculation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8" data-path="ldsc-and-gene-set-analysis.html"><a href="ldsc-and-gene-set-analysis.html"><i class="fa fa-check"></i><b>8</b> LDSC and Gene-Set Analysis</a>
<ul>
<li class="chapter" data-level="8.1" data-path="ldsc-and-gene-set-analysis.html"><a href="ldsc-and-gene-set-analysis.html#ldsc-linkage-disequilibrium-score-regression"><i class="fa fa-check"></i><b>8.1</b> LDSC (Linkage Disequilibrium Score Regression)</a></li>
<li class="chapter" data-level="8.2" data-path="ldsc-and-gene-set-analysis.html"><a href="ldsc-and-gene-set-analysis.html#magma-for-gene-level-analysis"><i class="fa fa-check"></i><b>8.2</b> MAGMA (for Gene-Level Analysis)</a>
<ul>
<li class="chapter" data-level="8.2.1" data-path="ldsc-and-gene-set-analysis.html"><a href="ldsc-and-gene-set-analysis.html#prepare-annotation-file"><i class="fa fa-check"></i><b>8.2.1</b> Prepare Annotation File</a></li>
<li class="chapter" data-level="8.2.2" data-path="ldsc-and-gene-set-analysis.html"><a href="ldsc-and-gene-set-analysis.html#part-1-gene-analysis"><i class="fa fa-check"></i><b>8.2.2</b> Part 1: Gene Analysis</a></li>
<li class="chapter" data-level="8.2.3" data-path="ldsc-and-gene-set-analysis.html"><a href="ldsc-and-gene-set-analysis.html#part-2-gene-set-analysis"><i class="fa fa-check"></i><b>8.2.3</b> Part 2: Gene-Set Analysis</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="9" data-path="perform-eqtl-analysis.html"><a href="perform-eqtl-analysis.html"><i class="fa fa-check"></i><b>9</b> Perform eQTL Analysis</a>
<ul>
<li class="chapter" data-level="9.1" data-path="perform-eqtl-analysis.html"><a href="perform-eqtl-analysis.html#filtering-snps-by-maf"><i class="fa fa-check"></i><b>9.1</b> Filtering SNPs by MAF</a></li>
<li class="chapter" data-level="9.2" data-path="perform-eqtl-analysis.html"><a href="perform-eqtl-analysis.html#gene-expression-profiling"><i class="fa fa-check"></i><b>9.2</b> Gene expression profiling</a></li>
<li class="chapter" data-level="9.3" data-path="perform-eqtl-analysis.html"><a href="perform-eqtl-analysis.html#covariates"><i class="fa fa-check"></i><b>9.3</b> Covariates</a></li>
<li class="chapter" data-level="9.4" data-path="perform-eqtl-analysis.html"><a href="perform-eqtl-analysis.html#using-principle-components-as-covariates"><i class="fa fa-check"></i><b>9.4</b> Using principle components as covariates</a></li>
<li class="chapter" data-level="9.5" data-path="perform-eqtl-analysis.html"><a href="perform-eqtl-analysis.html#eqtl-analysis"><i class="fa fa-check"></i><b>9.5</b> eQTL analysis</a>
<ul>
<li class="chapter" data-level="9.5.1" data-path="perform-eqtl-analysis.html"><a href="perform-eqtl-analysis.html#cis-eqtl-and-trans-eqtl-analysis"><i class="fa fa-check"></i><b>9.5.1</b> cis-eQTL and trans-eQTL analysis</a></li>
</ul></li>
<li class="chapter" data-level="9.6" data-path="perform-eqtl-analysis.html"><a href="perform-eqtl-analysis.html#multiple-testing-correction"><i class="fa fa-check"></i><b>9.6</b> Multiple testing correction</a></li>
<li class="chapter" data-level="9.7" data-path="perform-eqtl-analysis.html"><a href="perform-eqtl-analysis.html#gene-regional-manhanttan-plots"><i class="fa fa-check"></i><b>9.7</b> Gene regional Manhanttan plots</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">CSHL course - GWAS Hands-on Analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="gxg-and-gxe-interactions-power-and-sample-size-calculation" class="section level1 hasAnchor" number="7">
<h1><span class="header-section-number">Chapter 7</span> GxG and GxE Interactions, Power and Sample Size Calculation<a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#gxg-and-gxe-interactions-power-and-sample-size-calculation" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>In this section we select the top 5 SNPs for continuous Z-endoxifen ratio, and demonstrate how to use PLINK to perform tests of GxG [G(Gene) by G(Gene)] and GxE [G(Gene) by E(Environment)] interactions.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb1-1"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb1-1" tabindex="-1"></a><span class="co"># change the path below to where you saved &quot;cshl_gwas&quot; folder</span></span>
<span id="cb1-2"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb1-2" tabindex="-1"></a><span class="co">## run in R Terminal</span></span>
<span id="cb1-3"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb1-3" tabindex="-1"></a><span class="va">mydir</span><span class="op">=</span><span class="st">&quot;/path/to/cshl_gwas/Data/case_interaction&quot;</span></span>
<span id="cb1-4"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb1-4" tabindex="-1"></a><span class="bu">cd</span> <span class="va">$mydir</span></span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb2-1" tabindex="-1"></a><span class="co"># R codes- run in R Console</span></span>
<span id="cb2-2"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb2-2" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">&quot;/path/to/cshl_gwas/Data/case_interaction&quot;</span>)</span>
<span id="cb2-3"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb2-3" tabindex="-1"></a>mydir<span class="ot">&lt;-</span><span class="fu">getwd</span>()</span></code></pre></div>
<div id="prepare-files" class="section level2 hasAnchor" number="7.1">
<h2><span class="header-section-number">7.1</span> Prepare Files<a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#prepare-files" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We can obtain the top 5 SNPs and p-values for continuous Z-endoxifen ratio (using R)</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb3-1" tabindex="-1"></a><span class="co"># R codes- run in R Console</span></span>
<span id="cb3-2"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb3-2" tabindex="-1"></a>qassoc<span class="ot">=</span><span class="fu">fread</span>(<span class="st">&quot;../case_data/significant_snps_with_rs_for_z-endoxifen-ratio.txt&quot;</span>,<span class="at">header=</span>T) <span class="co">#load association results for continuous r_met</span></span>
<span id="cb3-3"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb3-3" tabindex="-1"></a>ordered_qassoc<span class="ot">=</span>qassoc[<span class="fu">order</span>(qassoc<span class="sc">$</span>FDR_BH),] <span class="co">#order association results based on p-value</span></span>
<span id="cb3-4"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb3-4" tabindex="-1"></a><span class="fu">head</span>(ordered_qassoc,<span class="dv">5</span>) <span class="co">#association results for the top 5 SNPs with the 5 smallest p-values</span></span></code></pre></div>
<pre><code>##      CHR       SNP     UNADJ    FDR_BH       BP
##    &lt;int&gt;    &lt;char&gt;     &lt;num&gt;     &lt;num&gt;    &lt;int&gt;
## 1:    22 rs8138080 3.318e-20 1.335e-17 42396371
## 2:    22 rs1062753 5.643e-19 9.347e-17 42392811
## 3:    22 rs4467371 1.365e-18 1.570e-16 42541328
## 4:    22    rs7245 4.102e-18 2.540e-16 42481849
## 5:    22 rs2284087 4.102e-18 2.540e-16 42485671</code></pre>
<p>We want to test Epistasis between the top 5 SNPs for continuous r_met using PLINK. As shown in the table above, the top 5 SNPs for continuous Z-endoxifen ratio are:</p>
<pre><code>chr22 rs8138080
chr22 rs4467371
chr22 rs2011944
chr22 rs1062753
chr22 rs4453786</code></pre>
<p>The following code generates <strong>map</strong> idx &amp; <strong>map</strong> file of the top 5 SNPs (using R)</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb6-1" tabindex="-1"></a><span class="co"># R codes- run in R Console</span></span>
<span id="cb6-2"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb6-2" tabindex="-1"></a>map<span class="ot">=</span><span class="fu">read.table</span>(<span class="st">&quot;genotype_1k.map&quot;</span>) <span class="co">#read map file</span></span>
<span id="cb6-3"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb6-3" tabindex="-1"></a>map<span class="ot">=</span><span class="fu">as.matrix</span>(map) <span class="co">#create a matrix from the MAP file</span></span>
<span id="cb6-4"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb6-4" tabindex="-1"></a>top5SNPs<span class="ot">=</span><span class="fu">c</span>(<span class="st">&quot;rs8138080&quot;</span>, <span class="st">&quot;rs4467371&quot;</span>,<span class="st">&quot;rs2011944&quot;</span>,<span class="st">&quot;rs1062753&quot;</span>,<span class="st">&quot;rs4453786&quot;</span>) <span class="co">#SNP ID of the top 5 SNPs</span></span>
<span id="cb6-5"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb6-5" tabindex="-1"></a>top5_idx<span class="ot">=</span><span class="fu">matrix</span>(<span class="dv">0</span>,<span class="dv">5</span>,<span class="dv">1</span>) <span class="co">#create an empty matrix</span></span>
<span id="cb6-6"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb6-6" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>){</span>
<span id="cb6-7"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb6-7" tabindex="-1"></a>  top5_idx[i]<span class="ot">=</span><span class="fu">which</span>(map[,<span class="dv">2</span>]<span class="sc">==</span>top5SNPs[i]) <span class="co">#index of the top 5 SNPs in the MAP file</span></span>
<span id="cb6-8"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb6-8" tabindex="-1"></a>}</span>
<span id="cb6-9"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb6-9" tabindex="-1"></a><span class="fu">as.vector</span>(top5_idx)</span></code></pre></div>
<pre><code>## [1] 948 943 927 922 928</code></pre>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb8-1" tabindex="-1"></a>top5SNPs.map<span class="ot">=</span>map[top5_idx,] <span class="co">#MAP file of the top 5 SNPs</span></span>
<span id="cb8-2"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb8-2" tabindex="-1"></a><span class="fu">write.table</span>(top5SNPs.map,<span class="st">&quot;top5SNPs.map&quot;</span>, <span class="at">row.names=</span>F, <span class="at">col.names=</span>F, <span class="at">quote=</span>F, <span class="at">sep=</span><span class="st">&quot; &quot;</span>) <span class="co">#save MAP file of the top 5 SNPs</span></span></code></pre></div>
<p>The following code generates <strong>ped</strong> file of the top 5 SNPs (in Linux) using the index of these 5 SNPs above. Since the first 6 columns are sample information and each SNP has two alleles in .ped file, the index of each pair of allele for each SNP= [6+2(top5_idx)-1, 6+2(top5_idx)].</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb9-1"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb9-1" tabindex="-1"></a><span class="co"># bash codes- run in terminal</span></span>
<span id="cb9-2"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb9-2" tabindex="-1"></a><span class="fu">awk</span> <span class="st">&#39;{print $1,$2,$3,$4,$5,$6, $1901,$1902, $1891,$1892, $1859, $1860, $1849,$1850, $1861, $1862}&#39;</span> <span class="dt">\</span></span>
<span id="cb9-3"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb9-3" tabindex="-1"></a>genotype_1k.ped <span class="op">&gt;</span> top5SNPs.ped</span></code></pre></div>
<p>Phenotype data:</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb10-1"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb10-1" tabindex="-1"></a><span class="co"># bash codes- run in terminal</span></span>
<span id="cb10-2"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb10-2" tabindex="-1"></a><span class="fu">paste</span> famID.txt samID.txt z_endoxifen_ratio.txt <span class="op">&gt;</span> ZR.txt</span></code></pre></div>
</div>
<div id="test-for-epistasis-gxg" class="section level2 hasAnchor" number="7.2">
<h2><span class="header-section-number">7.2</span> Test for Epistasis (GxG)<a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#test-for-epistasis-gxg" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>For disease-trait population-based samples, it is possible to test for epistasis. All pairwise combinations of SNPs can be tested using PLINK (computationally feasible). For more detailed information for this test, please refer: <a href="http://zzz.bwh.harvard.edu/plink/epi.shtml" class="uri">http://zzz.bwh.harvard.edu/plink/epi.shtml</a></p>
<p>PLINK produces a model based on allele dosage for each SNP, A and B, and fits the model in the form of</p>
<pre><code> Y ~ b0 + b1.A + b2.B + b3.AB + e </code></pre>
<div class="sourceCode" id="cb12"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb12-1"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb12-1" tabindex="-1"></a><span class="co">## # both Mac and Windows can run in Terminal</span></span>
<span id="cb12-2"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb12-2" tabindex="-1"></a><span class="co"># windows users</span></span>
<span id="cb12-3"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb12-3" tabindex="-1"></a><span class="ex">../../Tools/PLINK/plink_win64_20231211/plink.exe</span> <span class="dt">\</span></span>
<span id="cb12-4"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb12-4" tabindex="-1"></a>--noweb <span class="at">--file</span> top5SNPs <span class="at">--epistasis</span> <span class="at">--epi1</span> 1 <span class="at">--pheno</span> ZR.txt <span class="at">--out</span> younameit</span></code></pre></div>
<div class="sourceCode" id="cb13"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb13-1"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb13-1" tabindex="-1"></a><span class="co"># macOS users</span></span>
<span id="cb13-2"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb13-2" tabindex="-1"></a><span class="ex">../../Tools/PLINK/plink_mac_20231211/plink</span> <span class="dt">\</span></span>
<span id="cb13-3"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb13-3" tabindex="-1"></a>--noweb <span class="at">--file</span> top5SNPs <span class="at">--epistasis</span> <span class="at">--epi1</span> 1 <span class="at">--pheno</span> ZR.txt <span class="at">--out</span> younameit</span></code></pre></div>
<pre><code>## PLINK v1.90b7.2 64-bit (11 Dec 2023)           www.cog-genomics.org/plink/1.9/
## (C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3
## Logging to younameit.log.
## Options in effect:
##   --epi1 1
##   --epistasis
##   --file top5SNPs
##   --noweb
##   --out younameit
##   --pheno ZR.txt
## 
## Note: --noweb has no effect since no web check is implemented yet.
## 16384 MB RAM detected; reserving 8192 MB for main workspace.
## Scanning .ped file... 0%1%1%2%2%3%3%4%4%5%5%6%6%7%7%8%8%9%9%10%11%11%12%12%13%13%14%14%15%15%16%16%17%17%18%18%19%19%20%20%21%21%22%22%23%23%24%24%25%25%26%26%27%27%28%28%29%29%30%30%31%31%32%32%33%33%34%34%35%36%36%37%37%38%38%39%39%40%40%41%41%42%42%43%43%44%44%45%45%46%46%47%47%48%48%49%49%50%50%51%51%52%53%53%54%54%55%55%56%56%57%57%58%58%59%59%60%61%61%62%62%63%63%64%64%65%65%66%66%67%67%68%69%69%70%70%71%71%72%72%73%73%74%74%75%75%76%77%77%78%78%79%79%80%80%81%81%82%82%83%83%84%85%85%86%86%87%87%88%88%89%89%90%90%91%91%92%93%93%94%94%95%95%96%96%97%97%98%98%99%100%.ped scan complete (for binary autoconversion).
## Performing single-pass .bed write (5 variants, 192 people).
## 0%1%2%3%4%5%6%7%8%9%10%11%12%13%14%15%16%17%18%19%20%21%22%23%24%25%26%27%28%29%30%31%32%33%34%35%36%37%38%39%40%41%42%43%44%45%46%47%48%49%50%51%52%53%54%55%56%57%58%59%60%61%62%63%64%65%66%67%68%69%70%71%72%73%74%75%76%77%78%79%80%81%82%83%84%85%86%87%88%89%90%91%92%93%94%--file: younameit-temporary.bed + younameit-temporary.bim +
## younameit-temporary.fam written.
## 5 variants loaded from .bim file.
## 192 people (0 males, 192 females) loaded from .fam.
## 192 phenotype values present after --pheno.
## Using up to 8 threads (change this with --threads).
## Before main variant filters, 192 founders and 0 nonfounders present.
## Calculating allele frequencies... 0%1%2%3%4%5%6%7%8%9%10%11%12%13%14%15%16%17%18%19%20%21%22%23%24%25%26%27%28%29%30%31%32%33%34%35%36%37%38%39%40%41%42%43%44%45%46%47%48%49%50%51%52%53%54%55%56%57%58%59%60%61%62%63%64%65%66%67%68%69%70%71%72%73%74%75%76%77%78%79%80%81%82%83%84%85%86%87%88%89%90%91%92%93%94%95%96%97%98%99% done.
## Total genotyping rate is 0.998958.
## 5 variants and 192 people pass filters and QC.
## Phenotype data is quantitative.
## QT --epistasis to younameit.epi.qt ... 0% [processing]writing]             done.
## 9 valid tests performed, summary written to younameit.epi.qt.summary .</code></pre>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb15-1" tabindex="-1"></a><span class="co"># R codes- run in R Console</span></span>
<span id="cb15-2"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb15-2" tabindex="-1"></a>epi_res <span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="st">&#39;younameit.epi.qt&#39;</span>)</span>
<span id="cb15-3"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb15-3" tabindex="-1"></a>epi_res[<span class="sc">!</span><span class="fu">is.na</span>(epi_res<span class="sc">$</span>P),]</span></code></pre></div>
<pre><code>##     CHR1      SNP1  CHR2      SNP2    BETA_INT        STAT      P
##    &lt;int&gt;    &lt;char&gt; &lt;int&gt;    &lt;char&gt;       &lt;num&gt;       &lt;num&gt;  &lt;num&gt;
## 1:    22 rs1062753    22 rs8138080  0.00395402 0.000909368 0.9759
## 2:    22 rs1062753    22 rs4467371  0.05923540 0.186978000 0.6654
## 3:    22 rs1062753    22 rs4453786  0.05197290 0.140664000 0.7076
## 4:    22 rs1062753    22 rs2011944 -0.00625238 0.002135950 0.9631
## 5:    22 rs8138080    22 rs4467371  0.06445030 0.220641000 0.6386
## 6:    22 rs8138080    22 rs4453786  0.05797350 0.174841000 0.6758
## 7:    22 rs8138080    22 rs2011944  0.00117048 0.000074603 0.9931
## 8:    22 rs4467371    22 rs2011944 -0.01796740 0.015782700 0.9000
## 9:    22 rs4453786    22 rs2011944 -0.03609540 0.062551200 0.8025</code></pre>
<p>The output is in the form:</p>
<pre><code> CHR1    Chromosome of the first SNP

 SNP1    Identifier for the first SNP
     
 CHR2    Chromosome of the second SNP
     
 SNP2    Identifier for the second SNP
     
 OR_INT  Odds ratio for the interaction
     
 STAT    Chi-square statistic, df=1
     
 P       Asymptotic p-value</code></pre>
<div class="sourceCode" id="cb18"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb18-1"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb18-1" tabindex="-1"></a><span class="co">## # both Mac and Windows can run in Terminal</span></span>
<span id="cb18-2"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb18-2" tabindex="-1"></a><span class="co"># windows users</span></span>
<span id="cb18-3"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb18-3" tabindex="-1"></a><span class="ex">../../Tools/PLINK/plink_win64_20231211/plink.exe</span> <span class="dt">\</span></span>
<span id="cb18-4"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb18-4" tabindex="-1"></a>--noweb <span class="at">--file</span> top5SNPs <span class="at">--epistasis</span> <span class="at">--epi1</span> 0.05 <span class="at">--pheno</span> ZR.txt <span class="at">--out</span> younameit</span></code></pre></div>
<div class="sourceCode" id="cb19"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb19-1"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb19-1" tabindex="-1"></a><span class="co"># macOS users</span></span>
<span id="cb19-2"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb19-2" tabindex="-1"></a><span class="ex">../../Tools/PLINK/plink_mac_20231211/plink</span> <span class="dt">\</span></span>
<span id="cb19-3"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb19-3" tabindex="-1"></a>--noweb <span class="at">--file</span> top5SNPs <span class="at">--epistasis</span> <span class="at">--epi1</span> 0.05 <span class="at">--pheno</span> ZR.txt <span class="at">--out</span> younameit</span></code></pre></div>
<pre><code>## PLINK v1.90b7.2 64-bit (11 Dec 2023)           www.cog-genomics.org/plink/1.9/
## (C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3
## Logging to younameit.log.
## Options in effect:
##   --epi1 0.05
##   --epistasis
##   --file top5SNPs
##   --noweb
##   --out younameit
##   --pheno ZR.txt
## 
## Note: --noweb has no effect since no web check is implemented yet.
## 16384 MB RAM detected; reserving 8192 MB for main workspace.
## Scanning .ped file... 0%1%1%2%2%3%3%4%4%5%5%6%6%7%7%8%8%9%9%10%11%11%12%12%13%13%14%14%15%15%16%16%17%17%18%18%19%19%20%20%21%21%22%22%23%23%24%24%25%25%26%26%27%27%28%28%29%29%30%30%31%31%32%32%33%33%34%34%35%36%36%37%37%38%38%39%39%40%40%41%41%42%42%43%43%44%44%45%45%46%46%47%47%48%48%49%49%50%50%51%51%52%53%53%54%54%55%55%56%56%57%57%58%58%59%59%60%61%61%62%62%63%63%64%64%65%65%66%66%67%67%68%69%69%70%70%71%71%72%72%73%73%74%74%75%75%76%77%77%78%78%79%79%80%80%81%81%82%82%83%83%84%85%85%86%86%87%87%88%88%89%89%90%90%91%91%92%93%93%94%94%95%95%96%96%97%97%98%98%99%100%.ped scan complete (for binary autoconversion).
## Performing single-pass .bed write (5 variants, 192 people).
## 0%1%2%3%4%5%6%7%8%9%10%11%12%13%14%15%16%17%18%19%20%21%22%23%24%25%26%27%28%29%30%31%32%33%34%35%36%37%38%39%40%41%42%43%44%45%46%47%48%49%50%51%52%53%54%55%56%57%58%59%60%61%62%63%64%65%66%67%68%69%70%71%72%73%74%75%76%77%78%79%80%81%82%83%84%85%86%87%88%89%90%91%92%93%94%--file: younameit-temporary.bed + younameit-temporary.bim +
## younameit-temporary.fam written.
## 5 variants loaded from .bim file.
## 192 people (0 males, 192 females) loaded from .fam.
## 192 phenotype values present after --pheno.
## Using up to 8 threads (change this with --threads).
## Before main variant filters, 192 founders and 0 nonfounders present.
## Calculating allele frequencies... 0%1%2%3%4%5%6%7%8%9%10%11%12%13%14%15%16%17%18%19%20%21%22%23%24%25%26%27%28%29%30%31%32%33%34%35%36%37%38%39%40%41%42%43%44%45%46%47%48%49%50%51%52%53%54%55%56%57%58%59%60%61%62%63%64%65%66%67%68%69%70%71%72%73%74%75%76%77%78%79%80%81%82%83%84%85%86%87%88%89%90%91%92%93%94%95%96%97%98%99% done.
## Total genotyping rate is 0.998958.
## 5 variants and 192 people pass filters and QC.
## Phenotype data is quantitative.
## QT --epistasis to younameit.epi.qt ... 0% [processing]writing]             done.
## 9 valid tests performed, summary written to younameit.epi.qt.summary .</code></pre>
<p>It only records results that are significant, i.e., p&lt;=0.05, which prevents generating too many outputs.</p>
<p>The results are shown as below. There is nothing in the table so no Epistasis.</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb21-1" tabindex="-1"></a><span class="co"># R codes- run in R Console</span></span>
<span id="cb21-2"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb21-2" tabindex="-1"></a>epi_res <span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="st">&#39;younameit.epi.qt&#39;</span>)</span>
<span id="cb21-3"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb21-3" tabindex="-1"></a>epi_res</span></code></pre></div>
<pre><code>## Empty data.table (0 rows and 7 cols): CHR1,SNP1,CHR2,SNP2,BETA_INT,STAT...</code></pre>
</div>
<div id="gene-environment-interaction-gxe" class="section level2 hasAnchor" number="7.3">
<h2><span class="header-section-number">7.3</span> Gene-Environment Interaction (GxE)<a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#gene-environment-interaction-gxe" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>PLINK provides the ability to test for a difference in association with a quantitative trait between two environments (or, more generally, two groups). This test is simply based on comparing the difference between two regression coefficients.</p>
<p>We need to specify a covariate file for this test:</p>
<p><strong>cover_age.txt</strong></p>
<pre><code>Col 1 is family ID

Col 2 is sample ID

Col 3 is age (need to be in 2 groups: 1: old, 2: young)</code></pre>
<p>Our current covariate file <code>age.txt</code> is a continuous variable. So we need to manually separate it in 2 groups: old(recode as 1) and young(recode as 2). As median age is 55.5, we will assign the sample to “old” if its age is older than 55.5, otherwise to “young”.</p>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb24-1" tabindex="-1"></a><span class="co"># R codes- run in R Console</span></span>
<span id="cb24-2"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb24-2" tabindex="-1"></a>age<span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="st">&quot;age.txt&quot;</span>, <span class="at">header=</span>F) <span class="co"># read in age</span></span>
<span id="cb24-3"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb24-3" tabindex="-1"></a><span class="fu">names</span>(age)<span class="ot">&lt;-</span><span class="st">&quot;age&quot;</span></span>
<span id="cb24-4"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb24-4" tabindex="-1"></a>age_recode <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="dv">1</span>,<span class="dv">192</span>,<span class="dv">1</span>) <span class="co"># create a new data with all 1</span></span>
<span id="cb24-5"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb24-5" tabindex="-1"></a>idx <span class="ot">&lt;-</span> <span class="fu">which</span>(age<span class="sc">$</span>age<span class="sc">&lt;</span><span class="fl">55.5</span>) <span class="co"># which sample has age &lt; 55.5</span></span>
<span id="cb24-6"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb24-6" tabindex="-1"></a>age_recode[idx,]<span class="ot">&lt;-</span><span class="dv">2</span> <span class="co"># assign those with age &lt;55.5 as 2</span></span>
<span id="cb24-7"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb24-7" tabindex="-1"></a><span class="fu">fwrite</span>(age_recode,<span class="st">&quot;age_binary.txt&quot;</span>,<span class="at">quote=</span>F,<span class="at">col.name=</span>F, <span class="at">row.name=</span>F) <span class="co"># save data</span></span></code></pre></div>
<div class="sourceCode" id="cb25"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb25-1"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb25-1" tabindex="-1"></a><span class="co"># bash codes- run in terminal</span></span>
<span id="cb25-2"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb25-2" tabindex="-1"></a><span class="fu">paste</span> famID.txt samID.txt age_binary.txt <span class="op">&gt;</span> covar_age.txt</span></code></pre></div>
<div class="sourceCode" id="cb26"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb26-1"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb26-1" tabindex="-1"></a><span class="co">## # both Mac and Windows can run in Terminal</span></span>
<span id="cb26-2"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb26-2" tabindex="-1"></a><span class="co"># windows users</span></span>
<span id="cb26-3"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb26-3" tabindex="-1"></a><span class="ex">../../Tools/PLINK/plink_win64_20231211/plink.exe</span> <span class="dt">\</span></span>
<span id="cb26-4"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb26-4" tabindex="-1"></a>--noweb <span class="at">--file</span> top5SNPs <span class="at">--gxe</span> <span class="at">--covar</span> covar_age.txt <span class="dt">\</span></span>
<span id="cb26-5"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb26-5" tabindex="-1"></a>--pheno ZR.txt <span class="at">--out</span> younameit</span>
<span id="cb26-6"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb26-6" tabindex="-1"></a></span>
<span id="cb26-7"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb26-7" tabindex="-1"></a><span class="co"># macOS users</span></span>
<span id="cb26-8"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb26-8" tabindex="-1"></a><span class="ex">../../Tools/PLINK/plink_mac_20231211/plink</span> <span class="dt">\</span></span>
<span id="cb26-9"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb26-9" tabindex="-1"></a>--noweb <span class="at">--file</span> top5SNPs <span class="at">--gxe</span> <span class="at">--covar</span> covar_age.txt <span class="dt">\</span></span>
<span id="cb26-10"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb26-10" tabindex="-1"></a>--pheno ZR.txt <span class="at">--out</span> younameit</span></code></pre></div>
<pre><code>## bash: line 3: ../../Tools/PLINK/plink_win64_20231211/plink.exe: Permission denied
## PLINK v1.90b7.2 64-bit (11 Dec 2023)           www.cog-genomics.org/plink/1.9/
## (C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3
## Logging to younameit.log.
## Options in effect:
##   --covar covar_age.txt
##   --file top5SNPs
##   --gxe
##   --noweb
##   --out younameit
##   --pheno ZR.txt
## 
## Note: --noweb has no effect since no web check is implemented yet.
## 16384 MB RAM detected; reserving 8192 MB for main workspace.
## Scanning .ped file... 0%1%1%2%2%3%3%4%4%5%5%6%6%7%7%8%8%9%9%10%11%11%12%12%13%13%14%14%15%15%16%16%17%17%18%18%19%19%20%20%21%21%22%22%23%23%24%24%25%25%26%26%27%27%28%28%29%29%30%30%31%31%32%32%33%33%34%34%35%36%36%37%37%38%38%39%39%40%40%41%41%42%42%43%43%44%44%45%45%46%46%47%47%48%48%49%49%50%50%51%51%52%53%53%54%54%55%55%56%56%57%57%58%58%59%59%60%61%61%62%62%63%63%64%64%65%65%66%66%67%67%68%69%69%70%70%71%71%72%72%73%73%74%74%75%75%76%77%77%78%78%79%79%80%80%81%81%82%82%83%83%84%85%85%86%86%87%87%88%88%89%89%90%90%91%91%92%93%93%94%94%95%95%96%96%97%97%98%98%99%100%.ped scan complete (for binary autoconversion).
## Performing single-pass .bed write (5 variants, 192 people).
## 0%1%2%3%4%5%6%7%8%9%10%11%12%13%14%15%16%17%18%19%20%21%22%23%24%25%26%27%28%29%30%31%32%33%34%35%36%37%38%39%40%41%42%43%44%45%46%47%48%49%50%51%52%53%54%55%56%57%58%59%60%61%62%63%64%65%66%67%68%69%70%71%72%73%74%75%76%77%78%79%80%81%82%83%84%85%86%87%88%89%90%91%92%93%94%--file: younameit-temporary.bed + younameit-temporary.bim +
## younameit-temporary.fam written.
## 5 variants loaded from .bim file.
## 192 people (0 males, 192 females) loaded from .fam.
## 192 phenotype values present after --pheno.
## Using 1 thread (no multithreaded calculations invoked).
## --covar: 1 case/control covariate loaded for --gxe.
## Before main variant filters, 192 founders and 0 nonfounders present.
## Calculating allele frequencies... 0%1%2%3%4%5%6%7%8%9%10%11%12%13%14%15%16%17%18%19%20%21%22%23%24%25%26%27%28%29%30%31%32%33%34%35%36%37%38%39%40%41%42%43%44%45%46%47%48%49%50%51%52%53%54%55%56%57%58%59%60%61%62%63%64%65%66%67%68%69%70%71%72%73%74%75%76%77%78%79%80%81%82%83%84%85%86%87%88%89%90%91%92%93%94%95%96%97%98%99% done.
## Total genotyping rate is 0.998958.
## 5 variants and 192 people pass filters and QC.
## Phenotype data is quantitative.
## Writing --gxe report to younameit.qassoc.gxe ... 0%1%2%3%4%5%6%7%8%9%10%11%12%13%14%15%16%17%18%19%20%21%22%23%24%25%26%27%28%29%30%31%32%33%34%35%36%37%38%39%40%41%42%43%44%45%46%47%48%49%50%51%52%53%54%55%56%57%58%59%60%61%62%63%64%65%66%67%68%69%70%71%72%73%74%75%76%77%78%79%80%81%82%83%84%85%86%87%88%89%90%91%92%93%94%95%96%97%98%99%done.</code></pre>
<p>The output is in the form:</p>
<pre><code> CHR       Chromosome number

 SNP       SNP identifier

 NMISS1    Number of non-missing genotypes in the first group

 BETA1     Regression coefficient in the first group

 SE1       Standard error of coefficient in the first group

 NMISS2    Number of non-missing genotypes in the second group

 BETA2     Regression coefficient in the second group

 SE2       Standard error of coefficient in the second group

 Z_GXE     Z score, test for the interaction
     
 P_GXE     Asymptotic p-value for this test</code></pre>
<p>The results are shown as below:</p>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb29-1" tabindex="-1"></a><span class="co"># R codes- run in R Console</span></span>
<span id="cb29-2"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb29-2" tabindex="-1"></a>gxe_res <span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="st">&#39;younameit.qassoc.gxe&#39;</span>)</span>
<span id="cb29-3"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb29-3" tabindex="-1"></a>gxe_res</span></code></pre></div>
<pre><code>##      CHR       SNP NMISS1   BETA1    SE1 NMISS2   BETA2    SE2   Z_GXE  P_GXE
##    &lt;int&gt;    &lt;char&gt;  &lt;int&gt;   &lt;num&gt;  &lt;num&gt;  &lt;int&gt;   &lt;num&gt;  &lt;num&gt;   &lt;num&gt;  &lt;num&gt;
## 1:    22 rs1062753     96 -0.8618 0.1239     96 -0.9168 0.1229 0.31530 0.7525
## 2:    22 rs8138080     95 -0.9262 0.1211     96 -0.9368 0.1249 0.06062 0.9517
## 3:    22 rs4467371     96 -0.8660 0.1158     96 -0.9794 0.1346 0.63840 0.5232
## 4:    22 rs4453786     96 -0.8660 0.1158     96 -0.9166 0.1391 0.27920 0.7801
## 5:    22 rs2011944     96 -0.8660 0.1158     96 -0.9384 0.1331 0.41020 0.6817</code></pre>
<p>By looking at the P-value, there is still no GxE interaction.</p>
</div>
<div id="power-and-sample-size-calculation-for-gwas" class="section level2 hasAnchor" number="7.4">
<h2><span class="header-section-number">7.4</span> Power and sample size calculation for GWAS<a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#power-and-sample-size-calculation-for-gwas" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>In GWAS studies, it’s also crucial to perform power analysis and sample size calculation. This helps researchers to determine the appropriate number of participants to detect significant genetic associations with sufficient statistical power. A review paper of sample size calculation definitions and tools can be found at <a href="https://www.mdpi.com/2075-1729/13/1/235">Politi et. al, 2023</a>.</p>
<p>A number of factors must be considered for GWAS power analysis and sample size calculations.</p>
<ul>
<li><strong>Outcome</strong>: The phenotype of a study. It can be continuous or binary (case/control). For continuous pahenotype, variability is assessed with SD (standard deviation). While for binary phenotype, the paramater used is the proportion of individuals with disease.</li>
<li><strong>Effect size</strong>: The expected size of the genetic effect on the phenotype. This can be estimated based on previous studies or biological knowledge. The parameter used is odds ratio (OR) for binary phenotype or percentage of phenotypic variance for continuous phenotype.</li>
<li><strong>Significance level (alpha)</strong>: The threshold used to determine statistical significance. The most common level is 0.05 (5%).</li>
<li><strong>Power (1 - beta)</strong>: The probability of correctly rejecting the null hypothesis if the alternative hypothesis is true. The total power target is 0.8 (80%).</li>
<li><strong>Minor Allele frequency (MAF)</strong>: The frequency of a genetic variant in a population. This information can be obtained from benchmarking databases or pilot studies. A genetic variant can be classified as common (MAF &gt; 5%), uncommon (1–5% MAF) and rare (MAF &lt; 1%). Typically, GWAS studies are sufficiently powered only to test common genetic variants.</li>
<li><strong>Genetic Inheritance Model</strong>: The assumed genetic model to assess the clinical effect of an SNP. Three genetic models are usually adopted: a dominant, recessive and additive model.</li>
</ul>
<p>Below, we will use an R package <code>genpwr</code> to show you an example of power and sample size calculation. Package <code>genpwr</code> can calculate power, sample size or detectable effect size. It handle both continuous and binary outcomes, and is capable of considering GxE interactions when calculating.</p>
<p>Below, we will show you examples for calculating:</p>
<ul>
<li>power</li>
<li>sample size</li>
<li>detectable effect size (or odds ratio in the case of a binary outcome variable)
provided that two of the three above variables are entered into the appropriate genpwr function.</li>
</ul>
<p>First, we need to install this package from GitHub using “devtools”. The codes below show you how to install it on your laptop.</p>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb31-1" tabindex="-1"></a><span class="co"># R codes- run in R Console</span></span>
<span id="cb31-2"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb31-2" tabindex="-1"></a></span>
<span id="cb31-3"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb31-3" tabindex="-1"></a><span class="co"># installation</span></span>
<span id="cb31-4"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb31-4" tabindex="-1"></a><span class="co">#install.packages(&quot;devtools&quot;)</span></span>
<span id="cb31-5"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb31-5" tabindex="-1"></a><span class="co">#library(devtools)</span></span>
<span id="cb31-6"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb31-6" tabindex="-1"></a><span class="co">#install_github(&quot;camillemmoore/Power_Genetics&quot;, subdir=&quot;genpwr&quot;)</span></span>
<span id="cb31-7"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb31-7" tabindex="-1"></a></span>
<span id="cb31-8"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb31-8" tabindex="-1"></a><span class="co"># use package</span></span>
<span id="cb31-9"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb31-9" tabindex="-1"></a><span class="fu">library</span>(genpwr)</span></code></pre></div>
<p>We will determine the above parameters using our case study data.</p>
<div id="power-calculation-with-binary-outcome" class="section level3 hasAnchor" number="7.4.1">
<h3><span class="header-section-number">7.4.1</span> Power calculation with binary outcome:<a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#power-calculation-with-binary-outcome" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Based on Chapter 4, we have a case-control study with 192 individuals, including 95 cases and 97 controls (case rate of approximately 50%). The SNPs included in the GWAS analysis have MAF over a range of 0 to 0.5. (This can be calculated using plink <code>--freq</code>. Check Chapter 2.) But we will only focus on common alleles with MAF &gt;0.1. We want to calculate the statistical power to detect an odds ratio of 3, for all possible combinations of true and test models, assuming an alpha of 0.05.</p>
<p>We need to specify the calculation is for power with <code>calc="power"</code> and use <code>model="logistic"</code> for binary outcome. The function <code>genpwr.calc</code> calculates power for all alleles with MAF between 0.1 and 0.5 with an increment of 0.01 based on the study design. Then a plot is generated to visualize the power change vs MAF under different genetic models.</p>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb32-1" tabindex="-1"></a><span class="co"># R codes- run in R Console</span></span>
<span id="cb32-2"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb32-2" tabindex="-1"></a>pw <span class="ot">&lt;-</span> <span class="fu">genpwr.calc</span>(<span class="at">calc =</span> <span class="st">&quot;power&quot;</span>, <span class="at">model =</span> <span class="st">&quot;logistic&quot;</span>, </span>
<span id="cb32-3"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb32-3" tabindex="-1"></a>                  <span class="at">ge.interaction =</span> <span class="cn">NULL</span>, <span class="at">N=</span><span class="dv">192</span>, <span class="at">Case.Rate=</span><span class="fl">0.5</span>, <span class="at">k=</span><span class="cn">NULL</span>,</span>
<span id="cb32-4"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb32-4" tabindex="-1"></a>                  <span class="at">MAF=</span><span class="fu">seq</span>(<span class="fl">0.1</span>,<span class="fl">0.5</span>,<span class="fl">0.01</span>), <span class="at">OR=</span><span class="dv">3</span>, <span class="at">Alpha=</span><span class="fl">0.05</span>,</span>
<span id="cb32-5"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb32-5" tabindex="-1"></a>                  <span class="at">True.Model=</span><span class="fu">c</span>(<span class="st">&quot;Additive&quot;</span>,<span class="st">&quot;Dominant&quot;</span>, <span class="st">&quot;Recessive&quot;</span>), </span>
<span id="cb32-6"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb32-6" tabindex="-1"></a>                  <span class="at">Test.Model=</span><span class="fu">c</span>(<span class="st">&quot;Additive&quot;</span>,<span class="st">&quot;Dominant&quot;</span>, <span class="st">&quot;Recessive&quot;</span>))</span>
<span id="cb32-7"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb32-7" tabindex="-1"></a><span class="fu">power.plot</span>(pw)</span></code></pre></div>
<pre><code>## Warning: Use of `temp2$Power` is discouraged.
## ℹ Use `Power` instead.</code></pre>
<pre><code>## Warning: Use of `temp2$Test.Model` is discouraged.
## ℹ Use `Test.Model` instead.</code></pre>
<pre><code>## Warning: Use of `temp2$Power` is discouraged.
## ℹ Use `Power` instead.</code></pre>
<pre><code>## Warning: Use of `temp2$Test.Model` is discouraged.
## ℹ Use `Test.Model` instead.</code></pre>
<p><img src="cshl_gwas_files/figure-html/powerplot-1.png" width="672" /></p>
<div id="case-study-with-continuous-outcome" class="section level4 hasAnchor" number="7.4.1.1">
<h4><span class="header-section-number">7.4.1.1</span> Case study with continuous outcome:<a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#case-study-with-continuous-outcome" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>For continuous phenotype, we need to obtain the phenotype SD instead of case rate. The codes below read in the phenotype data from Chapter 3 and calculate the standard deviation for all metabolites.</p>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb37-1" tabindex="-1"></a><span class="co"># R codes- run in R Console</span></span>
<span id="cb37-2"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb37-2" tabindex="-1"></a><span class="co"># Read data</span></span>
<span id="cb37-3"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb37-3" tabindex="-1"></a>phenotype<span class="ot">=</span><span class="fu">read.table</span>(<span class="st">&quot;../case_data/phenotype.txt&quot;</span>, <span class="at">header=</span>T)</span>
<span id="cb37-4"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb37-4" tabindex="-1"></a><span class="fu">head</span>(phenotype)</span></code></pre></div>
<pre><code>##   sample_ID e.a.oh.tam e.endoxifen z.4.oh.tam_3.oh.tam z.a.oh.tam
## 1       3TX      0.390           0                2.10      0.000
## 2      23TX      0.546           0                1.77      0.041
## 3      41TX      0.531           0                1.46      0.000
## 4      54TX      0.496           0                2.19      0.063
## 5      76TX      0.339           0                1.67      0.079
## 6      94TX      0.303           0                1.10      0.064
##   z.endoxifen_3.oh.ndm.tam V4.oh.ndm.tam.gluc V4.oh.tam.o.gluc
## 1                     7.62              0.773            0.093
## 2                     4.31              0.888            0.132
## 3                     4.09              0.725            0.147
## 4                     2.95              0.329            0.080
## 5                     3.77              1.719            0.462
## 6                     1.21              0.104            0.027
##   V4prim.oh.ndm.tam V4prim.oh.tam ndm.tam tam.n.gluc tam.n.oxide tamoxifen
## 1             3.724          3.67  158.63       0.09        9.51    136.63
## 2             2.670          2.39  174.85       0.34        7.30    110.38
## 3             3.665          3.89  195.63       0.10        9.79    145.73
## 4             6.624          4.10  280.15       0.21       21.05    169.76
## 5             6.374          4.18  300.06       0.47       27.72    194.25
## 6             6.182          3.51  303.72       0.31       28.71    167.92
##   z.endoxifen_ratio
## 1       0.024134673
## 2       0.014308285
## 3       0.011294447
## 4       0.006079783
## 5       0.007010690
## 6       0.002365497</code></pre>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb39-1"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb39-1" tabindex="-1"></a><span class="co"># R codes- run in R Console</span></span>
<span id="cb39-2"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb39-2" tabindex="-1"></a><span class="co"># get sd</span></span>
<span id="cb39-3"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb39-3" tabindex="-1"></a><span class="fu">apply</span>(phenotype[,<span class="sc">-</span><span class="dv">1</span>],<span class="dv">2</span>,<span class="cf">function</span>(x){<span class="fu">sd</span>(x,<span class="at">na.rm=</span>T)})</span></code></pre></div>
<pre><code>##               e.a.oh.tam              e.endoxifen      z.4.oh.tam_3.oh.tam 
##             1.752448e-01             6.007286e-02             1.232891e+00 
##               z.a.oh.tam z.endoxifen_3.oh.ndm.tam       V4.oh.ndm.tam.gluc 
##             3.340708e-02             3.504672e+00             1.025989e+00 
##         V4.oh.tam.o.gluc        V4prim.oh.ndm.tam            V4prim.oh.tam 
##             1.575704e-01             2.156651e+00             1.371287e+00 
##                  ndm.tam               tam.n.gluc              tam.n.oxide 
##             1.040414e+02             2.775679e-01             7.363288e+00 
##                tamoxifen        z.endoxifen_ratio 
##             6.701619e+01             7.612795e-03</code></pre>
<p>We will use tamoxifen as an example of continuous phenotype. The SD of tamoxifen is approximately 67.</p>
<p>We still focus on common alleles with MAF &gt;0.1. and the goal is to calculate the statistical power to detect an effect size of 30 (~45% of the phenotype SD) under alpha of 0.05. Note that the effect size (ES) can’t exceed the total SD of outcome.</p>
<p>Similar as in binary study, we use <code>calc="power"</code> but specify continuous study model with <code>model="linear"</code>.</p>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb41-1"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb41-1" tabindex="-1"></a><span class="co"># R codes- run in R Console</span></span>
<span id="cb41-2"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb41-2" tabindex="-1"></a>pw <span class="ot">&lt;-</span> <span class="fu">genpwr.calc</span>(<span class="at">calc =</span> <span class="st">&quot;power&quot;</span>, <span class="at">model =</span> <span class="st">&quot;linear&quot;</span>,</span>
<span id="cb41-3"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb41-3" tabindex="-1"></a>   <span class="at">N=</span><span class="dv">192</span>, <span class="at">sd_y=</span><span class="dv">67</span>, <span class="at">k=</span><span class="cn">NULL</span>,</span>
<span id="cb41-4"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb41-4" tabindex="-1"></a>   <span class="at">MAF=</span><span class="fu">seq</span>(<span class="fl">0.1</span>, <span class="fl">0.5</span>, <span class="fl">0.01</span>), <span class="at">ES=</span><span class="dv">30</span>, <span class="at">Alpha=</span><span class="fl">0.05</span>,</span>
<span id="cb41-5"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb41-5" tabindex="-1"></a>   <span class="at">True.Model=</span><span class="fu">c</span>(<span class="st">&quot;Additive&quot;</span>,<span class="st">&quot;Dominant&quot;</span>, <span class="st">&quot;Recessive&quot;</span>), </span>
<span id="cb41-6"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb41-6" tabindex="-1"></a>   <span class="at">Test.Model=</span><span class="fu">c</span>(<span class="st">&quot;Additive&quot;</span>,<span class="st">&quot;Dominant&quot;</span>, <span class="st">&quot;Recessive&quot;</span>))</span>
<span id="cb41-7"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb41-7" tabindex="-1"></a></span>
<span id="cb41-8"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb41-8" tabindex="-1"></a><span class="fu">power.plot</span>(pw)</span></code></pre></div>
<pre><code>## Warning: Use of `temp2$Power` is discouraged.
## ℹ Use `Power` instead.</code></pre>
<pre><code>## Warning: Use of `temp2$Test.Model` is discouraged.
## ℹ Use `Test.Model` instead.</code></pre>
<pre><code>## Warning: Use of `temp2$Power` is discouraged.
## ℹ Use `Power` instead.</code></pre>
<pre><code>## Warning: Use of `temp2$Test.Model` is discouraged.
## ℹ Use `Test.Model` instead.</code></pre>
<p><img src="cshl_gwas_files/figure-html/pp2-1.png" width="672" /></p>
</div>
<div id="continuous-study-with-continuous-gxe-interaction" class="section level4 hasAnchor" number="7.4.1.2">
<h4><span class="header-section-number">7.4.1.2</span> Continuous study with continuous GxE interaction:<a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#continuous-study-with-continuous-gxe-interaction" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>In addition to power calculation for main effects model, <code>genpwr</code> is also capable to calculate power for studies with GxE interactions. We just show you a continuous study with continuous GxE interactions here. You can also practice for a binary study with continuous GxE interactions, or study with binary GxE interactions.</p>
<p>Besides the parameters above, we need to specify the effect size for both main effect environmental variables and GxE interactions. Also note that the sum of all effect size parameters shouldn’t exceed the total SD of phenotype. We also specify the type of GxE interactions with <code>ge.interaction="continuous"</code>.</p>
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb46-1"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb46-1" tabindex="-1"></a><span class="co"># R codes- run in R Console</span></span>
<span id="cb46-2"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb46-2" tabindex="-1"></a>pec <span class="ot">&lt;-</span> <span class="fu">genpwr.calc</span>(<span class="at">calc =</span> <span class="st">&quot;power&quot;</span>, <span class="at">model =</span> <span class="st">&quot;linear&quot;</span>,</span>
<span id="cb46-3"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb46-3" tabindex="-1"></a>                 <span class="at">ge.interaction =</span> <span class="st">&quot;continuous&quot;</span>,</span>
<span id="cb46-4"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb46-4" tabindex="-1"></a>                 <span class="at">N=</span><span class="dv">192</span>, <span class="at">sd_y=</span><span class="dv">67</span>, <span class="at">MAF=</span><span class="fu">seq</span>(<span class="fl">0.1</span>,<span class="fl">0.5</span>,<span class="fl">0.01</span>), <span class="at">ES_G=</span><span class="dv">30</span>, </span>
<span id="cb46-5"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb46-5" tabindex="-1"></a>                 <span class="at">ES_E=</span><span class="dv">5</span>, <span class="at">ES_GE=</span><span class="dv">20</span>, <span class="at">sd_e =</span><span class="dv">1</span>,  <span class="at">Alpha=</span><span class="fl">0.05</span>, </span>
<span id="cb46-6"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb46-6" tabindex="-1"></a>                   <span class="at">True.Model=</span><span class="fu">c</span>(<span class="st">&quot;Additive&quot;</span>,<span class="st">&quot;Dominant&quot;</span>, <span class="st">&quot;Recessive&quot;</span>), </span>
<span id="cb46-7"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb46-7" tabindex="-1"></a>                   <span class="at">Test.Model=</span><span class="fu">c</span>(<span class="st">&quot;Additive&quot;</span>,<span class="st">&quot;Dominant&quot;</span>, <span class="st">&quot;Recessive&quot;</span>))</span>
<span id="cb46-8"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb46-8" tabindex="-1"></a><span class="fu">power.plot</span>(pec)</span></code></pre></div>
<pre><code>## Warning: Use of `temp2$Power` is discouraged.
## ℹ Use `Power` instead.</code></pre>
<pre><code>## Warning: Use of `temp2$Test.Model` is discouraged.
## ℹ Use `Test.Model` instead.</code></pre>
<pre><code>## Warning: Use of `temp2$Power` is discouraged.
## ℹ Use `Power` instead.</code></pre>
<pre><code>## Warning: Use of `temp2$Test.Model` is discouraged.
## ℹ Use `Test.Model` instead.</code></pre>
<p><img src="cshl_gwas_files/figure-html/pp3-1.png" width="672" /></p>
</div>
</div>
<div id="sample-size-calculation" class="section level3 hasAnchor" number="7.4.2">
<h3><span class="header-section-number">7.4.2</span> Sample size calculation<a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#sample-size-calculation" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>For sample size calculation, we want to determine an appropriate study size that is sufficient to detect the effect size/OR under a desired power.</p>
<div id="case-study-with-binary-outcome" class="section level4 hasAnchor" number="7.4.2.1">
<h4><span class="header-section-number">7.4.2.1</span> Case study with binary outcome:<a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#case-study-with-binary-outcome" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>We still use the above binary case study with 192 individuals. Assuming the case rate is 50% and MAF of variants is 0.2-0.5, we want to calculate the sample size needed to detect an odds ratio of 3 with a power of 0.7 or 0.8.</p>
<p>We need to spcific calculation type with <code>calc="ss"</code> and <code>Power=c(0.7,0.8)</code>.</p>
<div class="sourceCode" id="cb51"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb51-1"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb51-1" tabindex="-1"></a><span class="co"># R codes- run in R Console</span></span>
<span id="cb51-2"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb51-2" tabindex="-1"></a>ss <span class="ot">&lt;-</span> <span class="fu">genpwr.calc</span>(<span class="at">calc =</span> <span class="st">&quot;ss&quot;</span>, <span class="at">model =</span> <span class="st">&quot;logistic&quot;</span>, <span class="at">ge.interaction =</span> <span class="cn">NULL</span>,</span>
<span id="cb51-3"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb51-3" tabindex="-1"></a>                  <span class="at">OR=</span><span class="dv">3</span>, <span class="at">Case.Rate=</span><span class="fl">0.5</span>, <span class="at">k=</span><span class="cn">NULL</span>,</span>
<span id="cb51-4"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb51-4" tabindex="-1"></a>                  <span class="at">MAF=</span><span class="fu">seq</span>(<span class="fl">0.2</span>, <span class="fl">0.5</span>, <span class="fl">0.01</span>), <span class="at">Power=</span><span class="fu">c</span>(<span class="fl">0.7</span>,<span class="fl">0.8</span>), <span class="at">Alpha=</span><span class="fl">0.05</span>,</span>
<span id="cb51-5"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb51-5" tabindex="-1"></a>                  <span class="at">True.Model=</span><span class="fu">c</span>(<span class="st">&quot;Additive&quot;</span>,<span class="st">&quot;Dominant&quot;</span>, <span class="st">&quot;Recessive&quot;</span>), </span>
<span id="cb51-6"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb51-6" tabindex="-1"></a>                  <span class="at">Test.Model=</span><span class="fu">c</span>(<span class="st">&quot;Additive&quot;</span>,<span class="st">&quot;Dominant&quot;</span>, <span class="st">&quot;Recessive&quot;</span>))</span>
<span id="cb51-7"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb51-7" tabindex="-1"></a><span class="fu">ss.plot</span>(ss)</span></code></pre></div>
<pre><code>## Warning: Use of `temp2$N_total` is discouraged.
## ℹ Use `N_total` instead.</code></pre>
<pre><code>## Warning: Use of `temp2$Test.Model` is discouraged.
## ℹ Use `Test.Model` instead.</code></pre>
<pre><code>## Warning: Use of `temp2$N_total` is discouraged.
## ℹ Use `N_total` instead.</code></pre>
<pre><code>## Warning: Use of `temp2$Test.Model` is discouraged.
## ℹ Use `Test.Model` instead.</code></pre>
<p><img src="cshl_gwas_files/figure-html/ss1-1.png" width="672" /></p>
<pre><code>## Warning: Use of `temp2$N_total` is discouraged.
## ℹ Use `N_total` instead.
## Use of `temp2$Test.Model` is discouraged.
## ℹ Use `Test.Model` instead.</code></pre>
<pre><code>## Warning: Use of `temp2$N_total` is discouraged.
## ℹ Use `N_total` instead.</code></pre>
<pre><code>## Warning: Use of `temp2$Test.Model` is discouraged.
## ℹ Use `Test.Model` instead.</code></pre>
<p><img src="cshl_gwas_files/figure-html/ss1-2.png" width="672" /></p>
<p>As a general rule, under the same assumptions, the dominant model needs the smallest sample size to achieve adequate statistical power, whereas the recessive model is the one that requires the largest number of participants.</p>
<p>We could check the dominant and additive model only and ignore the recessive model.</p>
<div class="sourceCode" id="cb59"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb59-1"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb59-1" tabindex="-1"></a><span class="co"># R codes- run in R Console</span></span>
<span id="cb59-2"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb59-2" tabindex="-1"></a>ss <span class="ot">&lt;-</span> <span class="fu">genpwr.calc</span>(<span class="at">calc =</span> <span class="st">&quot;ss&quot;</span>, <span class="at">model =</span> <span class="st">&quot;logistic&quot;</span>, <span class="at">ge.interaction =</span> <span class="cn">NULL</span>,</span>
<span id="cb59-3"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb59-3" tabindex="-1"></a>                  <span class="at">OR=</span><span class="dv">3</span>, <span class="at">Case.Rate=</span><span class="fl">0.5</span>, <span class="at">k=</span><span class="cn">NULL</span>,</span>
<span id="cb59-4"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb59-4" tabindex="-1"></a>                  <span class="at">MAF=</span><span class="fu">seq</span>(<span class="fl">0.2</span>, <span class="fl">0.5</span>, <span class="fl">0.01</span>), <span class="at">Power=</span><span class="fu">c</span>(<span class="fl">0.7</span>,<span class="fl">0.8</span>), <span class="at">Alpha=</span><span class="fl">0.05</span>,</span>
<span id="cb59-5"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb59-5" tabindex="-1"></a>                  <span class="at">True.Model=</span><span class="fu">c</span>(<span class="st">&quot;Additive&quot;</span>,<span class="st">&quot;Dominant&quot;</span>), </span>
<span id="cb59-6"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb59-6" tabindex="-1"></a>                  <span class="at">Test.Model=</span><span class="fu">c</span>(<span class="st">&quot;Additive&quot;</span>,<span class="st">&quot;Dominant&quot;</span>, <span class="st">&quot;Recessive&quot;</span>))</span>
<span id="cb59-7"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb59-7" tabindex="-1"></a><span class="fu">ss.plot</span>(ss)</span></code></pre></div>
<pre><code>## Warning: Use of `temp2$N_total` is discouraged.
## ℹ Use `N_total` instead.</code></pre>
<pre><code>## Warning: Use of `temp2$Test.Model` is discouraged.
## ℹ Use `Test.Model` instead.</code></pre>
<pre><code>## Warning: Use of `temp2$N_total` is discouraged.
## ℹ Use `N_total` instead.</code></pre>
<pre><code>## Warning: Use of `temp2$Test.Model` is discouraged.
## ℹ Use `Test.Model` instead.</code></pre>
<p><img src="cshl_gwas_files/figure-html/ss2-1.png" width="672" /></p>
<pre><code>## Warning: Use of `temp2$N_total` is discouraged.
## ℹ Use `N_total` instead.
## Use of `temp2$Test.Model` is discouraged.
## ℹ Use `Test.Model` instead.</code></pre>
<pre><code>## Warning: Use of `temp2$N_total` is discouraged.
## ℹ Use `N_total` instead.</code></pre>
<pre><code>## Warning: Use of `temp2$Test.Model` is discouraged.
## ℹ Use `Test.Model` instead.</code></pre>
<p><img src="cshl_gwas_files/figure-html/ss2-2.png" width="672" /></p>
<p>For both additive and dominant models, a sample size of about 200 individuals is sufficient to detect an OR of 3 if we test for additive or dominant genetic effects.</p>
</div>
<div id="case-study-with-continuous-outcome-1" class="section level4 hasAnchor" number="7.4.2.2">
<h4><span class="header-section-number">7.4.2.2</span> Case study with continuous outcome<a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#case-study-with-continuous-outcome-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Similar as above, we calculate the sample size needed to detect an effect size of 30 with 70% or 80% power. The continuous phenotype used is still tamoxifen with SD=76.</p>
<div class="sourceCode" id="cb67"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb67-1"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb67-1" tabindex="-1"></a><span class="co"># R codes- run in R Console</span></span>
<span id="cb67-2"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb67-2" tabindex="-1"></a>ss <span class="ot">&lt;-</span> <span class="fu">genpwr.calc</span>(<span class="at">calc =</span> <span class="st">&quot;ss&quot;</span>, <span class="at">model =</span> <span class="st">&quot;linear&quot;</span>,</span>
<span id="cb67-3"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb67-3" tabindex="-1"></a>                  <span class="at">sd_y=</span><span class="dv">67</span>, <span class="at">k=</span><span class="cn">NULL</span>, <span class="at">Power=</span><span class="fu">c</span>(<span class="fl">0.7</span>,<span class="fl">0.8</span>),</span>
<span id="cb67-4"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb67-4" tabindex="-1"></a>                  <span class="at">MAF=</span><span class="fu">seq</span>(<span class="fl">0.1</span>, <span class="fl">0.5</span>, <span class="fl">0.01</span>), <span class="at">ES=</span><span class="dv">30</span>, <span class="at">Alpha=</span><span class="fl">0.05</span>,</span>
<span id="cb67-5"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb67-5" tabindex="-1"></a>                  <span class="at">True.Model=</span><span class="fu">c</span>(<span class="st">&quot;Additive&quot;</span>,<span class="st">&quot;Dominant&quot;</span>), </span>
<span id="cb67-6"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb67-6" tabindex="-1"></a>                  <span class="at">Test.Model=</span><span class="fu">c</span>(<span class="st">&quot;Additive&quot;</span>,<span class="st">&quot;Dominant&quot;</span>, <span class="st">&quot;Recessive&quot;</span>))</span>
<span id="cb67-7"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb67-7" tabindex="-1"></a></span>
<span id="cb67-8"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb67-8" tabindex="-1"></a><span class="fu">ss.plot</span>(ss)</span></code></pre></div>
<pre><code>## Warning: Use of `temp2$N_total` is discouraged.
## ℹ Use `N_total` instead.</code></pre>
<pre><code>## Warning: Use of `temp2$Test.Model` is discouraged.
## ℹ Use `Test.Model` instead.</code></pre>
<pre><code>## Warning: Use of `temp2$N_total` is discouraged.
## ℹ Use `N_total` instead.</code></pre>
<pre><code>## Warning: Use of `temp2$Test.Model` is discouraged.
## ℹ Use `Test.Model` instead.</code></pre>
<p><img src="cshl_gwas_files/figure-html/ss3-1.png" width="672" /></p>
<pre><code>## Warning: Use of `temp2$N_total` is discouraged.
## ℹ Use `N_total` instead.
## Use of `temp2$Test.Model` is discouraged.
## ℹ Use `Test.Model` instead.</code></pre>
<pre><code>## Warning: Use of `temp2$N_total` is discouraged.
## ℹ Use `N_total` instead.</code></pre>
<pre><code>## Warning: Use of `temp2$Test.Model` is discouraged.
## ℹ Use `Test.Model` instead.</code></pre>
<p><img src="cshl_gwas_files/figure-html/ss3-2.png" width="672" />
#### Binary study with binary GxE interactions</p>
<div class="sourceCode" id="cb75"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb75-1"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb75-1" tabindex="-1"></a><span class="co"># R codes- run in R Console</span></span>
<span id="cb75-2"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb75-2" tabindex="-1"></a>sse <span class="ot">&lt;-</span> <span class="fu">genpwr.calc</span>(<span class="at">calc =</span> <span class="st">&quot;ss&quot;</span>, <span class="at">model =</span> <span class="st">&quot;logistic&quot;</span>,</span>
<span id="cb75-3"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb75-3" tabindex="-1"></a>                  <span class="at">Case.Rate=</span><span class="fl">0.5</span>, <span class="at">ge.interaction =</span> <span class="st">&quot;binary&quot;</span>, <span class="at">Power=</span><span class="fl">0.8</span>,</span>
<span id="cb75-4"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb75-4" tabindex="-1"></a>                  <span class="at">MAF=</span><span class="fu">seq</span>(<span class="fl">0.1</span>,<span class="fl">0.5</span>,<span class="fl">0.01</span>), <span class="at">OR_G=</span><span class="dv">3</span>, </span>
<span id="cb75-5"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb75-5" tabindex="-1"></a>                  <span class="at">OR_E=</span><span class="fl">0.5</span>, <span class="at">OR_GE=</span><span class="dv">4</span>, <span class="at">P_e =</span> <span class="fl">0.4</span>,  <span class="at">Alpha=</span><span class="fl">0.05</span>, </span>
<span id="cb75-6"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb75-6" tabindex="-1"></a>                  <span class="at">True.Model=</span><span class="fu">c</span>(<span class="st">&quot;Additive&quot;</span>,<span class="st">&quot;Dominant&quot;</span>), </span>
<span id="cb75-7"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb75-7" tabindex="-1"></a>                  <span class="at">Test.Model=</span><span class="fu">c</span>(<span class="st">&quot;Additive&quot;</span>,<span class="st">&quot;Dominant&quot;</span>, <span class="st">&quot;Recessive&quot;</span>))</span>
<span id="cb75-8"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#cb75-8" tabindex="-1"></a><span class="fu">ss.plot</span>(sse)</span></code></pre></div>
<pre><code>## Warning: Use of `temp2$N_total` is discouraged.
## ℹ Use `N_total` instead.</code></pre>
<pre><code>## Warning: Use of `temp2$Test.Model` is discouraged.
## ℹ Use `Test.Model` instead.</code></pre>
<pre><code>## Warning: Use of `temp2$N_total` is discouraged.
## ℹ Use `N_total` instead.</code></pre>
<pre><code>## Warning: Use of `temp2$Test.Model` is discouraged.
## ℹ Use `Test.Model` instead.</code></pre>
<p><img src="cshl_gwas_files/figure-html/ss4-1.png" width="672" /></p>

</div>
</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="gwas-with-binary-trait.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="ldsc-and-gene-set-analysis.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/06-gwas_gxe.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["cshl_gwas.pdf", "cshl_gwas.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
