<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>References | CSHL course - GWAS Hands-on Analysis</title>
  <meta name="description" content="This is the course material for the CSHL hands-on session in GWAS and eQTL." />
  <meta name="generator" content="bookdown 0.41 and GitBook 2.6.7" />

  <meta property="og:title" content="References | CSHL course - GWAS Hands-on Analysis" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This is the course material for the CSHL hands-on session in GWAS and eQTL." />
  <meta name="github-repo" content="rstudio/cshl_gwas" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="References | CSHL course - GWAS Hands-on Analysis" />
  
  <meta name="twitter:description" content="This is the course material for the CSHL hands-on session in GWAS and eQTL." />
  

<meta name="author" content="Min Zhang, Danni Liu- University of California, Irvine; Purdue University" />


<meta name="date" content="2025-06-09" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="perform-eqtl-analysis.html"/>

<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>



<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">CSHL Course- GWAS and EQTL Hands-on Analysis</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Welcome to the GWAS and eQTL Workshop</a>
<ul>
<li class="chapter" data-level="1.0.1" data-path="index.html"><a href="index.html#overview"><i class="fa fa-check"></i><b>1.0.1</b> Overview</a></li>
<li class="chapter" data-level="1.0.2" data-path="index.html"><a href="index.html#folder-structure"><i class="fa fa-check"></i><b>1.0.2</b> Folder structure:</a></li>
<li class="chapter" data-level="1.0.3" data-path="index.html"><a href="index.html#programming-languages"><i class="fa fa-check"></i><b>1.0.3</b> Programming Languages:</a></li>
<li class="chapter" data-level="1.0.4" data-path="index.html"><a href="index.html#required-gwas-tools"><i class="fa fa-check"></i><b>1.0.4</b> Required GWAS Tools</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="an-overview-of-regression-with-toy-examples.html"><a href="an-overview-of-regression-with-toy-examples.html"><i class="fa fa-check"></i><b>2</b> An Overview of Regression with Toy Examples</a>
<ul>
<li class="chapter" data-level="2.1" data-path="an-overview-of-regression-with-toy-examples.html"><a href="an-overview-of-regression-with-toy-examples.html#environment-setting-and-packages"><i class="fa fa-check"></i><b>2.1</b> Environment Setting and Packages</a></li>
<li class="chapter" data-level="2.2" data-path="an-overview-of-regression-with-toy-examples.html"><a href="an-overview-of-regression-with-toy-examples.html#toy-example-data"><i class="fa fa-check"></i><b>2.2</b> Toy Example Data</a></li>
<li class="chapter" data-level="2.3" data-path="an-overview-of-regression-with-toy-examples.html"><a href="an-overview-of-regression-with-toy-examples.html#logistic-regression"><i class="fa fa-check"></i><b>2.3</b> Logistic Regression</a></li>
<li class="chapter" data-level="2.4" data-path="an-overview-of-regression-with-toy-examples.html"><a href="an-overview-of-regression-with-toy-examples.html#roc-curve-and-auc-for-hoxb13"><i class="fa fa-check"></i><b>2.4</b> ROC curve and AUC for HOXB13</a></li>
<li class="chapter" data-level="2.5" data-path="an-overview-of-regression-with-toy-examples.html"><a href="an-overview-of-regression-with-toy-examples.html#practice"><i class="fa fa-check"></i><b>2.5</b> Practice</a></li>
<li class="chapter" data-level="2.6" data-path="an-overview-of-regression-with-toy-examples.html"><a href="an-overview-of-regression-with-toy-examples.html#comparison-of-hoxb13-roc-curve-and-nhlh1-roc-curve"><i class="fa fa-check"></i><b>2.6</b> Comparison of HOXB13 ROC curve and NHLH1 ROC curve</a></li>
<li class="chapter" data-level="2.7" data-path="an-overview-of-regression-with-toy-examples.html"><a href="an-overview-of-regression-with-toy-examples.html#linear-regression"><i class="fa fa-check"></i><b>2.7</b> Linear Regression</a>
<ul>
<li class="chapter" data-level="2.7.1" data-path="an-overview-of-regression-with-toy-examples.html"><a href="an-overview-of-regression-with-toy-examples.html#linear-regression-of-hoxb13-vs.-il17br"><i class="fa fa-check"></i><b>2.7.1</b> Linear regression of HOXB13 vs. IL17BR</a></li>
</ul></li>
<li class="chapter" data-level="2.8" data-path="an-overview-of-regression-with-toy-examples.html"><a href="an-overview-of-regression-with-toy-examples.html#variable-selectionforward-backward-and-stepwise-selection"><i class="fa fa-check"></i><b>2.8</b> Variable selection–Forward, backward and stepwise selection</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html"><i class="fa fa-check"></i><b>3</b> GWAS Tools and Data Preprocessing Workflow with Toy Example</a>
<ul>
<li class="chapter" data-level="3.1" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#environment-setting"><i class="fa fa-check"></i><b>3.1</b> Environment Setting</a></li>
<li class="chapter" data-level="3.2" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#gwas-tools"><i class="fa fa-check"></i><b>3.2</b> GWAS Tools</a></li>
<li class="chapter" data-level="3.3" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#quality-control"><i class="fa fa-check"></i><b>3.3</b> Quality Control</a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#an-example-of-summary-statistics"><i class="fa fa-check"></i><b>3.3.1</b> An Example of Summary Statistics</a></li>
<li class="chapter" data-level="3.3.2" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#an-example-of-inclusion-thresholds-for-qc"><i class="fa fa-check"></i><b>3.3.2</b> An Example of Inclusion Thresholds for QC</a></li>
<li class="chapter" data-level="3.3.3" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#remove-closely-related-individuals"><i class="fa fa-check"></i><b>3.3.3</b> Remove Closely Related Individuals</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#an-imputation-example"><i class="fa fa-check"></i><b>3.4</b> An Imputation Example</a></li>
<li class="chapter" data-level="3.5" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#population-stratification-correction-using-plink"><i class="fa fa-check"></i><b>3.5</b> Population Stratification Correction Using PLINK</a>
<ul>
<li class="chapter" data-level="3.5.1" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#pca-on-whole-genome-data-using-plink"><i class="fa fa-check"></i><b>3.5.1</b> PCA on whole genome data using PLINK</a></li>
<li class="chapter" data-level="3.5.2" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#test-the-significance-of-principle-components-pcs"><i class="fa fa-check"></i><b>3.5.2</b> Test the Significance of Principle Components (PCs)</a></li>
</ul></li>
<li class="chapter" data-level="3.6" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#simple-association-test"><i class="fa fa-check"></i><b>3.6</b> Simple Association Test</a>
<ul>
<li class="chapter" data-level="3.6.1" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#quantitative-trait-association-test-for-continuous-trait"><i class="fa fa-check"></i><b>3.6.1</b> Quantitative Trait Association Test for Continuous Trait</a></li>
<li class="chapter" data-level="3.6.2" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#casecontrol-asssociation-test-for-binary-trait"><i class="fa fa-check"></i><b>3.6.2</b> Case/Control Asssociation Test for Binary Trait</a></li>
</ul></li>
<li class="chapter" data-level="3.7" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#quantitative-trait-association-test-for-continuous-trait-with-covariates"><i class="fa fa-check"></i><b>3.7</b> Quantitative Trait Association Test for Continuous Trait with Covariates</a>
<ul>
<li class="chapter" data-level="3.7.1" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#covariates-adjustments-using-r"><i class="fa fa-check"></i><b>3.7.1</b> Covariates adjustments (using R):</a></li>
<li class="chapter" data-level="3.7.2" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example.html#association-test-using-plink"><i class="fa fa-check"></i><b>3.7.2</b> Association test using plink</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="gwas-case-study-data-preparation.html"><a href="gwas-case-study-data-preparation.html"><i class="fa fa-check"></i><b>4</b> GWAS Case Study-Data Preparation</a>
<ul>
<li class="chapter" data-level="4.1" data-path="gwas-case-study-data-preparation.html"><a href="gwas-case-study-data-preparation.html#environment-setting-and-packages-1"><i class="fa fa-check"></i><b>4.1</b> Environment setting and packages</a></li>
<li class="chapter" data-level="4.2" data-path="gwas-case-study-data-preparation.html"><a href="gwas-case-study-data-preparation.html#accessing-data"><i class="fa fa-check"></i><b>4.2</b> Accessing data</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="gwas-case-study-data-preparation.html"><a href="gwas-case-study-data-preparation.html#data-availability"><i class="fa fa-check"></i><b>4.2.1</b> Data availability</a></li>
<li class="chapter" data-level="4.2.2" data-path="gwas-case-study-data-preparation.html"><a href="gwas-case-study-data-preparation.html#data-files-documentation"><i class="fa fa-check"></i><b>4.2.2</b> Data files documentation</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="gwas-case-study-data-preparation.html"><a href="gwas-case-study-data-preparation.html#preprocessing-phenotype-data"><i class="fa fa-check"></i><b>4.3</b> Preprocessing Phenotype Data</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="gwas-case-study-data-preparation.html"><a href="gwas-case-study-data-preparation.html#extracting-phenotype-data"><i class="fa fa-check"></i><b>4.3.1</b> Extracting phenotype data</a></li>
<li class="chapter" data-level="4.3.2" data-path="gwas-case-study-data-preparation.html"><a href="gwas-case-study-data-preparation.html#summary-of-phenotypes-data"><i class="fa fa-check"></i><b>4.3.2</b> Summary of Phenotypes Data</a></li>
<li class="chapter" data-level="4.3.3" data-path="gwas-case-study-data-preparation.html"><a href="gwas-case-study-data-preparation.html#histogram-of-each-trait"><i class="fa fa-check"></i><b>4.3.3</b> Histogram of each trait</a></li>
<li class="chapter" data-level="4.3.4" data-path="gwas-case-study-data-preparation.html"><a href="gwas-case-study-data-preparation.html#handling-the-missing-values-in-the-phenotypes-and-covariates"><i class="fa fa-check"></i><b>4.3.4</b> Handling the missing values in the phenotypes and covariates</a></li>
<li class="chapter" data-level="4.3.5" data-path="gwas-case-study-data-preparation.html"><a href="gwas-case-study-data-preparation.html#checking-normality-assumptions-for-each-trait"><i class="fa fa-check"></i><b>4.3.5</b> Checking normality assumptions for each trait</a></li>
<li class="chapter" data-level="4.3.6" data-path="gwas-case-study-data-preparation.html"><a href="gwas-case-study-data-preparation.html#normalize-phenotype-data"><i class="fa fa-check"></i><b>4.3.6</b> Normalize phenotype data</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="gwas-case-study-data-preparation.html"><a href="gwas-case-study-data-preparation.html#covariates-adjustment"><i class="fa fa-check"></i><b>4.4</b> Covariates Adjustment</a></li>
<li class="chapter" data-level="4.5" data-path="gwas-case-study-data-preparation.html"><a href="gwas-case-study-data-preparation.html#prepare-plink-data"><i class="fa fa-check"></i><b>4.5</b> Prepare PLINK data</a>
<ul>
<li class="chapter" data-level="4.5.1" data-path="gwas-case-study-data-preparation.html"><a href="gwas-case-study-data-preparation.html#step-1-prepare-plink-ped-file"><i class="fa fa-check"></i><b>4.5.1</b> Step 1: prepare PLINK ped file</a></li>
<li class="chapter" data-level="4.5.2" data-path="gwas-case-study-data-preparation.html"><a href="gwas-case-study-data-preparation.html#step-2-prepare-plink-map-file"><i class="fa fa-check"></i><b>4.5.2</b> Step 2: prepare PLINK map file</a></li>
<li class="chapter" data-level="4.5.3" data-path="gwas-case-study-data-preparation.html"><a href="gwas-case-study-data-preparation.html#step-3-prepare-plink-alternate-phenotype-file-in-linux"><i class="fa fa-check"></i><b>4.5.3</b> Step 3 : prepare PLINK alternate phenotype file (in Linux)</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="gwas-with-continuous-trait.html"><a href="gwas-with-continuous-trait.html"><i class="fa fa-check"></i><b>5</b> GWAS with Continuous Trait</a>
<ul>
<li class="chapter" data-level="5.1" data-path="gwas-with-continuous-trait.html"><a href="gwas-with-continuous-trait.html#manhattan-plot-and-q-q-plot"><i class="fa fa-check"></i><b>5.1</b> Manhattan Plot and Q-Q Plot</a>
<ul>
<li class="chapter" data-level="5.1.1" data-path="gwas-with-continuous-trait.html"><a href="gwas-with-continuous-trait.html#generate-manhattan-plot"><i class="fa fa-check"></i><b>5.1.1</b> Generate manhattan plot</a></li>
<li class="chapter" data-level="5.1.2" data-path="gwas-with-continuous-trait.html"><a href="gwas-with-continuous-trait.html#generate-q-q-plot"><i class="fa fa-check"></i><b>5.1.2</b> Generate Q-Q plot</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="gwas-with-continuous-trait.html"><a href="gwas-with-continuous-trait.html#multiple-test-correction"><i class="fa fa-check"></i><b>5.2</b> Multiple-Test Correction</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="gwas-with-continuous-trait.html"><a href="gwas-with-continuous-trait.html#prepare-r-input-files-in-linux"><i class="fa fa-check"></i><b>5.2.1</b> Prepare R input files (in Linux)</a></li>
<li class="chapter" data-level="5.2.2" data-path="gwas-with-continuous-trait.html"><a href="gwas-with-continuous-trait.html#significant-snps-after-a-multiple-test-correction-of-fdr-using-r"><i class="fa fa-check"></i><b>5.2.2</b> Significant SNPs after a multiple-test correction of FDR (using R)</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="gwas-with-continuous-trait.html"><a href="gwas-with-continuous-trait.html#regional-association-plot"><i class="fa fa-check"></i><b>5.3</b> Regional Association Plot</a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="gwas-with-continuous-trait.html"><a href="gwas-with-continuous-trait.html#obtain-regional-snps-using-r"><i class="fa fa-check"></i><b>5.3.1</b> Obtain regional SNPs (using R)</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="gwas-with-continuous-trait.html"><a href="gwas-with-continuous-trait.html#distribution-of-z-endoxifen-ratio-for-rs8138080"><i class="fa fa-check"></i><b>5.4</b> Distribution of Z-endoxifen ratio for rs8138080</a>
<ul>
<li class="chapter" data-level="5.4.1" data-path="gwas-with-continuous-trait.html"><a href="gwas-with-continuous-trait.html#obtain-phenotype-data-z-endoxifen-ratio-using-r"><i class="fa fa-check"></i><b>5.4.1</b> Obtain phenotype data (Z-endoxifen ratio) (using R)</a></li>
<li class="chapter" data-level="5.4.2" data-path="gwas-with-continuous-trait.html"><a href="gwas-with-continuous-trait.html#obtain-genotype-for-significant-snps-using-r"><i class="fa fa-check"></i><b>5.4.2</b> Obtain genotype for significant SNPs (using R)</a></li>
<li class="chapter" data-level="5.4.3" data-path="gwas-with-continuous-trait.html"><a href="gwas-with-continuous-trait.html#create-box-plot-using-r"><i class="fa fa-check"></i><b>5.4.3</b> Create Box plot (using R)</a></li>
</ul></li>
<li class="chapter" data-level="5.5" data-path="gwas-with-continuous-trait.html"><a href="gwas-with-continuous-trait.html#proportion-of-variation-explained-by-rs8138080"><i class="fa fa-check"></i><b>5.5</b> Proportion of Variation Explained by rs8138080</a>
<ul>
<li class="chapter" data-level="5.5.1" data-path="gwas-with-continuous-trait.html"><a href="gwas-with-continuous-trait.html#recode-genotype-of-significant-snps-using-r"><i class="fa fa-check"></i><b>5.5.1</b> Recode genotype of significant SNPs (using R)</a></li>
<li class="chapter" data-level="5.5.2" data-path="gwas-with-continuous-trait.html"><a href="gwas-with-continuous-trait.html#calculate-marginal-r2-using-r"><i class="fa fa-check"></i><b>5.5.2</b> Calculate marginal <span class="math inline">\(R^2\)</span> (using R)</a></li>
</ul></li>
<li class="chapter" data-level="5.6" data-path="gwas-with-continuous-trait.html"><a href="gwas-with-continuous-trait.html#practice-time"><i class="fa fa-check"></i><b>5.6</b> PRACTICE TIME</a>
<ul>
<li class="chapter" data-level="5.6.1" data-path="gwas-with-continuous-trait.html"><a href="gwas-with-continuous-trait.html#practice-1"><i class="fa fa-check"></i><b>5.6.1</b> Practice 1</a></li>
<li class="chapter" data-level="5.6.2" data-path="gwas-with-continuous-trait.html"><a href="gwas-with-continuous-trait.html#practice-2"><i class="fa fa-check"></i><b>5.6.2</b> Practice 2</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="gwas-with-binary-trait.html"><a href="gwas-with-binary-trait.html"><i class="fa fa-check"></i><b>6</b> GWAS with Binary Trait</a>
<ul>
<li class="chapter" data-level="6.1" data-path="gwas-with-binary-trait.html"><a href="gwas-with-binary-trait.html#data-preparation"><i class="fa fa-check"></i><b>6.1</b> Data Preparation</a></li>
<li class="chapter" data-level="6.2" data-path="gwas-with-binary-trait.html"><a href="gwas-with-binary-trait.html#genome-wide-casecontrol-association-test-in-plink"><i class="fa fa-check"></i><b>6.2</b> Genome-Wide Case/Control Association Test (in PLINK)</a></li>
<li class="chapter" data-level="6.3" data-path="gwas-with-binary-trait.html"><a href="gwas-with-binary-trait.html#manhattan-plot-and-q-q-plot-1"><i class="fa fa-check"></i><b>6.3</b> Manhattan Plot and Q-Q Plot</a></li>
<li class="chapter" data-level="6.4" data-path="gwas-with-binary-trait.html"><a href="gwas-with-binary-trait.html#generate-manhattan-plot-1"><i class="fa fa-check"></i><b>6.4</b> Generate Manhattan plot</a></li>
<li class="chapter" data-level="6.5" data-path="gwas-with-binary-trait.html"><a href="gwas-with-binary-trait.html#generate-q-q-plot-1"><i class="fa fa-check"></i><b>6.5</b> Generate Q-Q plot</a></li>
<li class="chapter" data-level="6.6" data-path="gwas-with-binary-trait.html"><a href="gwas-with-binary-trait.html#multiple-test-correction-1"><i class="fa fa-check"></i><b>6.6</b> Multiple-Test Correction</a></li>
<li class="chapter" data-level="6.7" data-path="gwas-with-binary-trait.html"><a href="gwas-with-binary-trait.html#top-10-snps"><i class="fa fa-check"></i><b>6.7</b> Top 10 SNPs</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="gxg-and-gxe-interactions-power-and-sample-size-calculation.html"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html"><i class="fa fa-check"></i><b>7</b> GxG and GxE Interactions, Power and Sample Size Calculation</a>
<ul>
<li class="chapter" data-level="7.1" data-path="gxg-and-gxe-interactions-power-and-sample-size-calculation.html"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#prepare-files"><i class="fa fa-check"></i><b>7.1</b> Prepare Files</a></li>
<li class="chapter" data-level="7.2" data-path="gxg-and-gxe-interactions-power-and-sample-size-calculation.html"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#test-for-epistasis-gxg"><i class="fa fa-check"></i><b>7.2</b> Test for Epistasis (GxG)</a></li>
<li class="chapter" data-level="7.3" data-path="gxg-and-gxe-interactions-power-and-sample-size-calculation.html"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#gene-environment-interaction-gxe"><i class="fa fa-check"></i><b>7.3</b> Gene-Environment Interaction (GxE)</a></li>
<li class="chapter" data-level="7.4" data-path="gxg-and-gxe-interactions-power-and-sample-size-calculation.html"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#power-and-sample-size-calculation-for-gwas"><i class="fa fa-check"></i><b>7.4</b> Power and sample size calculation for GWAS</a>
<ul>
<li class="chapter" data-level="7.4.1" data-path="gxg-and-gxe-interactions-power-and-sample-size-calculation.html"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#power-calculation-with-binary-outcome"><i class="fa fa-check"></i><b>7.4.1</b> Power calculation with binary outcome:</a></li>
<li class="chapter" data-level="7.4.2" data-path="gxg-and-gxe-interactions-power-and-sample-size-calculation.html"><a href="gxg-and-gxe-interactions-power-and-sample-size-calculation.html#sample-size-calculation"><i class="fa fa-check"></i><b>7.4.2</b> Sample size calculation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8" data-path="ldsc-and-gene-set-analysis.html"><a href="ldsc-and-gene-set-analysis.html"><i class="fa fa-check"></i><b>8</b> LDSC and Gene-Set Analysis</a>
<ul>
<li class="chapter" data-level="8.1" data-path="ldsc-and-gene-set-analysis.html"><a href="ldsc-and-gene-set-analysis.html#ldsc-linkage-disequilibrium-score-regression"><i class="fa fa-check"></i><b>8.1</b> LDSC (Linkage Disequilibrium Score Regression)</a></li>
<li class="chapter" data-level="8.2" data-path="ldsc-and-gene-set-analysis.html"><a href="ldsc-and-gene-set-analysis.html#magma-for-gene-level-analysis"><i class="fa fa-check"></i><b>8.2</b> MAGMA (for Gene-Level Analysis)</a>
<ul>
<li class="chapter" data-level="8.2.1" data-path="ldsc-and-gene-set-analysis.html"><a href="ldsc-and-gene-set-analysis.html#prepare-annotation-file"><i class="fa fa-check"></i><b>8.2.1</b> Prepare Annotation File</a></li>
<li class="chapter" data-level="8.2.2" data-path="ldsc-and-gene-set-analysis.html"><a href="ldsc-and-gene-set-analysis.html#part-1-gene-analysis"><i class="fa fa-check"></i><b>8.2.2</b> Part 1: Gene Analysis</a></li>
<li class="chapter" data-level="8.2.3" data-path="ldsc-and-gene-set-analysis.html"><a href="ldsc-and-gene-set-analysis.html#part-2-gene-set-analysis"><i class="fa fa-check"></i><b>8.2.3</b> Part 2: Gene-Set Analysis</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="9" data-path="perform-eqtl-analysis.html"><a href="perform-eqtl-analysis.html"><i class="fa fa-check"></i><b>9</b> Perform eQTL Analysis</a>
<ul>
<li class="chapter" data-level="9.1" data-path="perform-eqtl-analysis.html"><a href="perform-eqtl-analysis.html#filtering-snps-by-maf"><i class="fa fa-check"></i><b>9.1</b> Filtering SNPs by MAF</a></li>
<li class="chapter" data-level="9.2" data-path="perform-eqtl-analysis.html"><a href="perform-eqtl-analysis.html#gene-expression-profiling"><i class="fa fa-check"></i><b>9.2</b> Gene expression profiling</a></li>
<li class="chapter" data-level="9.3" data-path="perform-eqtl-analysis.html"><a href="perform-eqtl-analysis.html#covariates"><i class="fa fa-check"></i><b>9.3</b> Covariates</a></li>
<li class="chapter" data-level="9.4" data-path="perform-eqtl-analysis.html"><a href="perform-eqtl-analysis.html#using-principle-components-as-covariates"><i class="fa fa-check"></i><b>9.4</b> Using principle components as covariates</a></li>
<li class="chapter" data-level="9.5" data-path="perform-eqtl-analysis.html"><a href="perform-eqtl-analysis.html#eqtl-analysis"><i class="fa fa-check"></i><b>9.5</b> eQTL analysis</a>
<ul>
<li class="chapter" data-level="9.5.1" data-path="perform-eqtl-analysis.html"><a href="perform-eqtl-analysis.html#cis-eqtl-and-trans-eqtl-analysis"><i class="fa fa-check"></i><b>9.5.1</b> cis-eQTL and trans-eQTL analysis</a></li>
</ul></li>
<li class="chapter" data-level="9.6" data-path="perform-eqtl-analysis.html"><a href="perform-eqtl-analysis.html#multiple-testing-correction"><i class="fa fa-check"></i><b>9.6</b> Multiple testing correction</a></li>
<li class="chapter" data-level="9.7" data-path="perform-eqtl-analysis.html"><a href="perform-eqtl-analysis.html#gene-regional-manhanttan-plots"><i class="fa fa-check"></i><b>9.7</b> Gene regional Manhanttan plots</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">CSHL course - GWAS Hands-on Analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="references" class="section level1 unnumbered hasAnchor">
<h1>References<a href="references.html#references" class="anchor-section" aria-label="Anchor link to header"></a></h1>

</div>
            </section>

          </div>
        </div>
      </div>
<a href="perform-eqtl-analysis.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/09-reference.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["cshl_gwas.pdf", "cshl_gwas.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
